<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUchicago" />

<meta name="date" content="2022-02-01" />

<title>220131_Griess_pH_perturbation3</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Denit_visualization_R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/Denit_visualization_R">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">220131_Griess_pH_perturbation3</h1>
<h4 class="author">KiseokUchicago</h4>
<h4 class="date">2022-02-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Denit_visualization_R/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210924code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210924code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210924)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  211020_df_full_innate.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  220519_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/120321_soil_buffering_capacity_Nichols-B.xlsx
    Untracked:  data/210911_TOC_sample_plate1.xlsx
    Untracked:  data/210922_Griess_sample_plate1-investigating.xlsx
    Untracked:  data/210922_Griess_sample_plate1.xlsx
    Untracked:  data/210927_TOC_sample_plate1_100ul.xlsx
    Untracked:  data/210927_TOC_sample_plate1_10ul.xlsx
    Untracked:  data/210928_Griess_sample_plate1.xlsx
    Untracked:  data/211005_Griess_sample_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate2.xlsx
    Untracked:  data/211008_Griess_blank_plate0.xlsx
    Untracked:  data/211008_Griess_blank_plate1.xlsx
    Untracked:  data/211017_Griess_plate0.xlsx
    Untracked:  data/211017_Griess_plate1.xlsx
    Untracked:  data/211017_Griess_plate2.xlsx
    Untracked:  data/211017_Griess_plate3.xlsx
    Untracked:  data/211017_Griess_plate4.xlsx
    Untracked:  data/211017_Griess_plate5.xlsx
    Untracked:  data/211017_Griess_plate6.xlsx
    Untracked:  data/211017_Griess_plate7.xlsx
    Untracked:  data/211017_Griess_plate8.xlsx
    Untracked:  data/211019_Griess_plate1.xlsx
    Untracked:  data/211020_Griess_plate0.xlsx
    Untracked:  data/211020_Griess_plate1.xlsx
    Untracked:  data/211020_df_full_innate.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate0.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate1.xlsx
    Untracked:  data/211025_Griess_plate0.xlsx
    Untracked:  data/211025_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate2.xlsx
    Untracked:  data/211028_Griess_plate2_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate3.xlsx
    Untracked:  data/211028_Griess_plate3_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate4.xlsx
    Untracked:  data/211028_Griess_plate4_no3_fit.xlsx
    Untracked:  data/211028_time_table.xlsx
    Untracked:  data/211127_Griess_SUP1.xlsx
    Untracked:  data/211127_Griess_SUP11.xlsx
    Untracked:  data/211127_Griess_SUP3.xlsx
    Untracked:  data/211127_Griess_SUP5.xlsx
    Untracked:  data/211127_Griess_SUP7.xlsx
    Untracked:  data/211127_Griess_SUP9.xlsx
    Untracked:  data/211127_Griess_plate0.xlsx
    Untracked:  data/211127_time_table.xlsx
    Untracked:  data/211128_Ammonia_sample_plate10.xlsx
    Untracked:  data/211128_Ammonia_sample_plate11.xlsx
    Untracked:  data/211128_Griess_plate1.xlsx
    Untracked:  data/211128_Griess_plate10.xlsx
    Untracked:  data/211128_Griess_plate11.xlsx
    Untracked:  data/211128_Griess_plate2.xlsx
    Untracked:  data/211128_Griess_plate3.xlsx
    Untracked:  data/211128_Griess_plate4.xlsx
    Untracked:  data/211128_Griess_plate5.xlsx
    Untracked:  data/211128_Griess_plate6.xlsx
    Untracked:  data/211128_Griess_plate7.xlsx
    Untracked:  data/211128_Griess_plate8.xlsx
    Untracked:  data/211128_Griess_plate9.xlsx
    Untracked:  data/211128_time_table.xlsx
    Untracked:  data/211201_pH_colorimetric.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_table.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_time.xlsx
    Untracked:  data/211208_pH_colorimetric_T3.xlsx
    Untracked:  data/211208_pH_colorimetric_T4.xlsx
    Untracked:  data/211208_pH_colorimetric_T5.xlsx
    Untracked:  data/211208_pH_colorimetric_plate1.xlsx
    Untracked:  data/211208_pH_colorimetric_plate2.xlsx
    Untracked:  data/211208_pH_colorimetric_plate3.xlsx
    Untracked:  data/211208_time_table.xlsx
    Untracked:  data/211209_OD600_T1.xlsx
    Untracked:  data/211209_OD600_T2.xlsx
    Untracked:  data/211209_OD600_T3.xlsx
    Untracked:  data/211209_time_table.xlsx
    Untracked:  data/211210_pH_colorimetric_T0.xlsx
    Untracked:  data/211210_pH_colorimetric_T1.xlsx
    Untracked:  data/211210_pH_colorimetric_T2.xlsx
    Untracked:  data/211210_pH_colorimetric_T3.xlsx
    Untracked:  data/211210_time_table.xlsx
    Untracked:  data/211218_pH_colorimetric_t0.xlsx
    Untracked:  data/211218_pH_colorimetric_t1.xlsx
    Untracked:  data/211218_pH_colorimetric_t2.xlsx
    Untracked:  data/211218_pH_colorimetric_t3.xlsx
    Untracked:  data/211218_pH_colorimetric_t4.xlsx
    Untracked:  data/211218_time_table.xlsx
    Untracked:  data/211220_Griess_plate1.xlsx
    Untracked:  data/211220_Griess_plate10.xlsx
    Untracked:  data/211220_Griess_plate11.xlsx
    Untracked:  data/211220_Griess_plate2.xlsx
    Untracked:  data/211220_Griess_plate3.xlsx
    Untracked:  data/211220_Griess_plate4.xlsx
    Untracked:  data/211220_Griess_plate5.xlsx
    Untracked:  data/211220_Griess_plate6.xlsx
    Untracked:  data/211220_Griess_plate7.xlsx
    Untracked:  data/211220_Griess_plate8.xlsx
    Untracked:  data/211220_Griess_plate9.xlsx
    Untracked:  data/211220_time_table_AU.xlsx
    Untracked:  data/211220_time_table_BN.xlsx
    Untracked:  data/211224_TOC_sample_plate1.xlsx
    Untracked:  data/220103_Griess_T0.xlsx
    Untracked:  data/220103_Griess_T1.xlsx
    Untracked:  data/220103_Griess_T11.xlsx
    Untracked:  data/220103_Griess_T12.xlsx
    Untracked:  data/220103_Griess_T2.xlsx
    Untracked:  data/220103_Griess_T3.xlsx
    Untracked:  data/220103_Griess_T4.xlsx
    Untracked:  data/220103_Griess_T5.xlsx
    Untracked:  data/220103_Griess_T6.xlsx
    Untracked:  data/220103_Griess_T7.xlsx
    Untracked:  data/220103_Griess_T8.xlsx
    Untracked:  data/220103_Griess_T9.xlsx
    Untracked:  data/220103_time_table.xlsx
    Untracked:  data/220106_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-2_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-3_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-4_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-5_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-6_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-7_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-8_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-9_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107_Ammonia_sample_plate1.xlsx
    Untracked:  data/220108_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate2.xlsx
    Untracked:  data/220111_Ammonia_sample_T0.xlsx
    Untracked:  data/220111_Ammonia_sample_T1.xlsx
    Untracked:  data/220111_Ammonia_sample_T2.xlsx
    Untracked:  data/220111_Ammonia_sample_T3.xlsx
    Untracked:  data/220111_TOC_sample_plate1.xlsx
    Untracked:  data/220112_Ammonia_sample_T11.xlsx
    Untracked:  data/220112_Ammonia_sample_T12.xlsx
    Untracked:  data/220112_Ammonia_sample_T4.xlsx
    Untracked:  data/220112_Ammonia_sample_T5.xlsx
    Untracked:  data/220112_Ammonia_sample_T6.xlsx
    Untracked:  data/220112_Ammonia_sample_T7.xlsx
    Untracked:  data/220112_Ammonia_sample_T8.xlsx
    Untracked:  data/220112_Ammonia_sample_T9.xlsx
    Untracked:  data/220113_pH_colorimetric_T0.xlsx
    Untracked:  data/220113_pH_colorimetric_T12.xlsx
    Untracked:  data/220113_pH_colorimetric_T2.xlsx
    Untracked:  data/220113_pH_colorimetric_T3.xlsx
    Untracked:  data/220113_pH_colorimetric_T8.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_using6.66.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_using6.66.xlsx
    Untracked:  data/220125_time_table.xlsx
    Untracked:  data/220128_Ammonia_sample_plate1.xlsx
    Untracked:  data/220128_Ammonia_sample_plate2.xlsx
    Untracked:  data/220128_Ammonia_sample_plate3.xlsx
    Untracked:  data/220128_Ammonia_sample_plate4.xlsx
    Untracked:  data/220128_Ammonia_sample_plate5.xlsx
    Untracked:  data/220128_Ammonia_sample_plate6.xlsx
    Untracked:  data/220128_Griess_plate1.xlsx
    Untracked:  data/220128_Griess_plate2.xlsx
    Untracked:  data/220128_Griess_plate3.xlsx
    Untracked:  data/220128_Griess_plate4.xlsx
    Untracked:  data/220128_Griess_plate5.xlsx
    Untracked:  data/220128_Griess_plate6.xlsx
    Untracked:  data/220128_time_table.xlsx
    Untracked:  data/220131_Ammonia_sample_plate1.xlsx
    Untracked:  data/220131_Ammonia_sample_plate2.xlsx
    Untracked:  data/220131_Ammonia_sample_plate3.xlsx
    Untracked:  data/220131_Ammonia_sample_plate4.xlsx
    Untracked:  data/220131_Ammonia_sample_plate5.xlsx
    Untracked:  data/220131_Ammonia_sample_plate6.xlsx
    Untracked:  data/220131_Griess_plate1.xlsx
    Untracked:  data/220131_Griess_plate2.xlsx
    Untracked:  data/220131_Griess_plate3.xlsx
    Untracked:  data/220131_Griess_plate4.xlsx
    Untracked:  data/220131_Griess_plate5.xlsx
    Untracked:  data/220131_Griess_plate6.xlsx
    Untracked:  data/220131_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220131_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220131_time_table.xlsx
    Untracked:  data/220327_Griess_pastplate9.xlsx
    Untracked:  data/220327_Griess_plate1.xlsx
    Untracked:  data/220327_Griess_plate10.xlsx
    Untracked:  data/220327_Griess_plate11.xlsx
    Untracked:  data/220327_Griess_plate12.xlsx
    Untracked:  data/220327_Griess_plate13.xlsx
    Untracked:  data/220327_Griess_plate14.xlsx
    Untracked:  data/220327_Griess_plate15.xlsx
    Untracked:  data/220327_Griess_plate16.xlsx
    Untracked:  data/220327_Griess_plate2.xlsx
    Untracked:  data/220327_Griess_plate3.xlsx
    Untracked:  data/220327_Griess_plate4.xlsx
    Untracked:  data/220327_Griess_plate5.xlsx
    Untracked:  data/220327_Griess_plate6.xlsx
    Untracked:  data/220327_Griess_plate7.xlsx
    Untracked:  data/220327_Griess_plate8.xlsx
    Untracked:  data/220327_Griess_plate9.xlsx
    Untracked:  data/220327_time_table.xlsx
    Untracked:  data/220329_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH3_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH4_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH6_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH7_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH3_half_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_base_treatment_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_150ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_300ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_60ul_mu_opt_T0.xlsx
    Untracked:  data/220408_pH_colorimetric_pH_limit_mu_opt_water.xlsx
    Untracked:  data/220412_pH_colorimetric_pH_half_mu_opt_WW.xlsx
    Untracked:  data/220414_Ammonia_sample_plate0.xlsx
    Untracked:  data/220418_Ammonia_sample_plate0.xlsx
    Untracked:  data/220419_Ammonia_sample_plate0.xlsx
    Untracked:  data/220420-2_Ammonia_bcf.xlsx
    Untracked:  data/220420_Ammonia_sample_plate1.xlsx
    Untracked:  data/220420_Ammonia_sample_plate10.xlsx
    Untracked:  data/220420_Ammonia_sample_plate11.xlsx
    Untracked:  data/220420_Ammonia_sample_plate12.xlsx
    Untracked:  data/220420_Ammonia_sample_plate13.xlsx
    Untracked:  data/220420_Ammonia_sample_plate14.xlsx
    Untracked:  data/220420_Ammonia_sample_plate15.xlsx
    Untracked:  data/220420_Ammonia_sample_plate16.xlsx
    Untracked:  data/220420_Ammonia_sample_plate2.xlsx
    Untracked:  data/220420_Ammonia_sample_plate3.xlsx
    Untracked:  data/220420_Ammonia_sample_plate4.xlsx
    Untracked:  data/220420_Ammonia_sample_plate5.xlsx
    Untracked:  data/220420_Ammonia_sample_plate6.xlsx
    Untracked:  data/220420_Ammonia_sample_plate7.xlsx
    Untracked:  data/220420_Ammonia_sample_plate8.xlsx
    Untracked:  data/220420_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_df_a_bcf.xlsx
    Untracked:  data/220421_Ammonia_df_pH_range.xlsx
    Untracked:  data/220421_Ammonia_sample_plate1.xlsx
    Untracked:  data/220421_Ammonia_sample_plate10.xlsx
    Untracked:  data/220421_Ammonia_sample_plate11.xlsx
    Untracked:  data/220421_Ammonia_sample_plate2.xlsx
    Untracked:  data/220421_Ammonia_sample_plate3.xlsx
    Untracked:  data/220421_Ammonia_sample_plate4.xlsx
    Untracked:  data/220421_Ammonia_sample_plate5.xlsx
    Untracked:  data/220421_Ammonia_sample_plate6.xlsx
    Untracked:  data/220421_Ammonia_sample_plate7.xlsx
    Untracked:  data/220421_Ammonia_sample_plate8.xlsx
    Untracked:  data/220421_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_sterile.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  data/220421_Griess_plate1.xlsx
    Untracked:  data/220421_Griess_plate10.xlsx
    Untracked:  data/220421_Griess_plate11.xlsx
    Untracked:  data/220421_Griess_plate2.xlsx
    Untracked:  data/220421_Griess_plate3.xlsx
    Untracked:  data/220421_Griess_plate4.xlsx
    Untracked:  data/220421_Griess_plate5.xlsx
    Untracked:  data/220421_Griess_plate6.xlsx
    Untracked:  data/220421_Griess_plate7.xlsx
    Untracked:  data/220421_Griess_plate8.xlsx
    Untracked:  data/220421_Griess_plate9.xlsx
    Untracked:  data/220421_nitrate_reduced_autoclaved_samples.xlsx
    Untracked:  data/220421_time_table.xlsx
    Untracked:  data/220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T4.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T7.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T9.xlsx
    Untracked:  data/220508_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate2.xlsx
    Untracked:  data/220513_TOC_sample_plate3.xlsx
    Untracked:  data/220517_time_table.xlsx
    Untracked:  data/220519_Griess_antibiotics_dose_experiment_remove_blanks.xlsx
    Untracked:  data/220519_Griess_crerar7_extreme.xlsx
    Untracked:  data/220519_Griess_plate1.xlsx
    Untracked:  data/220519_Griess_plate10.xlsx
    Untracked:  data/220519_Griess_plate11.xlsx
    Untracked:  data/220519_Griess_plate2.xlsx
    Untracked:  data/220519_Griess_plate3.xlsx
    Untracked:  data/220519_Griess_plate4.xlsx
    Untracked:  data/220519_Griess_plate5.xlsx
    Untracked:  data/220519_Griess_plate6.xlsx
    Untracked:  data/220519_Griess_plate7.xlsx
    Untracked:  data/220519_Griess_plate8.xlsx
    Untracked:  data/220519_Griess_plate9.xlsx
    Untracked:  data/220519_time_table.xlsx
    Untracked:  data/220520_TOC_sample_plate1.xlsx
    Untracked:  data/220520_TOC_sample_plate2.xlsx
    Untracked:  data/220520_TOC_sample_plate3.xlsx
    Untracked:  data/220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T10_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220526_TOC_sample_plate1.xlsx
    Untracked:  data/220527_TOC_sample_plate1.xlsx
    Untracked:  data/220601_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_control_OD600.xlsx
    Untracked:  data/220602-3_TOC_sample_plate1.xlsx
    Untracked:  data/220602-3_control_OD600.xlsx
    Untracked:  data/220602_TOC_sample_plate1.xlsx
    Untracked:  data/220603_100ul_OD600.xlsx
    Untracked:  data/220603_TOC_sample_plate1.xlsx
    Untracked:  data/220603_TOC_sample_plate2.xlsx
    Untracked:  data/220606_TOC_sample_plate1.xlsx
    Untracked:  data/220606_TOC_sample_plate2.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel_format_edited.xlsx
    Untracked:  data/OD434_OD585_C_WB_T5_220329.xlsx
    Untracked:  data/OD434_OD585_WW_C_T0_220329.xlsx
    Untracked:  data/OD434_OD585_pH3_half_area_220407.xlsx
    Untracked:  data/OD434_OD585_pH_base_treated_T0_220407.xlsx
    Untracked:  data/OD434_OD585_pH_half_150ul_220412.xlsx
    Untracked:  data/SPNa_86437_021122_edit.xlsx
    Untracked:  data/SPNa_86437_021122_edit2.xlsx
    Untracked:  data/TOC_TN_measurement_2_labs_211014.xlsx
    Untracked:  data/dry_weight_curve(9.5.21).xlsx
    Untracked:  data/pH_data(11.17.21)-Midway_etc.xlsx
    Untracked:  data/pH_data(12.6.21)-Midway_incubation_endpoint.xlsx
    Untracked:  data/pH_data(9.14.21).xlsx
    Untracked:  data/pH_data(9.28.21).xlsx
    Untracked:  data/pH_standard_150ul_220407.xlsx
    Untracked:  data/pH_standard_300ul_220407.xlsx
    Untracked:  data/pH_standard_60ul_220407.xlsx
    Untracked:  data/~$220513_TOC_sample_plate1.xlsx
    Untracked:  data/~$SPNa_86437_021122_edit.xlsx
    Untracked:  past_figures/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/220131_Griess_pH_perturbation3.Rmd</code>) and HTML (<code>docs/220131_Griess_pH_perturbation3.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Denit_visualization_R/5637859fccf4a0454a5c8692d81b5031f0a3c389/docs/220131_Griess_pH_perturbation3.html" target="_blank">5637859</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Denit_visualization_R/blob/f3a9f681647167de767498c4343f9f34f466bfc0/analysis/220131_Griess_pH_perturbation3.Rmd" target="_blank">f3a9f68</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-02-13
</td>
<td>
wflow_publish(c(“analysis/220131_Ammonia_pH_perturbation3.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Denit_visualization_R/9aa8ee6ddc642c766d56ab565be7ffa80d4cd891/docs/220131_Griess_pH_perturbation3.html" target="_blank">9aa8ee6</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-02-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Denit_visualization_R/blob/352308ad5b05fd4f7fa3bd5d2196d3ded667e204/analysis/220131_Griess_pH_perturbation3.Rmd" target="_blank">352308a</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-02-08
</td>
<td>
wflow_publish(“analysis/220131_Griess_pH_perturbation3.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="time-series-analysis---ph-perturbation-experiment-3-griess-assay" class="section level2">
<h2>Time series analysis - pH perturbation experiment-3 (Griess assay)</h2>
<p>Researcher: <strong>Kiseok Lee</strong><br />
Experiment Date: 1/27/22 - 1/31/22 (5 days)<br />
Analysis Date: 2/1/21 Lab: <strong>Seppe Kuehn</strong></p>
<pre class="r"><code># libraries
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(vegan)
library(tidyverse)
library(magrittr)
library(readxl)
library(reshape2)
library(gtools)
library(devtools)
library(openxlsx)
library(ape)
library(stringr)
library(tidyr)
library(ggrepel)
library(ggpubr)

## theme for ggplot
mytheme &lt;- theme_bw() + 
  theme(text = element_text(family=&quot;serif&quot;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10, family=&quot;serif&quot;))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(),panel.background=element_blank(),panel.border=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))

mytheme_2d &lt;- theme_bw() + 
  theme(text = element_text(family=&quot;serif&quot;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=13, family=&quot;serif&quot;))+
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))


# color collection
my_color_collection &lt;- c(
  &quot;#CBD588&quot;, &quot;#5F7FC7&quot;, &quot;orange&quot;, &quot;#AD6F3B&quot;, &quot;#673770&quot;, 
  &quot;#D14285&quot;, &quot;#652926&quot;, &quot;#C84248&quot;, &quot;#8569D5&quot;, &quot;#5E738F&quot;,
  &quot;#D1A33D&quot;, &quot;#8A7C64&quot;, &quot;#599861&quot;,&quot;#616163&quot;, &quot;#FFCDB2&quot;,
  &quot;#6D9F71&quot;, &quot;#242F40&quot;,
  &quot;#CCA43B&quot;, &quot;#F92A82&quot;, &quot;#ED7B84&quot;, &quot;#7EB77F&quot;, 
  &quot;#DEC4A1&quot;, &quot;#E5D1D0&quot;, &#39;#0E8482&#39;, &#39;#C9DAEA&#39;, &#39;#337357&#39;, 
  &#39;#95C623&#39;, &#39;#E55812&#39;, &#39;#04471C&#39;, &#39;#F2D7EE&#39;, &#39;#D3BCC0&#39;, 
  &#39;#A5668B&#39;, &#39;#69306D&#39;, &#39;#0E103D&#39;, &#39;#1A535C&#39;, &#39;#4ECDC4&#39;, 
  &#39;#F7FFF7&#39;, &#39;#FF6B6B&#39;, &#39;#FFE66D&#39;, &#39;#6699CC&#39;, &#39;#FFF275&#39;, 
  &#39;#FF8C42&#39;, &#39;#FF3C38&#39;, &#39;#A23E48&#39;, &#39;#000000&#39;, &#39;#CF5C36&#39;, 
  &#39;#EEE5E9&#39;, &#39;#7C7C7C&#39;, &#39;#EFC88B&#39;, &#39;#2E5266&#39;, &#39;#6E8898&#39;, 
  &#39;#9FB1BC&#39;, &#39;#D3D0CB&#39;, &#39;#E2C044&#39;, &#39;#5BC0EB&#39;, &#39;#FDE74C&#39;, 
  &#39;#9BC53D&#39;, &#39;#E55934&#39;, &#39;#FA7921&#39;, &quot;#CD9BCD&quot;, &quot;#508578&quot;, &quot;#DA5724&quot;)

# for git push, use this instead of using wflow_git_push()
# git push -u origin master (in the Git app / in the working directory)

# for making pdf file
library(rmarkdown)
# render(&quot;analysis/~~.Rmd&quot;, &quot;pdf_document&quot;)</code></pre>
</div>
<div id="import-data-table-from-python-code" class="section level2">
<h2>1. Import data table from python code</h2>
<p>We are going to use the vcl3 treated standard curve that is fitted with pure nitrate standards</p>
<pre class="r"><code># import file
df_plate1 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate1.xlsx&quot;)
df_plate2 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate2.xlsx&quot;)
df_plate3 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate3.xlsx&quot;)
df_plate4 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate4.xlsx&quot;)
df_plate5 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate5.xlsx&quot;)
df_plate6 &lt;- openxlsx::read.xlsx(&quot;data/220131_Griess_plate6.xlsx&quot;)

head(df_plate1)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20         T0   0.02780
2            HCl    3             1.5       20         T0   0.00015
3            HCl    5             2.5       20         T0   0.00050
4            HCl    7             3.5       20         T0   0.00030
5            HCl    9             4.5       20         T0   0.00000
6            HCl   11             5.5       20         T0   0.00220
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1      0.78505 0.02411308 0.7944467 0.7703336
2      0.75575 0.01023808 0.7634054 0.7531673
3      0.74260 0.01041350 0.7494993 0.7390858
4      0.73775 0.01031326 0.7443744 0.7340612
5      0.73765 0.01016290 0.7442688 0.7341059
6      0.75245 0.01126564 0.7599142 0.7486485</code></pre>
<pre class="r"><code>colnames(df_plate1)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate1)</code></pre>
<pre><code>[1] 96 15</code></pre>
<pre class="r"><code>head(df_plate2)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20         T2   0.02865
2            HCl    3             1.5       20         T2   0.00290
3            HCl    5             2.5       20         T2   0.00285
4            HCl    7             3.5       20         T2   0.00125
5            HCl    9             4.5       20         T2   0.00175
6            HCl   11             5.5       20         T2   0.00170
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1      0.71370 0.02454015 0.7189928 0.6944527
2      0.68220 0.01161656 0.6858275 0.6742109
3      0.65985 0.01159149 0.6623497 0.6507582
4      0.70290 0.01078943 0.7076118 0.6968224
5      0.68575 0.01104006 0.6895607 0.6785206
6      0.65705 0.01101500 0.6594115 0.6483965</code></pre>
<pre class="r"><code>colnames(df_plate2)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate2)</code></pre>
<pre><code>[1] 96 15</code></pre>
<pre class="r"><code>head(df_plate3)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20         T4   0.00150
2            HCl    3             1.5       20         T4   0.02665
3            HCl    5             2.5       20         T4   0.00205
4            HCl    7             3.5       20         T4   0.00090
5            HCl    9             4.5       20         T4   0.00000
6            HCl   11             5.5       20         T4   0.00075
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1      0.18310 0.01091474 0.1717280 0.1608133
2      0.50340 0.02353533 0.4992343 0.4756989
3      0.60920 0.01119045 0.6093078 0.5981173
4      0.63055 0.01061399 0.6316384 0.6210244
5      0.78325 0.01016290 0.7925374 0.7823745
6      0.83885 0.01053881 0.8516496 0.8411108</code></pre>
<pre class="r"><code>colnames(df_plate3)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate3)</code></pre>
<pre><code>[1] 96 15</code></pre>
<pre class="r"><code>head(df_plate4)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20         T6   0.00000
2            HCl    3             1.5       20         T6   0.01665
3            HCl    5             2.5       20         T6   0.00335
4            HCl    7             3.5       20         T6   0.00000
5            HCl    9             4.5       20         T6   0.00000
6            HCl   11             5.5       20         T6   0.00110
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1      0.00000 0.01016290 0.0000000 0.0000000
2      0.21715 0.01851390 0.2061476 0.1876337
3      0.45075 0.01184216 0.4448156 0.4329734
4      0.60770 0.01016290 0.6077404 0.5975775
5      0.66000 0.01016290 0.6625071 0.6523442
6      0.69785 0.01071424 0.7022937 0.6915795</code></pre>
<pre class="r"><code>colnames(df_plate4)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate4)</code></pre>
<pre><code>[1] 96 15</code></pre>
<pre class="r"><code>head(df_plate5)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20         T8   0.00140
2            HCl    3             1.5       20         T8   0.00175
3            HCl    5             2.5       20         T8   0.00110
4            HCl    7             3.5       20         T8   0.00000
5            HCl    9             4.5       20         T8   0.00090
6            HCl   11             5.5       20         T8   0.00205
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1       0.0000 0.01086462 0.0000000 0.0000000
2       0.0000 0.01104006 0.0000000 0.0000000
3       0.0000 0.01071424 0.0000000 0.0000000
4       0.5324 0.01016290 0.5293094 0.5191465
5       0.6421 0.01061399 0.6437357 0.6331217
6       0.6264 0.01119045 0.6272947 0.6161042</code></pre>
<pre class="r"><code>colnames(df_plate5)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate5)</code></pre>
<pre><code>[1] 96 15</code></pre>
<pre class="r"><code>head(df_plate6)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input               Soil Sample_type
1  A01             0             2 LaBaghWoods_pH6.66      Slurry
2  A02             0             2 LaBaghWoods_pH6.66      Slurry
3  A03             0             2 LaBaghWoods_pH6.66      Slurry
4  A04             0             2 LaBaghWoods_pH6.66      Slurry
5  A05             0             2 LaBaghWoods_pH6.66      Slurry
6  A06             0             2 LaBaghWoods_pH6.66      Slurry
  Titration_type Unit Concentration_M Added_ul Time_point NO2_OD540
1            HCl    1             0.5       20        T10   0.00445
2            HCl    3             1.5       20        T10   0.00395
3            HCl    5             2.5       20        T10   0.00240
4            HCl    7             3.5       20        T10   0.00135
5            HCl    9             4.5       20        T10   0.00110
6            HCl   11             5.5       20        T10   0.00185
  NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1       0.0000 0.01239366 0.0000000 0.0000000
2       0.0000 0.01214297 0.0000000 0.0000000
3       0.0000 0.01136590 0.0000000 0.0000000
4       0.5065 0.01083955 0.5024458 0.4916062
5       0.6484 0.01071424 0.6503391 0.6396249
6       0.7743 0.01109019 0.7830487 0.7719585</code></pre>
<pre class="r"><code>colnames(df_plate6)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_OD540&quot;       &quot;NO2NO3_OD540&quot;   
[13] &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;       &quot;NO3_mM&quot;         </code></pre>
<pre class="r"><code>dim(df_plate6)</code></pre>
<pre><code>[1] 48 15</code></pre>
<pre class="r"><code># X1 to Well
# df_p0 %&lt;&gt;% rename(Well = X1)
# df_p2 %&lt;&gt;% rename(Well = X1)
# df_p3 %&lt;&gt;% rename(Well = X1)
# df_p4 %&lt;&gt;% rename(Well = X1)
# df_p5 %&lt;&gt;% rename(Well = X1)
# df_p6 %&lt;&gt;% rename(Well = X1)
# df_p7 %&lt;&gt;% rename(Well = X1)
# df_p8 %&lt;&gt;% rename(Well = X1)
# df_p9 %&lt;&gt;% rename(Well = X1)
# df_p10 %&lt;&gt;% rename(Well = X1)
# df_p11 %&lt;&gt;% rename(Well = X1)

# remove wells that were contaminated during the experiment
df_plate1 %&lt;&gt;% filter(!(Well %in% c(&quot;A08&quot;))) # filter defect 
df_plate5 %&lt;&gt;% filter(!(Well %in% c(&quot;B04&quot;,&quot;D04&quot;,&quot;F04&quot;,&quot;H04&quot;))) # wrong sampling... I assume?
# df_p6 %&lt;&gt;% filter(!(Well %in% c(&quot;A11&quot;))) # filter burst issue
# df_p8 %&lt;&gt;% filter(!(Well %in% c(&quot;A07&quot;, &quot;A08&quot;, &quot;E08&quot;))) # filter burst issue
# df_p11 %&lt;&gt;% filter(!(Well %in% c(&quot;A01&quot;))) # filter burst issue  # G11 was removed because of abnormal value compared to other 2 replicates
# dim(df_p4)

# bind two dataframe
df_p &lt;- rbind(df_plate1, df_plate2, df_plate3, df_plate4, df_plate5, df_plate6)
dim(df_p)</code></pre>
<pre><code>[1] 523  15</code></pre>
<pre class="r"><code># remove NA
dim(df_p)</code></pre>
<pre><code>[1] 523  15</code></pre>
<pre class="r"><code>df_p &lt;- na.omit(df_p)
dim(df_p) # 528</code></pre>
<pre><code>[1] 523  15</code></pre>
<pre class="r"><code># multiply dilution factor which is 5/2
df_p %&lt;&gt;% select(-NO2_OD540, -NO2NO3_OD540)
df_p %&lt;&gt;% mutate(NO2_mM = NO2_mM * (5/2), NO2NO3_mM = NO2NO3_mM * (5/2), NO3_mM = NO3_mM * (5/2))

# Get the metadata for time point and left join
Time_table &lt;- openxlsx::read.xlsx(&quot;data/220131_time_table.xlsx&quot;)
# Time_table_BN &lt;- openxlsx::read.xlsx(&quot;data/220131_time_table_BN.xlsx&quot;)
# Time_table &lt;- rbind(Time_table_AU, Time_table_BN)

Time_table %&lt;&gt;% select(-Date) 
Time_table$Time_hours &lt;- round(Time_table$Time_hours, 1)
Time_table$Time_days &lt;- round(Time_table$Time_days, 1)
df_time &lt;- Time_table
dim(df_p)</code></pre>
<pre><code>[1] 523  13</code></pre>
<pre class="r"><code>df_p &lt;- df_p %&gt;% left_join(Time_table, by=(&quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p)</code></pre>
<pre><code>[1] 523  16</code></pre>
<pre class="r"><code>colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;      
[13] &quot;NO3_mM&quot;          &quot;Time_minutes&quot;    &quot;Time_hours&quot;      &quot;Time_days&quot;      </code></pre>
<pre class="r"><code># time_point order
df_p$Time_point &lt;-  factor(df_p$Time_point, levels = paste0(&quot;T&quot;,0:10))</code></pre>
</div>
<div id="get-average-and-standard-deviation-moisture-correction-blank-correction" class="section level2">
<h2>2. Get average and standard deviation &amp; Moisture correction &amp; Blank correction</h2>
<pre class="r"><code># plot to see
ggplot(df_p, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Nitrate_spike_in (mM)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Without averaging \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-3-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># average technical replicate (here only 1 replicate per pH perturbation)
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;      
[13] &quot;NO3_mM&quot;          &quot;Time_minutes&quot;    &quot;Time_hours&quot;      &quot;Time_days&quot;      </code></pre>
<pre class="r"><code>dim(df_p)</code></pre>
<pre><code>[1] 523  16</code></pre>
<pre class="r"><code># df_p &lt;- df_p %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Concentration_M, Added_ul, Sample_type, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(NO2_mM = mean(NO2_mM), Std_NO2_mM = sd(NO2_mM), NO3_mM = mean(NO3_mM), Std_NO3_mM = sd(NO3_mM)) %&gt;% ungroup()


# mols and molarity unit conversion
# test
df_p$Titration_type %&gt;% length()</code></pre>
<pre><code>[1] 523</code></pre>
<pre class="r"><code># Setting H_mM column
# df_soil &lt;- df_p %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;))

Added_Volume &lt;- 1.7 # ml
Soil_mg &lt;- 0.85
moisture_percent_1 = 23.6 # LaBaghWoods1
Added_Volume + Soil_mg*(moisture_percent_1/100)</code></pre>
<pre><code>[1] 1.9006</code></pre>
<pre class="r"><code># moisture_percent_2 = 33.2 # Allandale
df_p$Added_ul &lt;- ifelse(df_p$Titration_type == &quot;NaOH&quot;, -1*df_p$Added_ul, df_p$Added_ul) # HCl is +, NaOH is -
df_p %&lt;&gt;% mutate(H_mol = Concentration_M * Added_ul * 10^(-6)) # Calculate H mol 
df_p$Volume &lt;- ifelse(df_p$Soil == &quot;LaBaghWoods_pH6.66&quot;, Added_Volume + Soil_mg*(moisture_percent_1/100), 0) # Calc total volume
# df_p$Volume &lt;- ifelse(df_p$Soil == &quot;Allandale_pH4&quot;, Added_Volume + Soil_mg*(moisture_percent_2/100), df_p$Volume) # Calc total volume
# df_p$Volume %&gt;% unique()
df_p %&lt;&gt;% mutate(H_Molarity = H_mol / (Volume * 10^(-3)))
df_p %&lt;&gt;% mutate(H_mM = H_Molarity * 1000)
# openxlsx::write.xlsx(df_p, &quot;df_p.xlsx&quot;)

# how many levels of H_mM?
df_p %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange(H_mM)</code></pre>
<pre><code>         H_mM
1  -94.706935
2  -89.445438
3  -84.183942
4  -78.922446
5  -73.660949
6  -68.399453
7  -63.137956
8  -57.876460
9  -52.614964
10 -47.353467
11 -42.091971
12 -36.830475
13 -31.568978
14 -26.307482
15 -21.045985
16 -15.784489
17 -10.522993
18  -5.261496
19   0.000000
20   5.261496
21  10.522993
22  15.784489
23  21.045985
24  26.307482
25  31.568978
26  36.830475
27  42.091971
28  47.353467
29  52.614964
30  57.876460
31  63.137956
32  68.399453
33  73.660949
34  78.922446
35  84.183942
36  89.445438
37  94.706935</code></pre>
<pre class="r"><code># df_p %&gt;% filter(Soil == &quot;Allandale_pH4&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange()
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;      
[13] &quot;NO3_mM&quot;          &quot;Time_minutes&quot;    &quot;Time_hours&quot;      &quot;Time_days&quot;      
[17] &quot;H_mol&quot;           &quot;Volume&quot;          &quot;H_Molarity&quot;      &quot;H_mM&quot;           </code></pre>
<pre class="r"><code>## Moisture correction
dim(df_p)</code></pre>
<pre><code>[1] 523  20</code></pre>
<pre class="r"><code># Testing negative samples
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)  # Use Nitrate_Blank instead in the future
df_no3_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input               Soil   Sample_type
1   G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
2   G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
3   H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
4   H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
5   H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
6   H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
7   G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
8   G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
9   H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
10  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
11  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
12  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
13  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
14  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
15  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
16  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
17  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
18  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
19  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
20  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
21  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
22  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
23  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
24  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
25  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
26  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
27  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
28  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
29  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
30  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
31  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
32  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
33  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
   Titration_type Unit Concentration_M Added_ul Time_point     NO2_mM NO2NO3_mM
1         Nitrate    0               0       20         T0 0.02709889  1.902694
2         Nitrate    0               0       20         T1 0.02540724  1.962789
3         Nitrate    0               0       20         T0 0.02734952  2.078809
4         Nitrate    0               0       20         T0 0.02772547  1.942126
5         Nitrate    0               0       20         T1 0.02603375  1.911423
6         Nitrate    0               0       20         T1 0.02590845  2.070298
7         Nitrate    0               0       20         T2 0.03317792  1.967691
8         Nitrate    0               0       20         T3 0.02615906  1.870446
9         Nitrate    0               0       20         T2 0.04772785  1.990227
10        Nitrate    0               0       20         T2 0.02828942  1.997522
11        Nitrate    0               0       20         T3 0.02540724  1.829259
12        Nitrate    0               0       20         T3 0.02678560  1.814355
13        Nitrate    0               0       20         T4 0.02540724  2.178855
14        Nitrate    0               0       20         T5 0.02540724  2.112210
15        Nitrate    0               0       20         T4 0.02540724  2.128857
16        Nitrate    0               0       20         T4 0.02540724  2.021540
17        Nitrate    0               0       20         T5 0.02628437  2.093975
18        Nitrate    0               0       20         T5 0.02540724  1.950072
19        Nitrate    0               0       20         T6 0.02540724  2.066709
20        Nitrate    0               0       20         T7 0.02540724  2.077479
21        Nitrate    0               0       20         T6 0.02540724  2.000706
22        Nitrate    0               0       20         T6 0.02540724  1.976439
23        Nitrate    0               0       20         T7 0.02816410  2.099165
24        Nitrate    0               0       20         T7 0.02672295  1.902694
25        Nitrate    0               0       20         T8 0.02590845  2.126193
26        Nitrate    0               0       20         T9 0.02540724  2.139783
27        Nitrate    0               0       20         T8 0.02540724  2.077479
28        Nitrate    0               0       20         T8 0.02615906  1.982139
29        Nitrate    0               0       20         T9 0.02797611  2.135919
30        Nitrate    0               0       20         T9 0.02716154  1.890795
31        Nitrate    0               0       20        T10 0.02546989  2.310573
32        Nitrate    0               0       20        T10 0.02603375  2.267853
33        Nitrate    0               0       20        T10 0.02666029  2.070165
     NO3_mM Time_minutes Time_hours Time_days H_mol Volume H_Molarity H_mM
1  1.875596            0        0.0       0.0     0 1.9006          0    0
2  1.937381          215        3.6       0.1     0 1.9006          0    0
3  2.051460            0        0.0       0.0     0 1.9006          0    0
4  1.914401            0        0.0       0.0     0 1.9006          0    0
5  1.885390          215        3.6       0.1     0 1.9006          0    0
6  2.044390          215        3.6       0.1     0 1.9006          0    0
7  1.934513          385        6.4       0.3     0 1.9006          0    0
8  1.844287          580        9.7       0.4     0 1.9006          0    0
9  1.942500          385        6.4       0.3     0 1.9006          0    0
10 1.969233          385        6.4       0.3     0 1.9006          0    0
11 1.803851          580        9.7       0.4     0 1.9006          0    0
12 1.787570          580        9.7       0.4     0 1.9006          0    0
13 2.153447         1470       24.5       1.0     0 1.9006          0    0
14 2.086803         1855       30.9       1.3     0 1.9006          0    0
15 2.103450         1470       24.5       1.0     0 1.9006          0    0
16 1.996133         1470       24.5       1.0     0 1.9006          0    0
17 2.067690         1855       30.9       1.3     0 1.9006          0    0
18 1.924664         1855       30.9       1.3     0 1.9006          0    0
19 2.041301         2190       36.5       1.5     0 1.9006          0    0
20 2.052072         3010       50.2       2.1     0 1.9006          0    0
21 1.975298         2190       36.5       1.5     0 1.9006          0    0
22 1.951031         2190       36.5       1.5     0 1.9006          0    0
23 2.071001         3010       50.2       2.1     0 1.9006          0    0
24 1.875972         3010       50.2       2.1     0 1.9006          0    0
25 2.100285         3615       60.2       2.5     0 1.9006          0    0
26 2.114376         4535       75.6       3.1     0 1.9006          0    0
27 2.052072         3615       60.2       2.5     0 1.9006          0    0
28 1.955980         3615       60.2       2.5     0 1.9006          0    0
29 2.107943         4535       75.6       3.1     0 1.9006          0    0
30 1.863634         4535       75.6       3.1     0 1.9006          0    0
31 2.285103         5985       99.8       4.2     0 1.9006          0    0
32 2.241819         5985       99.8       4.2     0 1.9006          0    0
33 2.043505         5985       99.8       4.2     0 1.9006          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input               Soil   Sample_type
1   G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
2   G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
3   G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
4   G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
5   H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
6   H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
7   G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
8   G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
9   G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
10  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
11  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
12  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
13  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
14  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
15  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
16  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
17  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
18  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
19  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
20  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
21  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
22  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
23  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
24  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
25  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
26  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
27  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
28  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
29  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
30  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
31  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
32  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
   Titration_type Unit Concentration_M Added_ul Time_point   NO2_mM NO2NO3_mM
1         Nitrite    0               0       20         T0 2.089940  2.026984
2         Nitrite    0               0       20         T0 1.999982  2.031100
3         Nitrite    0               0       20         T1 1.971781  1.890663
4         Nitrite    0               0       20         T1 2.002740  1.951793
5         Nitrite    0               0       20         T0 2.066568  2.057138
6         Nitrite    0               0       20         T1 2.071239  1.990891
7         Nitrite    0               0       20         T2 2.022356  1.978294
8         Nitrite    0               0       20         T2 2.026937  1.963451
9         Nitrite    0               0       20         T3 1.999836  1.934448
10        Nitrite    0               0       20         T3 1.964614  1.878505
11        Nitrite    0               0       20         T2 2.028683  1.970739
12        Nitrite    0               0       20         T3 1.992218  1.893175
13        Nitrite    0               0       20         T4 2.142688  2.093176
14        Nitrite    0               0       20         T4 2.162380  2.149646
15        Nitrite    0               0       20         T5 2.065619  2.019284
16        Nitrite    0               0       20         T5 2.048267  1.998185
17        Nitrite    0               0       20         T4 2.143569  2.107417
18        Nitrite    0               0       20         T5 2.081243  2.030436
19        Nitrite    0               0       20         T6 2.088478  2.009595
20        Nitrite    0               0       20         T6 2.150620  2.056739
21        Nitrite    0               0       20         T7 2.176433  2.087721
22        Nitrite    0               0       20         T7 2.165395  2.088120
23        Nitrite    0               0       20         T6 2.080659  1.969282
24        Nitrite    0               0       20         T7 2.190724  2.079607
25        Nitrite    0               0       20         T8 2.128673  2.023664
26        Nitrite    0               0       20         T8 2.300075  2.200078
27        Nitrite    0               0       20         T9 2.248424  2.140316
28        Nitrite    0               0       20         T9 2.329560  2.216106
29        Nitrite    0               0       20         T9 2.287959  2.139250
30        Nitrite    0               0       20        T10 2.357474  2.265979
31        Nitrite    0               0       20        T10 2.448076  2.360199
32        Nitrite    0               0       20        T10 2.502329  2.363555
       NO3_mM Time_minutes Time_hours Time_days H_mol Volume H_Molarity H_mM
1  0.00000000            0        0.0       0.0     0 1.9006          0    0
2  0.03111824            0        0.0       0.0     0 1.9006          0    0
3  0.00000000          215        3.6       0.1     0 1.9006          0    0
4  0.00000000          215        3.6       0.1     0 1.9006          0    0
5  0.00000000            0        0.0       0.0     0 1.9006          0    0
6  0.00000000          215        3.6       0.1     0 1.9006          0    0
7  0.00000000          385        6.4       0.3     0 1.9006          0    0
8  0.00000000          385        6.4       0.3     0 1.9006          0    0
9  0.00000000          580        9.7       0.4     0 1.9006          0    0
10 0.00000000          580        9.7       0.4     0 1.9006          0    0
11 0.00000000          385        6.4       0.3     0 1.9006          0    0
12 0.00000000          580        9.7       0.4     0 1.9006          0    0
13 0.00000000         1470       24.5       1.0     0 1.9006          0    0
14 0.00000000         1470       24.5       1.0     0 1.9006          0    0
15 0.00000000         1855       30.9       1.3     0 1.9006          0    0
16 0.00000000         1855       30.9       1.3     0 1.9006          0    0
17 0.00000000         1470       24.5       1.0     0 1.9006          0    0
18 0.00000000         1855       30.9       1.3     0 1.9006          0    0
19 0.00000000         2190       36.5       1.5     0 1.9006          0    0
20 0.00000000         2190       36.5       1.5     0 1.9006          0    0
21 0.00000000         3010       50.2       2.1     0 1.9006          0    0
22 0.00000000         3010       50.2       2.1     0 1.9006          0    0
23 0.00000000         2190       36.5       1.5     0 1.9006          0    0
24 0.00000000         3010       50.2       2.1     0 1.9006          0    0
25 0.00000000         3615       60.2       2.5     0 1.9006          0    0
26 0.00000000         3615       60.2       2.5     0 1.9006          0    0
27 0.00000000         4535       75.6       3.1     0 1.9006          0    0
28 0.00000000         4535       75.6       3.1     0 1.9006          0    0
29 0.00000000         4535       75.6       3.1     0 1.9006          0    0
30 0.00000000         5985       99.8       4.2     0 1.9006          0    0
31 0.00000000         5985       99.8       4.2     0 1.9006          0    0
32 0.00000000         5985       99.8       4.2     0 1.9006          0    0</code></pre>
<pre class="r"><code># close to zero

# 1. Apply moisture correction factor (correcting for moisture in soil)
soil_spike_ratio = 0.5 # soil weight(0.85g) / spike in volume (1.7ml) 
moisture_percent_1 = 23.6 # LaBaghWoods1

mcf_1 = (soil_spike_ratio*(moisture_percent_1/100) + 1)
mcf_1</code></pre>
<pre><code>[1] 1.118</code></pre>
<pre class="r"><code># apply moisture factor to each soil
dim(df_p)</code></pre>
<pre><code>[1] 523  20</code></pre>
<pre class="r"><code>df_p %&gt;% select(Soil) %&gt;% unique()</code></pre>
<pre><code>                Soil
1 LaBaghWoods_pH6.66</code></pre>
<pre class="r"><code># this is a special case with no replicates
df_p_mcf_1 &lt;- df_p %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_1, NO2_mM = NO2_mM * mcf_1)
dim(df_p_mcf_1) #462</code></pre>
<pre><code>[1] 458  20</code></pre>
<pre class="r"><code>df_p_others_mcf &lt;- df_p %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;, &quot;Nitrate_Blank&quot;)) # Use Nitrate_Blank instead in the future
dim(df_p_others_mcf) #66</code></pre>
<pre><code>[1] 65 20</code></pre>
<pre class="r"><code>df_p_mcf &lt;- rbind(df_p_mcf_1, df_p_others_mcf)
dim(df_p_mcf) # 528</code></pre>
<pre><code>[1] 523  20</code></pre>
<pre class="r"><code># plot to see
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hours)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;After averaging with biological replicates \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-3-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. Apply blank correction factor (drying effect during incubation)
# Blank reads
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)
df_no3_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input               Soil   Sample_type
1   G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
2   G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
3   H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
4   H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
5   H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
6   H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
7   G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
8   G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
9   H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
10  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
11  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
12  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
13  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
14  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
15  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
16  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
17  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
18  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
19  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
20  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
21  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
22  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
23  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
24  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
25  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
26  G12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
27  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
28  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
29  H11             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
30  H12             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
31  G06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
32  H05             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
33  H06             0             2 LaBaghWoods_pH6.66 Nitrate_Blank
   Titration_type Unit Concentration_M Added_ul Time_point     NO2_mM NO2NO3_mM
1         Nitrate    0               0       20         T0 0.02709889  1.902694
2         Nitrate    0               0       20         T1 0.02540724  1.962789
3         Nitrate    0               0       20         T0 0.02734952  2.078809
4         Nitrate    0               0       20         T0 0.02772547  1.942126
5         Nitrate    0               0       20         T1 0.02603375  1.911423
6         Nitrate    0               0       20         T1 0.02590845  2.070298
7         Nitrate    0               0       20         T2 0.03317792  1.967691
8         Nitrate    0               0       20         T3 0.02615906  1.870446
9         Nitrate    0               0       20         T2 0.04772785  1.990227
10        Nitrate    0               0       20         T2 0.02828942  1.997522
11        Nitrate    0               0       20         T3 0.02540724  1.829259
12        Nitrate    0               0       20         T3 0.02678560  1.814355
13        Nitrate    0               0       20         T4 0.02540724  2.178855
14        Nitrate    0               0       20         T5 0.02540724  2.112210
15        Nitrate    0               0       20         T4 0.02540724  2.128857
16        Nitrate    0               0       20         T4 0.02540724  2.021540
17        Nitrate    0               0       20         T5 0.02628437  2.093975
18        Nitrate    0               0       20         T5 0.02540724  1.950072
19        Nitrate    0               0       20         T6 0.02540724  2.066709
20        Nitrate    0               0       20         T7 0.02540724  2.077479
21        Nitrate    0               0       20         T6 0.02540724  2.000706
22        Nitrate    0               0       20         T6 0.02540724  1.976439
23        Nitrate    0               0       20         T7 0.02816410  2.099165
24        Nitrate    0               0       20         T7 0.02672295  1.902694
25        Nitrate    0               0       20         T8 0.02590845  2.126193
26        Nitrate    0               0       20         T9 0.02540724  2.139783
27        Nitrate    0               0       20         T8 0.02540724  2.077479
28        Nitrate    0               0       20         T8 0.02615906  1.982139
29        Nitrate    0               0       20         T9 0.02797611  2.135919
30        Nitrate    0               0       20         T9 0.02716154  1.890795
31        Nitrate    0               0       20        T10 0.02546989  2.310573
32        Nitrate    0               0       20        T10 0.02603375  2.267853
33        Nitrate    0               0       20        T10 0.02666029  2.070165
     NO3_mM Time_minutes Time_hours Time_days H_mol Volume H_Molarity H_mM
1  1.875596            0        0.0       0.0     0 1.9006          0    0
2  1.937381          215        3.6       0.1     0 1.9006          0    0
3  2.051460            0        0.0       0.0     0 1.9006          0    0
4  1.914401            0        0.0       0.0     0 1.9006          0    0
5  1.885390          215        3.6       0.1     0 1.9006          0    0
6  2.044390          215        3.6       0.1     0 1.9006          0    0
7  1.934513          385        6.4       0.3     0 1.9006          0    0
8  1.844287          580        9.7       0.4     0 1.9006          0    0
9  1.942500          385        6.4       0.3     0 1.9006          0    0
10 1.969233          385        6.4       0.3     0 1.9006          0    0
11 1.803851          580        9.7       0.4     0 1.9006          0    0
12 1.787570          580        9.7       0.4     0 1.9006          0    0
13 2.153447         1470       24.5       1.0     0 1.9006          0    0
14 2.086803         1855       30.9       1.3     0 1.9006          0    0
15 2.103450         1470       24.5       1.0     0 1.9006          0    0
16 1.996133         1470       24.5       1.0     0 1.9006          0    0
17 2.067690         1855       30.9       1.3     0 1.9006          0    0
18 1.924664         1855       30.9       1.3     0 1.9006          0    0
19 2.041301         2190       36.5       1.5     0 1.9006          0    0
20 2.052072         3010       50.2       2.1     0 1.9006          0    0
21 1.975298         2190       36.5       1.5     0 1.9006          0    0
22 1.951031         2190       36.5       1.5     0 1.9006          0    0
23 2.071001         3010       50.2       2.1     0 1.9006          0    0
24 1.875972         3010       50.2       2.1     0 1.9006          0    0
25 2.100285         3615       60.2       2.5     0 1.9006          0    0
26 2.114376         4535       75.6       3.1     0 1.9006          0    0
27 2.052072         3615       60.2       2.5     0 1.9006          0    0
28 1.955980         3615       60.2       2.5     0 1.9006          0    0
29 2.107943         4535       75.6       3.1     0 1.9006          0    0
30 1.863634         4535       75.6       3.1     0 1.9006          0    0
31 2.285103         5985       99.8       4.2     0 1.9006          0    0
32 2.241819         5985       99.8       4.2     0 1.9006          0    0
33 2.043505         5985       99.8       4.2     0 1.9006          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input               Soil   Sample_type
1   G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
2   G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
3   G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
4   G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
5   H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
6   H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
7   G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
8   G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
9   G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
10  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
11  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
12  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
13  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
14  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
15  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
16  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
17  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
18  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
19  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
20  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
21  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
22  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
23  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
24  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
25  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
26  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
27  G10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
28  G11             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
29  H10             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
30  G04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
31  G05             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
32  H04             2             0 LaBaghWoods_pH6.66 Nitrite_Blank
   Titration_type Unit Concentration_M Added_ul Time_point   NO2_mM NO2NO3_mM
1         Nitrite    0               0       20         T0 2.089940  2.026984
2         Nitrite    0               0       20         T0 1.999982  2.031100
3         Nitrite    0               0       20         T1 1.971781  1.890663
4         Nitrite    0               0       20         T1 2.002740  1.951793
5         Nitrite    0               0       20         T0 2.066568  2.057138
6         Nitrite    0               0       20         T1 2.071239  1.990891
7         Nitrite    0               0       20         T2 2.022356  1.978294
8         Nitrite    0               0       20         T2 2.026937  1.963451
9         Nitrite    0               0       20         T3 1.999836  1.934448
10        Nitrite    0               0       20         T3 1.964614  1.878505
11        Nitrite    0               0       20         T2 2.028683  1.970739
12        Nitrite    0               0       20         T3 1.992218  1.893175
13        Nitrite    0               0       20         T4 2.142688  2.093176
14        Nitrite    0               0       20         T4 2.162380  2.149646
15        Nitrite    0               0       20         T5 2.065619  2.019284
16        Nitrite    0               0       20         T5 2.048267  1.998185
17        Nitrite    0               0       20         T4 2.143569  2.107417
18        Nitrite    0               0       20         T5 2.081243  2.030436
19        Nitrite    0               0       20         T6 2.088478  2.009595
20        Nitrite    0               0       20         T6 2.150620  2.056739
21        Nitrite    0               0       20         T7 2.176433  2.087721
22        Nitrite    0               0       20         T7 2.165395  2.088120
23        Nitrite    0               0       20         T6 2.080659  1.969282
24        Nitrite    0               0       20         T7 2.190724  2.079607
25        Nitrite    0               0       20         T8 2.128673  2.023664
26        Nitrite    0               0       20         T8 2.300075  2.200078
27        Nitrite    0               0       20         T9 2.248424  2.140316
28        Nitrite    0               0       20         T9 2.329560  2.216106
29        Nitrite    0               0       20         T9 2.287959  2.139250
30        Nitrite    0               0       20        T10 2.357474  2.265979
31        Nitrite    0               0       20        T10 2.448076  2.360199
32        Nitrite    0               0       20        T10 2.502329  2.363555
       NO3_mM Time_minutes Time_hours Time_days H_mol Volume H_Molarity H_mM
1  0.00000000            0        0.0       0.0     0 1.9006          0    0
2  0.03111824            0        0.0       0.0     0 1.9006          0    0
3  0.00000000          215        3.6       0.1     0 1.9006          0    0
4  0.00000000          215        3.6       0.1     0 1.9006          0    0
5  0.00000000            0        0.0       0.0     0 1.9006          0    0
6  0.00000000          215        3.6       0.1     0 1.9006          0    0
7  0.00000000          385        6.4       0.3     0 1.9006          0    0
8  0.00000000          385        6.4       0.3     0 1.9006          0    0
9  0.00000000          580        9.7       0.4     0 1.9006          0    0
10 0.00000000          580        9.7       0.4     0 1.9006          0    0
11 0.00000000          385        6.4       0.3     0 1.9006          0    0
12 0.00000000          580        9.7       0.4     0 1.9006          0    0
13 0.00000000         1470       24.5       1.0     0 1.9006          0    0
14 0.00000000         1470       24.5       1.0     0 1.9006          0    0
15 0.00000000         1855       30.9       1.3     0 1.9006          0    0
16 0.00000000         1855       30.9       1.3     0 1.9006          0    0
17 0.00000000         1470       24.5       1.0     0 1.9006          0    0
18 0.00000000         1855       30.9       1.3     0 1.9006          0    0
19 0.00000000         2190       36.5       1.5     0 1.9006          0    0
20 0.00000000         2190       36.5       1.5     0 1.9006          0    0
21 0.00000000         3010       50.2       2.1     0 1.9006          0    0
22 0.00000000         3010       50.2       2.1     0 1.9006          0    0
23 0.00000000         2190       36.5       1.5     0 1.9006          0    0
24 0.00000000         3010       50.2       2.1     0 1.9006          0    0
25 0.00000000         3615       60.2       2.5     0 1.9006          0    0
26 0.00000000         3615       60.2       2.5     0 1.9006          0    0
27 0.00000000         4535       75.6       3.1     0 1.9006          0    0
28 0.00000000         4535       75.6       3.1     0 1.9006          0    0
29 0.00000000         4535       75.6       3.1     0 1.9006          0    0
30 0.00000000         5985       99.8       4.2     0 1.9006          0    0
31 0.00000000         5985       99.8       4.2     0 1.9006          0    0
32 0.00000000         5985       99.8       4.2     0 1.9006          0    0</code></pre>
<pre class="r"><code># df_aero_blank$Nitrate_input &lt;- as.numeric(as.character(df_aero_blank$Nitrate_input))
df_no2_blank %&lt;&gt;% mutate(Correction_factor_NO2 = (Nitrite_input / NO2_mM)) 
df_no3_blank %&lt;&gt;% mutate(Correction_factor_NO3 = (Nitrate_input / NO3_mM)) 

# average the blanks by time point
cf_no2 &lt;- df_no2_blank %&gt;% select(Soil, Time_point, Correction_factor_NO2) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO2 = mean(Correction_factor_NO2)) %&gt;% ungroup()
cf_no3 &lt;- df_no3_blank %&gt;% select(Soil, Time_point, Correction_factor_NO3) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO3 = mean(Correction_factor_NO3)) %&gt;% ungroup()

# left join and multiply the correction factor
# left join to samples
df_sample_mcf &lt;- df_p_mcf %&gt;% filter(!(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;No_Nitrate&quot;)))
dim(df_sample_mcf) #462</code></pre>
<pre><code>[1] 425  20</code></pre>
<pre class="r"><code>df_sample_mcf$Sample_type %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Slurry&quot;</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_mcf %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_sample_bcf)</code></pre>
<pre><code>[1] 425  22</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_bcf %&gt;% mutate(NO3_mM = NO3_mM * Correction_factor_NO3, NO2_mM = NO2_mM * Correction_factor_NO2)
dim(df_sample_bcf) #462 22</code></pre>
<pre><code>[1] 425  22</code></pre>
<pre class="r"><code># merge dataframe with blank just for convenience
df_p_blanks_bcf &lt;- df_p_mcf %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;No_Nitrate&quot;)) %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p_blanks_bcf) #99 22</code></pre>
<pre><code>[1] 98 22</code></pre>
<pre class="r"><code>dim(df_sample_bcf) # 462</code></pre>
<pre><code>[1] 425  22</code></pre>
<pre class="r"><code>df_p_bcf &lt;- rbind(df_sample_bcf, df_p_blanks_bcf)
dim(df_p_bcf) # 528 22</code></pre>
<pre><code>[1] 523  22</code></pre>
<pre class="r"><code>## innate nitrate
df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot; &amp; Time_point == &quot;T0&quot;) %&gt;% select(NO3_mM) %&gt;% unlist() %&gt;% mean()</code></pre>
<pre><code>[1] 0.1913479</code></pre>
<pre class="r"><code># 0.19 mM</code></pre>
</div>
<div id="sanity-check-moisture-correction-blank-correction-factor" class="section level2">
<h2>3. Sanity check: moisture correction &amp; blank correction factor</h2>
<p>Blank correction<br />
Use the nitrate concentration of the blank and use the ratio.<br />
For example, multiply to nitrate concentration x (2mM / changed [NO3])<br />
</p>
<pre class="r"><code># without any correction
# nitrate blanks
dim(df_no3_blank)</code></pre>
<pre><code>[1] 33 21</code></pre>
<pre class="r"><code>ggplot(df_no3_blank, aes(x=Time_hours, y=NO3_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.7,0.5), limits=c(0, 2.7))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrate blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite blanks
dim(df_no2_blank)</code></pre>
<pre><code>[1] 32 21</code></pre>
<pre class="r"><code>ggplot(df_no2_blank, aes(x=Time_hours, y=NO2_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.7,0.5), limits=c(0, 2.7))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrite blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-4-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># without any correction
ggplot(df_p, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(Without any correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-4-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### What are those points with very high std?
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;            &quot;Nitrite_input&quot;   &quot;Nitrate_input&quot;   &quot;Soil&quot;           
 [5] &quot;Sample_type&quot;     &quot;Titration_type&quot;  &quot;Unit&quot;            &quot;Concentration_M&quot;
 [9] &quot;Added_ul&quot;        &quot;Time_point&quot;      &quot;NO2_mM&quot;          &quot;NO2NO3_mM&quot;      
[13] &quot;NO3_mM&quot;          &quot;Time_minutes&quot;    &quot;Time_hours&quot;      &quot;Time_days&quot;      
[17] &quot;H_mol&quot;           &quot;Volume&quot;          &quot;H_Molarity&quot;      &quot;H_mM&quot;           </code></pre>
<pre class="r"><code># df_p %&gt;% arrange(desc(Std_NO3_mM)) %&gt;% select(Soil, Titration_type, Concentration_M, Sample_type, Time_point, Std_NO3_mM) 
## based on this I corrected the wrongly removed E02 -&lt; removed E01 (filter burst). Eliminated point P11 G11.

# After moisture correction
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-4-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># After blank correction (evaporation) 
ggplot(df_p_bcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture + blank correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-4-5.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="nitrogen-dynamics" class="section level2">
<h2>4. Nitrogen dynamics</h2>
</div>
<div id="ph-perturbation-experiment." class="section level2">
<h2>4.1. pH perturbation experiment.</h2>
<ul>
<li>Is there a difference in pH perturbation levels?</li>
</ul>
<pre class="r"><code># pH color
col_pH &lt;- colorRampPalette(c(&quot;gold&quot;,&quot;red&quot;,&quot;purple&quot;))

library(colorRamps)
colorRamps::green2red</code></pre>
<pre><code>function (n) 
rgb.tables(n, red = c(1, 0, 2), green = c(0, 0, 2), blue = c(0, 
    0, 0, 0))
&lt;bytecode: 0x0000000022515c48&gt;
&lt;environment: namespace:colorRamps&gt;</code></pre>
<pre class="r"><code>plot(rep(1,13),col=col_pH(13),pch=19,cex=3)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-5-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>grad_pH &lt;- scale_colour_gradientn(colours = col_pH(100))

# Confer this page (https://stackoverflow.com/questions/21537782/how-to-set-fixed-continuous-colour-values-in-ggplot2)
# myPalette &lt;- colorRampPalette(rev(brewer.pal(11, &quot;Spectral&quot;)))
# sc &lt;- scale_colour_gradientn(colours = myPalette(100), limits=c(1, 8))

# 1. pH perturbation
# plot Nitrate
df_pH &lt;- df_p_bcf %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;))
dim(df_pH)</code></pre>
<pre><code>[1] 425  22</code></pre>
<pre class="r"><code># average the no acid base one.
df_plot &lt;- df_pH %&gt;% select(H_mM, NO3_mM, NO2_mM, Soil, Time_hours)
df_plot %&gt;% filter(H_mM == 0) %&gt;% dim()</code></pre>
<pre><code>[1] 33  5</code></pre>
<pre class="r"><code>df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Soil, Time_hours) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code># A tibble: 11 x 5
    H_mM Soil               Time_hours NO3_mM NO2_mM
   &lt;dbl&gt; &lt;chr&gt;                   &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1     0 LaBaghWoods_pH6.66        0   2.21   0.0303
 2     0 LaBaghWoods_pH6.66        3.6 2.09   0.0324
 3     0 LaBaghWoods_pH6.66        6.4 1.88   0.0747
 4     0 LaBaghWoods_pH6.66        9.7 1.68   0.0491
 5     0 LaBaghWoods_pH6.66       24.5 0.559  0.0318
 6     0 LaBaghWoods_pH6.66       30.9 0.0835 0.0328
 7     0 LaBaghWoods_pH6.66       36.5 0      0.0301
 8     0 LaBaghWoods_pH6.66       50.2 0      0.0323
 9     0 LaBaghWoods_pH6.66       60.2 0      0.0331
10     0 LaBaghWoods_pH6.66       75.6 0      0.0403
11     0 LaBaghWoods_pH6.66       99.8 0      0.0381</code></pre>
<pre class="r"><code>df_sub &lt;- df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Soil, Time_hours) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()
df_main &lt;- df_plot %&gt;% filter(H_mM != 0)
dim(df_plot)</code></pre>
<pre><code>[1] 425   5</code></pre>
<pre class="r"><code>dim(df_main)</code></pre>
<pre><code>[1] 392   5</code></pre>
<pre class="r"><code>df_plot2 &lt;- rbind(df_main, df_sub)

ggplot(df_plot2, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-5-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot Nitrite
ggplot(df_plot2, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-5-3.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="nitrite-peakheight-and-h_mm" class="section level2">
<h2>4.1.1. Nitrite peakheight and H_mM</h2>
<pre class="r"><code>colnames(df_plot2)</code></pre>
<pre><code>[1] &quot;H_mM&quot;       &quot;NO3_mM&quot;     &quot;NO2_mM&quot;     &quot;Soil&quot;       &quot;Time_hours&quot;</code></pre>
<pre class="r"><code>df_plot2 %&gt;% group_by(H_mM) %&gt;% summarize(Max_NO2_mM = max(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code># A tibble: 37 x 2
    H_mM Max_NO2_mM
   &lt;dbl&gt;      &lt;dbl&gt;
 1 -94.7     1.70  
 2 -89.4     1.45  
 3 -84.2     1.07  
 4 -78.9     0.0876
 5 -73.7     0.0895
 6 -68.4     1.17  
 7 -63.1     1.74  
 8 -57.9     1.30  
 9 -52.6     1.70  
10 -47.4     1.23  
# ... with 27 more rows</code></pre>
<pre class="r"><code># library(ggpmisc)
# x[ggpmisc:::find_peaks(df_plot2$NO2_mM)]

# get global peaks
df_plot2</code></pre>
<pre><code>          H_mM     NO3_mM     NO2_mM               Soil Time_hours
1     5.261496 2.21477315 0.06570583 LaBaghWoods_pH6.66        0.0
2    15.784489 2.16541863 0.02789778 LaBaghWoods_pH6.66        0.0
3    26.307482 2.12493322 0.02837579 LaBaghWoods_pH6.66        0.0
4    36.830475 2.11048697 0.02810264 LaBaghWoods_pH6.66        0.0
5    47.353467 2.11061553 0.02769292 LaBaghWoods_pH6.66        0.0
6    57.876460 2.15242679 0.03069779 LaBaghWoods_pH6.66        0.0
7     5.261496 2.28741092 0.03007993 LaBaghWoods_pH6.66        3.6
8    26.307482 2.14076713 0.02875853 LaBaghWoods_pH6.66        3.6
9    36.830475 1.84093055 0.02931490 LaBaghWoods_pH6.66        3.6
10   47.353467 1.94738972 0.03056678 LaBaghWoods_pH6.66        3.6
11   57.876460 1.90846253 0.02820219 LaBaghWoods_pH6.66        3.6
12   10.522993 2.19132655 0.02858066 LaBaghWoods_pH6.66        0.0
13   21.045985 2.13963436 0.02769292 LaBaghWoods_pH6.66        0.0
14   31.568978 2.15012346 0.02769292 LaBaghWoods_pH6.66        0.0
15   42.091971 2.12823669 0.02769292 LaBaghWoods_pH6.66        0.0
16   52.614964 2.22937721 0.02817092 LaBaghWoods_pH6.66        0.0
17   63.137956 2.26535315 0.02933186 LaBaghWoods_pH6.66        0.0
18   10.522993 2.02263211 0.03167965 LaBaghWoods_pH6.66        3.6
19   21.045985 2.03622658 0.02917580 LaBaghWoods_pH6.66        3.6
20   31.568978 1.92734473 0.02841082 LaBaghWoods_pH6.66        3.6
21   42.091971 2.22618893 0.02952354 LaBaghWoods_pH6.66        3.6
22   52.614964 1.87219957 0.02868899 LaBaghWoods_pH6.66        3.6
23   63.137956 2.08018309 0.02959308 LaBaghWoods_pH6.66        3.6
24   68.399453 2.19707882 0.02803435 LaBaghWoods_pH6.66        0.0
25   78.922446 2.23342482 0.03083438 LaBaghWoods_pH6.66        0.0
26   89.445438 2.13264991 0.02912699 LaBaghWoods_pH6.66        0.0
27   -5.261496 2.06855378 0.03144909 LaBaghWoods_pH6.66        0.0
28  -15.784489 2.13867774 0.04012587 LaBaghWoods_pH6.66        0.0
29  -26.307482 2.09039107 0.05161118 LaBaghWoods_pH6.66        0.0
30   68.399453 2.41387674 0.02820219 LaBaghWoods_pH6.66        3.6
31   78.922446 2.10119370 0.03105365 LaBaghWoods_pH6.66        3.6
32   89.445438 2.35398013 0.03161010 LaBaghWoods_pH6.66        3.6
33   -5.261496 1.91874988 0.04155977 LaBaghWoods_pH6.66        3.6
34  -15.784489 2.06292489 0.04559707 LaBaghWoods_pH6.66        3.6
35  -26.307482 1.82450989 0.05193350 LaBaghWoods_pH6.66        3.6
36   73.660949 2.29098970 0.02933186 LaBaghWoods_pH6.66        0.0
37   84.183942 2.24560320 0.03056119 LaBaghWoods_pH6.66        0.0
38   94.706935 2.21212638 0.03138079 LaBaghWoods_pH6.66        0.0
39  -10.522993 2.15803610 0.03431805 LaBaghWoods_pH6.66        0.0
40  -21.045985 2.26598649 0.05085891 LaBaghWoods_pH6.66        0.0
41  -31.568978 2.10694995 0.04887582 LaBaghWoods_pH6.66        0.0
42   73.660949 2.15737721 0.02938444 LaBaghWoods_pH6.66        3.6
43   84.183942 2.35718770 0.03056678 LaBaghWoods_pH6.66        3.6
44   94.706935 2.12791883 0.03300129 LaBaghWoods_pH6.66        3.6
45  -10.522993 2.24540847 0.04135097 LaBaghWoods_pH6.66        3.6
46  -21.045985 1.92710112 0.05067994 LaBaghWoods_pH6.66        3.6
47  -31.568978 1.99341582 0.05416229 LaBaghWoods_pH6.66        3.6
48  -36.830475 2.13937665 0.04969638 LaBaghWoods_pH6.66        0.0
49  -47.353467 2.15929298 0.05475743 LaBaghWoods_pH6.66        0.0
50  -57.876460 2.02932867 0.05626239 LaBaghWoods_pH6.66        0.0
51  -68.399453 1.96206020 0.06070968 LaBaghWoods_pH6.66        0.0
52  -78.922446 1.93601403 0.05284225 LaBaghWoods_pH6.66        0.0
53  -89.445438 1.92261133 0.05393661 LaBaghWoods_pH6.66        0.0
54  -36.830475 2.21845272 0.05144599 LaBaghWoods_pH6.66        3.6
55  -47.353467 1.89889353 0.05381402 LaBaghWoods_pH6.66        3.6
56  -57.876460 2.01074174 0.05520714 LaBaghWoods_pH6.66        3.6
57  -68.399453 1.65295962 0.06538061 LaBaghWoods_pH6.66        3.6
58  -78.922446 1.71458001 0.07263142 LaBaghWoods_pH6.66        3.6
59  -89.445438 1.35317579 0.08044366 LaBaghWoods_pH6.66        3.6
60  -42.091971 2.05233300 0.04826043 LaBaghWoods_pH6.66        0.0
61  -52.614964 2.05044214 0.05537308 LaBaghWoods_pH6.66        0.0
62  -63.137956 1.97974414 0.05715174 LaBaghWoods_pH6.66        0.0
63  -73.660949 2.03502055 0.05776748 LaBaghWoods_pH6.66        0.0
64  -84.183942 1.94064271 0.06365242 LaBaghWoods_pH6.66        0.0
65  -94.706935 2.09340960 0.06502133 LaBaghWoods_pH6.66        0.0
66  -42.091971 2.14078237 0.05590375 LaBaghWoods_pH6.66        3.6
67  -52.614964 1.82006814 0.05813309 LaBaghWoods_pH6.66        3.6
68  -63.137956 1.79189466 0.06217457 LaBaghWoods_pH6.66        3.6
69  -73.660949 1.74470709 0.06970281 LaBaghWoods_pH6.66        3.6
70  -84.183942 1.45223315 0.07535132 LaBaghWoods_pH6.66        3.6
71  -94.706935 1.42416426 0.08009481 LaBaghWoods_pH6.66        3.6
72    5.261496 1.99215790 0.06770988 LaBaghWoods_pH6.66        6.4
73   15.784489 1.93409091 0.03205179 LaBaghWoods_pH6.66        6.4
74   26.307482 1.86681274 0.03198263 LaBaghWoods_pH6.66        6.4
75   36.830475 1.99895582 0.02976962 LaBaghWoods_pH6.66        6.4
76   47.353467 1.94645412 0.03046115 LaBaghWoods_pH6.66        6.4
77   57.876460 1.86003790 0.03039200 LaBaghWoods_pH6.66        6.4
78    5.261496 1.68221850 0.06074843 LaBaghWoods_pH6.66        9.7
79   15.784489 1.82769758 0.04004772 LaBaghWoods_pH6.66        9.7
80   26.307482 2.20240053 0.03080099 LaBaghWoods_pH6.66        9.7
81   36.830475 2.02870782 0.03150666 LaBaghWoods_pH6.66        9.7
82   47.353467 1.91304252 0.02960141 LaBaghWoods_pH6.66        9.7
83   57.876460 1.86070558 0.02861360 LaBaghWoods_pH6.66        9.7
84   10.522993 1.83212331 0.04291385 LaBaghWoods_pH6.66        6.4
85   21.045985 1.94547132 0.02893983 LaBaghWoods_pH6.66        6.4
86   31.568978 1.86188034 0.02804093 LaBaghWoods_pH6.66        6.4
87   42.091971 1.80031750 0.02824836 LaBaghWoods_pH6.66        6.4
88   52.614964 1.95002646 0.02804093 LaBaghWoods_pH6.66        6.4
89   63.137956 1.96692729 0.02804093 LaBaghWoods_pH6.66        6.4
90   10.522993 1.75254834 0.08685623 LaBaghWoods_pH6.66        9.7
91   21.045985 1.83494337 0.03108325 LaBaghWoods_pH6.66        9.7
92   31.568978 1.86570547 0.02861360 LaBaghWoods_pH6.66        9.7
93   42.091971 1.95435993 0.02861360 LaBaghWoods_pH6.66        9.7
94   52.614964 1.85086900 0.02861360 LaBaghWoods_pH6.66        9.7
95   63.137956 1.92632921 0.03044816 LaBaghWoods_pH6.66        9.7
96   68.399453 2.22704140 0.02935472 LaBaghWoods_pH6.66        6.4
97   78.922446 2.00481532 0.03343508 LaBaghWoods_pH6.66        6.4
98   89.445438 1.96980900 0.03122187 LaBaghWoods_pH6.66        6.4
99   -5.261496 2.00850448 0.11720977 LaBaghWoods_pH6.66        6.4
100 -15.784489 1.79260349 0.19730597 LaBaghWoods_pH6.66        6.4
101 -26.307482 1.79953837 0.09553022 LaBaghWoods_pH6.66        6.4
102  68.399453 2.14430954 0.03009534 LaBaghWoods_pH6.66        9.7
103  78.922446 2.23540185 0.03277695 LaBaghWoods_pH6.66        9.7
104  89.445438 2.49557160 0.03475314 LaBaghWoods_pH6.66        9.7
105  -5.261496 1.69715735 0.13181208 LaBaghWoods_pH6.66        9.7
106 -15.784489 1.40411568 0.41859861 LaBaghWoods_pH6.66        9.7
107 -26.307482 1.61933413 0.18166488 LaBaghWoods_pH6.66        9.7
108  73.660949 2.07981960 0.03011538 LaBaghWoods_pH6.66        6.4
109  84.183942 1.93043662 0.03136019 LaBaghWoods_pH6.66        6.4
110  94.706935 2.04800716 0.03101440 LaBaghWoods_pH6.66        6.4
111 -10.522993 1.77674725 0.19513963 LaBaghWoods_pH6.66        6.4
112 -21.045985 1.81305172 0.14295800 LaBaghWoods_pH6.66        6.4
113 -31.568978 1.87200437 0.07117609 LaBaghWoods_pH6.66        6.4
114  73.660949 2.36011773 0.03122439 LaBaghWoods_pH6.66        9.7
115  84.183942 2.24183704 0.03418849 LaBaghWoods_pH6.66        9.7
116  94.706935 2.30688140 0.03256522 LaBaghWoods_pH6.66        9.7
117 -10.522993 1.55318951 0.40985389 LaBaghWoods_pH6.66        9.7
118 -21.045985 1.64789519 0.29983629 LaBaghWoods_pH6.66        9.7
119 -31.568978 1.76678123 0.11740410 LaBaghWoods_pH6.66        9.7
120 -36.830475 1.77394345 0.06271986 LaBaghWoods_pH6.66        6.4
121 -47.353467 1.74632853 0.06327423 LaBaghWoods_pH6.66        6.4
122 -57.876460 1.77449505 0.06667016 LaBaghWoods_pH6.66        6.4
123 -68.399453 1.56868075 0.07492045 LaBaghWoods_pH6.66        6.4
124 -78.922446 1.32134509 0.07713975 LaBaghWoods_pH6.66        6.4
125 -89.445438 1.20631763 0.07928999 LaBaghWoods_pH6.66        6.4
126 -36.830475 1.91376060 0.08218346 LaBaghWoods_pH6.66        9.7
127 -47.353467 1.85920822 0.06428361 LaBaghWoods_pH6.66        9.7
128 -57.876460 2.01551543 0.06838539 LaBaghWoods_pH6.66        9.7
129 -68.399453 1.50450449 0.07178074 LaBaghWoods_pH6.66        9.7
130 -78.922446 1.24869577 0.07645052 LaBaghWoods_pH6.66        9.7
131 -89.445438 0.99884789 0.08190030 LaBaghWoods_pH6.66        9.7
132 -42.091971 1.76212153 0.06209622 LaBaghWoods_pH6.66        6.4
133 -52.614964 1.64193598 0.06978952 LaBaghWoods_pH6.66        6.4
134 -63.137956 1.48520361 0.07422698 LaBaghWoods_pH6.66        6.4
135 -73.660949 1.42105864 0.07173076 LaBaghWoods_pH6.66        6.4
136 -84.183942 1.31329724 0.08345257 LaBaghWoods_pH6.66        6.4
137 -94.706935 1.23816115 0.08241182 LaBaghWoods_pH6.66        6.4
138 -42.091971 2.02730969 0.06916343 LaBaghWoods_pH6.66        9.7
139 -52.614964 1.82518173 0.06838539 LaBaghWoods_pH6.66        9.7
140 -63.137956 1.71225352 0.07192223 LaBaghWoods_pH6.66        9.7
141 -73.660949 1.41518213 0.07701665 LaBaghWoods_pH6.66        9.7
142 -84.183942 1.18291892 0.08182951 LaBaghWoods_pH6.66        9.7
143 -94.706935 0.99194496 0.08430728 LaBaghWoods_pH6.66        9.7
144   5.261496 0.43171887 0.02838484 LaBaghWoods_pH6.66       24.5
145  15.784489 1.27705984 0.06120590 LaBaghWoods_pH6.66       24.5
146  26.307482 1.60570392 0.02910183 LaBaghWoods_pH6.66       24.5
147  36.830475 1.66720020 0.02760271 LaBaghWoods_pH6.66       24.5
148  47.353467 2.10036020 0.02642959 LaBaghWoods_pH6.66       24.5
149  57.876460 2.25804331 0.02740718 LaBaghWoods_pH6.66       24.5
150   5.261496 0.00000000 0.03029344 LaBaghWoods_pH6.66       30.9
151  15.784489 0.94184123 0.06194423 LaBaghWoods_pH6.66       30.9
152  26.307482 1.36168338 0.02846156 LaBaghWoods_pH6.66       30.9
153  36.830475 1.67748826 0.02751178 LaBaghWoods_pH6.66       30.9
154  47.353467 1.77399774 0.03158267 LaBaghWoods_pH6.66       30.9
155  57.876460 1.80205908 0.02751178 LaBaghWoods_pH6.66       30.9
156  10.522993 0.84251664 0.05408748 LaBaghWoods_pH6.66       24.5
157  21.045985 1.43068465 0.03275249 LaBaghWoods_pH6.66       24.5
158  31.568978 1.64691837 0.02642959 LaBaghWoods_pH6.66       24.5
159  42.091971 1.85540718 0.02642959 LaBaghWoods_pH6.66       24.5
160  52.614964 2.06109089 0.02642959 LaBaghWoods_pH6.66       24.5
161  63.137956 2.15559304 0.02675544 LaBaghWoods_pH6.66       24.5
162  10.522993 0.35710705 0.03151481 LaBaghWoods_pH6.66       30.9
163  21.045985 1.17968574 0.03796252 LaBaghWoods_pH6.66       30.9
164  31.568978 1.57654306 0.02751178 LaBaghWoods_pH6.66       30.9
165  42.091971 1.82520542 0.02941139 LaBaghWoods_pH6.66       30.9
166  52.614964 1.84092635 0.02751178 LaBaghWoods_pH6.66       30.9
167  63.137956 1.95575526 0.03165052 LaBaghWoods_pH6.66       30.9
168  68.399453 2.06800135 0.02786341 LaBaghWoods_pH6.66       24.5
169  78.922446 2.21983668 0.02688579 LaBaghWoods_pH6.66       24.5
170  89.445438 2.23253060 0.02760271 LaBaghWoods_pH6.66       24.5
171  -5.261496 0.46343072 0.03549106 LaBaghWoods_pH6.66       24.5
172 -15.784489 0.00000000 0.31756757 LaBaghWoods_pH6.66       24.5
173 -26.307482 0.77928412 0.79408193 LaBaghWoods_pH6.66       24.5
174  68.399453 2.04134446 0.02778314 LaBaghWoods_pH6.66       30.9
175  78.922446 2.19805632 0.02934355 LaBaghWoods_pH6.66       30.9
176  89.445438 1.96309644 0.03002203 LaBaghWoods_pH6.66       30.9
177  -5.261496 0.01669228 0.03436505 LaBaghWoods_pH6.66       30.9
178 -15.784489 0.00000000 0.04380170 LaBaghWoods_pH6.66       30.9
179 -26.307482 0.00000000 0.68684041 LaBaghWoods_pH6.66       30.9
180  73.660949 2.34616689 0.02779824 LaBaghWoods_pH6.66       24.5
181  84.183942 2.17282388 0.02858038 LaBaghWoods_pH6.66       24.5
182  94.706935 2.16925992 0.03092705 LaBaghWoods_pH6.66       24.5
183 -10.522993 0.04850377 0.07355686 LaBaghWoods_pH6.66       24.5
184 -21.045985 0.29372756 0.84637617 LaBaghWoods_pH6.66       24.5
185 -31.568978 0.00000000 1.28382145 LaBaghWoods_pH6.66       24.5
186  73.660949 2.16899657 0.02751178 LaBaghWoods_pH6.66       30.9
187  84.183942 2.19992540 0.02751178 LaBaghWoods_pH6.66       30.9
188  94.706935 2.09784224 0.02839371 LaBaghWoods_pH6.66       30.9
189 -10.522993 0.00000000 0.03959184 LaBaghWoods_pH6.66       30.9
190 -21.045985 0.00000000 0.11104163 LaBaghWoods_pH6.66       30.9
191 -31.568978 0.00000000 0.05324406 LaBaghWoods_pH6.66       30.9
192 -36.830475 0.00000000 0.05291229 LaBaghWoods_pH6.66       24.5
193 -47.353467 0.00000000 1.23199660 LaBaghWoods_pH6.66       24.5
194 -57.876460 0.00000000 1.30175706 LaBaghWoods_pH6.66       24.5
195 -68.399453 1.63526211 0.07120351 LaBaghWoods_pH6.66       24.5
196 -78.922446 1.54955663 0.08362815 LaBaghWoods_pH6.66       24.5
197 -89.445438 1.35521984 0.08788106 LaBaghWoods_pH6.66       24.5
198 -36.830475 0.00000000 0.05310815 LaBaghWoods_pH6.66       30.9
199 -47.353467 0.00000000 0.05929288 LaBaghWoods_pH6.66       30.9
200 -57.876460 0.00000000 0.06296410 LaBaghWoods_pH6.66       30.9
201 -68.399453 1.32957899 0.29001540 LaBaghWoods_pH6.66       30.9
202 -78.922446 1.42321176 0.07956335 LaBaghWoods_pH6.66       30.9
203 -89.445438 1.11940225 0.08936794 LaBaghWoods_pH6.66       30.9
204 -42.091971 0.00000000 0.05545865 LaBaghWoods_pH6.66       24.5
205 -52.614964 0.00000000 1.69965253 LaBaghWoods_pH6.66       24.5
206 -63.137956 1.65412042 0.08781562 LaBaghWoods_pH6.66       24.5
207 -73.660949 1.53153939 0.08369357 LaBaghWoods_pH6.66       24.5
208 -84.183942 1.43351834 0.08873178 LaBaghWoods_pH6.66       24.5
209 -94.706935 1.26031825 0.08847001 LaBaghWoods_pH6.66       24.5
210 -42.091971 0.00000000 0.05603030 LaBaghWoods_pH6.66       30.9
211 -52.614964 0.00000000 0.06384804 LaBaghWoods_pH6.66       30.9
212 -63.137956 0.00000000 1.74445642 LaBaghWoods_pH6.66       30.9
213 -73.660949 1.58534676 0.08174161 LaBaghWoods_pH6.66       30.9
214 -84.183942 1.37055226 0.09038960 LaBaghWoods_pH6.66       30.9
215 -94.706935 1.11665654 0.09931500 LaBaghWoods_pH6.66       30.9
216   5.261496 0.00000000 0.02697399 LaBaghWoods_pH6.66       36.5
217  15.784489 0.52747287 0.04913892 LaBaghWoods_pH6.66       36.5
218  26.307482 1.21716806 0.03143102 LaBaghWoods_pH6.66       36.5
219  36.830475 1.67990038 0.02697399 LaBaghWoods_pH6.66       36.5
220  47.353467 1.83385970 0.02697399 LaBaghWoods_pH6.66       36.5
221  57.876460 1.94415727 0.02843735 LaBaghWoods_pH6.66       36.5
222   5.261496 0.00000000 0.02834206 LaBaghWoods_pH6.66       50.2
223  15.784489 0.00000000 0.02654055 LaBaghWoods_pH6.66       50.2
224  26.307482 0.11010524 0.03104475 LaBaghWoods_pH6.66       50.2
225  36.830475 1.58464262 0.02705525 LaBaghWoods_pH6.66       50.2
226  47.353467 1.68888462 0.02892117 LaBaghWoods_pH6.66       50.2
227  57.876460 1.69823196 0.02840641 LaBaghWoods_pH6.66       50.2
228  10.522993 0.00000000 0.02730656 LaBaghWoods_pH6.66       36.5
229  21.045985 0.78783577 0.03974985 LaBaghWoods_pH6.66       36.5
230  31.568978 1.45646247 0.02797172 LaBaghWoods_pH6.66       36.5
231  42.091971 1.61366782 0.02730656 LaBaghWoods_pH6.66       36.5
232  52.614964 1.79010132 0.02750610 LaBaghWoods_pH6.66       36.5
233  63.137956 2.12466662 0.02863691 LaBaghWoods_pH6.66       36.5
234  10.522993 0.00000000 0.02609021 LaBaghWoods_pH6.66       50.2
235  21.045985 0.00000000 0.02621887 LaBaghWoods_pH6.66       50.2
236  31.568978 1.15449242 0.02666922 LaBaghWoods_pH6.66       50.2
237  42.091971 1.63498389 0.02750562 LaBaghWoods_pH6.66       50.2
238  52.614964 1.89231892 0.02872813 LaBaghWoods_pH6.66       50.2
239  63.137956 2.09911203 0.02872813 LaBaghWoods_pH6.66       50.2
240  68.399453 1.94159936 0.02903604 LaBaghWoods_pH6.66       36.5
241  78.922446 2.12949782 0.02857039 LaBaghWoods_pH6.66       36.5
242  89.445438 2.10303159 0.02996738 LaBaghWoods_pH6.66       36.5
243  -5.261496 0.00000000 0.03322751 LaBaghWoods_pH6.66       36.5
244 -15.784489 0.00000000 0.04734056 LaBaghWoods_pH6.66       36.5
245 -26.307482 0.00000000 0.05093748 LaBaghWoods_pH6.66       36.5
246  68.399453 2.30587553 0.02744128 LaBaghWoods_pH6.66       50.2
247  78.922446 2.27987993 0.02859944 LaBaghWoods_pH6.66       50.2
248  89.445438 2.26989195 0.02956465 LaBaghWoods_pH6.66       50.2
249  -5.261496 0.00000000 0.03387666 LaBaghWoods_pH6.66       50.2
250 -15.784489 0.00000000 0.04108760 LaBaghWoods_pH6.66       50.2
251 -26.307482 0.00000000 0.04624043 LaBaghWoods_pH6.66       50.2
252  73.660949 2.09175828 0.02910256 LaBaghWoods_pH6.66       36.5
253  84.183942 2.20380266 0.03069918 LaBaghWoods_pH6.66       36.5
254  94.706935 2.17946967 0.03256212 LaBaghWoods_pH6.66       36.5
255 -10.522993 0.00000000 0.04394425 LaBaghWoods_pH6.66       36.5
256 -21.045985 0.00000000 0.04487650 LaBaghWoods_pH6.66       36.5
257 -31.568978 0.00000000 0.05153705 LaBaghWoods_pH6.66       36.5
258  73.660949 2.13253197 0.02744128 LaBaghWoods_pH6.66       50.2
259  84.183942 1.95032743 0.02756996 LaBaghWoods_pH6.66       50.2
260  94.706935 2.35847664 0.02995074 LaBaghWoods_pH6.66       50.2
261 -10.522993 0.00000000 0.04495205 LaBaghWoods_pH6.66       50.2
262 -21.045985 0.00000000 0.04649812 LaBaghWoods_pH6.66       50.2
263 -31.568978 0.00000000 0.04939742 LaBaghWoods_pH6.66       50.2
264 -36.830475 0.00000000 0.05280288 LaBaghWoods_pH6.66       36.5
265 -47.353467 0.00000000 0.05726744 LaBaghWoods_pH6.66       36.5
266 -57.876460 0.00000000 0.06179997 LaBaghWoods_pH6.66       36.5
267 -68.399453 0.00000000 1.17437834 LaBaghWoods_pH6.66       36.5
268 -78.922446 1.31733280 0.08762101 LaBaghWoods_pH6.66       36.5
269 -89.445438 1.16453852 0.08481661 LaBaghWoods_pH6.66       36.5
270 -36.830475 0.00000000 0.05216839 LaBaghWoods_pH6.66       50.2
271 -47.353467 0.00000000 0.05435977 LaBaghWoods_pH6.66       50.2
272 -57.876460 0.00000000 0.05990410 LaBaghWoods_pH6.66       50.2
273 -68.399453 0.00000000 0.06751482 LaBaghWoods_pH6.66       50.2
274 -78.922446 0.00000000 0.06951489 LaBaghWoods_pH6.66       50.2
275 -89.445438 0.99914325 0.09301995 LaBaghWoods_pH6.66       50.2
276 -42.091971 0.00000000 0.05620116 LaBaghWoods_pH6.66       36.5
277 -52.614964 0.00000000 0.06506689 LaBaghWoods_pH6.66       36.5
278 -63.137956 0.00000000 0.06820112 LaBaghWoods_pH6.66       36.5
279 -73.660949 1.29293341 0.08949089 LaBaghWoods_pH6.66       36.5
280 -84.183942 1.18814457 0.09156139 LaBaghWoods_pH6.66       36.5
281 -94.706935 1.05440239 0.09216255 LaBaghWoods_pH6.66       36.5
282 -42.091971 0.00000000 0.05055730 LaBaghWoods_pH6.66       50.2
283 -52.614964 0.00000000 0.05919482 LaBaghWoods_pH6.66       50.2
284 -63.137956 0.00000000 0.06512798 LaBaghWoods_pH6.66       50.2
285 -73.660949 0.00000000 0.07493582 LaBaghWoods_pH6.66       50.2
286 -84.183942 0.00000000 1.06861160 LaBaghWoods_pH6.66       50.2
287 -94.706935 1.00755369 0.09928998 LaBaghWoods_pH6.66       50.2
288   5.261496 0.00000000 0.02746795 LaBaghWoods_pH6.66       60.2
289  15.784489 0.00000000 0.02791151 LaBaghWoods_pH6.66       60.2
290  26.307482 0.00000000 0.02708777 LaBaghWoods_pH6.66       60.2
291  36.830475 1.42653199 0.02569386 LaBaghWoods_pH6.66       60.2
292  47.353467 1.73971754 0.02683432 LaBaghWoods_pH6.66       60.2
293  57.876460 1.69295628 0.02829171 LaBaghWoods_pH6.66       60.2
294   5.261496 0.00000000 0.03205397 LaBaghWoods_pH6.66       75.6
295  15.784489 0.00000000 0.02917537 LaBaghWoods_pH6.66       75.6
296  26.307482 0.00000000 0.02580756 LaBaghWoods_pH6.66       75.6
297  36.830475 1.45312662 0.02635860 LaBaghWoods_pH6.66       75.6
298  47.353467 1.54375015 0.02837927 LaBaghWoods_pH6.66       75.6
299  57.876460 1.98547847 0.02482798 LaBaghWoods_pH6.66       75.6
300  10.522993 0.00000000 0.02594729 LaBaghWoods_pH6.66       60.2
301  21.045985 0.00000000 0.02569386 LaBaghWoods_pH6.66       60.2
302  31.568978 0.44584098 0.02740459 LaBaghWoods_pH6.66       60.2
303  52.614964 1.74513253 0.02569386 LaBaghWoods_pH6.66       60.2
304  63.137956 1.97176886 0.02759468 LaBaghWoods_pH6.66       60.2
305  10.522993 0.00000000 0.02733828 LaBaghWoods_pH6.66       75.6
306  21.045985 0.00000000 0.02617491 LaBaghWoods_pH6.66       75.6
307  31.568978 0.00000000 0.02482798 LaBaghWoods_pH6.66       75.6
308  42.091971 1.48396437 0.02501164 LaBaghWoods_pH6.66       75.6
309  52.614964 1.91663676 0.02629737 LaBaghWoods_pH6.66       75.6
310  63.137956 1.78921830 0.02525653 LaBaghWoods_pH6.66       75.6
311  68.399453 1.92774876 0.02664424 LaBaghWoods_pH6.66       60.2
312  78.922446 2.03789847 0.02696105 LaBaghWoods_pH6.66       60.2
313  89.445438 2.13450650 0.02626408 LaBaghWoods_pH6.66       60.2
314  -5.261496 0.00000000 0.03494686 LaBaghWoods_pH6.66       60.2
315 -15.784489 0.00000000 0.04414230 LaBaghWoods_pH6.66       60.2
316 -26.307482 0.00000000 0.04598208 LaBaghWoods_pH6.66       60.2
317  68.399453 2.22476724 0.02525653 LaBaghWoods_pH6.66       75.6
318  78.922446 2.42127453 0.02684843 LaBaghWoods_pH6.66       75.6
319  89.445438 2.45722064 0.02905289 LaBaghWoods_pH6.66       75.6
320  -5.261496 0.00000000 0.03928366 LaBaghWoods_pH6.66       75.6
321 -15.784489 0.00000000 0.04124490 LaBaghWoods_pH6.66       75.6
322 -26.307482 0.00000000 0.04480033 LaBaghWoods_pH6.66       75.6
323  73.660949 2.09285324 0.02829171 LaBaghWoods_pH6.66       60.2
324  84.183942 2.13462774 0.02936901 LaBaghWoods_pH6.66       60.2
325  94.706935 2.03743004 0.02886204 LaBaghWoods_pH6.66       60.2
326 -21.045985 0.00000000 0.04376168 LaBaghWoods_pH6.66       60.2
327 -31.568978 0.00000000 0.04871046 LaBaghWoods_pH6.66       60.2
328  73.660949 2.45073004 0.02733828 LaBaghWoods_pH6.66       75.6
329  84.183942 2.45960510 0.02733828 LaBaghWoods_pH6.66       75.6
330  94.706935 2.44619214 0.02991028 LaBaghWoods_pH6.66       75.6
331 -10.522993 0.00000000 0.04247081 LaBaghWoods_pH6.66       75.6
332 -21.045985 0.00000000 0.04265470 LaBaghWoods_pH6.66       75.6
333 -31.568978 0.00000000 0.04418726 LaBaghWoods_pH6.66       75.6
334 -36.830475 0.00000000 0.04997965 LaBaghWoods_pH6.66       60.2
335 -47.353467 0.00000000 0.05289920 LaBaghWoods_pH6.66       60.2
336 -57.876460 0.00000000 0.05975611 LaBaghWoods_pH6.66       60.2
337 -68.399453 0.00000000 0.07017463 LaBaghWoods_pH6.66       60.2
338 -78.922446 0.00000000 0.08085499 LaBaghWoods_pH6.66       60.2
339 -89.445438 0.00000000 1.45055613 LaBaghWoods_pH6.66       60.2
340 -36.830475 0.00000000 0.41928339 LaBaghWoods_pH6.66       75.6
341 -47.353467 0.00000000 0.05099383 LaBaghWoods_pH6.66       75.6
342 -57.876460 0.00000000 0.05657645 LaBaghWoods_pH6.66       75.6
343 -68.399453 0.00000000 0.07057340 LaBaghWoods_pH6.66       75.6
344 -78.922446 0.00000000 0.07800727 LaBaghWoods_pH6.66       75.6
345 -89.445438 0.00000000 0.08716678 LaBaghWoods_pH6.66       75.6
346 -42.091971 0.00000000 0.04997965 LaBaghWoods_pH6.66       60.2
347 -52.614964 0.00000000 0.06115328 LaBaghWoods_pH6.66       60.2
348 -63.137956 0.00000000 0.06369394 LaBaghWoods_pH6.66       60.2
349 -84.183942 0.00000000 0.08040982 LaBaghWoods_pH6.66       60.2
350 -94.706935 0.48543913 0.08734333 LaBaghWoods_pH6.66       60.2
351 -42.091971 0.28194393 0.06148608 LaBaghWoods_pH6.66       75.6
352 -52.614964 0.00000000 0.05835599 LaBaghWoods_pH6.66       75.6
353 -63.137956 0.00000000 0.06572200 LaBaghWoods_pH6.66       75.6
354 -73.660949 0.00000000 0.07075767 LaBaghWoods_pH6.66       75.6
355 -84.183942 0.00000000 0.08974974 LaBaghWoods_pH6.66       75.6
356 -94.706935 0.00000000 1.69911297 LaBaghWoods_pH6.66       75.6
357   5.261496 0.00000000 0.02845803 LaBaghWoods_pH6.66       99.8
358  15.784489 0.00000000 0.02788239 LaBaghWoods_pH6.66       99.8
359  26.307482 0.00000000 0.02609810 LaBaghWoods_pH6.66       99.8
360  36.830475 1.25773935 0.02488952 LaBaghWoods_pH6.66       99.8
361  47.353467 1.63643460 0.02460178 LaBaghWoods_pH6.66       99.8
362  57.876460 1.97500066 0.02546502 LaBaghWoods_pH6.66       99.8
363  10.522993 0.00000000 0.02886098 LaBaghWoods_pH6.66       99.8
364  21.045985 0.00000000 0.02736435 LaBaghWoods_pH6.66       99.8
365  31.568978 0.00000000 0.02759459 LaBaghWoods_pH6.66       99.8
366  42.091971 1.50607012 0.02604055 LaBaghWoods_pH6.66       99.8
367  52.614964 1.85332931 0.02609810 LaBaghWoods_pH6.66       99.8
368  63.137956 1.90471785 0.02655854 LaBaghWoods_pH6.66       99.8
369  68.399453 1.78236039 0.02586789 LaBaghWoods_pH6.66       99.8
370  78.922446 2.37364544 0.02696144 LaBaghWoods_pH6.66       99.8
371  89.445438 1.94409108 0.02414141 LaBaghWoods_pH6.66       99.8
372  -5.261496 0.00000000 0.04424004 LaBaghWoods_pH6.66       99.8
373 -15.784489 0.00000000 0.04026396 LaBaghWoods_pH6.66       99.8
374 -26.307482 0.00000000 0.04193492 LaBaghWoods_pH6.66       99.8
375  73.660949 2.22518731 0.02471687 LaBaghWoods_pH6.66       99.8
376  84.183942 2.32464546 0.02586789 LaBaghWoods_pH6.66       99.8
377  94.706935 2.41742779 0.02886098 LaBaghWoods_pH6.66       99.8
378 -10.522993 0.00000000 0.03813236 LaBaghWoods_pH6.66       99.8
379 -21.045985 0.00000000 0.04216541 LaBaghWoods_pH6.66       99.8
380 -31.568978 0.00000000 0.04447057 LaBaghWoods_pH6.66       99.8
381 -36.830475 0.00000000 0.04856321 LaBaghWoods_pH6.66       99.8
382 -47.353467 0.00000000 0.04775611 LaBaghWoods_pH6.66       99.8
383 -57.876460 0.00000000 0.05110013 LaBaghWoods_pH6.66       99.8
384 -68.399453 0.00000000 0.05836764 LaBaghWoods_pH6.66       99.8
385 -78.922446 0.00000000 0.06610096 LaBaghWoods_pH6.66       99.8
386 -89.445438 0.00000000 0.07545625 LaBaghWoods_pH6.66       99.8
387 -42.091971 0.00000000 0.04902443 LaBaghWoods_pH6.66       99.8
388 -52.614964 0.00000000 0.05156144 LaBaghWoods_pH6.66       99.8
389 -63.137956 0.00000000 0.05859842 LaBaghWoods_pH6.66       99.8
390 -73.660949 0.00000000 0.06263772 LaBaghWoods_pH6.66       99.8
391 -84.183942 0.00000000 0.07383881 LaBaghWoods_pH6.66       99.8
392 -94.706935 0.00000000 1.15013070 LaBaghWoods_pH6.66       99.8
393   0.000000 2.20561768 0.03033354 LaBaghWoods_pH6.66        0.0
394   0.000000 2.09040112 0.03237525 LaBaghWoods_pH6.66        3.6
395   0.000000 1.87697175 0.07466739 LaBaghWoods_pH6.66        6.4
396   0.000000 1.68443369 0.04913502 LaBaghWoods_pH6.66        9.7
397   0.000000 0.55859031 0.03179629 LaBaghWoods_pH6.66       24.5
398   0.000000 0.08349005 0.03278156 LaBaghWoods_pH6.66       30.9
399   0.000000 0.00000000 0.03010045 LaBaghWoods_pH6.66       36.5
400   0.000000 0.00000000 0.03231064 LaBaghWoods_pH6.66       50.2
401   0.000000 0.00000000 0.03310857 LaBaghWoods_pH6.66       60.2
402   0.000000 0.00000000 0.04030512 LaBaghWoods_pH6.66       75.6
403   0.000000 0.00000000 0.03811318 LaBaghWoods_pH6.66       99.8</code></pre>
<pre class="r"><code>mat_H_time &lt;- dcast(df_plot2, H_mM ~ Time_hours, value.var = &quot;NO2_mM&quot;)
mat_H_time &lt;- tibble::column_to_rownames(mat_H_time, var = &quot;H_mM&quot;)
mat_H_time &lt;- as.matrix(mat_H_time) 
class(mat_H_time)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>## get rid of NA
# Substitute NA to average of NO2_mM of before and after.
na_to_average &lt;- function(mat_NO2){
  for (i in 1:dim(mat_NO2)[1]){
    # print(i)
    for (j in 1:dim(mat_NO2)[2]){
      # print(j)
      if (is.na(mat_NO2[i,j])){
        mat_NO2[i,j] &lt;- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2
      }
    }
  }
  return(mat_NO2)
}
mat_H_time &lt;- na_to_average(mat_H_time)
mat_H_time[is.na(mat_H_time)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code># maximum peak value
apply(mat_H_time, 1, max)</code></pre>
<pre><code>-94.7069346522151 -89.4454382826476 -84.1839419130801 -78.9224455435126 
       1.69911297        1.45055613        1.06861160        0.08762101 
-73.6609491739451 -68.3994528043776 -63.1379564348101 -57.8764600652426 
       0.08949089        1.17437834        1.74445642        1.30175706 
 -52.614963695675 -47.3534673261075   -42.09197095654 -36.8304745869725 
       1.69965253        1.23199660        0.06916343        0.41928339 
 -31.568978217405 -26.3074818478375   -21.04598547827 -15.7844891087025 
       1.28382145        0.79408193        0.84637617        0.41859861 
 -10.522992739135  -5.2614963695675                 0   5.2614963695675 
       0.40985389        0.13181208        0.07466739        0.06770988 
  10.522992739135  15.7844891087025    21.04598547827  26.3074818478375 
       0.08685623        0.06194423        0.03974985        0.03198263 
  31.568978217405  36.8304745869725    42.09197095654  47.3534673261075 
       0.02861360        0.03150666        0.02952354        0.03158267 
  52.614963695675  57.8764600652426  63.1379564348101  68.3994528043776 
       0.02872813        0.03069779        0.03165052        0.03009534 
 73.6609491739451  78.9224455435126  84.1839419130801  89.4454382826476 
       0.03122439        0.03343508        0.03418849        0.03475314 
 94.7069346522151 
       0.03300129 </code></pre>
<pre class="r"><code>## plot with peak height and H_mM
vec_peakheight &lt;- apply(mat_H_time, 1, max)
vec_H_mM &lt;- names(vec_peakheight)
plot(vec_H_mM, vec_peakheight)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-6-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_peakheight &lt;- data.frame(H_mM = vec_H_mM, NO2_mM = vec_peakheight)
df_peakheight$H_mM &lt;- as.numeric(df_peakheight$H_mM)
## plot
# linear regression
fit.peakheight &lt;- lm(NO2_mM ~ H_mM, df_peakheight)
summary(fit.peakheight)</code></pre>
<pre><code>
Call:
lm(formula = NO2_mM ~ H_mM, data = df_peakheight)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.96289 -0.25157 -0.02186  0.23540  0.84893 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  0.451150   0.067955   6.639 1.12e-07 ***
H_mM        -0.007594   0.001210  -6.278 3.33e-07 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4134 on 35 degrees of freedom
Multiple R-squared:  0.5297,    Adjusted R-squared:  0.5162 
F-statistic: 39.42 on 1 and 35 DF,  p-value: 3.327e-07</code></pre>
<pre class="r"><code># Plot H_mM with peak time

## 1. linear regression
ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak (mM)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peak height \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_abline(slope = coef(fit.peakheight)[[2]], intercept = coef(fit.peakheight)[[1]],
              color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=0,y=1, label= paste0(&quot;y = &quot;, round(coef(fit.peakheight)[[1]],3),&quot;+&quot;,round(coef(fit.peakheight)[[2]],3),&quot;x&quot;, &quot;,  R^2: &quot;, round(summary(fit.peakheight)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-6-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. quadratic regression

df_peakheight$H_mM_2 &lt;- (df_peakheight$H_mM)^2 
quad.peakheight &lt;- lm(NO2_mM ~ H_mM + H_mM_2, df_peakheight)
summary(quad.peakheight)</code></pre>
<pre><code>
Call:
lm(formula = NO2_mM ~ H_mM + H_mM_2, data = df_peakheight)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.08824 -0.15076  0.00301  0.12251  0.86473 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  3.224e-01  9.916e-02   3.252  0.00259 ** 
H_mM        -7.594e-03  1.176e-03  -6.458 2.21e-07 ***
H_mM_2       4.079e-05  2.343e-05   1.741  0.09073 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4019 on 34 degrees of freedom
Multiple R-squared:  0.5682,    Adjusted R-squared:  0.5428 
F-statistic: 22.37 on 2 and 34 DF,  p-value: 6.316e-07</code></pre>
<pre class="r"><code>perturbValues &lt;- seq(-100, 100, 0.1)
peakPredict &lt;- predict(quad.peakheight,list(H_mM=perturbValues, H_mM_2=perturbValues^2))
df_quad &lt;- data.frame(H_mM = perturbValues, NO2_mM = peakPredict)
plot(perturbValues, peakPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-6-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak height (mM)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peak height \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  # annotate(&quot;text&quot;,x=0,y=30, label= paste0(&quot;y = &quot;, round(coef(fit.no2peak)[[1]],3),&quot;+&quot;,round(coef(fit.no2peak)[[2]],3),&quot;x&quot;), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-6-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak height (mM)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peak height \n&quot;) +
  geom_line(data = df_quad, aes(x = H_mM, y = NO2_mM), color = &quot;blue&quot;, size = 1) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=0,y=1, label= paste0(&quot;y = &quot;, round(coef(quad.peakheight)[[1]],3), &quot;+&quot;, round(coef(quad.peakheight)[[2]],3),&quot;x+&quot;,round(coef(quad.peakheight)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(quad.peakheight)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-6-5.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-on-the-nitrate-nitrite-time-readings." class="section level2">
<h2>4.1.1.2. PCA on the nitrate, nitrite time readings.</h2>
<ol style="list-style-type: decimal">
<li>First plot PCA with only the nitrate</li>
</ol>
<pre class="r"><code>## make the above values to matrices

mat_NO2 &lt;- dcast(df_plot2, H_mM ~ Time_hours, value.var = &quot;NO2_mM&quot;)
mat_NO2 &lt;- tibble::column_to_rownames(mat_NO2, var = &quot;H_mM&quot;)
mat_NO2 &lt;- as.matrix(mat_NO2) 
class(mat_NO2)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO2)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO2[is.na(mat_NO2)]</code></pre>
<pre><code>[1] NA NA NA NA</code></pre>
<pre class="r"><code>mat_NO2 &lt;- na_to_average(mat_NO2)
mat_NO2[is.na(mat_NO2)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>mat_NO3 &lt;- dcast(df_plot2, H_mM ~ Time_hours, value.var = &quot;NO3_mM&quot;)
mat_NO3 &lt;- tibble::column_to_rownames(mat_NO3, var = &quot;H_mM&quot;)
mat_NO3 &lt;- as.matrix(mat_NO3) 
class(mat_NO3)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO3)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO3[is.na(mat_NO3)]</code></pre>
<pre><code>[1] NA NA NA NA</code></pre>
<pre class="r"><code>mat_NO3 &lt;- na_to_average(mat_NO3)
mat_NO3[is.na(mat_NO3)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
any(is.na(mat_NO3))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO3)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.8659735</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.08607356</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                         PC1          PC2          PC3          PC4
-94.7069346522151 -0.4877701  0.987693657 -0.963985913  0.030791842
-89.4454382826476 -0.6103223  1.216632593 -0.921019411  0.023526255
-84.1839419130801 -0.8643860  1.402347012 -0.210040638 -0.003381064
-78.9224455435126 -0.7473050  1.448718316 -0.032633068 -0.106511084
-73.6609491739451 -0.6842638  1.430266785  0.134715500 -0.224006455
-68.3994528043776 -1.2007402  0.921215711  0.350990850  0.598310614
-63.1379564348101 -1.6310814  0.243546856  0.273245570  0.952211245
-57.8764600652426 -2.1103097 -0.768668897 -0.016298926 -0.210663315
-52.614963695675  -2.1392072 -0.651002451 -0.238705444 -0.095835618
-47.3534673261075 -2.1266924 -0.706578675 -0.148993076 -0.146340058
-42.09197095654   -1.9868530 -0.896186594  0.048680267 -0.199308429
-36.8304745869725 -2.1103024 -0.796623279  0.011431749 -0.246991865
-31.568978217405  -2.1267622 -0.726286763 -0.126109977 -0.163864401
-26.3074818478375 -1.9004841 -0.247124377  0.004213579  0.392896999
-21.04598547827   -2.0455416 -0.533771265 -0.115056969  0.057183589
-15.7844891087025 -2.1594387 -0.634337387 -0.328079202 -0.084005449
-10.522992739135  -2.1251492 -0.687626490 -0.160182150 -0.138582233
-5.2614963695675  -1.9761720 -0.479089715  0.032243388  0.126472555
0                 -1.9219104 -0.418367843  0.102691852  0.130953555
5.2614963695675   -1.9790604 -0.588279361  0.151621869  0.012923719
10.522992739135   -1.7362029 -0.142112546  0.256629759  0.222571936
15.7844891087025  -1.1927424  0.469910368  0.545416586 -0.024733641
21.04598547827    -0.9668901  0.740813591  0.614973981 -0.146309387
26.3074818478375  -0.6161101  0.936335930  0.829545571 -0.446826450
31.568978217405    0.1073815  1.043978566  0.044697062 -0.557966948
36.8304745869725   1.8986971 -0.018975738 -0.122427381 -0.156542821
42.09197095654     2.1644325 -0.025845369 -0.015826580 -0.042307225
47.3534673261075   2.4656246  0.080107925 -0.083774127  0.114381989
52.614963695675    2.7799687 -0.114619016 -0.219152903  0.215304874
57.8764600652426   2.8620512  0.002669176 -0.085023261  0.325114482
63.1379564348101   3.1221307 -0.026964350 -0.203629159 -0.036705484
68.3994528043776   3.2977690 -0.395454566  0.060316650 -0.172628391
73.6609491739451   3.7541158 -0.350055723  0.217898698  0.078096959
78.9224455435126   3.8045603 -0.361254608 -0.014325510  0.043442287
84.1839419130801   3.7251184 -0.387325017  0.178638082 -0.047292884
89.4454382826476   3.6237434 -0.488431400  0.176760106 -0.068204258
94.7069346522151   3.8401045 -0.479255055 -0.029447423 -0.005175442
                           PC5          PC6          PC7          PC8
-94.7069346522151  0.249078903 -0.071643705 -0.006976872  0.069115864
-89.4454382826476  0.194775301  0.103562632 -0.085903452  0.067178985
-84.1839419130801 -0.442696406  0.064483934 -0.003771303 -0.047971329
-78.9224455435126 -0.401511128  0.101653064  0.128605180  0.107382590
-73.6609491739451 -0.368854870  0.008339189  0.035302173  0.042661035
-68.3994528043776 -0.031574718 -0.482305113 -0.391219047  0.059054184
-63.1379564348101  0.283804546  0.397434434  0.035749944  0.135148784
-57.8764600652426 -0.044518527  0.119729746 -0.206307384 -0.021874549
-52.614963695675  -0.132154296  0.110042168 -0.224659032 -0.054957805
-47.3534673261075 -0.082471204  0.061507360 -0.150001553 -0.087374602
-42.09197095654   -0.123647532  0.115823729 -0.191224925  0.136166974
-36.8304745869725 -0.035858302  0.025126146 -0.031137174  0.105910503
-31.568978217405  -0.061211447 -0.025072107 -0.034824886 -0.082389445
-26.3074818478375  0.103810802  0.093704127  0.071907739 -0.090851768
-21.04598547827   -0.007417599 -0.021941170  0.069550673 -0.091485558
-15.7844891087025 -0.141446528 -0.169704094  0.174131343  0.038737714
-10.522992739135  -0.084576812 -0.127494133  0.185984379  0.154552181
-5.2614963695675   0.084963491 -0.013369331  0.069294090 -0.173850222
0                  0.088041896 -0.051191837  0.140998932  0.003069423
5.2614963695675    0.117897003 -0.094220538  0.276566008  0.053204039
10.522992739135    0.107615679 -0.110632437  0.004019975  0.006351534
15.7844891087025   0.068739591 -0.111388483  0.049303549 -0.061880609
21.04598547827     0.022786780 -0.085362370  0.069458297 -0.090168770
26.3074818478375   0.097652831  0.218095853 -0.004020723 -0.031818915
31.568978217405    0.680072352  0.003320227 -0.035983659 -0.058689571
36.8304745869725   0.075912888  0.018568968 -0.129899734 -0.177568495
42.09197095654     0.018609339 -0.130113250  0.010679805  0.167497334
47.3534673261075   0.046705465  0.019040384  0.098767856 -0.130357794
52.614963695675   -0.048684574 -0.083439990  0.017291829 -0.136586085
57.8764600652426  -0.236891782  0.108629988  0.118468256 -0.092024918
63.1379564348101   0.118202946 -0.018411294  0.191866703 -0.081854137
68.3994528043776   0.283977033 -0.200884228  0.058055739  0.159900746
73.6609491739451  -0.054532098  0.007754915 -0.076050951 -0.041751255
78.9224455435126  -0.108582075 -0.028331904 -0.099739051 -0.054540683
84.1839419130801  -0.339003151 -0.004750792  0.068375519  0.130929361
89.4454382826476   0.159600815  0.193550121 -0.118845942  0.246479352
94.7069346522151  -0.056614611  0.059889793 -0.083812302 -0.075344095
                            PC9         PC10          PC11
-94.7069346522151  0.1431280355  0.029220202 -0.0971058766
-89.4454382826476 -0.2408060607 -0.099193616  0.0408542792
-84.1839419130801 -0.0077972875  0.086029212 -0.0010923979
-78.9224455435126 -0.0040273512  0.042647866  0.0384850111
-73.6609491739451  0.0088616028  0.007143611 -0.0347124673
-68.3994528043776  0.0342041899 -0.010982253  0.0349512782
-63.1379564348101  0.0440027970 -0.020771741  0.0132583194
-57.8764600652426  0.0633342315 -0.019925053  0.0701104608
-52.614963695675   0.0679245915 -0.008822235  0.0074925943
-47.3534673261075  0.0511953659 -0.026935478 -0.0574429687
-42.09197095654   -0.0638295354  0.066277229 -0.0091870010
-36.8304745869725  0.0558039353 -0.075638571  0.0106176638
-31.568978217405  -0.0006835979  0.021614181  0.0347810818
-26.3074818478375 -0.0290255543  0.050722401  0.0054704413
-21.04598547827   -0.0038799680 -0.016667078 -0.1251433235
-15.7844891087025 -0.0348812340  0.022011647  0.0239512473
-10.522992739135  -0.0037767223 -0.039406829  0.0208770747
-5.2614963695675  -0.0655781073  0.094073494  0.0822049241
0                 -0.0209319049 -0.018064748 -0.0409443455
5.2614963695675   -0.0500100183 -0.015180801  0.0068739914
10.522992739135    0.0035585905 -0.039686457 -0.0575454593
15.7844891087025  -0.0322693009 -0.006156342 -0.0212712691
21.04598547827    -0.0452202593  0.013726224 -0.0009215644
26.3074818478375  -0.0083787569 -0.070116983 -0.0080828503
31.568978217405    0.0420144966 -0.036967701  0.0031419613
36.8304745869725   0.0140199518  0.158993712  0.0108517494
42.09197095654     0.1511504905 -0.077677896  0.0745890106
47.3534673261075   0.1801868689  0.049444746  0.0979482729
52.614963695675   -0.0353867807  0.014601006 -0.0430010015
57.8764600652426  -0.0544844272 -0.009675658  0.0150350935
63.1379564348101   0.1528647391 -0.057968652  0.0100999220
68.3994528043776  -0.1612153981  0.137934525  0.0216471628
73.6609491739451  -0.0250255455  0.011049602 -0.0971667248
78.9224455435126  -0.0838388818 -0.124935056 -0.0032589480
84.1839419130801   0.0693515050 -0.036655928 -0.0313798945
89.4454382826476   0.0085814010  0.149051304 -0.0295411661
94.7069346522151  -0.1191361007 -0.143111884  0.0345557185</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                         PC1       PC2         PC3          PC4         PC5
-94.7069346522151 -0.4877701 0.9876937 -0.96398591  0.030791842  0.24907890
-89.4454382826476 -0.6103223 1.2166326 -0.92101941  0.023526255  0.19477530
-84.1839419130801 -0.8643860 1.4023470 -0.21004064 -0.003381064 -0.44269641
-78.9224455435126 -0.7473050 1.4487183 -0.03263307 -0.106511084 -0.40151113
-73.6609491739451 -0.6842638 1.4302668  0.13471550 -0.224006455 -0.36885487
-68.3994528043776 -1.2007402 0.9212157  0.35099085  0.598310614 -0.03157472
                           PC6          PC7         PC8          PC9
-94.7069346522151 -0.071643705 -0.006976872  0.06911586  0.143128036
-89.4454382826476  0.103562632 -0.085903452  0.06717898 -0.240806061
-84.1839419130801  0.064483934 -0.003771303 -0.04797133 -0.007797288
-78.9224455435126  0.101653064  0.128605180  0.10738259 -0.004027351
-73.6609491739451  0.008339189  0.035302173  0.04266103  0.008861603
-68.3994528043776 -0.482305113 -0.391219047  0.05905418  0.034204190
                          PC10         PC11
-94.7069346522151  0.029220202 -0.097105877
-89.4454382826476 -0.099193616  0.040854279
-84.1839419130801  0.086029212 -0.001092398
-78.9224455435126  0.042647866  0.038485011
-73.6609491739451  0.007143611 -0.034712467
-68.3994528043776 -0.010982253  0.034951278</code></pre>
<pre class="r"><code>df_out$H_mM &lt;- rownames(df_out)
df_out$H_mM &lt;- as.numeric(df_out$H_mM)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrate dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-7-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(H_mM, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrate dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-7-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (col_pH(100)),
          main = &quot;Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-7-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>PCA with only nitrite dynamics</li>
</ol>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
any(is.na(mat_NO2))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO2)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.379731</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.2085221</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                          PC1          PC2           PC3          PC4
-94.7069346522151  0.30706698  1.906284117  0.1396445823 -0.098036512
-89.4454382826476  0.18490145  0.051265043 -0.0135384577  1.359810706
-84.1839419130801  0.16295414  0.029020863 -0.0008707845  0.086221218
-78.9224455435126  0.15542772 -0.030674858  0.0259843439  0.001702017
-73.6609491739451  0.15458380 -0.038587600  0.0231868062 -0.005393560
-68.3994528043776  0.17482521  0.008601835 -0.3219592826  0.056509041
-63.1379564348101  0.07336192  0.062997482 -1.6152818461 -0.077492017
-57.8764600652426 -1.05657385  0.034734603  0.1056340858  0.006310598
-52.614963695675  -1.45184458  0.067968923  0.1241648933  0.016621582
-47.3534673261075 -0.98762168  0.021664808  0.1063189359 -0.002471267
-42.09197095654    0.18208971 -0.061734654  0.0519428434 -0.032110025
-36.8304745869725  0.20972423  0.234363200  0.0768445667 -0.050678305
-31.568978217405  -1.04232320  0.017984576  0.1144111620 -0.005025727
-26.3074818478375 -0.58850008  0.021410575 -0.5366722423 -0.040921063
-21.04598547827   -0.62099985 -0.012044136  0.0325849161 -0.019061543
-15.7844891087025 -0.09932573 -0.058076434  0.0698508397 -0.026077158
-10.522992739135   0.14389030 -0.077952426  0.0625587365 -0.031983800
-5.2614963695675   0.19634761 -0.086862713  0.0730075688 -0.047323034
0                  0.20479748 -0.092096576  0.0767856192 -0.051417293
5.2614963695675    0.20631246 -0.104561881  0.0783419157 -0.056485654
10.522992739135    0.18010991 -0.107120471  0.0786509807 -0.058125785
15.7844891087025   0.17462253 -0.103540352  0.0471029618 -0.056451158
21.04598547827     0.20405403 -0.110759973  0.0706182195 -0.059068086
26.3074818478375   0.20794141 -0.112613488  0.0806598950 -0.057464102
31.568978217405    0.21072993 -0.113306113  0.0820721142 -0.057887067
36.8304745869725   0.20932715 -0.113414747  0.0822055488 -0.059508689
42.09197095654     0.21055188 -0.113881565  0.0802339949 -0.058998964
47.3534673261075   0.21055637 -0.111566369  0.0781899144 -0.058463025
52.614963695675    0.21074581 -0.112892904  0.0821664300 -0.059463299
57.8764600652426   0.20963972 -0.114154354  0.0818681721 -0.056674395
63.1379564348101   0.21012975 -0.113008033  0.0778038706 -0.057541382
68.3994528043776   0.20916404 -0.113680171  0.0816688277 -0.058424169
73.6609491739451   0.20928844 -0.112435353  0.0819134850 -0.056777415
78.9224455435126   0.20994983 -0.111452068  0.0800597597 -0.057988067
84.1839419130801   0.20841309 -0.111521456  0.0816604820 -0.055490680
89.4454382826476   0.20924682 -0.111032580  0.0793096400 -0.058592144
94.7069346522151   0.20643529 -0.107324754  0.0808765016 -0.055779775
                           PC5          PC6          PC7           PC8
-94.7069346522151  0.041256399 -0.041323029  0.007609868 -0.0314683948
-89.4454382826476  0.111114684 -0.077944776  0.009076176 -0.0006704661
-84.1839419130801 -0.226117775  0.967192845  0.017329626 -0.0009924796
-78.9224455435126 -0.014185687 -0.002320559 -0.007190761 -0.0073458522
-73.6609491739451 -0.017785161  0.003304401 -0.005595854 -0.0082170301
-68.3994528043776 -1.034668390 -0.240037246  0.005439979 -0.0008110285
-63.1379564348101  0.217947786  0.015849032  0.035847945 -0.0005465139
-57.8764600652426 -0.014125499 -0.002782306  0.068460722  0.0001258015
-52.614963695675  -0.024258962 -0.003620103  0.088570304  0.0031659379
-47.3534673261075 -0.008507822 -0.006822287  0.069396351 -0.0007663159
-42.09197095654    0.014749532 -0.012188366  0.002234533 -0.0035088238
-36.8304745869725  0.028191181 -0.021387036 -0.006165640  0.1914012254
-31.568978217405  -0.004240152 -0.008680276  0.019856981 -0.0016414516
-26.3074818478375  0.087024509 -0.003866018 -0.060547022  0.0008778572
-21.04598547827    0.016828550 -0.004319344 -0.197898131 -0.0026614419
-15.7844891087025  0.016235062 -0.008170577 -0.357930035 -0.0058946163
-10.522992739135   0.022875303 -0.004509159 -0.361969609 -0.0053986004
-5.2614963695675   0.035691803 -0.019794222 -0.078861452 -0.0134591677
0                  0.039546559 -0.023445930  0.013456631 -0.0083308447
5.2614963695675    0.041648388 -0.025144995  0.004582759 -0.0030043348
10.522992739135    0.042489160 -0.027952015 -0.008266164 -0.0074382211
15.7844891087025   0.025541934 -0.033428136  0.040220817 -0.0055795651
21.04598547827     0.031852884 -0.031975130  0.047502765 -0.0071591370
26.3074818478375   0.037738922 -0.025557223  0.046312432 -0.0063457502
31.568978217405    0.042026936 -0.029239319  0.049547124 -0.0082221357
36.8304745869725   0.042679254 -0.028399069  0.046246511 -0.0050751167
42.09197095654     0.042584149 -0.028127860  0.049488817 -0.0067797534
47.3534673261075   0.042957756 -0.026684555  0.047836137 -0.0036932523
52.614963695675    0.041892328 -0.027024326  0.049583021 -0.0061369739
57.8764600652426   0.041144631 -0.027572273  0.048727310 -0.0062786928
63.1379564348101   0.041434930 -0.027256976  0.048001820 -0.0069527764
68.3994528043776   0.040744079 -0.028590291  0.047774265 -0.0065219636
73.6609491739451   0.040774632 -0.028678918  0.046438754 -0.0043073271
78.9224455435126   0.041058390 -0.027144882  0.043692120 -0.0063247604
84.1839419130801   0.039236031 -0.028850193  0.043269185 -0.0051422038
89.4454382826476   0.039615400 -0.026529481  0.042823873 -0.0028060326
94.7069346522151   0.037008277 -0.026979402  0.045097842 -0.0060897977
                           PC9          PC10          PC11
-94.7069346522151  0.002831682 -4.121459e-04  5.420780e-04
-89.4454382826476  0.002657588 -8.091714e-04  5.102723e-04
-84.1839419130801  0.003274893 -1.070176e-03  5.574536e-04
-78.9224455435126 -0.034494851  1.354813e-02 -9.727671e-03
-73.6609491739451 -0.031622872  1.774938e-02 -3.945611e-03
-68.3994528043776  0.003450423 -1.487710e-03  7.823119e-04
-63.1379564348101 -0.003713209 -2.583856e-03 -5.969752e-04
-57.8764600652426 -0.007019251 -1.989450e-03  1.454124e-03
-52.614963695675  -0.003853832 -6.946613e-03 -2.064272e-03
-47.3534673261075 -0.006499229 -1.248424e-03  1.286890e-03
-42.09197095654   -0.020263438  1.111511e-02 -2.469457e-03
-36.8304745869725 -0.001083598 -7.601781e-04 -7.418136e-04
-31.568978217405   0.006699009  4.138404e-03 -2.697379e-03
-26.3074818478375  0.012800209  4.506240e-03  2.052393e-03
-21.04598547827    0.008114243  6.793649e-03  3.314527e-03
-15.7844891087025  0.005413110 -2.717448e-05  2.205310e-04
-10.522992739135   0.005301870 -3.443406e-03 -1.070944e-03
-5.2614963695675  -0.027085353 -2.236400e-02 -6.643257e-03
0                 -0.018603693 -2.174514e-02 -1.949859e-03
5.2614963695675   -0.026211168  4.692833e-04  2.676692e-02
10.522992739135    0.014662833  6.397135e-03 -1.058254e-03
15.7844891087025   0.010481944 -2.304588e-04 -1.144428e-03
21.04598547827     0.009010617 -2.860347e-04 -8.100530e-04
26.3074818478375   0.006553387 -2.116368e-03 -1.093902e-04
31.568978217405    0.008600107 -4.475774e-04 -2.869831e-04
36.8304745869725   0.007742878 -2.467443e-04 -6.324857e-04
42.09197095654     0.007966231  2.495152e-07 -1.051072e-03
47.3534673261075   0.006533495 -7.613651e-04 -1.821018e-03
52.614963695675    0.008359330 -1.797664e-04 -1.477938e-04
57.8764600652426   0.005917194 -8.076350e-04  2.027795e-03
63.1379564348101   0.008056641  1.296101e-03  2.210435e-04
68.3994528043776   0.008231734 -9.834468e-04  5.488525e-05
73.6609491739451   0.007034632 -4.424503e-05  2.567331e-04
78.9224455435126   0.003720948 -9.604183e-06  2.621907e-04
84.1839419130801   0.006030576  1.135338e-03  4.735829e-04
89.4454382826476   0.006544867  1.196263e-03 -1.338226e-03
94.7069346522151   0.004460056  2.655416e-03 -4.767926e-04</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                        PC1          PC2           PC3          PC4         PC5
-94.7069346522151 0.3070670  1.906284117  0.1396445823 -0.098036512  0.04125640
-89.4454382826476 0.1849015  0.051265043 -0.0135384577  1.359810706  0.11111468
-84.1839419130801 0.1629541  0.029020863 -0.0008707845  0.086221218 -0.22611778
-78.9224455435126 0.1554277 -0.030674858  0.0259843439  0.001702017 -0.01418569
-73.6609491739451 0.1545838 -0.038587600  0.0231868062 -0.005393560 -0.01778516
-68.3994528043776 0.1748252  0.008601835 -0.3219592826  0.056509041 -1.03466839
                           PC6          PC7           PC8          PC9
-94.7069346522151 -0.041323029  0.007609868 -0.0314683948  0.002831682
-89.4454382826476 -0.077944776  0.009076176 -0.0006704661  0.002657588
-84.1839419130801  0.967192845  0.017329626 -0.0009924796  0.003274893
-78.9224455435126 -0.002320559 -0.007190761 -0.0073458522 -0.034494851
-73.6609491739451  0.003304401 -0.005595854 -0.0082170301 -0.031622872
-68.3994528043776 -0.240037246  0.005439979 -0.0008110285  0.003450423
                           PC10          PC11
-94.7069346522151 -0.0004121459  0.0005420780
-89.4454382826476 -0.0008091714  0.0005102723
-84.1839419130801 -0.0010701760  0.0005574536
-78.9224455435126  0.0135481275 -0.0097276714
-73.6609491739451  0.0177493753 -0.0039456108
-68.3994528043776 -0.0014877097  0.0007823119</code></pre>
<pre class="r"><code>df_out$H_mM &lt;- rownames(df_out)
df_out$H_mM &lt;- as.numeric(df_out$H_mM)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrite dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-8-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(H_mM, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrite dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-8-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (col_pH(100)),
          main = &quot;Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-8-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>PCA with both nitrate and nitrite dynamics</li>
</ol>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
dim(mat_NO2)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>dim(mat_NO3)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO23 &lt;- cbind(mat_NO3, mat_NO2)
any(is.na(mat_NO23))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>dim(mat_NO23)</code></pre>
<pre><code>[1] 37 22</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO23)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.8038253</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.08763187</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                         PC1         PC2           PC3          PC4         PC5
-94.7069346522151 -0.4992033  1.22461003 -1.559679e+00  0.773841950  0.72121099
-89.4454382826476 -0.6042167  1.34358109 -8.825410e-01  0.157478694 -0.05963203
-84.1839419130801 -0.8532461  1.45557951 -1.814776e-01 -0.272806913 -0.24587812
-78.9224455435126 -0.7267533  1.41449549 -1.181899e-02 -0.327845112 -0.16433552
-73.6609491739451 -0.6638339  1.39185370  8.333104e-02 -0.438601225 -0.16193342
-68.3994528043776 -1.1957537  1.02399917  5.045083e-01  0.306397050 -0.35171953
-63.1379564348101 -1.6545465  0.38491630  7.932897e-01  1.555016386 -0.66691862
-57.8764600652426 -2.1911087 -0.98861571 -3.652997e-01 -0.303645161 -0.46680871
-52.614963695675  -2.2544913 -0.98528774 -6.780421e-01 -0.308345248 -0.80359201
-47.3534673261075 -2.2008567 -0.91139792 -4.154737e-01 -0.246671456 -0.41886912
-42.09197095654   -1.9598897 -0.76762460  2.127180e-01 -0.022746147  0.49023369
-36.8304745869725 -2.0870000 -0.63577278  4.509045e-02  0.049800200  0.64226052
-31.568978217405  -2.2060661 -0.94703794 -4.114083e-01 -0.275457235 -0.41688012
-26.3074818478375 -1.9567286 -0.33175634  6.638058e-02  0.485606226 -0.52018956
-21.04598547827   -2.0928230 -0.64717647 -1.872324e-01 -0.028150004 -0.17915353
-15.7844891087025 -2.1615480 -0.60013028 -1.260128e-01  0.009946359  0.36229368
-10.522992739135  -2.1058599 -0.58379902  8.688799e-02  0.019157501  0.55976379
-5.2614963695675  -1.9470734 -0.37377047  2.566097e-01  0.106699300  0.39898671
0                 -1.8902453 -0.31335983  3.101230e-01  0.093283049  0.40011299
5.2614963695675   -1.9469092 -0.47393078  3.550578e-01  0.042308055  0.49787793
10.522992739135   -1.7067824 -0.06783606  3.994217e-01  0.057361885  0.25560907
15.7844891087025  -1.1652824  0.49320010  5.143014e-01 -0.238198944  0.07708980
21.04598547827    -0.9364429  0.74627194  5.300085e-01 -0.384655537  0.03996884
26.3074818478375  -0.5862569  0.91670005  6.017218e-01 -0.612950741 -0.01914256
31.568978217405    0.1340640  1.01109298  1.944531e-02 -0.430051782  0.10860959
36.8304745869725   1.9147282 -0.02712814 -6.829243e-02 -0.094880004  0.06229914
42.09197095654     2.1794903 -0.03388802  2.395823e-02 -0.038989104  0.06229729
47.3534673261075   2.4791860  0.06004411 -1.949296e-02  0.051793182 -0.03828107
52.614963695675    2.7918153 -0.12819771 -1.020998e-01  0.144291881 -0.02598254
57.8764600652426   2.8736348 -0.02057743  1.087546e-06  0.143708579 -0.13023788
63.1379564348101   3.1324220 -0.05414602 -1.382747e-01  0.024629251  0.01867606
68.3994528043776   3.3069285 -0.40391794  4.212447e-02 -0.034391351  0.14214305
73.6609491739451   3.7610387 -0.37474025  1.434285e-01  0.014412302 -0.07796786
78.9224455435126   3.8110238 -0.38586065 -2.144749e-02  0.035884061 -0.05131171
84.1839419130801   3.7319488 -0.41128598  1.061037e-01 -0.077865510 -0.03403964
89.4454382826476   3.6308720 -0.50085595  1.105935e-01  0.023819324  0.01458136
94.7069346522151   3.8457653 -0.49825046 -3.651211e-02  0.040816242 -0.02114097
                            PC6          PC7           PC8          PC9
-94.7069346522151 -8.328154e-01  0.049029667 -0.0171307914 -0.038414743
-89.4454382826476  1.227015e+00  0.254350390 -0.3525740163 -0.207020519
-84.1839419130801  1.289358e-01 -0.488381816  0.6457795623 -0.016643934
-78.9224455435126  2.413284e-02 -0.131699922  0.2910483581 -0.009221755
-73.6609491739451 -6.715483e-02 -0.088798501  0.2088622874 -0.053845169
-68.3994528043776 -2.235905e-01 -0.809327805 -0.8650825467 -0.048744675
-63.1379564348101  1.901911e-05  0.400661793  0.2712491189 -0.097125415
-57.8764600652426 -1.537407e-01  0.100862199 -0.0695015594 -0.141662430
-52.614963695675  -1.575625e-01  0.012087591 -0.0516966657 -0.034525235
-47.3534673261075 -9.072577e-02  0.023477362 -0.0356251107 -0.035425622
-42.09197095654    1.115378e-01 -0.052992716  0.0597972077 -0.363867491
-36.8304745869725 -6.879346e-02  0.007940295  0.0467612768 -0.291079464
-31.568978217405  -8.944011e-02  0.026749136 -0.0424315855  0.059114023
-26.3074818478375 -2.307647e-02  0.128055034  0.0772757727  0.159099100
-21.04598547827    1.704743e-02 -0.036031588  0.0171784520  0.177308795
-15.7844891087025  2.417076e-01 -0.211638206  0.1210247136  0.181760470
-10.522992739135   2.288429e-01 -0.143949538  0.1168248079  0.064782498
-5.2614963695675   1.368138e-01 -0.049942493 -0.0009280291  0.138253353
0                  9.608877e-02 -0.032851296 -0.0078312392  0.090456491
5.2614963695675    9.890144e-02  0.022345132  0.0052260445  0.106560698
10.522992739135    1.007528e-02 -0.049832690 -0.0972908425  0.079162929
15.7844891087025  -1.565612e-01  0.090603030 -0.0857740158  0.037075973
21.04598547827    -1.964527e-01  0.135647162 -0.0443161632  0.019930189
26.3074818478375  -3.080821e-01  0.421576135  0.0003561799 -0.211895097
31.568978217405   -9.196525e-02  0.661250911 -0.2753329389  0.261348276
36.8304745869725   3.366991e-02  0.080366450 -0.0293352893  0.064186691
42.09197095654    -3.799657e-03 -0.015232169 -0.0382829278  0.027315122
47.3534673261075   2.009838e-02 -0.003435627  0.0345916085  0.207140595
52.614963695675    1.056914e-01 -0.159546816  0.0146056138  0.185180918
57.8764600652426   9.198805e-02 -0.221205304  0.1777293311  0.082046672
63.1379564348101   5.313295e-02  0.077398923  0.0262384047  0.245653606
68.3994528043776  -1.511091e-02  0.172105236 -0.1857281933  0.060280720
73.6609491739451  -7.984637e-02 -0.043434581 -0.0233189843 -0.098109681
78.9224455435126   2.246429e-02 -0.102825775 -0.0020118445 -0.068935706
84.1839419130801  -5.759856e-02 -0.171612205  0.1599634475 -0.203398336
89.4454382826476  -7.188755e-02  0.180029463 -0.0563795908 -0.235129585
94.7069346522151   4.004081e-02 -0.031796865  0.0060601471 -0.091612261
                          PC10          PC11        PC12         PC13
-94.7069346522151 -0.063139774  0.0572408125 -0.04140560 -0.022142260
-89.4454382826476 -0.088919108  0.0895484500 -0.06592186  0.006748642
-84.1839419130801  0.521023664  0.1348847079 -0.10814436  0.016619489
-78.9224455435126 -0.260256485 -0.1877184771  0.09102783  0.026935853
-73.6609491739451 -0.217144306 -0.1656957889  0.06588249 -0.040409010
-68.3994528043776  0.105014110 -0.0965765014  0.02912862  0.004550228
-63.1379564348101  0.045294078 -0.0974372704  0.02637484  0.012831827
-57.8764600652426  0.047425558  0.0124581765  0.02182354  0.052090264
-52.614963695675  -0.018737891  0.0146342344  0.02190417  0.053603950
-47.3534673261075  0.012416582  0.0343786481  0.05003541  0.024637346
-42.09197095654    0.119114301 -0.0234883002  0.17324823 -0.035527784
-36.8304745869725  0.058696820 -0.0337282029  0.07603981  0.082354110
-31.568978217405  -0.012107582 -0.0090402332 -0.05630522 -0.008826081
-26.3074818478375 -0.035448828  0.0161984233 -0.05559287 -0.079508889
-21.04598547827   -0.062504324  0.0003236607 -0.08117216 -0.091181635
-15.7844891087025 -0.041093500 -0.2440143209 -0.02974887 -0.118529668
-10.522992739135  -0.014583087 -0.2718693424 -0.06738069 -0.050336168
-5.2614963695675   0.020125074  0.1915295125  0.11920979 -0.061786623
0                 -0.029577524  0.1397385543 -0.01029286  0.090682894
5.2614963695675   -0.028062558  0.0631315722 -0.11000420  0.105355478
10.522992739135   -0.046748046  0.1868935613 -0.03825310  0.069309575
15.7844891087025  -0.098145607  0.1479881015 -0.09836434 -0.016805871
21.04598547827    -0.126182699  0.1242424537 -0.08796313 -0.052590951
26.3074818478375  -0.073091466  0.0672305096 -0.03007858 -0.043909878
31.568978217405    0.296278083 -0.1463151432  0.10873299  0.005061007
36.8304745869725   0.131705833  0.0196511545  0.22661986 -0.111701369
42.09197095654     0.005974522 -0.1560625512 -0.04071797  0.189916029
47.3534673261075  -0.024000152  0.0467492022  0.11019425  0.129529688
52.614963695675   -0.038319446  0.0945977027  0.08072624 -0.037237726
57.8764600652426  -0.197983277  0.1426980814  0.06142454  0.009405457
63.1379564348101   0.010617469 -0.0704204755  0.04091323  0.135562282
68.3994528043776   0.165560869 -0.1251809491 -0.23425148 -0.082605774
73.6609491739451  -0.031301554  0.1106948105 -0.05418369 -0.075223693
78.9224455435126  -0.023656655  0.0402265018  0.02596430 -0.079356785
84.1839419130801  -0.157463779 -0.1005721998 -0.10779045  0.059210045
89.4454382826476   0.139912837 -0.0592716497 -0.06670704  0.025377613
94.7069346522151   0.009307847  0.0523525745  0.05502831 -0.092101610
                          PC14          PC15         PC16          PC17
-94.7069346522151 -0.009766361  9.666016e-03 -0.002000591  0.0008609207
-89.4454382826476  0.016678306  8.729520e-03  0.017471985  0.0154580084
-84.1839419130801  0.028315983  8.238651e-03  0.003500728  0.0006884242
-78.9224455435126 -0.126402186 -2.299080e-02 -0.008218579 -0.0693552248
-73.6609491739451 -0.027489657 -6.092106e-02 -0.029682259  0.0284538972
-68.3994528043776  0.016877557  2.721103e-02  0.027012782 -0.0238669121
-63.1379564348101  0.024298505 -2.447045e-02 -0.009078655  0.0075465502
-57.8764600652426  0.007615347  8.717673e-03  0.028062894 -0.0057199122
-52.614963695675  -0.039708706 -6.411429e-03 -0.038372746 -0.0085873725
-47.3534673261075  0.021763473 -2.571451e-02 -0.028992067  0.0095874278
-42.09197095654   -0.087495692 -3.790451e-02 -0.029431287  0.0019714793
-36.8304745869725  0.118230637 -4.133017e-02  0.053050594  0.0062946597
-31.568978217405  -0.001071562 -5.467246e-02  0.066935445 -0.0124307785
-26.3074818478375 -0.013957491  2.467845e-03  0.044566264 -0.0006769268
-21.04598547827    0.038120362  1.107570e-01 -0.063312428  0.0173457081
-15.7844891087025  0.025685007  3.751156e-02  0.008043804  0.0255971760
-10.522992739135   0.057186353  8.227256e-02 -0.021807851  0.0092978522
-5.2614963695675  -0.059143475  2.544343e-02  0.075512046 -0.0087073384
0                 -0.039840643 -1.436782e-02 -0.045832588 -0.0405297059
5.2614963695675   -0.022065499 -4.870981e-02  0.047540639 -0.0927169082
10.522992739135   -0.037481378  4.263595e-03 -0.148698185  0.0544247048
15.7844891087025   0.009626345 -4.192710e-02 -0.005328323  0.0559166801
21.04598547827     0.015300546 -4.852956e-02  0.045464565  0.0501053598
26.3074818478375   0.086264869  1.166274e-01  0.042688884 -0.0405628387
31.568978217405    0.045122470  3.288795e-05 -0.072871056 -0.0423380603
36.8304745869725  -0.064136422 -2.882331e-02  0.062503224  0.0925986928
42.09197095654     0.021591344 -5.067319e-02 -0.061085428  0.0672360932
47.3534673261075  -0.011980005  7.183885e-02  0.096058703  0.0609150959
52.614963695675   -0.017181080 -5.812509e-02 -0.050637387  0.0140386786
57.8764600652426  -0.047719611  4.252577e-02 -0.001689337 -0.0985084731
63.1379564348101   0.089234247  3.591125e-02  0.056277594 -0.0131721772
68.3994528043776  -0.129887413 -1.294940e-01  0.048159928 -0.0313924958
73.6609491739451   0.015756333  3.874380e-02 -0.027057885  0.0599612616
78.9224455435126   0.117951979 -5.822079e-02 -0.067281693 -0.0251089718
84.1839419130801   0.046416018 -1.459141e-03  0.047006066  0.0488045022
89.4454382826476  -0.203727909  1.504301e-01 -0.028540363 -0.0121291457
94.7069346522151   0.137019408 -2.664368e-02 -0.029937436 -0.1012999309
                           PC18          PC19          PC20          PC21
-94.7069346522151  0.0086072913 -0.0170846500 -0.0007263724 -6.548803e-05
-89.4454382826476 -0.0011570940 -0.0010910700 -0.0003958548 -1.207613e-04
-84.1839419130801  0.0018598951 -0.0004523609 -0.0004959438 -2.556783e-04
-78.9224455435126  0.0072105567  0.0199197204  0.0017616575 -4.389731e-04
-73.6609491739451 -0.0335533558 -0.0026143110  0.0103684714  8.546471e-03
-68.3994528043776 -0.0112294164 -0.0020461288 -0.0004483077 -2.747759e-04
-63.1379564348101  0.0011794863 -0.0105027947  0.0029241838 -1.598957e-03
-57.8764600652426  0.0628285004 -0.0130416625  0.0033617667  5.422505e-03
-52.614963695675   0.0196401420  0.0138153040  0.0064426022 -4.710545e-03
-47.3534673261075 -0.0590592575 -0.0235947340  0.0021710281 -2.722035e-03
-42.09197095654   -0.0149657442 -0.0517223070 -0.0132891826  2.150231e-03
-36.8304745869725 -0.0404574709  0.1079714326  0.0017707703 -2.913390e-04
-31.568978217405   0.0055636266 -0.0060177776 -0.0103955824 -2.689026e-03
-26.3074818478375  0.0009980187  0.0238587630 -0.0082830544  3.119888e-03
-21.04598547827   -0.0710520845  0.0141701191  0.0029076585  6.838688e-03
-15.7844891087025  0.0301518653  0.0017882094 -0.0025307743 -1.650730e-03
-10.522992739135   0.0394077830 -0.0125413631 -0.0013300870 -4.933013e-03
-5.2614963695675   0.0635841112 -0.0075795864  0.0213515792 -1.403878e-03
0                 -0.0390138533 -0.0205139849  0.0122279016 -1.458940e-02
5.2614963695675   -0.0439455827 -0.0317668837  0.0002704584  1.605734e-02
10.522992739135    0.0273602267  0.0151585334 -0.0122244433  3.206487e-03
15.7844891087025   0.0174912027  0.0052160175 -0.0035651019 -1.513949e-03
21.04598547827     0.0189426058  0.0017565581 -0.0041815342 -2.581990e-03
26.3074818478375   0.0118430998 -0.0153922762 -0.0036141142 -4.093085e-03
31.568978217405   -0.0161728684  0.0050370569  0.0002340283 -1.098821e-03
36.8304745869725  -0.0262641019 -0.0067815303 -0.0056786944 -1.049585e-03
42.09197095654     0.0809355088 -0.0053286868  0.0015198957  8.569138e-04
47.3534673261075   0.0426975218  0.0053753951 -0.0009465356  6.343744e-03
52.614963695675   -0.0335359855  0.0149738137 -0.0029903873 -3.032425e-03
57.8764600652426   0.0098739429  0.0270684662 -0.0129185638 -5.094109e-03
63.1379564348101  -0.0615382602 -0.0241337997 -0.0035866428 -1.909456e-03
68.3994528043776   0.0029839708  0.0183011733  0.0025188510 -2.476999e-03
73.6609491739451  -0.0463615105  0.0003294328  0.0117182603  4.139161e-03
78.9224455435126   0.0376612241 -0.0034384796  0.0065905780  3.086871e-03
84.1839419130801  -0.0410800457 -0.0344592660 -0.0001594900 -6.470265e-03
89.4454382826476  -0.0025461254  0.0226798592  0.0013566982  1.921265e-03
94.7069346522151   0.0511121772 -0.0073162016 -0.0017357223  3.375717e-03
                           PC22
-94.7069346522151 -0.0002212823
-89.4454382826476 -0.0001598275
-84.1839419130801 -0.0001881770
-78.9224455435126  0.0032371898
-73.6609491739451  0.0012111725
-68.3994528043776 -0.0002423063
-63.1379564348101 -0.0005745341
-57.8764600652426 -0.0008879745
-52.614963695675  -0.0043844344
-47.3534673261075 -0.0006622194
-42.09197095654    0.0043155225
-36.8304745869725  0.0002867665
-31.568978217405   0.0054508796
-26.3074818478375  0.0008565065
-21.04598547827    0.0028089540
-15.7844891087025 -0.0031428337
-10.522992739135  -0.0014513786
-5.2614963695675   0.0033558833
0                  0.0002962713
5.2614963695675   -0.0074302904
10.522992739135    0.0033392977
15.7844891087025   0.0007130555
21.04598547827    -0.0011024537
26.3074818478375  -0.0019822036
31.568978217405   -0.0009231903
36.8304745869725  -0.0064532303
42.09197095654    -0.0021393465
47.3534673261075   0.0024502498
52.614963695675   -0.0024320982
57.8764600652426  -0.0043724514
63.1379564348101   0.0061304039
68.3994528043776   0.0027131276
73.6609491739451   0.0007925574
78.9224455435126  -0.0009706162
84.1839419130801  -0.0012188114
89.4454382826476  -0.0002708559
94.7069346522151   0.0032526777</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                         PC1      PC2         PC3        PC4         PC5
-94.7069346522151 -0.4992033 1.224610 -1.55967877  0.7738419  0.72121099
-89.4454382826476 -0.6042167 1.343581 -0.88254100  0.1574787 -0.05963203
-84.1839419130801 -0.8532461 1.455580 -0.18147763 -0.2728069 -0.24587812
-78.9224455435126 -0.7267533 1.414495 -0.01181899 -0.3278451 -0.16433552
-73.6609491739451 -0.6638339 1.391854  0.08333104 -0.4386012 -0.16193342
-68.3994528043776 -1.1957537 1.023999  0.50450829  0.3063970 -0.35171953
                          PC6         PC7         PC8          PC9        PC10
-94.7069346522151 -0.83281538  0.04902967 -0.01713079 -0.038414743 -0.06313977
-89.4454382826476  1.22701550  0.25435039 -0.35257402 -0.207020519 -0.08891911
-84.1839419130801  0.12893585 -0.48838182  0.64577956 -0.016643934  0.52102366
-78.9224455435126  0.02413284 -0.13169992  0.29104836 -0.009221755 -0.26025649
-73.6609491739451 -0.06715483 -0.08879850  0.20886229 -0.053845169 -0.21714431
-68.3994528043776 -0.22359046 -0.80932781 -0.86508255 -0.048744675  0.10501411
                         PC11        PC12         PC13         PC14
-94.7069346522151  0.05724081 -0.04140560 -0.022142260 -0.009766361
-89.4454382826476  0.08954845 -0.06592186  0.006748642  0.016678306
-84.1839419130801  0.13488471 -0.10814436  0.016619489  0.028315983
-78.9224455435126 -0.18771848  0.09102783  0.026935853 -0.126402186
-73.6609491739451 -0.16569579  0.06588249 -0.040409010 -0.027489657
-68.3994528043776 -0.09657650  0.02912862  0.004550228  0.016877557
                          PC15         PC16          PC17         PC18
-94.7069346522151  0.009666016 -0.002000591  0.0008609207  0.008607291
-89.4454382826476  0.008729520  0.017471985  0.0154580084 -0.001157094
-84.1839419130801  0.008238651  0.003500728  0.0006884242  0.001859895
-78.9224455435126 -0.022990802 -0.008218579 -0.0693552248  0.007210557
-73.6609491739451 -0.060921059 -0.029682259  0.0284538972 -0.033553356
-68.3994528043776  0.027211029  0.027012782 -0.0238669121 -0.011229416
                           PC19          PC20          PC21          PC22
-94.7069346522151 -0.0170846500 -0.0007263724 -6.548803e-05 -0.0002212823
-89.4454382826476 -0.0010910700 -0.0003958548 -1.207613e-04 -0.0001598275
-84.1839419130801 -0.0004523609 -0.0004959438 -2.556783e-04 -0.0001881770
-78.9224455435126  0.0199197204  0.0017616575 -4.389731e-04  0.0032371898
-73.6609491739451 -0.0026143110  0.0103684714  8.546471e-03  0.0012111725
-68.3994528043776 -0.0020461288 -0.0004483077 -2.747759e-04 -0.0002423063</code></pre>
<pre class="r"><code>df_out$H_mM &lt;- rownames(df_out)
df_out$H_mM &lt;- as.numeric(df_out$H_mM)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrite + Nitrate dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-9-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(H_mM, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = H_mM))+
  scale_colour_gradientn(colours = col_pH(100)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis (Nitrite + Nitrate dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-9-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (col_pH(100)),
          main = &quot;Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-9-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
</div>
<div id="using-inferred-ph-pca-on-the-nitrate-nitrite-time-readings." class="section level2">
<h2>4.1.1.3. (using inferred pH) PCA on the nitrate, nitrite time readings.</h2>
<ol style="list-style-type: decimal">
<li>(using inferred pH) First plot PCA with only the nitrate</li>
</ol>
<pre class="r"><code>## make the above values to matrices
df_plot2$pH &lt;- (6.5 / (1+ exp(1*(0.07*(df_plot2$H_mM+125))-10))) + 2.5
mat_NO2 &lt;- dcast(df_plot2, pH ~ Time_hours, value.var = &quot;NO2_mM&quot;)
mat_NO2 &lt;- tibble::column_to_rownames(mat_NO2, var = &quot;pH&quot;)
mat_NO2 &lt;- as.matrix(mat_NO2) 
class(mat_NO2)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO2)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO2[is.na(mat_NO2)]</code></pre>
<pre><code>[1] NA NA NA NA</code></pre>
<pre class="r"><code>mat_NO2 &lt;- na_to_average(mat_NO2)
mat_NO2[is.na(mat_NO2)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>mat_NO3 &lt;- dcast(df_plot2, pH ~ Time_hours, value.var = &quot;NO3_mM&quot;)
mat_NO3 &lt;- tibble::column_to_rownames(mat_NO3, var = &quot;pH&quot;)
mat_NO3 &lt;- as.matrix(mat_NO3) 
class(mat_NO3)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO3)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO3[is.na(mat_NO3)]</code></pre>
<pre><code>[1] NA NA NA NA</code></pre>
<pre class="r"><code>mat_NO3 &lt;- na_to_average(mat_NO3)
mat_NO3[is.na(mat_NO3)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
any(is.na(mat_NO3))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO3)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.8659735</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.08607356</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                        PC1          PC2          PC3          PC4          PC5
2.52982874216965 -3.8401045  0.479255055  0.029447423  0.005175442  0.056614611
2.54302304249866 -3.6237434  0.488431400 -0.176760106  0.068204258 -0.159600815
2.56199772134572 -3.7251184  0.387325017 -0.178638082  0.047292884  0.339003151
2.58922526312282 -3.8045603  0.361254608  0.014325510 -0.043442287  0.108582075
2.62817226680648 -3.7541158  0.350055723 -0.217898698 -0.078096959  0.054532098
2.68363274717552 -3.2977690  0.395454566 -0.060316650  0.172628391 -0.283977033
2.7621039756057  -3.1221307  0.026964350  0.203629159  0.036705484 -0.118202946
2.87213246874345 -2.8620512 -0.002669176  0.085023261 -0.325114482  0.236891782
3.02446632998257 -2.7799687  0.114619016  0.219152903 -0.215304874  0.048684574
3.23171258546232 -2.4656246 -0.080107925  0.083774127 -0.114381989 -0.046705465
3.5070518892221  -2.1644325  0.025845369  0.015826580  0.042307225 -0.018609339
3.86154407990938 -1.8986971  0.018975738  0.122427381  0.156542821 -0.075912888
4.29993149900709 -0.1073815 -1.043978566 -0.044697062  0.557966948 -0.680072352
4.81585455700761  0.6161101 -0.936335930 -0.829545571  0.446826450 -0.097652831
5.38876782542826  0.9668901 -0.740813591 -0.614973981  0.146309387 -0.022786780
5.98535166498929  1.1927424 -0.469910368 -0.545416586  0.024733641 -0.068739591
6.56640618671601  1.7362029  0.142112546 -0.256629759 -0.222571936 -0.107615679
7.09662707522223  1.9790604  0.588279361 -0.151621869 -0.012923719 -0.117897003
7.55244909763549  1.9219104  0.418367843 -0.102691852 -0.130953555 -0.088041896
7.92464780541878  1.9761720  0.479089715 -0.032243388 -0.126472555 -0.084963491
8.2159958447781   2.1251492  0.687626490  0.160182150  0.138582233  0.084576812
8.43660629576058  2.1594387  0.634337387  0.328079202  0.084005449  0.141446528
8.59948897711833  2.0455416  0.533771265  0.115056969 -0.057183589  0.007417599
8.71752137494786  1.9004841  0.247124377 -0.004213579 -0.392896999 -0.103810802
8.80189884276433  2.1267622  0.726286763  0.126109977  0.163864401  0.061211447
8.86163332665687  2.1103024  0.796623279 -0.011431749  0.246991865  0.035858302
8.90363119822312  1.9868530  0.896186594 -0.048680267  0.199308429  0.123647532
8.93301585011944  2.1266924  0.706578675  0.148993076  0.146340058  0.082471204
8.95350562621574  2.1392072  0.651002451  0.238705444  0.095835618  0.132154296
8.96775919824282  2.1103097  0.768668897  0.016298926  0.210663315  0.044518527
8.97765824405471  1.6310814 -0.243546856 -0.273245570 -0.952211245 -0.283804546
8.98452521098818  1.2007402 -0.921215711 -0.350990850 -0.598310614  0.031574718
8.9892850367663   0.6842638 -1.430266785 -0.134715500  0.224006455  0.368854870
8.99258248229834  0.7473050 -1.448718316  0.032633068  0.106511084  0.401511128
8.99486596777732  0.8643860 -1.402347012  0.210040638  0.003381064  0.442696406
8.9964468661431   0.6103223 -1.216632593  0.921019411 -0.023526255 -0.194775301
8.99754115021365  0.4877701 -0.987693657  0.963985913 -0.030791842 -0.249078903
                          PC6          PC7          PC8           PC9
2.52982874216965 -0.059889793  0.083812302  0.075344095  0.1191361007
2.54302304249866 -0.193550121  0.118845942 -0.246479352 -0.0085814010
2.56199772134572  0.004750792 -0.068375519 -0.130929361 -0.0693515050
2.58922526312282  0.028331904  0.099739051  0.054540683  0.0838388818
2.62817226680648 -0.007754915  0.076050951  0.041751255  0.0250255455
2.68363274717552  0.200884228 -0.058055739 -0.159900746  0.1612153981
2.7621039756057   0.018411294 -0.191866703  0.081854137 -0.1528647391
2.87213246874345 -0.108629988 -0.118468256  0.092024918  0.0544844272
3.02446632998257  0.083439990 -0.017291829  0.136586085  0.0353867807
3.23171258546232 -0.019040384 -0.098767856  0.130357794 -0.1801868689
3.5070518892221   0.130113250 -0.010679805 -0.167497334 -0.1511504905
3.86154407990938 -0.018568968  0.129899734  0.177568495 -0.0140199518
4.29993149900709 -0.003320227  0.035983659  0.058689571 -0.0420144966
4.81585455700761 -0.218095853  0.004020723  0.031818915  0.0083787569
5.38876782542826  0.085362370 -0.069458297  0.090168770  0.0452202593
5.98535166498929  0.111388483 -0.049303549  0.061880609  0.0322693009
6.56640618671601  0.110632437 -0.004019975 -0.006351534 -0.0035585905
7.09662707522223  0.094220538 -0.276566008 -0.053204039  0.0500100183
7.55244909763549  0.051191837 -0.140998932 -0.003069423  0.0209319049
7.92464780541878  0.013369331 -0.069294090  0.173850222  0.0655781073
8.2159958447781   0.127494133 -0.185984379 -0.154552181  0.0037767223
8.43660629576058  0.169704094 -0.174131343 -0.038737714  0.0348812340
8.59948897711833  0.021941170 -0.069550673  0.091485558  0.0038799680
8.71752137494786 -0.093704127 -0.071907739  0.090851768  0.0290255543
8.80189884276433  0.025072107  0.034824886  0.082389445  0.0006835979
8.86163332665687 -0.025126146  0.031137174 -0.105910503 -0.0558039353
8.90363119822312 -0.115823729  0.191224925 -0.136166974  0.0638295354
8.93301585011944 -0.061507360  0.150001553  0.087374602 -0.0511953659
8.95350562621574 -0.110042168  0.224659032  0.054957805 -0.0679245915
8.96775919824282 -0.119729746  0.206307384  0.021874549 -0.0633342315
8.97765824405471 -0.397434434 -0.035749944 -0.135148784 -0.0440027970
8.98452521098818  0.482305113  0.391219047 -0.059054184 -0.0342041899
8.9892850367663  -0.008339189 -0.035302173 -0.042661035 -0.0088616028
8.99258248229834 -0.101653064 -0.128605180 -0.107382590  0.0040273512
8.99486596777732 -0.064483934  0.003771303  0.047971329  0.0077972875
8.9964468661431  -0.103562632  0.085903452 -0.067178985  0.2408060607
8.99754115021365  0.071643705  0.006976872 -0.069115864 -0.1431280355
                         PC10          PC11
2.52982874216965  0.143111884  0.0345557185
2.54302304249866 -0.149051304 -0.0295411661
2.56199772134572  0.036655928 -0.0313798945
2.58922526312282  0.124935056 -0.0032589480
2.62817226680648 -0.011049602 -0.0971667248
2.68363274717552 -0.137934525  0.0216471628
2.7621039756057   0.057968652  0.0100999220
2.87213246874345  0.009675658  0.0150350935
3.02446632998257 -0.014601006 -0.0430010015
3.23171258546232 -0.049444746  0.0979482729
3.5070518892221   0.077677896  0.0745890106
3.86154407990938 -0.158993712  0.0108517494
4.29993149900709  0.036967701  0.0031419613
4.81585455700761  0.070116983 -0.0080828503
5.38876782542826 -0.013726224 -0.0009215644
5.98535166498929  0.006156342 -0.0212712691
6.56640618671601  0.039686457 -0.0575454593
7.09662707522223  0.015180801  0.0068739914
7.55244909763549  0.018064748 -0.0409443455
7.92464780541878 -0.094073494  0.0822049241
8.2159958447781   0.039406829  0.0208770747
8.43660629576058 -0.022011647  0.0239512473
8.59948897711833  0.016667078 -0.1251433235
8.71752137494786 -0.050722401  0.0054704413
8.80189884276433 -0.021614181  0.0347810818
8.86163332665687  0.075638571  0.0106176638
8.90363119822312 -0.066277229 -0.0091870010
8.93301585011944  0.026935478 -0.0574429687
8.95350562621574  0.008822235  0.0074925943
8.96775919824282  0.019925053  0.0701104608
8.97765824405471  0.020771741  0.0132583194
8.98452521098818  0.010982253  0.0349512782
8.9892850367663  -0.007143611 -0.0347124673
8.99258248229834 -0.042647866  0.0384850111
8.99486596777732 -0.086029212 -0.0010923979
8.9964468661431   0.099193616  0.0408542792
8.99754115021365 -0.029220202 -0.0971058766</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                       PC1       PC2         PC3          PC4         PC5
2.52982874216965 -3.840105 0.4792551  0.02944742  0.005175442  0.05661461
2.54302304249866 -3.623743 0.4884314 -0.17676011  0.068204258 -0.15960082
2.56199772134572 -3.725118 0.3873250 -0.17863808  0.047292884  0.33900315
2.58922526312282 -3.804560 0.3612546  0.01432551 -0.043442287  0.10858207
2.62817226680648 -3.754116 0.3500557 -0.21789870 -0.078096959  0.05453210
2.68363274717552 -3.297769 0.3954546 -0.06031665  0.172628391 -0.28397703
                          PC6         PC7         PC8          PC9        PC10
2.52982874216965 -0.059889793  0.08381230  0.07534409  0.119136101  0.14311188
2.54302304249866 -0.193550121  0.11884594 -0.24647935 -0.008581401 -0.14905130
2.56199772134572  0.004750792 -0.06837552 -0.13092936 -0.069351505  0.03665593
2.58922526312282  0.028331904  0.09973905  0.05454068  0.083838882  0.12493506
2.62817226680648 -0.007754915  0.07605095  0.04175126  0.025025546 -0.01104960
2.68363274717552  0.200884228 -0.05805574 -0.15990075  0.161215398 -0.13793452
                         PC11
2.52982874216965  0.034555718
2.54302304249866 -0.029541166
2.56199772134572 -0.031379894
2.58922526312282 -0.003258948
2.62817226680648 -0.097166725
2.68363274717552  0.021647163</code></pre>
<pre class="r"><code>df_out$pH &lt;- rownames(df_out)
df_out$pH &lt;- as.numeric(df_out$pH)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrate dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-10-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(pH, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrate dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-10-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (rev(col_pH(100))),
          main = &quot;[Using inferred pH] Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-10-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>(using inferred pH) PCA with only nitrite dynamics</li>
</ol>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
any(is.na(mat_NO2))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO2)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.379731</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.2085221</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                         PC1          PC2           PC3          PC4
2.52982874216965 -0.20643529  0.107324754 -0.0808765016  0.055779775
2.54302304249866 -0.20924682  0.111032580 -0.0793096400  0.058592144
2.56199772134572 -0.20841309  0.111521456 -0.0816604820  0.055490680
2.58922526312282 -0.20994983  0.111452068 -0.0800597597  0.057988067
2.62817226680648 -0.20928844  0.112435353 -0.0819134850  0.056777415
2.68363274717552 -0.20916404  0.113680171 -0.0816688277  0.058424169
2.7621039756057  -0.21012975  0.113008033 -0.0778038706  0.057541382
2.87213246874345 -0.20963972  0.114154354 -0.0818681721  0.056674395
3.02446632998257 -0.21074581  0.112892904 -0.0821664300  0.059463299
3.23171258546232 -0.21055637  0.111566369 -0.0781899144  0.058463025
3.5070518892221  -0.21055188  0.113881565 -0.0802339949  0.058998964
3.86154407990938 -0.20932715  0.113414747 -0.0822055488  0.059508689
4.29993149900709 -0.21072993  0.113306113 -0.0820721142  0.057887067
4.81585455700761 -0.20794141  0.112613488 -0.0806598950  0.057464102
5.38876782542826 -0.20405403  0.110759973 -0.0706182195  0.059068086
5.98535166498929 -0.17462253  0.103540352 -0.0471029618  0.056451158
6.56640618671601 -0.18010991  0.107120471 -0.0786509807  0.058125785
7.09662707522223 -0.20631246  0.104561881 -0.0783419157  0.056485654
7.55244909763549 -0.20479748  0.092096576 -0.0767856192  0.051417293
7.92464780541878 -0.19634761  0.086862713 -0.0730075688  0.047323034
8.2159958447781  -0.14389030  0.077952426 -0.0625587365  0.031983800
8.43660629576058  0.09932573  0.058076434 -0.0698508397  0.026077158
8.59948897711833  0.62099985  0.012044136 -0.0325849161  0.019061543
8.71752137494786  0.58850008 -0.021410575  0.5366722423  0.040921063
8.80189884276433  1.04232320 -0.017984576 -0.1144111620  0.005025727
8.86163332665687 -0.20972423 -0.234363200 -0.0768445667  0.050678305
8.90363119822312 -0.18208971  0.061734654 -0.0519428434  0.032110025
8.93301585011944  0.98762168 -0.021664808 -0.1063189359  0.002471267
8.95350562621574  1.45184458 -0.067968923 -0.1241648933 -0.016621582
8.96775919824282  1.05657385 -0.034734603 -0.1056340858 -0.006310598
8.97765824405471 -0.07336192 -0.062997482  1.6152818461  0.077492017
8.98452521098818 -0.17482521 -0.008601835  0.3219592826 -0.056509041
8.9892850367663  -0.15458380  0.038587600 -0.0231868062  0.005393560
8.99258248229834 -0.15542772  0.030674858 -0.0259843439 -0.001702017
8.99486596777732 -0.16295414 -0.029020863  0.0008707845 -0.086221218
8.9964468661431  -0.18490145 -0.051265043  0.0135384577 -1.359810706
8.99754115021365 -0.30706698 -1.906284117 -0.1396445823  0.098036512
                          PC5          PC6          PC7           PC8
2.52982874216965 -0.037008277  0.026979402 -0.045097842  0.0060897977
2.54302304249866 -0.039615400  0.026529481 -0.042823873  0.0028060326
2.56199772134572 -0.039236031  0.028850193 -0.043269185  0.0051422038
2.58922526312282 -0.041058390  0.027144882 -0.043692120  0.0063247604
2.62817226680648 -0.040774632  0.028678918 -0.046438754  0.0043073271
2.68363274717552 -0.040744079  0.028590291 -0.047774265  0.0065219636
2.7621039756057  -0.041434930  0.027256976 -0.048001820  0.0069527764
2.87213246874345 -0.041144631  0.027572273 -0.048727310  0.0062786928
3.02446632998257 -0.041892328  0.027024326 -0.049583021  0.0061369739
3.23171258546232 -0.042957756  0.026684555 -0.047836137  0.0036932523
3.5070518892221  -0.042584149  0.028127860 -0.049488817  0.0067797534
3.86154407990938 -0.042679254  0.028399069 -0.046246511  0.0050751167
4.29993149900709 -0.042026936  0.029239319 -0.049547124  0.0082221357
4.81585455700761 -0.037738922  0.025557223 -0.046312432  0.0063457502
5.38876782542826 -0.031852884  0.031975130 -0.047502765  0.0071591370
5.98535166498929 -0.025541934  0.033428136 -0.040220817  0.0055795651
6.56640618671601 -0.042489160  0.027952015  0.008266164  0.0074382211
7.09662707522223 -0.041648388  0.025144995 -0.004582759  0.0030043348
7.55244909763549 -0.039546559  0.023445930 -0.013456631  0.0083308447
7.92464780541878 -0.035691803  0.019794222  0.078861452  0.0134591677
8.2159958447781  -0.022875303  0.004509159  0.361969609  0.0053986004
8.43660629576058 -0.016235062  0.008170577  0.357930035  0.0058946163
8.59948897711833 -0.016828550  0.004319344  0.197898131  0.0026614419
8.71752137494786 -0.087024509  0.003866018  0.060547022 -0.0008778572
8.80189884276433  0.004240152  0.008680276 -0.019856981  0.0016414516
8.86163332665687 -0.028191181  0.021387036  0.006165640 -0.1914012254
8.90363119822312 -0.014749532  0.012188366 -0.002234533  0.0035088238
8.93301585011944  0.008507822  0.006822287 -0.069396351  0.0007663159
8.95350562621574  0.024258962  0.003620103 -0.088570304 -0.0031659379
8.96775919824282  0.014125499  0.002782306 -0.068460722 -0.0001258015
8.97765824405471 -0.217947786 -0.015849032 -0.035847945  0.0005465139
8.98452521098818  1.034668390  0.240037246 -0.005439979  0.0008110285
8.9892850367663   0.017785161 -0.003304401  0.005595854  0.0082170301
8.99258248229834  0.014185687  0.002320559  0.007190761  0.0073458522
8.99486596777732  0.226117775 -0.967192845 -0.017329626  0.0009924796
8.9964468661431  -0.111114684  0.077944776 -0.009076176  0.0006704661
8.99754115021365 -0.041256399  0.041323029 -0.007609868  0.0314683948
                          PC9          PC10          PC11
2.52982874216965 -0.004460056 -2.655416e-03  4.767926e-04
2.54302304249866 -0.006544867 -1.196263e-03  1.338226e-03
2.56199772134572 -0.006030576 -1.135338e-03 -4.735829e-04
2.58922526312282 -0.003720948  9.604183e-06 -2.621907e-04
2.62817226680648 -0.007034632  4.424503e-05 -2.567331e-04
2.68363274717552 -0.008231734  9.834468e-04 -5.488525e-05
2.7621039756057  -0.008056641 -1.296101e-03 -2.210435e-04
2.87213246874345 -0.005917194  8.076350e-04 -2.027795e-03
3.02446632998257 -0.008359330  1.797664e-04  1.477938e-04
3.23171258546232 -0.006533495  7.613651e-04  1.821018e-03
3.5070518892221  -0.007966231 -2.495152e-07  1.051072e-03
3.86154407990938 -0.007742878  2.467443e-04  6.324857e-04
4.29993149900709 -0.008600107  4.475774e-04  2.869831e-04
4.81585455700761 -0.006553387  2.116368e-03  1.093902e-04
5.38876782542826 -0.009010617  2.860347e-04  8.100530e-04
5.98535166498929 -0.010481944  2.304588e-04  1.144428e-03
6.56640618671601 -0.014662833 -6.397135e-03  1.058254e-03
7.09662707522223  0.026211168 -4.692833e-04 -2.676692e-02
7.55244909763549  0.018603693  2.174514e-02  1.949859e-03
7.92464780541878  0.027085353  2.236400e-02  6.643257e-03
8.2159958447781  -0.005301870  3.443406e-03  1.070944e-03
8.43660629576058 -0.005413110  2.717448e-05 -2.205310e-04
8.59948897711833 -0.008114243 -6.793649e-03 -3.314527e-03
8.71752137494786 -0.012800209 -4.506240e-03 -2.052393e-03
8.80189884276433 -0.006699009 -4.138404e-03  2.697379e-03
8.86163332665687  0.001083598  7.601781e-04  7.418136e-04
8.90363119822312  0.020263438 -1.111511e-02  2.469457e-03
8.93301585011944  0.006499229  1.248424e-03 -1.286890e-03
8.95350562621574  0.003853832  6.946613e-03  2.064272e-03
8.96775919824282  0.007019251  1.989450e-03 -1.454124e-03
8.97765824405471  0.003713209  2.583856e-03  5.969752e-04
8.98452521098818 -0.003450423  1.487710e-03 -7.823119e-04
8.9892850367663   0.031622872 -1.774938e-02  3.945611e-03
8.99258248229834  0.034494851 -1.354813e-02  9.727671e-03
8.99486596777732 -0.003274893  1.070176e-03 -5.574536e-04
8.9964468661431  -0.002657588  8.091714e-04 -5.102723e-04
8.99754115021365 -0.002831682  4.121459e-04 -5.420780e-04</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                        PC1       PC2         PC3        PC4         PC5
2.52982874216965 -0.2064353 0.1073248 -0.08087650 0.05577978 -0.03700828
2.54302304249866 -0.2092468 0.1110326 -0.07930964 0.05859214 -0.03961540
2.56199772134572 -0.2084131 0.1115215 -0.08166048 0.05549068 -0.03923603
2.58922526312282 -0.2099498 0.1114521 -0.08005976 0.05798807 -0.04105839
2.62817226680648 -0.2092884 0.1124354 -0.08191348 0.05677741 -0.04077463
2.68363274717552 -0.2091640 0.1136802 -0.08166883 0.05842417 -0.04074408
                        PC6         PC7         PC8          PC9          PC10
2.52982874216965 0.02697940 -0.04509784 0.006089798 -0.004460056 -2.655416e-03
2.54302304249866 0.02652948 -0.04282387 0.002806033 -0.006544867 -1.196263e-03
2.56199772134572 0.02885019 -0.04326919 0.005142204 -0.006030576 -1.135338e-03
2.58922526312282 0.02714488 -0.04369212 0.006324760 -0.003720948  9.604183e-06
2.62817226680648 0.02867892 -0.04643875 0.004307327 -0.007034632  4.424503e-05
2.68363274717552 0.02859029 -0.04777427 0.006521964 -0.008231734  9.834468e-04
                          PC11
2.52982874216965  4.767926e-04
2.54302304249866  1.338226e-03
2.56199772134572 -4.735829e-04
2.58922526312282 -2.621907e-04
2.62817226680648 -2.567331e-04
2.68363274717552 -5.488525e-05</code></pre>
<pre class="r"><code>df_out$pH &lt;- rownames(df_out)
df_out$pH &lt;- as.numeric(df_out$pH)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrite dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-11-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(pH, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrite dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-11-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (rev(col_pH(100))),
          main = &quot;[Using inferred pH] Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-11-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>(using inferred pH) PCA with both nitrate and nitrite dynamics</li>
</ol>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)
dim(mat_NO2)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>dim(mat_NO3)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO23 &lt;- cbind(mat_NO3, mat_NO2)
any(is.na(mat_NO23))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>dim(mat_NO23)</code></pre>
<pre><code>[1] 37 22</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(mat_NO23)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.8038253</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.08763187</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out</code></pre>
<pre><code>                        PC1         PC2           PC3          PC4         PC5
2.52982874216965 -3.8457653  0.49825046  3.651211e-02 -0.040816242  0.02114097
2.54302304249866 -3.6308720  0.50085595 -1.105935e-01 -0.023819324 -0.01458136
2.56199772134572 -3.7319488  0.41128598 -1.061037e-01  0.077865510  0.03403964
2.58922526312282 -3.8110238  0.38586065  2.144749e-02 -0.035884061  0.05131171
2.62817226680648 -3.7610387  0.37474025 -1.434285e-01 -0.014412302  0.07796786
2.68363274717552 -3.3069285  0.40391794 -4.212447e-02  0.034391351 -0.14214305
2.7621039756057  -3.1324220  0.05414602  1.382747e-01 -0.024629251 -0.01867606
2.87213246874345 -2.8736348  0.02057743 -1.087546e-06 -0.143708579  0.13023788
3.02446632998257 -2.7918153  0.12819771  1.020998e-01 -0.144291881  0.02598254
3.23171258546232 -2.4791860 -0.06004411  1.949296e-02 -0.051793182  0.03828107
3.5070518892221  -2.1794903  0.03388802 -2.395823e-02  0.038989104 -0.06229729
3.86154407990938 -1.9147282  0.02712814  6.829243e-02  0.094880004 -0.06229914
4.29993149900709 -0.1340640 -1.01109298 -1.944531e-02  0.430051782 -0.10860959
4.81585455700761  0.5862569 -0.91670005 -6.017218e-01  0.612950741  0.01914256
5.38876782542826  0.9364429 -0.74627194 -5.300085e-01  0.384655537 -0.03996884
5.98535166498929  1.1652824 -0.49320010 -5.143014e-01  0.238198944 -0.07708980
6.56640618671601  1.7067824  0.06783606 -3.994217e-01 -0.057361885 -0.25560907
7.09662707522223  1.9469092  0.47393078 -3.550578e-01 -0.042308055 -0.49787793
7.55244909763549  1.8902453  0.31335983 -3.101230e-01 -0.093283049 -0.40011299
7.92464780541878  1.9470734  0.37377047 -2.566097e-01 -0.106699300 -0.39898671
8.2159958447781   2.1058599  0.58379902 -8.688799e-02 -0.019157501 -0.55976379
8.43660629576058  2.1615480  0.60013028  1.260128e-01 -0.009946359 -0.36229368
8.59948897711833  2.0928230  0.64717647  1.872324e-01  0.028150004  0.17915353
8.71752137494786  1.9567286  0.33175634 -6.638058e-02 -0.485606226  0.52018956
8.80189884276433  2.2060661  0.94703794  4.114083e-01  0.275457235  0.41688012
8.86163332665687  2.0870000  0.63577278 -4.509045e-02 -0.049800200 -0.64226052
8.90363119822312  1.9598897  0.76762460 -2.127180e-01  0.022746147 -0.49023369
8.93301585011944  2.2008567  0.91139792  4.154737e-01  0.246671456  0.41886912
8.95350562621574  2.2544913  0.98528774  6.780421e-01  0.308345248  0.80359201
8.96775919824282  2.1911087  0.98861571  3.652997e-01  0.303645161  0.46680871
8.97765824405471  1.6545465 -0.38491630 -7.932897e-01 -1.555016386  0.66691862
8.98452521098818  1.1957537 -1.02399917 -5.045083e-01 -0.306397050  0.35171953
8.9892850367663   0.6638339 -1.39185370 -8.333104e-02  0.438601225  0.16193342
8.99258248229834  0.7267533 -1.41449549  1.181899e-02  0.327845112  0.16433552
8.99486596777732  0.8532461 -1.45557951  1.814776e-01  0.272806913  0.24587812
8.9964468661431   0.6042167 -1.34358109  8.825410e-01 -0.157478694  0.05963203
8.99754115021365  0.4992033 -1.22461003  1.559679e+00 -0.773841950 -0.72121099
                           PC6          PC7           PC8          PC9
2.52982874216965 -4.004081e-02  0.031796865 -0.0060601471  0.091612261
2.54302304249866  7.188755e-02 -0.180029463  0.0563795908  0.235129585
2.56199772134572  5.759856e-02  0.171612205 -0.1599634475  0.203398336
2.58922526312282 -2.246429e-02  0.102825775  0.0020118445  0.068935706
2.62817226680648  7.984637e-02  0.043434581  0.0233189843  0.098109681
2.68363274717552  1.511091e-02 -0.172105236  0.1857281933 -0.060280720
2.7621039756057  -5.313295e-02 -0.077398923 -0.0262384047 -0.245653606
2.87213246874345 -9.198805e-02  0.221205304 -0.1777293311 -0.082046672
3.02446632998257 -1.056914e-01  0.159546816 -0.0146056138 -0.185180918
3.23171258546232 -2.009838e-02  0.003435627 -0.0345916085 -0.207140595
3.5070518892221   3.799657e-03  0.015232169  0.0382829278 -0.027315122
3.86154407990938 -3.366991e-02 -0.080366450  0.0293352893 -0.064186691
4.29993149900709  9.196525e-02 -0.661250911  0.2753329389 -0.261348276
4.81585455700761  3.080821e-01 -0.421576135 -0.0003561799  0.211895097
5.38876782542826  1.964527e-01 -0.135647162  0.0443161632 -0.019930189
5.98535166498929  1.565612e-01 -0.090603030  0.0857740158 -0.037075973
6.56640618671601 -1.007528e-02  0.049832690  0.0972908425 -0.079162929
7.09662707522223 -9.890144e-02 -0.022345132 -0.0052260445 -0.106560698
7.55244909763549 -9.608877e-02  0.032851296  0.0078312392 -0.090456491
7.92464780541878 -1.368138e-01  0.049942493  0.0009280291 -0.138253353
8.2159958447781  -2.288429e-01  0.143949538 -0.1168248079 -0.064782498
8.43660629576058 -2.417076e-01  0.211638206 -0.1210247136 -0.181760470
8.59948897711833 -1.704743e-02  0.036031588 -0.0171784520 -0.177308795
8.71752137494786  2.307647e-02 -0.128055034 -0.0772757727 -0.159099100
8.80189884276433  8.944011e-02 -0.026749136  0.0424315855 -0.059114023
8.86163332665687  6.879346e-02 -0.007940295 -0.0467612768  0.291079464
8.90363119822312 -1.115378e-01  0.052992716 -0.0597972077  0.363867491
8.93301585011944  9.072577e-02 -0.023477362  0.0356251107  0.035425622
8.95350562621574  1.575625e-01 -0.012087591  0.0516966657  0.034525235
8.96775919824282  1.537407e-01 -0.100862199  0.0695015594  0.141662430
8.97765824405471 -1.901911e-05 -0.400661793 -0.2712491189  0.097125415
8.98452521098818  2.235905e-01  0.809327805  0.8650825467  0.048744675
8.9892850367663   6.715483e-02  0.088798501 -0.2088622874  0.053845169
8.99258248229834 -2.413284e-02  0.131699922 -0.2910483581  0.009221755
8.99486596777732 -1.289358e-01  0.488381816 -0.6457795623  0.016643934
8.9964468661431  -1.227015e+00 -0.254350390  0.3525740163  0.207020519
8.99754115021365  8.328154e-01 -0.049029667  0.0171307914  0.038414743
                         PC10          PC11        PC12         PC13
2.52982874216965  0.009307847  0.0523525745  0.05502831 -0.092101610
2.54302304249866  0.139912837 -0.0592716497 -0.06670704  0.025377613
2.56199772134572 -0.157463779 -0.1005721998 -0.10779045  0.059210045
2.58922526312282 -0.023656655  0.0402265018  0.02596430 -0.079356785
2.62817226680648 -0.031301554  0.1106948105 -0.05418369 -0.075223693
2.68363274717552  0.165560869 -0.1251809491 -0.23425148 -0.082605774
2.7621039756057   0.010617469 -0.0704204755  0.04091323  0.135562282
2.87213246874345 -0.197983277  0.1426980814  0.06142454  0.009405457
3.02446632998257 -0.038319446  0.0945977027  0.08072624 -0.037237726
3.23171258546232 -0.024000152  0.0467492022  0.11019425  0.129529688
3.5070518892221   0.005974522 -0.1560625512 -0.04071797  0.189916029
3.86154407990938  0.131705833  0.0196511545  0.22661986 -0.111701369
4.29993149900709  0.296278083 -0.1463151432  0.10873299  0.005061007
4.81585455700761 -0.073091466  0.0672305096 -0.03007858 -0.043909878
5.38876782542826 -0.126182699  0.1242424537 -0.08796313 -0.052590951
5.98535166498929 -0.098145607  0.1479881015 -0.09836434 -0.016805871
6.56640618671601 -0.046748046  0.1868935613 -0.03825310  0.069309575
7.09662707522223 -0.028062558  0.0631315722 -0.11000420  0.105355478
7.55244909763549 -0.029577524  0.1397385543 -0.01029286  0.090682894
7.92464780541878  0.020125074  0.1915295125  0.11920979 -0.061786623
8.2159958447781  -0.014583087 -0.2718693424 -0.06738069 -0.050336168
8.43660629576058 -0.041093500 -0.2440143209 -0.02974887 -0.118529668
8.59948897711833 -0.062504324  0.0003236607 -0.08117216 -0.091181635
8.71752137494786 -0.035448828  0.0161984233 -0.05559287 -0.079508889
8.80189884276433 -0.012107582 -0.0090402332 -0.05630522 -0.008826081
8.86163332665687  0.058696820 -0.0337282029  0.07603981  0.082354110
8.90363119822312  0.119114301 -0.0234883002  0.17324823 -0.035527784
8.93301585011944  0.012416582  0.0343786481  0.05003541  0.024637346
8.95350562621574 -0.018737891  0.0146342344  0.02190417  0.053603950
8.96775919824282  0.047425558  0.0124581765  0.02182354  0.052090264
8.97765824405471  0.045294078 -0.0974372704  0.02637484  0.012831827
8.98452521098818  0.105014110 -0.0965765014  0.02912862  0.004550228
8.9892850367663  -0.217144306 -0.1656957889  0.06588249 -0.040409010
8.99258248229834 -0.260256485 -0.1877184771  0.09102783  0.026935853
8.99486596777732  0.521023664  0.1348847079 -0.10814436  0.016619489
8.9964468661431  -0.088919108  0.0895484500 -0.06592186  0.006748642
8.99754115021365 -0.063139774  0.0572408125 -0.04140560 -0.022142260
                         PC14          PC15         PC16          PC17
2.52982874216965 -0.137019408  2.664368e-02  0.029937436  0.1012999309
2.54302304249866  0.203727909 -1.504301e-01  0.028540363  0.0121291457
2.56199772134572 -0.046416018  1.459141e-03 -0.047006066 -0.0488045022
2.58922526312282 -0.117951979  5.822079e-02  0.067281693  0.0251089718
2.62817226680648 -0.015756333 -3.874380e-02  0.027057885 -0.0599612616
2.68363274717552  0.129887413  1.294940e-01 -0.048159928  0.0313924958
2.7621039756057  -0.089234247 -3.591125e-02 -0.056277594  0.0131721772
2.87213246874345  0.047719611 -4.252577e-02  0.001689337  0.0985084731
3.02446632998257  0.017181080  5.812509e-02  0.050637387 -0.0140386786
3.23171258546232  0.011980005 -7.183885e-02 -0.096058703 -0.0609150959
3.5070518892221  -0.021591344  5.067319e-02  0.061085428 -0.0672360932
3.86154407990938  0.064136422  2.882331e-02 -0.062503224 -0.0925986928
4.29993149900709 -0.045122470 -3.288795e-05  0.072871056  0.0423380603
4.81585455700761 -0.086264869 -1.166274e-01 -0.042688884  0.0405628387
5.38876782542826 -0.015300546  4.852956e-02 -0.045464565 -0.0501053598
5.98535166498929 -0.009626345  4.192710e-02  0.005328323 -0.0559166801
6.56640618671601  0.037481378 -4.263595e-03  0.148698185 -0.0544247048
7.09662707522223  0.022065499  4.870981e-02 -0.047540639  0.0927169082
7.55244909763549  0.039840643  1.436782e-02  0.045832588  0.0405297059
7.92464780541878  0.059143475 -2.544343e-02 -0.075512046  0.0087073384
8.2159958447781  -0.057186353 -8.227256e-02  0.021807851 -0.0092978522
8.43660629576058 -0.025685007 -3.751156e-02 -0.008043804 -0.0255971760
8.59948897711833 -0.038120362 -1.107570e-01  0.063312428 -0.0173457081
8.71752137494786  0.013957491 -2.467845e-03 -0.044566264  0.0006769268
8.80189884276433  0.001071562  5.467246e-02 -0.066935445  0.0124307785
8.86163332665687 -0.118230637  4.133017e-02 -0.053050594 -0.0062946597
8.90363119822312  0.087495692  3.790451e-02  0.029431287 -0.0019714793
8.93301585011944 -0.021763473  2.571451e-02  0.028992067 -0.0095874278
8.95350562621574  0.039708706  6.411429e-03  0.038372746  0.0085873725
8.96775919824282 -0.007615347 -8.717673e-03 -0.028062894  0.0057199122
8.97765824405471 -0.024298505  2.447045e-02  0.009078655 -0.0075465502
8.98452521098818 -0.016877557 -2.721103e-02 -0.027012782  0.0238669121
8.9892850367663   0.027489657  6.092106e-02  0.029682259 -0.0284538972
8.99258248229834  0.126402186  2.299080e-02  0.008218579  0.0693552248
8.99486596777732 -0.028315983 -8.238651e-03 -0.003500728 -0.0006884242
8.9964468661431  -0.016678306 -8.729520e-03 -0.017471985 -0.0154580084
8.99754115021365  0.009766361 -9.666016e-03  0.002000591 -0.0008609207
                          PC18          PC19          PC20          PC21
2.52982874216965 -0.0511121772  0.0073162016 -0.0017357223  3.375717e-03
2.54302304249866  0.0025461254 -0.0226798592  0.0013566982  1.921265e-03
2.56199772134572  0.0410800457  0.0344592660 -0.0001594900 -6.470265e-03
2.58922526312282 -0.0376612241  0.0034384796  0.0065905780  3.086871e-03
2.62817226680648  0.0463615105 -0.0003294328  0.0117182603  4.139161e-03
2.68363274717552 -0.0029839708 -0.0183011733  0.0025188510 -2.476999e-03
2.7621039756057   0.0615382602  0.0241337997 -0.0035866428 -1.909456e-03
2.87213246874345 -0.0098739429 -0.0270684662 -0.0129185638 -5.094109e-03
3.02446632998257  0.0335359855 -0.0149738137 -0.0029903873 -3.032425e-03
3.23171258546232 -0.0426975218 -0.0053753951 -0.0009465356  6.343744e-03
3.5070518892221  -0.0809355088  0.0053286868  0.0015198957  8.569138e-04
3.86154407990938  0.0262641019  0.0067815303 -0.0056786944 -1.049585e-03
4.29993149900709  0.0161728684 -0.0050370569  0.0002340283 -1.098821e-03
4.81585455700761 -0.0118430998  0.0153922762 -0.0036141142 -4.093085e-03
5.38876782542826 -0.0189426058 -0.0017565581 -0.0041815342 -2.581990e-03
5.98535166498929 -0.0174912027 -0.0052160175 -0.0035651019 -1.513949e-03
6.56640618671601 -0.0273602267 -0.0151585334 -0.0122244433  3.206487e-03
7.09662707522223  0.0439455827  0.0317668837  0.0002704584  1.605734e-02
7.55244909763549  0.0390138533  0.0205139849  0.0122279016 -1.458940e-02
7.92464780541878 -0.0635841112  0.0075795864  0.0213515792 -1.403878e-03
8.2159958447781  -0.0394077830  0.0125413631 -0.0013300870 -4.933013e-03
8.43660629576058 -0.0301518653 -0.0017882094 -0.0025307743 -1.650730e-03
8.59948897711833  0.0710520845 -0.0141701191  0.0029076585  6.838688e-03
8.71752137494786 -0.0009980187 -0.0238587630 -0.0082830544  3.119888e-03
8.80189884276433 -0.0055636266  0.0060177776 -0.0103955824 -2.689026e-03
8.86163332665687  0.0404574709 -0.1079714326  0.0017707703 -2.913390e-04
8.90363119822312  0.0149657442  0.0517223070 -0.0132891826  2.150231e-03
8.93301585011944  0.0590592575  0.0235947340  0.0021710281 -2.722035e-03
8.95350562621574 -0.0196401420 -0.0138153040  0.0064426022 -4.710545e-03
8.96775919824282 -0.0628285004  0.0130416625  0.0033617667  5.422505e-03
8.97765824405471 -0.0011794863  0.0105027947  0.0029241838 -1.598957e-03
8.98452521098818  0.0112294164  0.0020461288 -0.0004483077 -2.747759e-04
8.9892850367663   0.0335533558  0.0026143110  0.0103684714  8.546471e-03
8.99258248229834 -0.0072105567 -0.0199197204  0.0017616575 -4.389731e-04
8.99486596777732 -0.0018598951  0.0004523609 -0.0004959438 -2.556783e-04
8.9964468661431   0.0011570940  0.0010910700 -0.0003958548 -1.207613e-04
8.99754115021365 -0.0086072913  0.0170846500 -0.0007263724 -6.548803e-05
                          PC22
2.52982874216965  0.0032526777
2.54302304249866 -0.0002708559
2.56199772134572 -0.0012188114
2.58922526312282 -0.0009706162
2.62817226680648  0.0007925574
2.68363274717552  0.0027131276
2.7621039756057   0.0061304039
2.87213246874345 -0.0043724514
3.02446632998257 -0.0024320982
3.23171258546232  0.0024502498
3.5070518892221  -0.0021393465
3.86154407990938 -0.0064532303
4.29993149900709 -0.0009231903
4.81585455700761 -0.0019822036
5.38876782542826 -0.0011024537
5.98535166498929  0.0007130555
6.56640618671601  0.0033392977
7.09662707522223 -0.0074302904
7.55244909763549  0.0002962713
7.92464780541878  0.0033558833
8.2159958447781  -0.0014513786
8.43660629576058 -0.0031428337
8.59948897711833  0.0028089540
8.71752137494786  0.0008565065
8.80189884276433  0.0054508796
8.86163332665687  0.0002867665
8.90363119822312  0.0043155225
8.93301585011944 -0.0006622194
8.95350562621574 -0.0043844344
8.96775919824282 -0.0008879745
8.97765824405471 -0.0005745341
8.98452521098818 -0.0002423063
8.9892850367663   0.0012111725
8.99258248229834  0.0032371898
8.99486596777732 -0.0001881770
8.9964468661431  -0.0001598275
8.99754115021365 -0.0002212823</code></pre>
<pre class="r"><code>head(df_out)</code></pre>
<pre><code>                       PC1       PC2         PC3         PC4         PC5
2.52982874216965 -3.845765 0.4982505  0.03651211 -0.04081624  0.02114097
2.54302304249866 -3.630872 0.5008559 -0.11059350 -0.02381932 -0.01458136
2.56199772134572 -3.731949 0.4112860 -0.10610368  0.07786551  0.03403964
2.58922526312282 -3.811024 0.3858607  0.02144749 -0.03588406  0.05131171
2.62817226680648 -3.761039 0.3747403 -0.14342854 -0.01441230  0.07796786
2.68363274717552 -3.306929 0.4039179 -0.04212447  0.03439135 -0.14214305
                         PC6         PC7          PC8         PC9         PC10
2.52982874216965 -0.04004081  0.03179686 -0.006060147  0.09161226  0.009307847
2.54302304249866  0.07188755 -0.18002946  0.056379591  0.23512958  0.139912837
2.56199772134572  0.05759856  0.17161220 -0.159963448  0.20339834 -0.157463779
2.58922526312282 -0.02246429  0.10282578  0.002011845  0.06893571 -0.023656655
2.62817226680648  0.07984637  0.04343458  0.023318984  0.09810968 -0.031301554
2.68363274717552  0.01511091 -0.17210524  0.185728193 -0.06028072  0.165560869
                        PC11        PC12        PC13        PC14         PC15
2.52982874216965  0.05235257  0.05502831 -0.09210161 -0.13701941  0.026643679
2.54302304249866 -0.05927165 -0.06670704  0.02537761  0.20372791 -0.150430112
2.56199772134572 -0.10057220 -0.10779045  0.05921004 -0.04641602  0.001459141
2.58922526312282  0.04022650  0.02596430 -0.07935679 -0.11795198  0.058220785
2.62817226680648  0.11069481 -0.05418369 -0.07522369 -0.01575633 -0.038743802
2.68363274717552 -0.12518095 -0.23425148 -0.08260577  0.12988741  0.129494019
                        PC16        PC17         PC18          PC19
2.52982874216965  0.02993744  0.10129993 -0.051112177  0.0073162016
2.54302304249866  0.02854036  0.01212915  0.002546125 -0.0226798592
2.56199772134572 -0.04700607 -0.04880450  0.041080046  0.0344592660
2.58922526312282  0.06728169  0.02510897 -0.037661224  0.0034384796
2.62817226680648  0.02705789 -0.05996126  0.046361510 -0.0003294328
2.68363274717552 -0.04815993  0.03139250 -0.002983971 -0.0183011733
                         PC20         PC21          PC22
2.52982874216965 -0.001735722  0.003375717  0.0032526777
2.54302304249866  0.001356698  0.001921265 -0.0002708559
2.56199772134572 -0.000159490 -0.006470265 -0.0012188114
2.58922526312282  0.006590578  0.003086871 -0.0009706162
2.62817226680648  0.011718260  0.004139161  0.0007925574
2.68363274717552  0.002518851 -0.002476999  0.0027131276</code></pre>
<pre class="r"><code>df_out$pH &lt;- rownames(df_out)
df_out$pH &lt;- as.numeric(df_out$pH)

# plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2))+
  geom_point(size = 4, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-12-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Label with mM
ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf(&quot;%0.1f&quot;, round(pH, digits = 1))))+
  geom_point(size = 5, alpha=0.9, aes(col = pH))+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n&quot;) +
  ggrepel::geom_label_repel() +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-12-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization
library(&quot;plot3D&quot;)
rownames(df_out) &lt;- as.numeric(rownames(df_out))
scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = &quot;b2&quot;, pch = 16, cex = 2,col = (rev(col_pH(100))),
          main = &quot;[Using inferred pH] Principle component analysis (Nitrate dynamics)&quot;, xlab = &quot;PC1&quot;,
          ylab =&quot;PC2&quot;, zlab = &quot;PC3&quot;,
          theta = 15, phi = 50)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-12-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## is this correct? I don&#39;t think so.. doesn&#39;t agree with the 2d plot.... let&#39;s come back to this some time later..</code></pre>
<p>Let’s take a closer look at the basic pH curves</p>
<pre class="r"><code>df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$H_mM &gt; -70, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (H_mM &gt; -70) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 60

df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$H_mM &gt; -60, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (H_mM &gt; -60) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 50
df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$H_mM &gt; -50, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (H_mM &gt; -50) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##

df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$pH &lt; 8.9, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (inferred pH &gt; 8.9) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 3 clusters - first cluster
df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$pH &lt; 8.9, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (inferred pH &gt; 8.9) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 3 clusters - second cluster
df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$pH &gt; 8.9 | df_plot3$pH &lt; 4, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation (4 &lt; inferred pH &lt; 8.9) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-6.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 3 clusters - third cluster
df_plot3 &lt;- df_plot2
df_plot3$NO3_mM &lt;- ifelse(df_plot3$pH &gt; 4, NA, df_plot3$NO3_mM)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.25), limits=c(0, 2.5))+
  ggtitle(&quot;pH perturbation (inferred pH &lt; 4) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-13-7.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="peak-time" class="section level2">
<h2>4.1.2. Peak time</h2>
<pre class="r"><code># maximum peak time
vec_peaktime &lt;- apply(mat_H_time, 1, which.max)
vec_hr &lt;- df_time$Time_hours

vec_hr[vec_peaktime]</code></pre>
<pre><code> [1] 75.6 60.2 50.2 36.5 36.5 36.5 30.9 24.5 24.5 24.5  9.7 75.6 24.5 24.5 24.5
[16]  9.7  9.7  9.7  6.4  6.4  9.7 30.9 36.5  6.4  9.7  9.7  3.6 30.9 50.2  0.0
[31] 30.9  9.7  9.7  6.4  9.7  9.7  3.6</code></pre>
<pre class="r"><code>names(vec_peaktime)</code></pre>
<pre><code> [1] &quot;-94.7069346522151&quot; &quot;-89.4454382826476&quot; &quot;-84.1839419130801&quot;
 [4] &quot;-78.9224455435126&quot; &quot;-73.6609491739451&quot; &quot;-68.3994528043776&quot;
 [7] &quot;-63.1379564348101&quot; &quot;-57.8764600652426&quot; &quot;-52.614963695675&quot; 
[10] &quot;-47.3534673261075&quot; &quot;-42.09197095654&quot;   &quot;-36.8304745869725&quot;
[13] &quot;-31.568978217405&quot;  &quot;-26.3074818478375&quot; &quot;-21.04598547827&quot;  
[16] &quot;-15.7844891087025&quot; &quot;-10.522992739135&quot;  &quot;-5.2614963695675&quot; 
[19] &quot;0&quot;                 &quot;5.2614963695675&quot;   &quot;10.522992739135&quot;  
[22] &quot;15.7844891087025&quot;  &quot;21.04598547827&quot;    &quot;26.3074818478375&quot; 
[25] &quot;31.568978217405&quot;   &quot;36.8304745869725&quot;  &quot;42.09197095654&quot;   
[28] &quot;47.3534673261075&quot;  &quot;52.614963695675&quot;   &quot;57.8764600652426&quot; 
[31] &quot;63.1379564348101&quot;  &quot;68.3994528043776&quot;  &quot;73.6609491739451&quot; 
[34] &quot;78.9224455435126&quot;  &quot;84.1839419130801&quot;  &quot;89.4454382826476&quot; 
[37] &quot;94.7069346522151&quot; </code></pre>
<pre class="r"><code>plot(names(vec_peaktime), vec_hr[vec_peaktime])</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-14-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_peak_NO2 &lt;- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime])
df_peak_NO2$H_mM &lt;- as.numeric(df_peak_NO2$H_mM)


# 1. linear regression
fit.no2peak &lt;- lm(Time_hours ~ H_mM, df_peak_NO2)
summary(fit.no2peak)</code></pre>
<pre><code>
Call:
lm(formula = Time_hours ~ H_mM, data = df_peak_NO2)

Residuals:
    Min      1Q  Median      3Q     Max 
-22.317 -11.296  -3.237   4.435  44.653 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 23.45676    2.61881   8.957 1.39e-10 ***
H_mM        -0.20338    0.04662  -4.363 0.000108 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 15.93 on 35 degrees of freedom
Multiple R-squared:  0.3523,    Adjusted R-squared:  0.3338 
F-statistic: 19.03 on 1 and 35 DF,  p-value: 0.000108</code></pre>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]],
              color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=0,y=30, label= paste0(&quot;y = &quot;, round(coef(fit.no2peak)[[1]],3),&quot;+&quot;,round(coef(fit.no2peak)[[2]],3),&quot;x&quot;, &quot;,  R^2: &quot;, round(summary(fit.no2peak)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-14-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. quadratic regression

df_peak_NO2$H_mM_2 &lt;- (df_peak_NO2$H_mM)^2 
quadraticModel &lt;- lm(Time_hours ~ H_mM + H_mM_2, df_peak_NO2)
summary(quadraticModel)</code></pre>
<pre><code>
Call:
lm(formula = Time_hours ~ H_mM + H_mM_2, data = df_peak_NO2)

Residuals:
    Min      1Q  Median      3Q     Max 
-19.574  -8.889  -5.168   1.830  48.219 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 17.2013365  3.7194731   4.625 5.24e-05 ***
H_mM        -0.2033795  0.0441127  -4.610 5.46e-05 ***
H_mM_2       0.0019821  0.0008789   2.255   0.0307 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 15.07 on 34 degrees of freedom
Multiple R-squared:  0.4366,    Adjusted R-squared:  0.4034 
F-statistic: 13.17 on 2 and 34 DF,  p-value: 5.815e-05</code></pre>
<pre class="r"><code>perturbValues &lt;- seq(-100, 100, 0.1)
peakPredict &lt;- predict(quadraticModel,list(H_mM=perturbValues, H_mM_2=perturbValues^2))
df_quad &lt;- data.frame(H_mM = perturbValues, Time_hours = peakPredict)
plot(perturbValues, peakPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-14-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  # annotate(&quot;text&quot;,x=0,y=30, label= paste0(&quot;y = &quot;, round(coef(fit.no2peak)[[1]],3),&quot;+&quot;,round(coef(fit.no2peak)[[2]],3),&quot;x&quot;), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-14-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks \n&quot;) +
  geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = &quot;blue&quot;, size = 1) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=0,y=30, label= paste0(&quot;y = &quot;, round(coef(quadraticModel)[[1]],3), &quot;+&quot;, round(coef(quadraticModel)[[2]],3),&quot;x+&quot;,round(coef(quadraticModel)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(quadraticModel)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-14-5.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="get-rid-of-noise-by-peak-threshold" class="section level2">
<h2>4.1.3. Get rid of noise by peak threshold</h2>
<pre class="r"><code># maximum peak value
apply(mat_H_time, 1, max) %&gt;% length()</code></pre>
<pre><code>[1] 37</code></pre>
<pre class="r"><code>vec_max &lt;- apply(mat_H_time, 1, max)
hist(vec_max, breaks = 50, xlab = &quot;NO2 mM at peak&quot;, main = &quot;Histogram of peak NO2 concentration&quot;)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sort(vec_max)</code></pre>
<pre><code>  31.568978217405   52.614963695675    42.09197095654  68.3994528043776 
       0.02861360        0.02872813        0.02952354        0.03009534 
 57.8764600652426  73.6609491739451  36.8304745869725  47.3534673261075 
       0.03069779        0.03122439        0.03150666        0.03158267 
 63.1379564348101  26.3074818478375  94.7069346522151  78.9224455435126 
       0.03165052        0.03198263        0.03300129        0.03343508 
 84.1839419130801  89.4454382826476    21.04598547827  15.7844891087025 
       0.03418849        0.03475314        0.03974985        0.06194423 
  5.2614963695675   -42.09197095654                 0   10.522992739135 
       0.06770988        0.06916343        0.07466739        0.08685623 
-78.9224455435126 -73.6609491739451  -5.2614963695675  -10.522992739135 
       0.08762101        0.08949089        0.13181208        0.40985389 
-15.7844891087025 -36.8304745869725 -26.3074818478375   -21.04598547827 
       0.41859861        0.41928339        0.79408193        0.84637617 
-84.1839419130801 -68.3994528043776 -47.3534673261075  -31.568978217405 
       1.06861160        1.17437834        1.23199660        1.28382145 
-57.8764600652426 -89.4454382826476 -94.7069346522151  -52.614963695675 
       1.30175706        1.45055613        1.69911297        1.69965253 
-63.1379564348101 
       1.74445642 </code></pre>
<pre class="r"><code>min_peak_NO2_mM &lt;- 0.05

vec_max[vec_max &gt; min_peak_NO2_mM]</code></pre>
<pre><code>-94.7069346522151 -89.4454382826476 -84.1839419130801 -78.9224455435126 
       1.69911297        1.45055613        1.06861160        0.08762101 
-73.6609491739451 -68.3994528043776 -63.1379564348101 -57.8764600652426 
       0.08949089        1.17437834        1.74445642        1.30175706 
 -52.614963695675 -47.3534673261075   -42.09197095654 -36.8304745869725 
       1.69965253        1.23199660        0.06916343        0.41928339 
 -31.568978217405 -26.3074818478375   -21.04598547827 -15.7844891087025 
       1.28382145        0.79408193        0.84637617        0.41859861 
 -10.522992739135  -5.2614963695675                 0   5.2614963695675 
       0.40985389        0.13181208        0.07466739        0.06770988 
  10.522992739135  15.7844891087025 
       0.08685623        0.06194423 </code></pre>
<pre class="r"><code># maximum peak time
vec_peaktime &lt;- apply(mat_H_time, 1, which.max)
vec_hr &lt;- df_time$Time_hours

vec_hr[vec_peaktime]</code></pre>
<pre><code> [1] 75.6 60.2 50.2 36.5 36.5 36.5 30.9 24.5 24.5 24.5  9.7 75.6 24.5 24.5 24.5
[16]  9.7  9.7  9.7  6.4  6.4  9.7 30.9 36.5  6.4  9.7  9.7  3.6 30.9 50.2  0.0
[31] 30.9  9.7  9.7  6.4  9.7  9.7  3.6</code></pre>
<pre class="r"><code>names(vec_peaktime)</code></pre>
<pre><code> [1] &quot;-94.7069346522151&quot; &quot;-89.4454382826476&quot; &quot;-84.1839419130801&quot;
 [4] &quot;-78.9224455435126&quot; &quot;-73.6609491739451&quot; &quot;-68.3994528043776&quot;
 [7] &quot;-63.1379564348101&quot; &quot;-57.8764600652426&quot; &quot;-52.614963695675&quot; 
[10] &quot;-47.3534673261075&quot; &quot;-42.09197095654&quot;   &quot;-36.8304745869725&quot;
[13] &quot;-31.568978217405&quot;  &quot;-26.3074818478375&quot; &quot;-21.04598547827&quot;  
[16] &quot;-15.7844891087025&quot; &quot;-10.522992739135&quot;  &quot;-5.2614963695675&quot; 
[19] &quot;0&quot;                 &quot;5.2614963695675&quot;   &quot;10.522992739135&quot;  
[22] &quot;15.7844891087025&quot;  &quot;21.04598547827&quot;    &quot;26.3074818478375&quot; 
[25] &quot;31.568978217405&quot;   &quot;36.8304745869725&quot;  &quot;42.09197095654&quot;   
[28] &quot;47.3534673261075&quot;  &quot;52.614963695675&quot;   &quot;57.8764600652426&quot; 
[31] &quot;63.1379564348101&quot;  &quot;68.3994528043776&quot;  &quot;73.6609491739451&quot; 
[34] &quot;78.9224455435126&quot;  &quot;84.1839419130801&quot;  &quot;89.4454382826476&quot; 
[37] &quot;94.7069346522151&quot; </code></pre>
<pre class="r"><code>plot(names(vec_peaktime), vec_hr[vec_peaktime])</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_peak_NO2 &lt;- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime])
df_peak_NO2$H_mM &lt;- as.numeric(df_peak_NO2$H_mM)

df_peak_NO2_threshold &lt;- df_peak_NO2[vec_max &gt; min_peak_NO2_mM, ]
dim(df_peak_NO2_threshold)</code></pre>
<pre><code>[1] 22  2</code></pre>
<pre class="r"><code>df_threshold &lt;- na.omit(df_peak_NO2_threshold)
dim(df_threshold)</code></pre>
<pre><code>[1] 22  2</code></pre>
<pre class="r"><code># 1. linear regression
fit.no2peak &lt;- lm(Time_hours ~ H_mM, df_threshold)
summary(fit.no2peak)</code></pre>
<pre><code>
Call:
lm(formula = Time_hours ~ H_mM, data = df_threshold)

Residuals:
    Min      1Q  Median      3Q     Max 
-20.536  -8.024  -4.955   2.206  47.545 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 12.78859    5.07953   2.518 0.020446 *  
H_mM        -0.41450    0.09828  -4.218 0.000423 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 15.39 on 20 degrees of freedom
Multiple R-squared:  0.4707,    Adjusted R-squared:  0.4443 
F-statistic: 17.79 on 1 and 20 DF,  p-value: 0.0004228</code></pre>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks (NO2_mM &gt; 0.02) \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]],
              color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=-40,y=40, label= paste0(&quot;y = &quot;, round(coef(fit.no2peak)[[1]],3),&quot;+&quot;,round(coef(fit.no2peak)[[2]],3),&quot;x&quot;, &quot;,  R^2: &quot;, round(summary(fit.no2peak)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. quadratic regression

df_threshold$H_mM_2 &lt;- (df_threshold$H_mM)^2 
quadraticModel_th &lt;- lm(Time_hours ~ H_mM + H_mM_2, df_threshold)
summary(quadraticModel_th)</code></pre>
<pre><code>
Call:
lm(formula = Time_hours ~ H_mM + H_mM_2, data = df_threshold)

Residuals:
    Min      1Q  Median      3Q     Max 
-14.954  -7.035  -5.311   3.786  53.127 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept) 15.021231   5.097467   2.947  0.00828 **
H_mM        -0.016691   0.268583  -0.062  0.95110   
H_mM_2       0.005041   0.003184   1.583  0.12993   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 14.84 on 19 degrees of freedom
Multiple R-squared:  0.5324,    Adjusted R-squared:  0.4832 
F-statistic: 10.82 on 2 and 19 DF,  p-value: 0.0007306</code></pre>
<pre class="r"><code>perturbValues &lt;- seq(-100, 10, 0.1)
peakPredict &lt;- predict(quadraticModel_th,list(H_mM=perturbValues, H_mM_2=perturbValues^2))
df_quad &lt;- data.frame(H_mM = perturbValues, Time_hours = peakPredict)
plot(perturbValues, peakPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks (NO2_mM &gt; 0.02) \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  # annotate(&quot;text&quot;,x=0,y=30, label= paste0(&quot;y = &quot;, round(coef(fit.no2peak)[[1]],3),&quot;+&quot;,round(coef(fit.no2peak)[[2]],3),&quot;x&quot;), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot H_mM with peak time
ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) +
  geom_point(size=2.5, shape=16, color = &quot;blue&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_line(size=1)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n NO2 (Nitrite) peak time (hr)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with pH perturbation and nitrite peaks (NO2_mM &gt; 0.02) \n&quot;) +
  geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = &quot;blue&quot;, size = 1) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]],
              # color = &quot;maroon2&quot;) +
  # show equation
  # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;))) +
  annotate(&quot;text&quot;,x=-45,y=40, label= paste0(&quot;y = &quot;, round(coef(quadraticModel_th)[[1]],3), &quot;+&quot;, round(coef(quadraticModel_th)[[2]],3),&quot;x+&quot;,round(coef(quadraticModel_th)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(quadraticModel_th)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-15-6.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="area-under-curve-calculation" class="section level2">
<h2>4.2. Area under curve calculation</h2>
<pre class="r"><code>df_plot2$Time_hours</code></pre>
<pre><code>  [1]  0.0  0.0  0.0  0.0  0.0  0.0  3.6  3.6  3.6  3.6  3.6  0.0  0.0  0.0  0.0
 [16]  0.0  0.0  3.6  3.6  3.6  3.6  3.6  3.6  0.0  0.0  0.0  0.0  0.0  0.0  3.6
 [31]  3.6  3.6  3.6  3.6  3.6  0.0  0.0  0.0  0.0  0.0  0.0  3.6  3.6  3.6  3.6
 [46]  3.6  3.6  0.0  0.0  0.0  0.0  0.0  0.0  3.6  3.6  3.6  3.6  3.6  3.6  0.0
 [61]  0.0  0.0  0.0  0.0  0.0  3.6  3.6  3.6  3.6  3.6  3.6  6.4  6.4  6.4  6.4
 [76]  6.4  6.4  9.7  9.7  9.7  9.7  9.7  9.7  6.4  6.4  6.4  6.4  6.4  6.4  9.7
 [91]  9.7  9.7  9.7  9.7  9.7  6.4  6.4  6.4  6.4  6.4  6.4  9.7  9.7  9.7  9.7
[106]  9.7  9.7  6.4  6.4  6.4  6.4  6.4  6.4  9.7  9.7  9.7  9.7  9.7  9.7  6.4
[121]  6.4  6.4  6.4  6.4  6.4  9.7  9.7  9.7  9.7  9.7  9.7  6.4  6.4  6.4  6.4
[136]  6.4  6.4  9.7  9.7  9.7  9.7  9.7  9.7 24.5 24.5 24.5 24.5 24.5 24.5 30.9
[151] 30.9 30.9 30.9 30.9 30.9 24.5 24.5 24.5 24.5 24.5 24.5 30.9 30.9 30.9 30.9
[166] 30.9 30.9 24.5 24.5 24.5 24.5 24.5 24.5 30.9 30.9 30.9 30.9 30.9 30.9 24.5
[181] 24.5 24.5 24.5 24.5 24.5 30.9 30.9 30.9 30.9 30.9 30.9 24.5 24.5 24.5 24.5
[196] 24.5 24.5 30.9 30.9 30.9 30.9 30.9 30.9 24.5 24.5 24.5 24.5 24.5 24.5 30.9
[211] 30.9 30.9 30.9 30.9 30.9 36.5 36.5 36.5 36.5 36.5 36.5 50.2 50.2 50.2 50.2
[226] 50.2 50.2 36.5 36.5 36.5 36.5 36.5 36.5 50.2 50.2 50.2 50.2 50.2 50.2 36.5
[241] 36.5 36.5 36.5 36.5 36.5 50.2 50.2 50.2 50.2 50.2 50.2 36.5 36.5 36.5 36.5
[256] 36.5 36.5 50.2 50.2 50.2 50.2 50.2 50.2 36.5 36.5 36.5 36.5 36.5 36.5 50.2
[271] 50.2 50.2 50.2 50.2 50.2 36.5 36.5 36.5 36.5 36.5 36.5 50.2 50.2 50.2 50.2
[286] 50.2 50.2 60.2 60.2 60.2 60.2 60.2 60.2 75.6 75.6 75.6 75.6 75.6 75.6 60.2
[301] 60.2 60.2 60.2 60.2 75.6 75.6 75.6 75.6 75.6 75.6 60.2 60.2 60.2 60.2 60.2
[316] 60.2 75.6 75.6 75.6 75.6 75.6 75.6 60.2 60.2 60.2 60.2 60.2 75.6 75.6 75.6
[331] 75.6 75.6 75.6 60.2 60.2 60.2 60.2 60.2 60.2 75.6 75.6 75.6 75.6 75.6 75.6
[346] 60.2 60.2 60.2 60.2 60.2 75.6 75.6 75.6 75.6 75.6 75.6 99.8 99.8 99.8 99.8
[361] 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8
[376] 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8
[391] 99.8 99.8  0.0  3.6  6.4  9.7 24.5 30.9 36.5 50.2 60.2 75.6 99.8</code></pre>
<pre class="r"><code>require(pracma)
colnames(df_plot2)</code></pre>
<pre><code>[1] &quot;H_mM&quot;       &quot;NO3_mM&quot;     &quot;NO2_mM&quot;     &quot;Soil&quot;       &quot;Time_hours&quot;
[6] &quot;pH&quot;        </code></pre>
<pre class="r"><code>df_corr &lt;- df_plot2 %&gt;% select(H_mM, Time_hours, NO3_mM, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(auc = trapz(Time_hours, NO3_mM)) %&gt;% ungroup()

plot(df_corr$H_mM, df_corr$auc)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_corr$H_mM_2 &lt;- (df_corr$H_mM)^2
fit.no3 &lt;- lm(auc ~ H_mM + H_mM_2, df_corr)
summary(fit.no3)</code></pre>
<pre><code>
Call:
lm(formula = auc ~ H_mM + H_mM_2, data = df_corr)

Residuals:
    Min      1Q  Median      3Q     Max 
-49.092 -15.109  -1.848   9.520  45.576 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 56.047602   5.522171   10.15 7.97e-12 ***
H_mM         1.076282   0.065493   16.43  &lt; 2e-16 ***
H_mM_2       0.013061   0.001305   10.01 1.14e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 22.38 on 34 degrees of freedom
Multiple R-squared:  0.9159,    Adjusted R-squared:  0.9109 
F-statistic: 185.1 on 2 and 34 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>perturbH &lt;- seq(-100, 100, 0.1)
aucPredict &lt;- predict(fit.no3,list(H_mM=perturbH, H_mM_2=perturbH^2))
df_auc_quad &lt;- data.frame(H_mM = perturbH, Time_hours = aucPredict)
plot(perturbH, aucPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (1) Plot fitted linear regression line
ggplot(df_corr, aes(x=H_mM, y=auc)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Area under curve (NO3-)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and area under nitrate curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=0,y=100, label= paste0(&quot;y = &quot;, round(coef(fit.no3)[[1]],3), &quot;+&quot;, round(coef(fit.no3)[[2]],3),&quot;x+&quot;,round(coef(fit.no3)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.no3)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 220213 Use the inferred pH values for x axis
# y &lt;- (6.5 / (1+ exp(1*(0.07*(x+125))-10))) + 2.5
df_corr$inferred_pH &lt;-  (6.5 / (1+ exp(1*(0.07*(df_corr$H_mM+125))-10))) + 2.5

plot(df_corr$inferred_pH, df_corr$auc)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_corr$inferred_pH_2 &lt;- (df_corr$inferred_pH)^2
fit.no3 &lt;- lm(auc ~ inferred_pH + inferred_pH_2, df_corr)
summary(fit.no3)</code></pre>
<pre><code>
Call:
lm(formula = auc ~ inferred_pH + inferred_pH_2, data = df_corr)

Residuals:
     Min       1Q   Median       3Q      Max 
-18.8860  -8.2231  -0.4106   6.8251  23.7550 

Coefficients:
               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    470.4813    15.0101   31.34   &lt;2e-16 ***
inferred_pH   -116.8025     6.2044  -18.83   &lt;2e-16 ***
inferred_pH_2    7.7646     0.5248   14.80   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 10.98 on 34 degrees of freedom
Multiple R-squared:  0.9797,    Adjusted R-squared:  0.9785 
F-statistic: 822.1 on 2 and 34 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>perturbpH &lt;- seq(2, 11, 0.01)
aucPredict &lt;- predict(fit.no3,list(inferred_pH=perturbpH, inferred_pH_2=perturbpH^2))
df_auc_quad &lt;- data.frame(H_mM = perturbpH, auc = aucPredict)
plot(perturbpH, aucPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (1) Plot fitted linear regression line
ggplot(df_corr, aes(x=inferred_pH, y=auc)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;inferred pH \n&quot;) +
  ylab(&quot;\n Area under curve (NO3-)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbed pH and area under nitrate curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_auc_quad, aes(x = H_mM, y = auc), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=7,y=100, label= paste0(&quot;y = &quot;, round(coef(fit.no3)[[1]],3), &quot;+&quot;, round(coef(fit.no3)[[2]],3),&quot;x+&quot;,round(coef(fit.no3)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.no3)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-6.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## NO2 area under curve
colnames(df_plot2)</code></pre>
<pre><code>[1] &quot;H_mM&quot;       &quot;NO3_mM&quot;     &quot;NO2_mM&quot;     &quot;Soil&quot;       &quot;Time_hours&quot;
[6] &quot;pH&quot;        </code></pre>
<pre class="r"><code>df_corr_no2 &lt;- df_plot2 %&gt;% select(H_mM, Time_hours, NO3_mM, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(auc = trapz(Time_hours, NO2_mM)) %&gt;% ungroup()

plot(df_corr_no2$H_mM, df_corr_no2$auc)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-7.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit.no2 &lt;- lm(auc ~ H_mM, df_corr_no2)
summary(fit.no2)</code></pre>
<pre><code>
Call:
lm(formula = auc ~ H_mM, data = df_corr_no2)

Residuals:
     Min       1Q   Median       3Q      Max 
-12.6569  -3.4259  -0.6752   3.3469  31.2322 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   9.7789     1.1799   8.288 9.05e-10 ***
H_mM         -0.1335     0.0210  -6.357 2.63e-07 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 7.177 on 35 degrees of freedom
Multiple R-squared:  0.5358,    Adjusted R-squared:  0.5226 
F-statistic: 40.41 on 1 and 35 DF,  p-value: 2.626e-07</code></pre>
<pre class="r"><code>df_corr_no2$H_mM_2 &lt;- (df_corr_no2$H_mM)^2 
fit.no2.quad &lt;- lm(auc ~ H_mM + H_mM_2, df_corr_no2)
summary(fit.no2.quad)</code></pre>
<pre><code>
Call:
lm(formula = auc ~ H_mM + H_mM_2, data = df_corr_no2)

Residuals:
     Min       1Q   Median       3Q      Max 
-16.1529  -1.4455  -0.1061   0.4766  24.6182 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  6.1884239  1.5958777   3.878 0.000459 ***
H_mM        -0.1335115  0.0189270  -7.054 3.82e-08 ***
H_mM_2       0.0011377  0.0003771   3.017 0.004809 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 6.468 on 34 degrees of freedom
Multiple R-squared:  0.6339,    Adjusted R-squared:  0.6123 
F-statistic: 29.43 on 2 and 34 DF,  p-value: 3.819e-08</code></pre>
<pre class="r"><code>perturbH &lt;- seq(-100, 100, 0.1)
aucPredict &lt;- predict(fit.no2.quad,list(H_mM=perturbH, H_mM_2=perturbH^2))
df_auc_quad &lt;- data.frame(H_mM = perturbH, Time_hours = aucPredict)
plot(perturbH, aucPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-8.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (1) Plot fitted linear regression line
ggplot(df_corr_no2, aes(x=H_mM, y=auc)) +
  geom_point(size=2.5, shape=21, color = &quot;brown&quot;) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Area under curve (NO2-)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and area under nitrite curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=0,y=20, label= paste0(&quot;y = &quot;, round(coef(fit.no2.quad)[[1]],3), &quot;+&quot;, round(coef(fit.no2.quad)[[2]],3),&quot;x+&quot;,round(coef(fit.no2.quad)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.no2.quad)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-16-9.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="get-ir-dynamics-reduction-rate-of-the-nitrite" class="section level2">
<h2>5. Get Ir dynamics (Reduction rate of the nitrite)</h2>
</div>
<div id="i-p" class="section level2">
<h2>5.1. I p</h2>
<pre class="r"><code>df_plot2</code></pre>
<pre><code>          H_mM     NO3_mM     NO2_mM               Soil Time_hours       pH
1     5.261496 2.21477315 0.06570583 LaBaghWoods_pH6.66        0.0 7.096627
2    15.784489 2.16541863 0.02789778 LaBaghWoods_pH6.66        0.0 5.985352
3    26.307482 2.12493322 0.02837579 LaBaghWoods_pH6.66        0.0 4.815855
4    36.830475 2.11048697 0.02810264 LaBaghWoods_pH6.66        0.0 3.861544
5    47.353467 2.11061553 0.02769292 LaBaghWoods_pH6.66        0.0 3.231713
6    57.876460 2.15242679 0.03069779 LaBaghWoods_pH6.66        0.0 2.872132
7     5.261496 2.28741092 0.03007993 LaBaghWoods_pH6.66        3.6 7.096627
8    26.307482 2.14076713 0.02875853 LaBaghWoods_pH6.66        3.6 4.815855
9    36.830475 1.84093055 0.02931490 LaBaghWoods_pH6.66        3.6 3.861544
10   47.353467 1.94738972 0.03056678 LaBaghWoods_pH6.66        3.6 3.231713
11   57.876460 1.90846253 0.02820219 LaBaghWoods_pH6.66        3.6 2.872132
12   10.522993 2.19132655 0.02858066 LaBaghWoods_pH6.66        0.0 6.566406
13   21.045985 2.13963436 0.02769292 LaBaghWoods_pH6.66        0.0 5.388768
14   31.568978 2.15012346 0.02769292 LaBaghWoods_pH6.66        0.0 4.299931
15   42.091971 2.12823669 0.02769292 LaBaghWoods_pH6.66        0.0 3.507052
16   52.614964 2.22937721 0.02817092 LaBaghWoods_pH6.66        0.0 3.024466
17   63.137956 2.26535315 0.02933186 LaBaghWoods_pH6.66        0.0 2.762104
18   10.522993 2.02263211 0.03167965 LaBaghWoods_pH6.66        3.6 6.566406
19   21.045985 2.03622658 0.02917580 LaBaghWoods_pH6.66        3.6 5.388768
20   31.568978 1.92734473 0.02841082 LaBaghWoods_pH6.66        3.6 4.299931
21   42.091971 2.22618893 0.02952354 LaBaghWoods_pH6.66        3.6 3.507052
22   52.614964 1.87219957 0.02868899 LaBaghWoods_pH6.66        3.6 3.024466
23   63.137956 2.08018309 0.02959308 LaBaghWoods_pH6.66        3.6 2.762104
24   68.399453 2.19707882 0.02803435 LaBaghWoods_pH6.66        0.0 2.683633
25   78.922446 2.23342482 0.03083438 LaBaghWoods_pH6.66        0.0 2.589225
26   89.445438 2.13264991 0.02912699 LaBaghWoods_pH6.66        0.0 2.543023
27   -5.261496 2.06855378 0.03144909 LaBaghWoods_pH6.66        0.0 7.924648
28  -15.784489 2.13867774 0.04012587 LaBaghWoods_pH6.66        0.0 8.436606
29  -26.307482 2.09039107 0.05161118 LaBaghWoods_pH6.66        0.0 8.717521
30   68.399453 2.41387674 0.02820219 LaBaghWoods_pH6.66        3.6 2.683633
31   78.922446 2.10119370 0.03105365 LaBaghWoods_pH6.66        3.6 2.589225
32   89.445438 2.35398013 0.03161010 LaBaghWoods_pH6.66        3.6 2.543023
33   -5.261496 1.91874988 0.04155977 LaBaghWoods_pH6.66        3.6 7.924648
34  -15.784489 2.06292489 0.04559707 LaBaghWoods_pH6.66        3.6 8.436606
35  -26.307482 1.82450989 0.05193350 LaBaghWoods_pH6.66        3.6 8.717521
36   73.660949 2.29098970 0.02933186 LaBaghWoods_pH6.66        0.0 2.628172
37   84.183942 2.24560320 0.03056119 LaBaghWoods_pH6.66        0.0 2.561998
38   94.706935 2.21212638 0.03138079 LaBaghWoods_pH6.66        0.0 2.529829
39  -10.522993 2.15803610 0.03431805 LaBaghWoods_pH6.66        0.0 8.215996
40  -21.045985 2.26598649 0.05085891 LaBaghWoods_pH6.66        0.0 8.599489
41  -31.568978 2.10694995 0.04887582 LaBaghWoods_pH6.66        0.0 8.801899
42   73.660949 2.15737721 0.02938444 LaBaghWoods_pH6.66        3.6 2.628172
43   84.183942 2.35718770 0.03056678 LaBaghWoods_pH6.66        3.6 2.561998
44   94.706935 2.12791883 0.03300129 LaBaghWoods_pH6.66        3.6 2.529829
45  -10.522993 2.24540847 0.04135097 LaBaghWoods_pH6.66        3.6 8.215996
46  -21.045985 1.92710112 0.05067994 LaBaghWoods_pH6.66        3.6 8.599489
47  -31.568978 1.99341582 0.05416229 LaBaghWoods_pH6.66        3.6 8.801899
48  -36.830475 2.13937665 0.04969638 LaBaghWoods_pH6.66        0.0 8.861633
49  -47.353467 2.15929298 0.05475743 LaBaghWoods_pH6.66        0.0 8.933016
50  -57.876460 2.02932867 0.05626239 LaBaghWoods_pH6.66        0.0 8.967759
51  -68.399453 1.96206020 0.06070968 LaBaghWoods_pH6.66        0.0 8.984525
52  -78.922446 1.93601403 0.05284225 LaBaghWoods_pH6.66        0.0 8.992582
53  -89.445438 1.92261133 0.05393661 LaBaghWoods_pH6.66        0.0 8.996447
54  -36.830475 2.21845272 0.05144599 LaBaghWoods_pH6.66        3.6 8.861633
55  -47.353467 1.89889353 0.05381402 LaBaghWoods_pH6.66        3.6 8.933016
56  -57.876460 2.01074174 0.05520714 LaBaghWoods_pH6.66        3.6 8.967759
57  -68.399453 1.65295962 0.06538061 LaBaghWoods_pH6.66        3.6 8.984525
58  -78.922446 1.71458001 0.07263142 LaBaghWoods_pH6.66        3.6 8.992582
59  -89.445438 1.35317579 0.08044366 LaBaghWoods_pH6.66        3.6 8.996447
60  -42.091971 2.05233300 0.04826043 LaBaghWoods_pH6.66        0.0 8.903631
61  -52.614964 2.05044214 0.05537308 LaBaghWoods_pH6.66        0.0 8.953506
62  -63.137956 1.97974414 0.05715174 LaBaghWoods_pH6.66        0.0 8.977658
63  -73.660949 2.03502055 0.05776748 LaBaghWoods_pH6.66        0.0 8.989285
64  -84.183942 1.94064271 0.06365242 LaBaghWoods_pH6.66        0.0 8.994866
65  -94.706935 2.09340960 0.06502133 LaBaghWoods_pH6.66        0.0 8.997541
66  -42.091971 2.14078237 0.05590375 LaBaghWoods_pH6.66        3.6 8.903631
67  -52.614964 1.82006814 0.05813309 LaBaghWoods_pH6.66        3.6 8.953506
68  -63.137956 1.79189466 0.06217457 LaBaghWoods_pH6.66        3.6 8.977658
69  -73.660949 1.74470709 0.06970281 LaBaghWoods_pH6.66        3.6 8.989285
70  -84.183942 1.45223315 0.07535132 LaBaghWoods_pH6.66        3.6 8.994866
71  -94.706935 1.42416426 0.08009481 LaBaghWoods_pH6.66        3.6 8.997541
72    5.261496 1.99215790 0.06770988 LaBaghWoods_pH6.66        6.4 7.096627
73   15.784489 1.93409091 0.03205179 LaBaghWoods_pH6.66        6.4 5.985352
74   26.307482 1.86681274 0.03198263 LaBaghWoods_pH6.66        6.4 4.815855
75   36.830475 1.99895582 0.02976962 LaBaghWoods_pH6.66        6.4 3.861544
76   47.353467 1.94645412 0.03046115 LaBaghWoods_pH6.66        6.4 3.231713
77   57.876460 1.86003790 0.03039200 LaBaghWoods_pH6.66        6.4 2.872132
78    5.261496 1.68221850 0.06074843 LaBaghWoods_pH6.66        9.7 7.096627
79   15.784489 1.82769758 0.04004772 LaBaghWoods_pH6.66        9.7 5.985352
80   26.307482 2.20240053 0.03080099 LaBaghWoods_pH6.66        9.7 4.815855
81   36.830475 2.02870782 0.03150666 LaBaghWoods_pH6.66        9.7 3.861544
82   47.353467 1.91304252 0.02960141 LaBaghWoods_pH6.66        9.7 3.231713
83   57.876460 1.86070558 0.02861360 LaBaghWoods_pH6.66        9.7 2.872132
84   10.522993 1.83212331 0.04291385 LaBaghWoods_pH6.66        6.4 6.566406
85   21.045985 1.94547132 0.02893983 LaBaghWoods_pH6.66        6.4 5.388768
86   31.568978 1.86188034 0.02804093 LaBaghWoods_pH6.66        6.4 4.299931
87   42.091971 1.80031750 0.02824836 LaBaghWoods_pH6.66        6.4 3.507052
88   52.614964 1.95002646 0.02804093 LaBaghWoods_pH6.66        6.4 3.024466
89   63.137956 1.96692729 0.02804093 LaBaghWoods_pH6.66        6.4 2.762104
90   10.522993 1.75254834 0.08685623 LaBaghWoods_pH6.66        9.7 6.566406
91   21.045985 1.83494337 0.03108325 LaBaghWoods_pH6.66        9.7 5.388768
92   31.568978 1.86570547 0.02861360 LaBaghWoods_pH6.66        9.7 4.299931
93   42.091971 1.95435993 0.02861360 LaBaghWoods_pH6.66        9.7 3.507052
94   52.614964 1.85086900 0.02861360 LaBaghWoods_pH6.66        9.7 3.024466
95   63.137956 1.92632921 0.03044816 LaBaghWoods_pH6.66        9.7 2.762104
96   68.399453 2.22704140 0.02935472 LaBaghWoods_pH6.66        6.4 2.683633
97   78.922446 2.00481532 0.03343508 LaBaghWoods_pH6.66        6.4 2.589225
98   89.445438 1.96980900 0.03122187 LaBaghWoods_pH6.66        6.4 2.543023
99   -5.261496 2.00850448 0.11720977 LaBaghWoods_pH6.66        6.4 7.924648
100 -15.784489 1.79260349 0.19730597 LaBaghWoods_pH6.66        6.4 8.436606
101 -26.307482 1.79953837 0.09553022 LaBaghWoods_pH6.66        6.4 8.717521
102  68.399453 2.14430954 0.03009534 LaBaghWoods_pH6.66        9.7 2.683633
103  78.922446 2.23540185 0.03277695 LaBaghWoods_pH6.66        9.7 2.589225
104  89.445438 2.49557160 0.03475314 LaBaghWoods_pH6.66        9.7 2.543023
105  -5.261496 1.69715735 0.13181208 LaBaghWoods_pH6.66        9.7 7.924648
106 -15.784489 1.40411568 0.41859861 LaBaghWoods_pH6.66        9.7 8.436606
107 -26.307482 1.61933413 0.18166488 LaBaghWoods_pH6.66        9.7 8.717521
108  73.660949 2.07981960 0.03011538 LaBaghWoods_pH6.66        6.4 2.628172
109  84.183942 1.93043662 0.03136019 LaBaghWoods_pH6.66        6.4 2.561998
110  94.706935 2.04800716 0.03101440 LaBaghWoods_pH6.66        6.4 2.529829
111 -10.522993 1.77674725 0.19513963 LaBaghWoods_pH6.66        6.4 8.215996
112 -21.045985 1.81305172 0.14295800 LaBaghWoods_pH6.66        6.4 8.599489
113 -31.568978 1.87200437 0.07117609 LaBaghWoods_pH6.66        6.4 8.801899
114  73.660949 2.36011773 0.03122439 LaBaghWoods_pH6.66        9.7 2.628172
115  84.183942 2.24183704 0.03418849 LaBaghWoods_pH6.66        9.7 2.561998
116  94.706935 2.30688140 0.03256522 LaBaghWoods_pH6.66        9.7 2.529829
117 -10.522993 1.55318951 0.40985389 LaBaghWoods_pH6.66        9.7 8.215996
118 -21.045985 1.64789519 0.29983629 LaBaghWoods_pH6.66        9.7 8.599489
119 -31.568978 1.76678123 0.11740410 LaBaghWoods_pH6.66        9.7 8.801899
120 -36.830475 1.77394345 0.06271986 LaBaghWoods_pH6.66        6.4 8.861633
121 -47.353467 1.74632853 0.06327423 LaBaghWoods_pH6.66        6.4 8.933016
122 -57.876460 1.77449505 0.06667016 LaBaghWoods_pH6.66        6.4 8.967759
123 -68.399453 1.56868075 0.07492045 LaBaghWoods_pH6.66        6.4 8.984525
124 -78.922446 1.32134509 0.07713975 LaBaghWoods_pH6.66        6.4 8.992582
125 -89.445438 1.20631763 0.07928999 LaBaghWoods_pH6.66        6.4 8.996447
126 -36.830475 1.91376060 0.08218346 LaBaghWoods_pH6.66        9.7 8.861633
127 -47.353467 1.85920822 0.06428361 LaBaghWoods_pH6.66        9.7 8.933016
128 -57.876460 2.01551543 0.06838539 LaBaghWoods_pH6.66        9.7 8.967759
129 -68.399453 1.50450449 0.07178074 LaBaghWoods_pH6.66        9.7 8.984525
130 -78.922446 1.24869577 0.07645052 LaBaghWoods_pH6.66        9.7 8.992582
131 -89.445438 0.99884789 0.08190030 LaBaghWoods_pH6.66        9.7 8.996447
132 -42.091971 1.76212153 0.06209622 LaBaghWoods_pH6.66        6.4 8.903631
133 -52.614964 1.64193598 0.06978952 LaBaghWoods_pH6.66        6.4 8.953506
134 -63.137956 1.48520361 0.07422698 LaBaghWoods_pH6.66        6.4 8.977658
135 -73.660949 1.42105864 0.07173076 LaBaghWoods_pH6.66        6.4 8.989285
136 -84.183942 1.31329724 0.08345257 LaBaghWoods_pH6.66        6.4 8.994866
137 -94.706935 1.23816115 0.08241182 LaBaghWoods_pH6.66        6.4 8.997541
138 -42.091971 2.02730969 0.06916343 LaBaghWoods_pH6.66        9.7 8.903631
139 -52.614964 1.82518173 0.06838539 LaBaghWoods_pH6.66        9.7 8.953506
140 -63.137956 1.71225352 0.07192223 LaBaghWoods_pH6.66        9.7 8.977658
141 -73.660949 1.41518213 0.07701665 LaBaghWoods_pH6.66        9.7 8.989285
142 -84.183942 1.18291892 0.08182951 LaBaghWoods_pH6.66        9.7 8.994866
143 -94.706935 0.99194496 0.08430728 LaBaghWoods_pH6.66        9.7 8.997541
144   5.261496 0.43171887 0.02838484 LaBaghWoods_pH6.66       24.5 7.096627
145  15.784489 1.27705984 0.06120590 LaBaghWoods_pH6.66       24.5 5.985352
146  26.307482 1.60570392 0.02910183 LaBaghWoods_pH6.66       24.5 4.815855
147  36.830475 1.66720020 0.02760271 LaBaghWoods_pH6.66       24.5 3.861544
148  47.353467 2.10036020 0.02642959 LaBaghWoods_pH6.66       24.5 3.231713
149  57.876460 2.25804331 0.02740718 LaBaghWoods_pH6.66       24.5 2.872132
150   5.261496 0.00000000 0.03029344 LaBaghWoods_pH6.66       30.9 7.096627
151  15.784489 0.94184123 0.06194423 LaBaghWoods_pH6.66       30.9 5.985352
152  26.307482 1.36168338 0.02846156 LaBaghWoods_pH6.66       30.9 4.815855
153  36.830475 1.67748826 0.02751178 LaBaghWoods_pH6.66       30.9 3.861544
154  47.353467 1.77399774 0.03158267 LaBaghWoods_pH6.66       30.9 3.231713
155  57.876460 1.80205908 0.02751178 LaBaghWoods_pH6.66       30.9 2.872132
156  10.522993 0.84251664 0.05408748 LaBaghWoods_pH6.66       24.5 6.566406
157  21.045985 1.43068465 0.03275249 LaBaghWoods_pH6.66       24.5 5.388768
158  31.568978 1.64691837 0.02642959 LaBaghWoods_pH6.66       24.5 4.299931
159  42.091971 1.85540718 0.02642959 LaBaghWoods_pH6.66       24.5 3.507052
160  52.614964 2.06109089 0.02642959 LaBaghWoods_pH6.66       24.5 3.024466
161  63.137956 2.15559304 0.02675544 LaBaghWoods_pH6.66       24.5 2.762104
162  10.522993 0.35710705 0.03151481 LaBaghWoods_pH6.66       30.9 6.566406
163  21.045985 1.17968574 0.03796252 LaBaghWoods_pH6.66       30.9 5.388768
164  31.568978 1.57654306 0.02751178 LaBaghWoods_pH6.66       30.9 4.299931
165  42.091971 1.82520542 0.02941139 LaBaghWoods_pH6.66       30.9 3.507052
166  52.614964 1.84092635 0.02751178 LaBaghWoods_pH6.66       30.9 3.024466
167  63.137956 1.95575526 0.03165052 LaBaghWoods_pH6.66       30.9 2.762104
168  68.399453 2.06800135 0.02786341 LaBaghWoods_pH6.66       24.5 2.683633
169  78.922446 2.21983668 0.02688579 LaBaghWoods_pH6.66       24.5 2.589225
170  89.445438 2.23253060 0.02760271 LaBaghWoods_pH6.66       24.5 2.543023
171  -5.261496 0.46343072 0.03549106 LaBaghWoods_pH6.66       24.5 7.924648
172 -15.784489 0.00000000 0.31756757 LaBaghWoods_pH6.66       24.5 8.436606
173 -26.307482 0.77928412 0.79408193 LaBaghWoods_pH6.66       24.5 8.717521
174  68.399453 2.04134446 0.02778314 LaBaghWoods_pH6.66       30.9 2.683633
175  78.922446 2.19805632 0.02934355 LaBaghWoods_pH6.66       30.9 2.589225
176  89.445438 1.96309644 0.03002203 LaBaghWoods_pH6.66       30.9 2.543023
177  -5.261496 0.01669228 0.03436505 LaBaghWoods_pH6.66       30.9 7.924648
178 -15.784489 0.00000000 0.04380170 LaBaghWoods_pH6.66       30.9 8.436606
179 -26.307482 0.00000000 0.68684041 LaBaghWoods_pH6.66       30.9 8.717521
180  73.660949 2.34616689 0.02779824 LaBaghWoods_pH6.66       24.5 2.628172
181  84.183942 2.17282388 0.02858038 LaBaghWoods_pH6.66       24.5 2.561998
182  94.706935 2.16925992 0.03092705 LaBaghWoods_pH6.66       24.5 2.529829
183 -10.522993 0.04850377 0.07355686 LaBaghWoods_pH6.66       24.5 8.215996
184 -21.045985 0.29372756 0.84637617 LaBaghWoods_pH6.66       24.5 8.599489
185 -31.568978 0.00000000 1.28382145 LaBaghWoods_pH6.66       24.5 8.801899
186  73.660949 2.16899657 0.02751178 LaBaghWoods_pH6.66       30.9 2.628172
187  84.183942 2.19992540 0.02751178 LaBaghWoods_pH6.66       30.9 2.561998
188  94.706935 2.09784224 0.02839371 LaBaghWoods_pH6.66       30.9 2.529829
189 -10.522993 0.00000000 0.03959184 LaBaghWoods_pH6.66       30.9 8.215996
190 -21.045985 0.00000000 0.11104163 LaBaghWoods_pH6.66       30.9 8.599489
191 -31.568978 0.00000000 0.05324406 LaBaghWoods_pH6.66       30.9 8.801899
192 -36.830475 0.00000000 0.05291229 LaBaghWoods_pH6.66       24.5 8.861633
193 -47.353467 0.00000000 1.23199660 LaBaghWoods_pH6.66       24.5 8.933016
194 -57.876460 0.00000000 1.30175706 LaBaghWoods_pH6.66       24.5 8.967759
195 -68.399453 1.63526211 0.07120351 LaBaghWoods_pH6.66       24.5 8.984525
196 -78.922446 1.54955663 0.08362815 LaBaghWoods_pH6.66       24.5 8.992582
197 -89.445438 1.35521984 0.08788106 LaBaghWoods_pH6.66       24.5 8.996447
198 -36.830475 0.00000000 0.05310815 LaBaghWoods_pH6.66       30.9 8.861633
199 -47.353467 0.00000000 0.05929288 LaBaghWoods_pH6.66       30.9 8.933016
200 -57.876460 0.00000000 0.06296410 LaBaghWoods_pH6.66       30.9 8.967759
201 -68.399453 1.32957899 0.29001540 LaBaghWoods_pH6.66       30.9 8.984525
202 -78.922446 1.42321176 0.07956335 LaBaghWoods_pH6.66       30.9 8.992582
203 -89.445438 1.11940225 0.08936794 LaBaghWoods_pH6.66       30.9 8.996447
204 -42.091971 0.00000000 0.05545865 LaBaghWoods_pH6.66       24.5 8.903631
205 -52.614964 0.00000000 1.69965253 LaBaghWoods_pH6.66       24.5 8.953506
206 -63.137956 1.65412042 0.08781562 LaBaghWoods_pH6.66       24.5 8.977658
207 -73.660949 1.53153939 0.08369357 LaBaghWoods_pH6.66       24.5 8.989285
208 -84.183942 1.43351834 0.08873178 LaBaghWoods_pH6.66       24.5 8.994866
209 -94.706935 1.26031825 0.08847001 LaBaghWoods_pH6.66       24.5 8.997541
210 -42.091971 0.00000000 0.05603030 LaBaghWoods_pH6.66       30.9 8.903631
211 -52.614964 0.00000000 0.06384804 LaBaghWoods_pH6.66       30.9 8.953506
212 -63.137956 0.00000000 1.74445642 LaBaghWoods_pH6.66       30.9 8.977658
213 -73.660949 1.58534676 0.08174161 LaBaghWoods_pH6.66       30.9 8.989285
214 -84.183942 1.37055226 0.09038960 LaBaghWoods_pH6.66       30.9 8.994866
215 -94.706935 1.11665654 0.09931500 LaBaghWoods_pH6.66       30.9 8.997541
216   5.261496 0.00000000 0.02697399 LaBaghWoods_pH6.66       36.5 7.096627
217  15.784489 0.52747287 0.04913892 LaBaghWoods_pH6.66       36.5 5.985352
218  26.307482 1.21716806 0.03143102 LaBaghWoods_pH6.66       36.5 4.815855
219  36.830475 1.67990038 0.02697399 LaBaghWoods_pH6.66       36.5 3.861544
220  47.353467 1.83385970 0.02697399 LaBaghWoods_pH6.66       36.5 3.231713
221  57.876460 1.94415727 0.02843735 LaBaghWoods_pH6.66       36.5 2.872132
222   5.261496 0.00000000 0.02834206 LaBaghWoods_pH6.66       50.2 7.096627
223  15.784489 0.00000000 0.02654055 LaBaghWoods_pH6.66       50.2 5.985352
224  26.307482 0.11010524 0.03104475 LaBaghWoods_pH6.66       50.2 4.815855
225  36.830475 1.58464262 0.02705525 LaBaghWoods_pH6.66       50.2 3.861544
226  47.353467 1.68888462 0.02892117 LaBaghWoods_pH6.66       50.2 3.231713
227  57.876460 1.69823196 0.02840641 LaBaghWoods_pH6.66       50.2 2.872132
228  10.522993 0.00000000 0.02730656 LaBaghWoods_pH6.66       36.5 6.566406
229  21.045985 0.78783577 0.03974985 LaBaghWoods_pH6.66       36.5 5.388768
230  31.568978 1.45646247 0.02797172 LaBaghWoods_pH6.66       36.5 4.299931
231  42.091971 1.61366782 0.02730656 LaBaghWoods_pH6.66       36.5 3.507052
232  52.614964 1.79010132 0.02750610 LaBaghWoods_pH6.66       36.5 3.024466
233  63.137956 2.12466662 0.02863691 LaBaghWoods_pH6.66       36.5 2.762104
234  10.522993 0.00000000 0.02609021 LaBaghWoods_pH6.66       50.2 6.566406
235  21.045985 0.00000000 0.02621887 LaBaghWoods_pH6.66       50.2 5.388768
236  31.568978 1.15449242 0.02666922 LaBaghWoods_pH6.66       50.2 4.299931
237  42.091971 1.63498389 0.02750562 LaBaghWoods_pH6.66       50.2 3.507052
238  52.614964 1.89231892 0.02872813 LaBaghWoods_pH6.66       50.2 3.024466
239  63.137956 2.09911203 0.02872813 LaBaghWoods_pH6.66       50.2 2.762104
240  68.399453 1.94159936 0.02903604 LaBaghWoods_pH6.66       36.5 2.683633
241  78.922446 2.12949782 0.02857039 LaBaghWoods_pH6.66       36.5 2.589225
242  89.445438 2.10303159 0.02996738 LaBaghWoods_pH6.66       36.5 2.543023
243  -5.261496 0.00000000 0.03322751 LaBaghWoods_pH6.66       36.5 7.924648
244 -15.784489 0.00000000 0.04734056 LaBaghWoods_pH6.66       36.5 8.436606
245 -26.307482 0.00000000 0.05093748 LaBaghWoods_pH6.66       36.5 8.717521
246  68.399453 2.30587553 0.02744128 LaBaghWoods_pH6.66       50.2 2.683633
247  78.922446 2.27987993 0.02859944 LaBaghWoods_pH6.66       50.2 2.589225
248  89.445438 2.26989195 0.02956465 LaBaghWoods_pH6.66       50.2 2.543023
249  -5.261496 0.00000000 0.03387666 LaBaghWoods_pH6.66       50.2 7.924648
250 -15.784489 0.00000000 0.04108760 LaBaghWoods_pH6.66       50.2 8.436606
251 -26.307482 0.00000000 0.04624043 LaBaghWoods_pH6.66       50.2 8.717521
252  73.660949 2.09175828 0.02910256 LaBaghWoods_pH6.66       36.5 2.628172
253  84.183942 2.20380266 0.03069918 LaBaghWoods_pH6.66       36.5 2.561998
254  94.706935 2.17946967 0.03256212 LaBaghWoods_pH6.66       36.5 2.529829
255 -10.522993 0.00000000 0.04394425 LaBaghWoods_pH6.66       36.5 8.215996
256 -21.045985 0.00000000 0.04487650 LaBaghWoods_pH6.66       36.5 8.599489
257 -31.568978 0.00000000 0.05153705 LaBaghWoods_pH6.66       36.5 8.801899
258  73.660949 2.13253197 0.02744128 LaBaghWoods_pH6.66       50.2 2.628172
259  84.183942 1.95032743 0.02756996 LaBaghWoods_pH6.66       50.2 2.561998
260  94.706935 2.35847664 0.02995074 LaBaghWoods_pH6.66       50.2 2.529829
261 -10.522993 0.00000000 0.04495205 LaBaghWoods_pH6.66       50.2 8.215996
262 -21.045985 0.00000000 0.04649812 LaBaghWoods_pH6.66       50.2 8.599489
263 -31.568978 0.00000000 0.04939742 LaBaghWoods_pH6.66       50.2 8.801899
264 -36.830475 0.00000000 0.05280288 LaBaghWoods_pH6.66       36.5 8.861633
265 -47.353467 0.00000000 0.05726744 LaBaghWoods_pH6.66       36.5 8.933016
266 -57.876460 0.00000000 0.06179997 LaBaghWoods_pH6.66       36.5 8.967759
267 -68.399453 0.00000000 1.17437834 LaBaghWoods_pH6.66       36.5 8.984525
268 -78.922446 1.31733280 0.08762101 LaBaghWoods_pH6.66       36.5 8.992582
269 -89.445438 1.16453852 0.08481661 LaBaghWoods_pH6.66       36.5 8.996447
270 -36.830475 0.00000000 0.05216839 LaBaghWoods_pH6.66       50.2 8.861633
271 -47.353467 0.00000000 0.05435977 LaBaghWoods_pH6.66       50.2 8.933016
272 -57.876460 0.00000000 0.05990410 LaBaghWoods_pH6.66       50.2 8.967759
273 -68.399453 0.00000000 0.06751482 LaBaghWoods_pH6.66       50.2 8.984525
274 -78.922446 0.00000000 0.06951489 LaBaghWoods_pH6.66       50.2 8.992582
275 -89.445438 0.99914325 0.09301995 LaBaghWoods_pH6.66       50.2 8.996447
276 -42.091971 0.00000000 0.05620116 LaBaghWoods_pH6.66       36.5 8.903631
277 -52.614964 0.00000000 0.06506689 LaBaghWoods_pH6.66       36.5 8.953506
278 -63.137956 0.00000000 0.06820112 LaBaghWoods_pH6.66       36.5 8.977658
279 -73.660949 1.29293341 0.08949089 LaBaghWoods_pH6.66       36.5 8.989285
280 -84.183942 1.18814457 0.09156139 LaBaghWoods_pH6.66       36.5 8.994866
281 -94.706935 1.05440239 0.09216255 LaBaghWoods_pH6.66       36.5 8.997541
282 -42.091971 0.00000000 0.05055730 LaBaghWoods_pH6.66       50.2 8.903631
283 -52.614964 0.00000000 0.05919482 LaBaghWoods_pH6.66       50.2 8.953506
284 -63.137956 0.00000000 0.06512798 LaBaghWoods_pH6.66       50.2 8.977658
285 -73.660949 0.00000000 0.07493582 LaBaghWoods_pH6.66       50.2 8.989285
286 -84.183942 0.00000000 1.06861160 LaBaghWoods_pH6.66       50.2 8.994866
287 -94.706935 1.00755369 0.09928998 LaBaghWoods_pH6.66       50.2 8.997541
288   5.261496 0.00000000 0.02746795 LaBaghWoods_pH6.66       60.2 7.096627
289  15.784489 0.00000000 0.02791151 LaBaghWoods_pH6.66       60.2 5.985352
290  26.307482 0.00000000 0.02708777 LaBaghWoods_pH6.66       60.2 4.815855
291  36.830475 1.42653199 0.02569386 LaBaghWoods_pH6.66       60.2 3.861544
292  47.353467 1.73971754 0.02683432 LaBaghWoods_pH6.66       60.2 3.231713
293  57.876460 1.69295628 0.02829171 LaBaghWoods_pH6.66       60.2 2.872132
294   5.261496 0.00000000 0.03205397 LaBaghWoods_pH6.66       75.6 7.096627
295  15.784489 0.00000000 0.02917537 LaBaghWoods_pH6.66       75.6 5.985352
296  26.307482 0.00000000 0.02580756 LaBaghWoods_pH6.66       75.6 4.815855
297  36.830475 1.45312662 0.02635860 LaBaghWoods_pH6.66       75.6 3.861544
298  47.353467 1.54375015 0.02837927 LaBaghWoods_pH6.66       75.6 3.231713
299  57.876460 1.98547847 0.02482798 LaBaghWoods_pH6.66       75.6 2.872132
300  10.522993 0.00000000 0.02594729 LaBaghWoods_pH6.66       60.2 6.566406
301  21.045985 0.00000000 0.02569386 LaBaghWoods_pH6.66       60.2 5.388768
302  31.568978 0.44584098 0.02740459 LaBaghWoods_pH6.66       60.2 4.299931
303  52.614964 1.74513253 0.02569386 LaBaghWoods_pH6.66       60.2 3.024466
304  63.137956 1.97176886 0.02759468 LaBaghWoods_pH6.66       60.2 2.762104
305  10.522993 0.00000000 0.02733828 LaBaghWoods_pH6.66       75.6 6.566406
306  21.045985 0.00000000 0.02617491 LaBaghWoods_pH6.66       75.6 5.388768
307  31.568978 0.00000000 0.02482798 LaBaghWoods_pH6.66       75.6 4.299931
308  42.091971 1.48396437 0.02501164 LaBaghWoods_pH6.66       75.6 3.507052
309  52.614964 1.91663676 0.02629737 LaBaghWoods_pH6.66       75.6 3.024466
310  63.137956 1.78921830 0.02525653 LaBaghWoods_pH6.66       75.6 2.762104
311  68.399453 1.92774876 0.02664424 LaBaghWoods_pH6.66       60.2 2.683633
312  78.922446 2.03789847 0.02696105 LaBaghWoods_pH6.66       60.2 2.589225
313  89.445438 2.13450650 0.02626408 LaBaghWoods_pH6.66       60.2 2.543023
314  -5.261496 0.00000000 0.03494686 LaBaghWoods_pH6.66       60.2 7.924648
315 -15.784489 0.00000000 0.04414230 LaBaghWoods_pH6.66       60.2 8.436606
316 -26.307482 0.00000000 0.04598208 LaBaghWoods_pH6.66       60.2 8.717521
317  68.399453 2.22476724 0.02525653 LaBaghWoods_pH6.66       75.6 2.683633
318  78.922446 2.42127453 0.02684843 LaBaghWoods_pH6.66       75.6 2.589225
319  89.445438 2.45722064 0.02905289 LaBaghWoods_pH6.66       75.6 2.543023
320  -5.261496 0.00000000 0.03928366 LaBaghWoods_pH6.66       75.6 7.924648
321 -15.784489 0.00000000 0.04124490 LaBaghWoods_pH6.66       75.6 8.436606
322 -26.307482 0.00000000 0.04480033 LaBaghWoods_pH6.66       75.6 8.717521
323  73.660949 2.09285324 0.02829171 LaBaghWoods_pH6.66       60.2 2.628172
324  84.183942 2.13462774 0.02936901 LaBaghWoods_pH6.66       60.2 2.561998
325  94.706935 2.03743004 0.02886204 LaBaghWoods_pH6.66       60.2 2.529829
326 -21.045985 0.00000000 0.04376168 LaBaghWoods_pH6.66       60.2 8.599489
327 -31.568978 0.00000000 0.04871046 LaBaghWoods_pH6.66       60.2 8.801899
328  73.660949 2.45073004 0.02733828 LaBaghWoods_pH6.66       75.6 2.628172
329  84.183942 2.45960510 0.02733828 LaBaghWoods_pH6.66       75.6 2.561998
330  94.706935 2.44619214 0.02991028 LaBaghWoods_pH6.66       75.6 2.529829
331 -10.522993 0.00000000 0.04247081 LaBaghWoods_pH6.66       75.6 8.215996
332 -21.045985 0.00000000 0.04265470 LaBaghWoods_pH6.66       75.6 8.599489
333 -31.568978 0.00000000 0.04418726 LaBaghWoods_pH6.66       75.6 8.801899
334 -36.830475 0.00000000 0.04997965 LaBaghWoods_pH6.66       60.2 8.861633
335 -47.353467 0.00000000 0.05289920 LaBaghWoods_pH6.66       60.2 8.933016
336 -57.876460 0.00000000 0.05975611 LaBaghWoods_pH6.66       60.2 8.967759
337 -68.399453 0.00000000 0.07017463 LaBaghWoods_pH6.66       60.2 8.984525
338 -78.922446 0.00000000 0.08085499 LaBaghWoods_pH6.66       60.2 8.992582
339 -89.445438 0.00000000 1.45055613 LaBaghWoods_pH6.66       60.2 8.996447
340 -36.830475 0.00000000 0.41928339 LaBaghWoods_pH6.66       75.6 8.861633
341 -47.353467 0.00000000 0.05099383 LaBaghWoods_pH6.66       75.6 8.933016
342 -57.876460 0.00000000 0.05657645 LaBaghWoods_pH6.66       75.6 8.967759
343 -68.399453 0.00000000 0.07057340 LaBaghWoods_pH6.66       75.6 8.984525
344 -78.922446 0.00000000 0.07800727 LaBaghWoods_pH6.66       75.6 8.992582
345 -89.445438 0.00000000 0.08716678 LaBaghWoods_pH6.66       75.6 8.996447
346 -42.091971 0.00000000 0.04997965 LaBaghWoods_pH6.66       60.2 8.903631
347 -52.614964 0.00000000 0.06115328 LaBaghWoods_pH6.66       60.2 8.953506
348 -63.137956 0.00000000 0.06369394 LaBaghWoods_pH6.66       60.2 8.977658
349 -84.183942 0.00000000 0.08040982 LaBaghWoods_pH6.66       60.2 8.994866
350 -94.706935 0.48543913 0.08734333 LaBaghWoods_pH6.66       60.2 8.997541
351 -42.091971 0.28194393 0.06148608 LaBaghWoods_pH6.66       75.6 8.903631
352 -52.614964 0.00000000 0.05835599 LaBaghWoods_pH6.66       75.6 8.953506
353 -63.137956 0.00000000 0.06572200 LaBaghWoods_pH6.66       75.6 8.977658
354 -73.660949 0.00000000 0.07075767 LaBaghWoods_pH6.66       75.6 8.989285
355 -84.183942 0.00000000 0.08974974 LaBaghWoods_pH6.66       75.6 8.994866
356 -94.706935 0.00000000 1.69911297 LaBaghWoods_pH6.66       75.6 8.997541
357   5.261496 0.00000000 0.02845803 LaBaghWoods_pH6.66       99.8 7.096627
358  15.784489 0.00000000 0.02788239 LaBaghWoods_pH6.66       99.8 5.985352
359  26.307482 0.00000000 0.02609810 LaBaghWoods_pH6.66       99.8 4.815855
360  36.830475 1.25773935 0.02488952 LaBaghWoods_pH6.66       99.8 3.861544
361  47.353467 1.63643460 0.02460178 LaBaghWoods_pH6.66       99.8 3.231713
362  57.876460 1.97500066 0.02546502 LaBaghWoods_pH6.66       99.8 2.872132
363  10.522993 0.00000000 0.02886098 LaBaghWoods_pH6.66       99.8 6.566406
364  21.045985 0.00000000 0.02736435 LaBaghWoods_pH6.66       99.8 5.388768
365  31.568978 0.00000000 0.02759459 LaBaghWoods_pH6.66       99.8 4.299931
366  42.091971 1.50607012 0.02604055 LaBaghWoods_pH6.66       99.8 3.507052
367  52.614964 1.85332931 0.02609810 LaBaghWoods_pH6.66       99.8 3.024466
368  63.137956 1.90471785 0.02655854 LaBaghWoods_pH6.66       99.8 2.762104
369  68.399453 1.78236039 0.02586789 LaBaghWoods_pH6.66       99.8 2.683633
370  78.922446 2.37364544 0.02696144 LaBaghWoods_pH6.66       99.8 2.589225
371  89.445438 1.94409108 0.02414141 LaBaghWoods_pH6.66       99.8 2.543023
372  -5.261496 0.00000000 0.04424004 LaBaghWoods_pH6.66       99.8 7.924648
373 -15.784489 0.00000000 0.04026396 LaBaghWoods_pH6.66       99.8 8.436606
374 -26.307482 0.00000000 0.04193492 LaBaghWoods_pH6.66       99.8 8.717521
375  73.660949 2.22518731 0.02471687 LaBaghWoods_pH6.66       99.8 2.628172
376  84.183942 2.32464546 0.02586789 LaBaghWoods_pH6.66       99.8 2.561998
377  94.706935 2.41742779 0.02886098 LaBaghWoods_pH6.66       99.8 2.529829
378 -10.522993 0.00000000 0.03813236 LaBaghWoods_pH6.66       99.8 8.215996
379 -21.045985 0.00000000 0.04216541 LaBaghWoods_pH6.66       99.8 8.599489
380 -31.568978 0.00000000 0.04447057 LaBaghWoods_pH6.66       99.8 8.801899
381 -36.830475 0.00000000 0.04856321 LaBaghWoods_pH6.66       99.8 8.861633
382 -47.353467 0.00000000 0.04775611 LaBaghWoods_pH6.66       99.8 8.933016
383 -57.876460 0.00000000 0.05110013 LaBaghWoods_pH6.66       99.8 8.967759
384 -68.399453 0.00000000 0.05836764 LaBaghWoods_pH6.66       99.8 8.984525
385 -78.922446 0.00000000 0.06610096 LaBaghWoods_pH6.66       99.8 8.992582
386 -89.445438 0.00000000 0.07545625 LaBaghWoods_pH6.66       99.8 8.996447
387 -42.091971 0.00000000 0.04902443 LaBaghWoods_pH6.66       99.8 8.903631
388 -52.614964 0.00000000 0.05156144 LaBaghWoods_pH6.66       99.8 8.953506
389 -63.137956 0.00000000 0.05859842 LaBaghWoods_pH6.66       99.8 8.977658
390 -73.660949 0.00000000 0.06263772 LaBaghWoods_pH6.66       99.8 8.989285
391 -84.183942 0.00000000 0.07383881 LaBaghWoods_pH6.66       99.8 8.994866
392 -94.706935 0.00000000 1.15013070 LaBaghWoods_pH6.66       99.8 8.997541
393   0.000000 2.20561768 0.03033354 LaBaghWoods_pH6.66        0.0 7.552449
394   0.000000 2.09040112 0.03237525 LaBaghWoods_pH6.66        3.6 7.552449
395   0.000000 1.87697175 0.07466739 LaBaghWoods_pH6.66        6.4 7.552449
396   0.000000 1.68443369 0.04913502 LaBaghWoods_pH6.66        9.7 7.552449
397   0.000000 0.55859031 0.03179629 LaBaghWoods_pH6.66       24.5 7.552449
398   0.000000 0.08349005 0.03278156 LaBaghWoods_pH6.66       30.9 7.552449
399   0.000000 0.00000000 0.03010045 LaBaghWoods_pH6.66       36.5 7.552449
400   0.000000 0.00000000 0.03231064 LaBaghWoods_pH6.66       50.2 7.552449
401   0.000000 0.00000000 0.03310857 LaBaghWoods_pH6.66       60.2 7.552449
402   0.000000 0.00000000 0.04030512 LaBaghWoods_pH6.66       75.6 7.552449
403   0.000000 0.00000000 0.03811318 LaBaghWoods_pH6.66       99.8 7.552449</code></pre>
<pre class="r"><code># Nitrite produced I_produced = A - A(t)
df_plot2 %&gt;% filter(Time_hours == 0)</code></pre>
<pre><code>         H_mM   NO3_mM     NO2_mM               Soil Time_hours       pH
1    5.261496 2.214773 0.06570583 LaBaghWoods_pH6.66          0 7.096627
2   15.784489 2.165419 0.02789778 LaBaghWoods_pH6.66          0 5.985352
3   26.307482 2.124933 0.02837579 LaBaghWoods_pH6.66          0 4.815855
4   36.830475 2.110487 0.02810264 LaBaghWoods_pH6.66          0 3.861544
5   47.353467 2.110616 0.02769292 LaBaghWoods_pH6.66          0 3.231713
6   57.876460 2.152427 0.03069779 LaBaghWoods_pH6.66          0 2.872132
7   10.522993 2.191327 0.02858066 LaBaghWoods_pH6.66          0 6.566406
8   21.045985 2.139634 0.02769292 LaBaghWoods_pH6.66          0 5.388768
9   31.568978 2.150123 0.02769292 LaBaghWoods_pH6.66          0 4.299931
10  42.091971 2.128237 0.02769292 LaBaghWoods_pH6.66          0 3.507052
11  52.614964 2.229377 0.02817092 LaBaghWoods_pH6.66          0 3.024466
12  63.137956 2.265353 0.02933186 LaBaghWoods_pH6.66          0 2.762104
13  68.399453 2.197079 0.02803435 LaBaghWoods_pH6.66          0 2.683633
14  78.922446 2.233425 0.03083438 LaBaghWoods_pH6.66          0 2.589225
15  89.445438 2.132650 0.02912699 LaBaghWoods_pH6.66          0 2.543023
16  -5.261496 2.068554 0.03144909 LaBaghWoods_pH6.66          0 7.924648
17 -15.784489 2.138678 0.04012587 LaBaghWoods_pH6.66          0 8.436606
18 -26.307482 2.090391 0.05161118 LaBaghWoods_pH6.66          0 8.717521
19  73.660949 2.290990 0.02933186 LaBaghWoods_pH6.66          0 2.628172
20  84.183942 2.245603 0.03056119 LaBaghWoods_pH6.66          0 2.561998
21  94.706935 2.212126 0.03138079 LaBaghWoods_pH6.66          0 2.529829
22 -10.522993 2.158036 0.03431805 LaBaghWoods_pH6.66          0 8.215996
23 -21.045985 2.265986 0.05085891 LaBaghWoods_pH6.66          0 8.599489
24 -31.568978 2.106950 0.04887582 LaBaghWoods_pH6.66          0 8.801899
25 -36.830475 2.139377 0.04969638 LaBaghWoods_pH6.66          0 8.861633
26 -47.353467 2.159293 0.05475743 LaBaghWoods_pH6.66          0 8.933016
27 -57.876460 2.029329 0.05626239 LaBaghWoods_pH6.66          0 8.967759
28 -68.399453 1.962060 0.06070968 LaBaghWoods_pH6.66          0 8.984525
29 -78.922446 1.936014 0.05284225 LaBaghWoods_pH6.66          0 8.992582
30 -89.445438 1.922611 0.05393661 LaBaghWoods_pH6.66          0 8.996447
31 -42.091971 2.052333 0.04826043 LaBaghWoods_pH6.66          0 8.903631
32 -52.614964 2.050442 0.05537308 LaBaghWoods_pH6.66          0 8.953506
33 -63.137956 1.979744 0.05715174 LaBaghWoods_pH6.66          0 8.977658
34 -73.660949 2.035021 0.05776748 LaBaghWoods_pH6.66          0 8.989285
35 -84.183942 1.940643 0.06365242 LaBaghWoods_pH6.66          0 8.994866
36 -94.706935 2.093410 0.06502133 LaBaghWoods_pH6.66          0 8.997541
37   0.000000 2.205618 0.03033354 LaBaghWoods_pH6.66          0 7.552449</code></pre>
<pre class="r"><code>df_plot2 %&gt;% group_by(H_mM, Soil, Time_hours)</code></pre>
<pre><code># A tibble: 403 x 6
# Groups:   H_mM, Soil, Time_hours [403]
    H_mM NO3_mM NO2_mM Soil               Time_hours    pH
   &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                   &lt;dbl&gt; &lt;dbl&gt;
 1  5.26   2.21 0.0657 LaBaghWoods_pH6.66        0    7.10
 2 15.8    2.17 0.0279 LaBaghWoods_pH6.66        0    5.99
 3 26.3    2.12 0.0284 LaBaghWoods_pH6.66        0    4.82
 4 36.8    2.11 0.0281 LaBaghWoods_pH6.66        0    3.86
 5 47.4    2.11 0.0277 LaBaghWoods_pH6.66        0    3.23
 6 57.9    2.15 0.0307 LaBaghWoods_pH6.66        0    2.87
 7  5.26   2.29 0.0301 LaBaghWoods_pH6.66        3.6  7.10
 8 26.3    2.14 0.0288 LaBaghWoods_pH6.66        3.6  4.82
 9 36.8    1.84 0.0293 LaBaghWoods_pH6.66        3.6  3.86
10 47.4    1.95 0.0306 LaBaghWoods_pH6.66        3.6  3.23
# ... with 393 more rows</code></pre>
<pre class="r"><code>mat_NO3 &lt;- dcast(df_plot2, H_mM ~ Time_hours, value.var = &quot;NO3_mM&quot;)
mat_NO3 &lt;- tibble::column_to_rownames(mat_NO3, var = &quot;H_mM&quot;)
mat_NO3 &lt;- as.matrix(mat_NO3) 
class(mat_NO3)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO3)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code># NA
mat_NO3 &lt;- na_to_average(mat_NO3)
mat_NO3[is.na(mat_NO3)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>mat_NO3[,2:dim(mat_NO3)[2]]</code></pre>
<pre><code>                       3.6      6.4       9.7       24.5       30.9      36.5
-94.7069346522151 1.424164 1.238161 0.9919450 1.26031825 1.11665654 1.0544024
-89.4454382826476 1.353176 1.206318 0.9988479 1.35521984 1.11940225 1.1645385
-84.1839419130801 1.452233 1.313297 1.1829189 1.43351834 1.37055226 1.1881446
-78.9224455435126 1.714580 1.321345 1.2486958 1.54955663 1.42321176 1.3173328
-73.6609491739451 1.744707 1.421059 1.4151821 1.53153939 1.58534676 1.2929334
-68.3994528043776 1.652960 1.568681 1.5045045 1.63526211 1.32957899 0.0000000
-63.1379564348101 1.791895 1.485204 1.7122535 1.65412042 0.00000000 0.0000000
-57.8764600652426 2.010742 1.774495 2.0155154 0.00000000 0.00000000 0.0000000
-52.614963695675  1.820068 1.641936 1.8251817 0.00000000 0.00000000 0.0000000
-47.3534673261075 1.898894 1.746329 1.8592082 0.00000000 0.00000000 0.0000000
-42.09197095654   2.140782 1.762122 2.0273097 0.00000000 0.00000000 0.0000000
-36.8304745869725 2.218453 1.773943 1.9137606 0.00000000 0.00000000 0.0000000
-31.568978217405  1.993416 1.872004 1.7667812 0.00000000 0.00000000 0.0000000
-26.3074818478375 1.824510 1.799538 1.6193341 0.77928412 0.00000000 0.0000000
-21.04598547827   1.927101 1.813052 1.6478952 0.29372756 0.00000000 0.0000000
-15.7844891087025 2.062925 1.792603 1.4041157 0.00000000 0.00000000 0.0000000
-10.522992739135  2.245408 1.776747 1.5531895 0.04850377 0.00000000 0.0000000
-5.2614963695675  1.918750 2.008504 1.6971574 0.46343072 0.01669228 0.0000000
0                 2.090401 1.876972 1.6844337 0.55859031 0.08349005 0.0000000
5.2614963695675   2.287411 1.992158 1.6822185 0.43171887 0.00000000 0.0000000
10.522992739135   2.022632 1.832123 1.7525483 0.84251664 0.35710705 0.0000000
15.7844891087025  2.049755 1.934091 1.8276976 1.27705984 0.94184123 0.5274729
21.04598547827    2.036227 1.945471 1.8349434 1.43068465 1.17968574 0.7878358
26.3074818478375  2.140767 1.866813 2.2024005 1.60570392 1.36168338 1.2171681
31.568978217405   1.927345 1.861880 1.8657055 1.64691837 1.57654306 1.4564625
36.8304745869725  1.840931 1.998956 2.0287078 1.66720020 1.67748826 1.6799004
42.09197095654    2.226189 1.800317 1.9543599 1.85540718 1.82520542 1.6136678
47.3534673261075  1.947390 1.946454 1.9130425 2.10036020 1.77399774 1.8338597
52.614963695675   1.872200 1.950026 1.8508690 2.06109089 1.84092635 1.7901013
57.8764600652426  1.908463 1.860038 1.8607056 2.25804331 1.80205908 1.9441573
63.1379564348101  2.080183 1.966927 1.9263292 2.15559304 1.95575526 2.1246666
68.3994528043776  2.413877 2.227041 2.1443095 2.06800135 2.04134446 1.9415994
73.6609491739451  2.157377 2.079820 2.3601177 2.34616689 2.16899657 2.0917583
78.9224455435126  2.101194 2.004815 2.2354018 2.21983668 2.19805632 2.1294978
84.1839419130801  2.357188 1.930437 2.2418370 2.17282388 2.19992540 2.2038027
89.4454382826476  2.353980 1.969809 2.4955716 2.23253060 1.96309644 2.1030316
94.7069346522151  2.127919 2.048007 2.3068814 2.16925992 2.09784224 2.1794697
                       50.2      60.2      75.6     99.8
-94.7069346522151 1.0075537 0.4854391 0.0000000 0.000000
-89.4454382826476 0.9991432 0.0000000 0.0000000 0.000000
-84.1839419130801 0.0000000 0.0000000 0.0000000 0.000000
-78.9224455435126 0.0000000 0.0000000 0.0000000 0.000000
-73.6609491739451 0.0000000 0.0000000 0.0000000 0.000000
-68.3994528043776 0.0000000 0.0000000 0.0000000 0.000000
-63.1379564348101 0.0000000 0.0000000 0.0000000 0.000000
-57.8764600652426 0.0000000 0.0000000 0.0000000 0.000000
-52.614963695675  0.0000000 0.0000000 0.0000000 0.000000
-47.3534673261075 0.0000000 0.0000000 0.0000000 0.000000
-42.09197095654   0.0000000 0.0000000 0.2819439 0.000000
-36.8304745869725 0.0000000 0.0000000 0.0000000 0.000000
-31.568978217405  0.0000000 0.0000000 0.0000000 0.000000
-26.3074818478375 0.0000000 0.0000000 0.0000000 0.000000
-21.04598547827   0.0000000 0.0000000 0.0000000 0.000000
-15.7844891087025 0.0000000 0.0000000 0.0000000 0.000000
-10.522992739135  0.0000000 0.0000000 0.0000000 0.000000
-5.2614963695675  0.0000000 0.0000000 0.0000000 0.000000
0                 0.0000000 0.0000000 0.0000000 0.000000
5.2614963695675   0.0000000 0.0000000 0.0000000 0.000000
10.522992739135   0.0000000 0.0000000 0.0000000 0.000000
15.7844891087025  0.0000000 0.0000000 0.0000000 0.000000
21.04598547827    0.0000000 0.0000000 0.0000000 0.000000
26.3074818478375  0.1101052 0.0000000 0.0000000 0.000000
31.568978217405   1.1544924 0.4458410 0.0000000 0.000000
36.8304745869725  1.5846426 1.4265320 1.4531266 1.257739
42.09197095654    1.6349839 1.5594741 1.4839644 1.506070
47.3534673261075  1.6888846 1.7397175 1.5437502 1.636435
52.614963695675   1.8923189 1.7451325 1.9166368 1.853329
57.8764600652426  1.6982320 1.6929563 1.9854785 1.975001
63.1379564348101  2.0991120 1.9717689 1.7892183 1.904718
68.3994528043776  2.3058755 1.9277488 2.2247672 1.782360
73.6609491739451  2.1325320 2.0928532 2.4507300 2.225187
78.9224455435126  2.2798799 2.0378985 2.4212745 2.373645
84.1839419130801  1.9503274 2.1346277 2.4596051 2.324645
89.4454382826476  2.2698919 2.1345065 2.4572206 1.944091
94.7069346522151  2.3584766 2.0374300 2.4461921 2.417428</code></pre>
<pre class="r"><code>mat_NO3[,1] - mat_NO3[,1]</code></pre>
<pre><code>-94.7069346522151 -89.4454382826476 -84.1839419130801 -78.9224455435126 
                0                 0                 0                 0 
-73.6609491739451 -68.3994528043776 -63.1379564348101 -57.8764600652426 
                0                 0                 0                 0 
 -52.614963695675 -47.3534673261075   -42.09197095654 -36.8304745869725 
                0                 0                 0                 0 
 -31.568978217405 -26.3074818478375   -21.04598547827 -15.7844891087025 
                0                 0                 0                 0 
 -10.522992739135  -5.2614963695675                 0   5.2614963695675 
                0                 0                 0                 0 
  10.522992739135  15.7844891087025    21.04598547827  26.3074818478375 
                0                 0                 0                 0 
  31.568978217405  36.8304745869725    42.09197095654  47.3534673261075 
                0                 0                 0                 0 
  52.614963695675  57.8764600652426  63.1379564348101  68.3994528043776 
                0                 0                 0                 0 
 73.6609491739451  78.9224455435126  84.1839419130801  89.4454382826476 
                0                 0                 0                 0 
 94.7069346522151 
                0 </code></pre>
<pre class="r"><code>mat_NO3_consumed &lt;- mat_NO3
for (j in 1:dim(mat_NO3_consumed)[2]){
  mat_NO3_consumed[,j] = mat_NO3[,1] - mat_NO3_consumed[,j]
}

mat_NO3_consumed</code></pre>
<pre><code>                  0         3.6         6.4          9.7        24.5       30.9
-94.7069346522151 0  0.66924535  0.85524845  1.101464645  0.83309135 0.97675307
-89.4454382826476 0  0.56943554  0.71629371  0.923763440  0.56739149 0.80320908
-84.1839419130801 0  0.48840957  0.62734548  0.757723790  0.50712438 0.57009045
-78.9224455435126 0  0.22143401  0.61466894  0.687318255  0.38645739 0.51280227
-73.6609491739451 0  0.29031347  0.61396192  0.619838420  0.50348117 0.44967379
-68.3994528043776 0  0.30910058  0.39337944  0.457555710  0.32679809 0.63248121
-63.1379564348101 0  0.18784948  0.49454053  0.267490614  0.32562372 1.97974414
-57.8764600652426 0  0.01858693  0.25483362  0.013813243  2.02932867 2.02932867
-52.614963695675  0  0.23037400  0.40850616  0.225260402  2.05044214 2.05044214
-47.3534673261075 0  0.26039945  0.41296445  0.300084755  2.15929298 2.15929298
-42.09197095654   0 -0.08844938  0.29021147  0.025023306  2.05233300 2.05233300
-36.8304745869725 0 -0.07907606  0.36543320  0.225616053  2.13937665 2.13937665
-31.568978217405  0  0.11353413  0.23494558  0.340168721  2.10694995 2.10694995
-26.3074818478375 0  0.26588118  0.29085270  0.471056937  1.31110695 2.09039107
-21.04598547827   0  0.33888537  0.45293476  0.618091292  1.97225893 2.26598649
-15.7844891087025 0  0.07575285  0.34607425  0.734562054  2.13867774 2.13867774
-10.522992739135  0 -0.08737237  0.38128885  0.604846590  2.10953233 2.15803610
-5.2614963695675  0  0.14980391  0.06004930  0.371396430  1.60512307 2.05186150
0                 0  0.11521655  0.32864593  0.521183986  1.64702737 2.12212763
5.2614963695675   0 -0.07263776  0.22261525  0.532554651  1.78305428 2.21477315
10.522992739135   0  0.16869444  0.35920324  0.438778212  1.34880991 1.83421950
15.7844891087025  0  0.11566386  0.23132772  0.337721056  0.88835879 1.22357740
21.04598547827    0  0.10340778  0.19416304  0.304690992  0.70894971 0.95994862
26.3074818478375  0 -0.01583391  0.25812049 -0.077467308  0.51922930 0.76324984
31.568978217405   0  0.22277873  0.28824312  0.284417991  0.50320508 0.57358040
36.8304745869725  0  0.26955642  0.11153115  0.081779147  0.44328677 0.43299871
42.09197095654    0 -0.09795223  0.32791920  0.173876760  0.27282952 0.30303128
47.3534673261075  0  0.16322582  0.16416141  0.197573017  0.01025533 0.33661779
52.614963695675   0  0.35717764  0.27935075  0.378508205  0.16828631 0.38845086
57.8764600652426  0  0.24396425  0.29238888  0.291721205 -0.10561652 0.35036770
63.1379564348101  0  0.18517006  0.29842586  0.339023934  0.10976010 0.30959788
68.3994528043776  0 -0.21679792 -0.02996258  0.052769282  0.12907747 0.15573436
73.6609491739451  0  0.13361249  0.21117010 -0.069128029 -0.05517719 0.12199313
78.9224455435126  0  0.13223112  0.22860950 -0.001977029  0.01358813 0.03536849
84.1839419130801  0 -0.11158451  0.31516657  0.003766151  0.07277932 0.04567780
89.4454382826476  0 -0.22133022  0.16284092 -0.362921681 -0.09988068 0.16955348
94.7069346522151  0  0.08420755  0.16411921 -0.094755024  0.04286646 0.11428414
                        36.5        50.2         60.2        75.6        99.8
-94.7069346522151 1.03900721  1.08585591  1.607970470  2.09340960  2.09340960
-89.4454382826476 0.75807281  0.92346808  1.922611331  1.92261133  1.92261133
-84.1839419130801 0.75249814  1.94064271  1.940642713  1.94064271  1.94064271
-78.9224455435126 0.61868123  1.93601403  1.936014026  1.93601403  1.93601403
-73.6609491739451 0.74208715  2.03502055  2.035020555  2.03502055  2.03502055
-68.3994528043776 1.96206020  1.96206020  1.962060197  1.96206020  1.96206020
-63.1379564348101 1.97974414  1.97974414  1.979744137  1.97974414  1.97974414
-57.8764600652426 2.02932867  2.02932867  2.029328671  2.02932867  2.02932867
-52.614963695675  2.05044214  2.05044214  2.050442136  2.05044214  2.05044214
-47.3534673261075 2.15929298  2.15929298  2.159292978  2.15929298  2.15929298
-42.09197095654   2.05233300  2.05233300  2.052332997  1.77038907  2.05233300
-36.8304745869725 2.13937665  2.13937665  2.139376654  2.13937665  2.13937665
-31.568978217405  2.10694995  2.10694995  2.106949948  2.10694995  2.10694995
-26.3074818478375 2.09039107  2.09039107  2.090391069  2.09039107  2.09039107
-21.04598547827   2.26598649  2.26598649  2.265986485  2.26598649  2.26598649
-15.7844891087025 2.13867774  2.13867774  2.138677737  2.13867774  2.13867774
-10.522992739135  2.15803610  2.15803610  2.158036098  2.15803610  2.15803610
-5.2614963695675  2.06855378  2.06855378  2.068553784  2.06855378  2.06855378
0                 2.20561768  2.20561768  2.205617677  2.20561768  2.20561768
5.2614963695675   2.21477315  2.21477315  2.214773153  2.21477315  2.21477315
10.522992739135   2.19132655  2.19132655  2.191326549  2.19132655  2.19132655
15.7844891087025  1.63794576  2.16541863  2.165418631  2.16541863  2.16541863
21.04598547827    1.35179859  2.13963436  2.139634360  2.13963436  2.13963436
26.3074818478375  0.90776516  2.01482798  2.124933224  2.12493322  2.12493322
31.568978217405   0.69366098  0.99563104  1.704282479  2.15012346  2.15012346
36.8304745869725  0.43058659  0.52584435  0.683954978  0.65736035  0.85274762
42.09197095654    0.51456887  0.49325281  0.568762564  0.64427232  0.62216657
47.3534673261075  0.27675584  0.42173092  0.370897989  0.56686538  0.47418093
52.614963695675   0.43927588  0.33705829  0.484244679  0.31274045  0.37604790
57.8764600652426  0.20826952  0.45419483  0.459470504  0.16694832  0.17742613
63.1379564348101  0.14068653  0.16624112  0.293584284  0.47613485  0.36063530
68.3994528043776  0.25547946 -0.10879671  0.269330065 -0.02768842  0.41471843
73.6609491739451  0.19923142  0.15845773  0.198136458 -0.15974034  0.06580239
78.9224455435126  0.10392700 -0.04645511  0.195526348 -0.18784972 -0.14022062
84.1839419130801  0.04180053  0.29527576  0.110975455 -0.21400191 -0.07904226
89.4454382826476  0.02961833 -0.13724203 -0.001856588 -0.32457073  0.18855884
94.7069346522151  0.03265671 -0.14635026  0.174696343 -0.23406576 -0.20530141</code></pre>
<pre class="r"><code># make negative values into 0.
mat_NO3_consumed[mat_NO3_consumed&lt;0] &lt;- 0
mat_NO3_consumed</code></pre>
<pre><code>                  0        3.6       6.4         9.7       24.5       30.9
-94.7069346522151 0 0.66924535 0.8552485 1.101464645 0.83309135 0.97675307
-89.4454382826476 0 0.56943554 0.7162937 0.923763440 0.56739149 0.80320908
-84.1839419130801 0 0.48840957 0.6273455 0.757723790 0.50712438 0.57009045
-78.9224455435126 0 0.22143401 0.6146689 0.687318255 0.38645739 0.51280227
-73.6609491739451 0 0.29031347 0.6139619 0.619838420 0.50348117 0.44967379
-68.3994528043776 0 0.30910058 0.3933794 0.457555710 0.32679809 0.63248121
-63.1379564348101 0 0.18784948 0.4945405 0.267490614 0.32562372 1.97974414
-57.8764600652426 0 0.01858693 0.2548336 0.013813243 2.02932867 2.02932867
-52.614963695675  0 0.23037400 0.4085062 0.225260402 2.05044214 2.05044214
-47.3534673261075 0 0.26039945 0.4129644 0.300084755 2.15929298 2.15929298
-42.09197095654   0 0.00000000 0.2902115 0.025023306 2.05233300 2.05233300
-36.8304745869725 0 0.00000000 0.3654332 0.225616053 2.13937665 2.13937665
-31.568978217405  0 0.11353413 0.2349456 0.340168721 2.10694995 2.10694995
-26.3074818478375 0 0.26588118 0.2908527 0.471056937 1.31110695 2.09039107
-21.04598547827   0 0.33888537 0.4529348 0.618091292 1.97225893 2.26598649
-15.7844891087025 0 0.07575285 0.3460742 0.734562054 2.13867774 2.13867774
-10.522992739135  0 0.00000000 0.3812888 0.604846590 2.10953233 2.15803610
-5.2614963695675  0 0.14980391 0.0600493 0.371396430 1.60512307 2.05186150
0                 0 0.11521655 0.3286459 0.521183986 1.64702737 2.12212763
5.2614963695675   0 0.00000000 0.2226153 0.532554651 1.78305428 2.21477315
10.522992739135   0 0.16869444 0.3592032 0.438778212 1.34880991 1.83421950
15.7844891087025  0 0.11566386 0.2313277 0.337721056 0.88835879 1.22357740
21.04598547827    0 0.10340778 0.1941630 0.304690992 0.70894971 0.95994862
26.3074818478375  0 0.00000000 0.2581205 0.000000000 0.51922930 0.76324984
31.568978217405   0 0.22277873 0.2882431 0.284417991 0.50320508 0.57358040
36.8304745869725  0 0.26955642 0.1115312 0.081779147 0.44328677 0.43299871
42.09197095654    0 0.00000000 0.3279192 0.173876760 0.27282952 0.30303128
47.3534673261075  0 0.16322582 0.1641614 0.197573017 0.01025533 0.33661779
52.614963695675   0 0.35717764 0.2793507 0.378508205 0.16828631 0.38845086
57.8764600652426  0 0.24396425 0.2923889 0.291721205 0.00000000 0.35036770
63.1379564348101  0 0.18517006 0.2984259 0.339023934 0.10976010 0.30959788
68.3994528043776  0 0.00000000 0.0000000 0.052769282 0.12907747 0.15573436
73.6609491739451  0 0.13361249 0.2111701 0.000000000 0.00000000 0.12199313
78.9224455435126  0 0.13223112 0.2286095 0.000000000 0.01358813 0.03536849
84.1839419130801  0 0.00000000 0.3151666 0.003766151 0.07277932 0.04567780
89.4454382826476  0 0.00000000 0.1628409 0.000000000 0.00000000 0.16955348
94.7069346522151  0 0.08420755 0.1641192 0.000000000 0.04286646 0.11428414
                        36.5      50.2      60.2      75.6       99.8
-94.7069346522151 1.03900721 1.0858559 1.6079705 2.0934096 2.09340960
-89.4454382826476 0.75807281 0.9234681 1.9226113 1.9226113 1.92261133
-84.1839419130801 0.75249814 1.9406427 1.9406427 1.9406427 1.94064271
-78.9224455435126 0.61868123 1.9360140 1.9360140 1.9360140 1.93601403
-73.6609491739451 0.74208715 2.0350206 2.0350206 2.0350206 2.03502055
-68.3994528043776 1.96206020 1.9620602 1.9620602 1.9620602 1.96206020
-63.1379564348101 1.97974414 1.9797441 1.9797441 1.9797441 1.97974414
-57.8764600652426 2.02932867 2.0293287 2.0293287 2.0293287 2.02932867
-52.614963695675  2.05044214 2.0504421 2.0504421 2.0504421 2.05044214
-47.3534673261075 2.15929298 2.1592930 2.1592930 2.1592930 2.15929298
-42.09197095654   2.05233300 2.0523330 2.0523330 1.7703891 2.05233300
-36.8304745869725 2.13937665 2.1393767 2.1393767 2.1393767 2.13937665
-31.568978217405  2.10694995 2.1069499 2.1069499 2.1069499 2.10694995
-26.3074818478375 2.09039107 2.0903911 2.0903911 2.0903911 2.09039107
-21.04598547827   2.26598649 2.2659865 2.2659865 2.2659865 2.26598649
-15.7844891087025 2.13867774 2.1386777 2.1386777 2.1386777 2.13867774
-10.522992739135  2.15803610 2.1580361 2.1580361 2.1580361 2.15803610
-5.2614963695675  2.06855378 2.0685538 2.0685538 2.0685538 2.06855378
0                 2.20561768 2.2056177 2.2056177 2.2056177 2.20561768
5.2614963695675   2.21477315 2.2147732 2.2147732 2.2147732 2.21477315
10.522992739135   2.19132655 2.1913265 2.1913265 2.1913265 2.19132655
15.7844891087025  1.63794576 2.1654186 2.1654186 2.1654186 2.16541863
21.04598547827    1.35179859 2.1396344 2.1396344 2.1396344 2.13963436
26.3074818478375  0.90776516 2.0148280 2.1249332 2.1249332 2.12493322
31.568978217405   0.69366098 0.9956310 1.7042825 2.1501235 2.15012346
36.8304745869725  0.43058659 0.5258444 0.6839550 0.6573603 0.85274762
42.09197095654    0.51456887 0.4932528 0.5687626 0.6442723 0.62216657
47.3534673261075  0.27675584 0.4217309 0.3708980 0.5668654 0.47418093
52.614963695675   0.43927588 0.3370583 0.4842447 0.3127404 0.37604790
57.8764600652426  0.20826952 0.4541948 0.4594705 0.1669483 0.17742613
63.1379564348101  0.14068653 0.1662411 0.2935843 0.4761349 0.36063530
68.3994528043776  0.25547946 0.0000000 0.2693301 0.0000000 0.41471843
73.6609491739451  0.19923142 0.1584577 0.1981365 0.0000000 0.06580239
78.9224455435126  0.10392700 0.0000000 0.1955263 0.0000000 0.00000000
84.1839419130801  0.04180053 0.2952758 0.1109755 0.0000000 0.00000000
89.4454382826476  0.02961833 0.0000000 0.0000000 0.0000000 0.18855884
94.7069346522151  0.03265671 0.0000000 0.1746963 0.0000000 0.00000000</code></pre>
<pre class="r"><code># This is NO2 produced
mat_NO2_produced = mat_NO3_consumed
df_Ip &lt;- melt(mat_NO2_produced) 
names(df_Ip) &lt;- c(&quot;H_mM&quot;,&quot;Time_hours&quot;,&quot;NO2_mM&quot;)

# plot I p = Nitrite produced (accumulative) = Nitrate reduced 
ggplot(df_Ip, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_p (Nitrite produced) = A0 - A(t) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-17-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="i-r" class="section level2">
<h2>5.1. I r</h2>
<pre class="r"><code># plot I r = Nitrite reduced (accumulative) = I p - I
mat_NO2_produced[is.na(mat_NO2_produced)] </code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>mat_NO2_produced</code></pre>
<pre><code>                  0        3.6       6.4         9.7       24.5       30.9
-94.7069346522151 0 0.66924535 0.8552485 1.101464645 0.83309135 0.97675307
-89.4454382826476 0 0.56943554 0.7162937 0.923763440 0.56739149 0.80320908
-84.1839419130801 0 0.48840957 0.6273455 0.757723790 0.50712438 0.57009045
-78.9224455435126 0 0.22143401 0.6146689 0.687318255 0.38645739 0.51280227
-73.6609491739451 0 0.29031347 0.6139619 0.619838420 0.50348117 0.44967379
-68.3994528043776 0 0.30910058 0.3933794 0.457555710 0.32679809 0.63248121
-63.1379564348101 0 0.18784948 0.4945405 0.267490614 0.32562372 1.97974414
-57.8764600652426 0 0.01858693 0.2548336 0.013813243 2.02932867 2.02932867
-52.614963695675  0 0.23037400 0.4085062 0.225260402 2.05044214 2.05044214
-47.3534673261075 0 0.26039945 0.4129644 0.300084755 2.15929298 2.15929298
-42.09197095654   0 0.00000000 0.2902115 0.025023306 2.05233300 2.05233300
-36.8304745869725 0 0.00000000 0.3654332 0.225616053 2.13937665 2.13937665
-31.568978217405  0 0.11353413 0.2349456 0.340168721 2.10694995 2.10694995
-26.3074818478375 0 0.26588118 0.2908527 0.471056937 1.31110695 2.09039107
-21.04598547827   0 0.33888537 0.4529348 0.618091292 1.97225893 2.26598649
-15.7844891087025 0 0.07575285 0.3460742 0.734562054 2.13867774 2.13867774
-10.522992739135  0 0.00000000 0.3812888 0.604846590 2.10953233 2.15803610
-5.2614963695675  0 0.14980391 0.0600493 0.371396430 1.60512307 2.05186150
0                 0 0.11521655 0.3286459 0.521183986 1.64702737 2.12212763
5.2614963695675   0 0.00000000 0.2226153 0.532554651 1.78305428 2.21477315
10.522992739135   0 0.16869444 0.3592032 0.438778212 1.34880991 1.83421950
15.7844891087025  0 0.11566386 0.2313277 0.337721056 0.88835879 1.22357740
21.04598547827    0 0.10340778 0.1941630 0.304690992 0.70894971 0.95994862
26.3074818478375  0 0.00000000 0.2581205 0.000000000 0.51922930 0.76324984
31.568978217405   0 0.22277873 0.2882431 0.284417991 0.50320508 0.57358040
36.8304745869725  0 0.26955642 0.1115312 0.081779147 0.44328677 0.43299871
42.09197095654    0 0.00000000 0.3279192 0.173876760 0.27282952 0.30303128
47.3534673261075  0 0.16322582 0.1641614 0.197573017 0.01025533 0.33661779
52.614963695675   0 0.35717764 0.2793507 0.378508205 0.16828631 0.38845086
57.8764600652426  0 0.24396425 0.2923889 0.291721205 0.00000000 0.35036770
63.1379564348101  0 0.18517006 0.2984259 0.339023934 0.10976010 0.30959788
68.3994528043776  0 0.00000000 0.0000000 0.052769282 0.12907747 0.15573436
73.6609491739451  0 0.13361249 0.2111701 0.000000000 0.00000000 0.12199313
78.9224455435126  0 0.13223112 0.2286095 0.000000000 0.01358813 0.03536849
84.1839419130801  0 0.00000000 0.3151666 0.003766151 0.07277932 0.04567780
89.4454382826476  0 0.00000000 0.1628409 0.000000000 0.00000000 0.16955348
94.7069346522151  0 0.08420755 0.1641192 0.000000000 0.04286646 0.11428414
                        36.5      50.2      60.2      75.6       99.8
-94.7069346522151 1.03900721 1.0858559 1.6079705 2.0934096 2.09340960
-89.4454382826476 0.75807281 0.9234681 1.9226113 1.9226113 1.92261133
-84.1839419130801 0.75249814 1.9406427 1.9406427 1.9406427 1.94064271
-78.9224455435126 0.61868123 1.9360140 1.9360140 1.9360140 1.93601403
-73.6609491739451 0.74208715 2.0350206 2.0350206 2.0350206 2.03502055
-68.3994528043776 1.96206020 1.9620602 1.9620602 1.9620602 1.96206020
-63.1379564348101 1.97974414 1.9797441 1.9797441 1.9797441 1.97974414
-57.8764600652426 2.02932867 2.0293287 2.0293287 2.0293287 2.02932867
-52.614963695675  2.05044214 2.0504421 2.0504421 2.0504421 2.05044214
-47.3534673261075 2.15929298 2.1592930 2.1592930 2.1592930 2.15929298
-42.09197095654   2.05233300 2.0523330 2.0523330 1.7703891 2.05233300
-36.8304745869725 2.13937665 2.1393767 2.1393767 2.1393767 2.13937665
-31.568978217405  2.10694995 2.1069499 2.1069499 2.1069499 2.10694995
-26.3074818478375 2.09039107 2.0903911 2.0903911 2.0903911 2.09039107
-21.04598547827   2.26598649 2.2659865 2.2659865 2.2659865 2.26598649
-15.7844891087025 2.13867774 2.1386777 2.1386777 2.1386777 2.13867774
-10.522992739135  2.15803610 2.1580361 2.1580361 2.1580361 2.15803610
-5.2614963695675  2.06855378 2.0685538 2.0685538 2.0685538 2.06855378
0                 2.20561768 2.2056177 2.2056177 2.2056177 2.20561768
5.2614963695675   2.21477315 2.2147732 2.2147732 2.2147732 2.21477315
10.522992739135   2.19132655 2.1913265 2.1913265 2.1913265 2.19132655
15.7844891087025  1.63794576 2.1654186 2.1654186 2.1654186 2.16541863
21.04598547827    1.35179859 2.1396344 2.1396344 2.1396344 2.13963436
26.3074818478375  0.90776516 2.0148280 2.1249332 2.1249332 2.12493322
31.568978217405   0.69366098 0.9956310 1.7042825 2.1501235 2.15012346
36.8304745869725  0.43058659 0.5258444 0.6839550 0.6573603 0.85274762
42.09197095654    0.51456887 0.4932528 0.5687626 0.6442723 0.62216657
47.3534673261075  0.27675584 0.4217309 0.3708980 0.5668654 0.47418093
52.614963695675   0.43927588 0.3370583 0.4842447 0.3127404 0.37604790
57.8764600652426  0.20826952 0.4541948 0.4594705 0.1669483 0.17742613
63.1379564348101  0.14068653 0.1662411 0.2935843 0.4761349 0.36063530
68.3994528043776  0.25547946 0.0000000 0.2693301 0.0000000 0.41471843
73.6609491739451  0.19923142 0.1584577 0.1981365 0.0000000 0.06580239
78.9224455435126  0.10392700 0.0000000 0.1955263 0.0000000 0.00000000
84.1839419130801  0.04180053 0.2952758 0.1109755 0.0000000 0.00000000
89.4454382826476  0.02961833 0.0000000 0.0000000 0.0000000 0.18855884
94.7069346522151  0.03265671 0.0000000 0.1746963 0.0000000 0.00000000</code></pre>
<pre class="r"><code>dim(mat_NO2_produced)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code>mat_NO2 &lt;- dcast(df_plot2, H_mM ~ Time_hours, value.var = &quot;NO2_mM&quot;)
mat_NO2 &lt;- tibble::column_to_rownames(mat_NO2, var = &quot;H_mM&quot;)
mat_NO2 &lt;- as.matrix(mat_NO2) 
class(mat_NO2)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>dim(mat_NO2)</code></pre>
<pre><code>[1] 37 11</code></pre>
<pre class="r"><code># Substitute NA to average of NO2_mM of before and after.
mat_NO2[is.na(mat_NO2)] </code></pre>
<pre><code>[1] NA NA NA NA</code></pre>
<pre class="r"><code>for (i in 1:dim(mat_NO2)[1]){
  # print(i)
  for (j in 1:dim(mat_NO2)[2]){
    # print(j)
    if (is.na(mat_NO2[i,j])){
      mat_NO2[i,j] &lt;- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2
    }
  }
}
mat_NO2[is.na(mat_NO2)] # no more NA</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>mat_Ir &lt;- mat_NO2_produced - mat_NO2

df_Ir &lt;- melt(mat_Ir) 
names(df_Ir) &lt;- c(&quot;H_mM&quot;,&quot;Time_hours&quot;,&quot;NO2_mM&quot;)

# plot I r = reduced Nitrite (accumulative)
ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-18-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="aac-area-above-curve" class="section level2">
<h2>5.1.1. AAC (area above curve)</h2>
<pre class="r"><code>df_Ir$Time_hours</code></pre>
<pre><code>  [1]  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
 [16]  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
 [31]  0.0  0.0  0.0  0.0  0.0  0.0  0.0  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6
 [46]  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6
 [61]  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  3.6  6.4
 [76]  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4
 [91]  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4  6.4
[106]  6.4  6.4  6.4  6.4  6.4  6.4  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7
[121]  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7
[136]  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7  9.7 24.5 24.5
[151] 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5
[166] 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5 24.5
[181] 24.5 24.5 24.5 24.5 24.5 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9
[196] 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9
[211] 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 30.9 36.5 36.5 36.5
[226] 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5
[241] 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5 36.5
[256] 36.5 36.5 36.5 36.5 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2
[271] 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2
[286] 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 50.2 60.2 60.2 60.2 60.2
[301] 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2
[316] 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2 60.2
[331] 60.2 60.2 60.2 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6
[346] 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6
[361] 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 75.6 99.8 99.8 99.8 99.8 99.8
[376] 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8
[391] 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8 99.8
[406] 99.8 99.8</code></pre>
<pre class="r"><code>require(pracma)
colnames(df_Ir)</code></pre>
<pre><code>[1] &quot;H_mM&quot;       &quot;Time_hours&quot; &quot;NO2_mM&quot;    </code></pre>
<pre class="r"><code># Let&#39;s get the area above curve
# what is the total area?
time_max = max(df_Ir$Time_hours)
upper_limit_mean = mean(df_Ir %&gt;% filter(Time_hours == time_max) %&gt;% filter(NO2_mM &gt; 1.8) %&gt;% select(NO2_mM) %&gt;% unlist())
total_area = time_max * upper_limit_mean

# correlation

df_corr_aac &lt;- df_Ir %&gt;% select(H_mM, Time_hours, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(aac = total_area - trapz(Time_hours, NO2_mM)) %&gt;% ungroup()

is.na(df_corr_aac) ## no NA</code></pre>
<pre><code>       H_mM   aac
 [1,] FALSE FALSE
 [2,] FALSE FALSE
 [3,] FALSE FALSE
 [4,] FALSE FALSE
 [5,] FALSE FALSE
 [6,] FALSE FALSE
 [7,] FALSE FALSE
 [8,] FALSE FALSE
 [9,] FALSE FALSE
[10,] FALSE FALSE
[11,] FALSE FALSE
[12,] FALSE FALSE
[13,] FALSE FALSE
[14,] FALSE FALSE
[15,] FALSE FALSE
[16,] FALSE FALSE
[17,] FALSE FALSE
[18,] FALSE FALSE
[19,] FALSE FALSE
[20,] FALSE FALSE
[21,] FALSE FALSE
[22,] FALSE FALSE
[23,] FALSE FALSE
[24,] FALSE FALSE
[25,] FALSE FALSE
[26,] FALSE FALSE
[27,] FALSE FALSE
[28,] FALSE FALSE
[29,] FALSE FALSE
[30,] FALSE FALSE
[31,] FALSE FALSE
[32,] FALSE FALSE
[33,] FALSE FALSE
[34,] FALSE FALSE
[35,] FALSE FALSE
[36,] FALSE FALSE
[37,] FALSE FALSE</code></pre>
<pre class="r"><code>plot(df_corr$H_mM, df_corr_aac$aac)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-19-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_corr_aac$H_mM_2 &lt;- (df_corr_aac$H_mM)^2
fit.aac &lt;- lm(aac ~ H_mM + H_mM_2, df_corr_aac)
summary(fit.aac)</code></pre>
<pre><code>
Call:
lm(formula = aac ~ H_mM + H_mM_2, data = df_corr_aac)

Residuals:
    Min      1Q  Median      3Q     Max 
-52.951 -14.672  -0.528   9.912  54.300 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 52.690630   6.182973   8.522 5.92e-10 ***
H_mM         0.783882   0.073330  10.690 2.06e-12 ***
H_mM_2       0.014371   0.001461   9.836 1.78e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 25.06 on 34 degrees of freedom
Multiple R-squared:  0.8612,    Adjusted R-squared:  0.8531 
F-statistic: 105.5 on 2 and 34 DF,  p-value: 2.622e-15</code></pre>
<pre class="r"><code>perturbH &lt;- seq(-100, 100, 0.1)
aacPredict &lt;- predict(fit.aac,list(H_mM=perturbH, H_mM_2=perturbH^2))
df_aac_quad &lt;- data.frame(H_mM = perturbH, Time_hours = aacPredict)
plot(perturbH, aacPredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-19-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (1) Plot fitted quadratic regression line
ggplot(df_corr_aac, aes(x=H_mM, y=aac)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Area above I_r curve (NO2-)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and area under nitrate curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_aac_quad, aes(x = H_mM, y = Time_hours), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=0,y=100, label= paste0(&quot;y = &quot;, round(coef(fit.aac)[[1]],3), &quot;+&quot;, round(coef(fit.aac)[[2]],3),&quot;x+&quot;,round(coef(fit.aac)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.aac)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-19-3.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="average-slope-max-no2_mm-time" class="section level2">
<h2>5.1.2. Average slope (Max NO2_mM / Time)</h2>
<pre class="r"><code>df_Ir_av_slope &lt;- df_Ir %&gt;% select(H_mM, Time_hours, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(average_slope = max(NO2_mM) trapz(Time_hours, NO2_mM)) %&gt;% ungroup()

df_Ir %&gt;% select(H_mM, Time_hours, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(max(NO2_mM))

## get peaks
# https://stackoverflow.com/questions/55342232/how-to-find-local-maximum-in-r-from-graph 
# plot I r = reduced Nitrite (accumulative)
library(ggpmisc)
ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  stat_peaks(col = &quot;green&quot;, size = 4) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))

## plot in a different way
df_Ir_filt &lt;- df_Ir %&gt;% filter(H_mM &lt; 15 &amp; H_mM &gt; -15)
ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  # stat_peaks(col = &quot;green&quot;, size = 4) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))


## plot in a different way
df_Ir_filt &lt;- df_Ir %&gt;% filter(H_mM &lt; 30 &amp; H_mM &gt; -30)
ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  # stat_peaks(col = &quot;green&quot;, size = 4) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))


## plot in a different way
df_Ir_filt &lt;- df_Ir %&gt;% filter(H_mM &lt; 50 &amp; H_mM &gt;-50)
ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  # stat_peaks(col = &quot;green&quot;, size = 4) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))

## plot in a different way
df_Ir_filt &lt;- df_Ir %&gt;% filter(H_mM &lt; 70 &amp; H_mM &gt; -70)
ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;I_r (Nitrite reduced) = I_p - I(t) \n&quot;) +
  # stat_peaks(col = &quot;green&quot;, size = 4) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))



## get peaks coordinates
mat_Ir
x = colnames(mat_Ir)
y = mat_Ir[1,]
x[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)]
y[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)]

plot(x, y)

for (i in 1:dim(mat_Ir)[1]){
  print(i)
  x = colnames(mat_Ir)
  y = mat_Ir[i,]
  x_max = x[ggpmisc:::find_peaks(y, span = 3)]
  y_max = y[ggpmisc:::find_peaks(y, span = 3)]
  # print(x_max)
  print(y_max)
  plot(x, y)
}

## Max
vec_H_mM &lt;- rownames(mat_Ir)
vec_max_NO_mM &lt;- rep(-1, dim(mat_Ir)[1])
vec_max_time &lt;- rep(-1, dim(mat_Ir)[1])

for (i in 1:dim(mat_Ir)[1]){
  y = mat_Ir[i,]
  max_j = max(y)
  # save
  vec_max_NO_mM[i] &lt;- max_j
  vec_max_time[i] &lt;- which.max(y)
  iter_j &lt;- which.max(y) -1
  # get the more early points if the difference with max is not big (10%)
  for (k in 1:iter_j){
    print(k)
    if (abs(max_j - y[which.max(y)-k])/max_j &lt; 0.1){
      max_j = y[which.max(y)-k]
      # print(max_j)
      vec_max_NO_mM[i] &lt;- max_j
      print(names(max_j))
      vec_max_time[i] &lt;- names(max_j)
    } else{ break }
  }
}

df_ave_slope &lt;- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time)

# check
i = 3
plot(colnames(mat_Ir), mat_Ir[i,])
points(vec_max_time[i], vec_max_NO_mM[i], col=&#39;red&#39;)

for (i in 1:dim(mat_Ir)[1]){
  plot(colnames(mat_Ir), mat_Ir[i,])
  points(vec_max_time[i], vec_max_NO_mM[i], col=&#39;red&#39;)
}

# get average slope
df_ave_slope$H_mM &lt;- as.numeric(df_ave_slope$H_mM)
df_ave_slope$average_slope &lt;- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours)

# plot
plot(vec_H_mM, df_ave_slope$average_slope)

# quadratic fit
df_ave_slope$H_mM_2 &lt;- (df_ave_slope$H_mM)^2
fit.aveslope &lt;- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope)
summary(fit.aveslope)

perturbH &lt;- seq(-100, 100, 0.1)
aveslopePredict &lt;- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2))
df_avs_quad &lt;- data.frame(H_mM = perturbH, average_slope = aveslopePredict)
plot(perturbH, aveslopePredict)

# (1) Plot fitted quadratic regression line
ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Average slope of I_r curve (NO2_mM/Time_hours)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and average slope of I_r curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=0,y=0,label= paste0(&quot;y = &quot;, round(coef(fit.aveslope)[[1]],3), &quot;+&quot;, round(coef(fit.aveslope)[[2]],3),&quot;x+&quot;,round(coef(fit.aveslope)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.aveslope)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d


## restrict the samples with maximum over 1.5 mM NO2
df_ave_slope %&lt;&gt;% filter(NO2_mM &gt; 1.5)

# quadratic fit
df_ave_slope$H_mM_2 &lt;- (df_ave_slope$H_mM)^2
fit.aveslope &lt;- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope)
summary(fit.aveslope)

perturbH &lt;- seq(-100, 100, 0.1)
aveslopePredict &lt;- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2))
df_avs_quad &lt;- data.frame(H_mM = perturbH, average_slope = aveslopePredict)
plot(perturbH, aveslopePredict)

# (1) Plot fitted quadratic regression line
ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Average slope of I_r curve (NO2_mM/Time_hours)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and average slope of I_r curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  # geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  # annotate(&quot;text&quot;,x=0,y=0,label= paste0(&quot;y = &quot;, round(coef(fit.aveslope)[[1]],3), &quot;+&quot;, round(coef(fit.aveslope)[[2]],3),&quot;x+&quot;,round(coef(fit.aveslope)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.aveslope)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d
</code></pre>
<pre><code>Error: &lt;text&gt;:1:123: unexpected symbol
1: df_Ir_av_slope &lt;- df_Ir %&gt;% select(H_mM, Time_hours, NO2_mM) %&gt;% group_by(H_mM) %&gt;% summarize(average_slope = max(NO2_mM) trapz
                                                                                                                              ^</code></pre>
</div>
<div id="got-a-new-idea.-start-from-the-end" class="section level2">
<h2>5.1.3. Got a new idea. Start from the end</h2>
<pre class="r"><code>## Max
vec_H_mM &lt;- rownames(mat_Ir)
vec_max_NO_mM &lt;- rep(-1, dim(mat_Ir)[1])
vec_max_time &lt;- rep(-1, dim(mat_Ir)[1])

for (i in 1:dim(mat_Ir)[1]){
  x = colnames(mat_Ir)
  y = mat_Ir[i,]
  max_j = y[dim(mat_Ir)[2]]
  # save
  vec_max_NO_mM[i] &lt;- max_j
  vec_max_time[i] &lt;- x[dim(mat_Ir)[2]]
  iter_j &lt;- dim(mat_Ir)[2] -1
  # get the more early points if the difference with max is not big (10%)
  for (k in 1:iter_j){
    print(k)
    if ( (y[dim(mat_Ir)[2]-k] &gt; max_j) || (abs(max_j - y[dim(mat_Ir)[2]-k])/max_j &lt; 0.1) ){
      max_j = y[dim(mat_Ir)[2] - k]
      print(max_j)
      vec_max_NO_mM[i] &lt;- max_j
      print(names(max_j))
      vec_max_time[i] &lt;- names(max_j)
    } else{ break }
  }
}</code></pre>
<pre><code>[1] 1
[1] 1
    75.6 
1.835445 
[1] &quot;75.6&quot;
[1] 2
[1] 1
    75.6 
1.850893 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.860233 
[1] &quot;60.2&quot;
[1] 3
[1] 1
    75.6 
1.858007 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.855159 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.866499 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
1.964263 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.962174 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.960085 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
1.891487 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.891886 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.894545 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
1.914022 
[1] &quot;75.6&quot;
[1] 2
   60.2 
1.91605 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.914616 
[1] &quot;50.2&quot;
[1] 4
    36.5 
1.911543 
[1] &quot;36.5&quot;
[1] 5
[1] 1
    75.6 
1.972752 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.969573 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.969425 
[1] &quot;50.2&quot;
[1] 4
    36.5 
1.967529 
[1] &quot;36.5&quot;
[1] 5
    30.9 
1.966365 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
1.992086 
[1] &quot;75.6&quot;
[1] 2
    60.2 
1.989289 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.991247 
[1] &quot;50.2&quot;
[1] 4
    36.5 
1.985375 
[1] &quot;36.5&quot;
[1] 5
    30.9 
1.986594 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.108299 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.106394 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.104933 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.102026 
[1] &quot;36.5&quot;
[1] 5
30.9 
 2.1 
[1] &quot;30.9&quot;
[1] 6
[1] 1
[1] 1
[1] 1
    75.6 
2.062763 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.058239 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.057553 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.055413 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.053706 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.045591 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.044409 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.044151 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.039454 
[1] &quot;36.5&quot;
[1] 5
[1] 1
    75.6 
2.223332 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.222225 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.219488 
[1] &quot;50.2&quot;
[1] 4
   36.5 
2.22111 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.154945 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.097433 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.094535 
[1] &quot;60.2&quot;
[1] 3
   50.2 
2.09759 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.091337 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.094876 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.115565 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.114325 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.113084 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.114092 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.118444 
[1] &quot;30.9&quot;
[1] 6
    24.5 
2.035975 
[1] &quot;24.5&quot;
[1] 7
[1] 1
   75.6 
2.02927 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.033607 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.034677 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.035326 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.017496 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.165313 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.172509 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.173307 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.175517 
[1] &quot;36.5&quot;
[1] 5
    30.9 
2.089346 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.182719 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.187305 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.186431 
[1] &quot;50.2&quot;
[1] 4
    36.5 
2.187799 
[1] &quot;36.5&quot;
[1] 5
   30.9 
2.18448 
[1] &quot;30.9&quot;
[1] 6
[1] 1
    75.6 
2.163988 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.165379 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.165236 
[1] &quot;50.2&quot;
[1] 4
   36.5 
2.16402 
[1] &quot;36.5&quot;
[1] 5
[1] 1
    75.6 
2.136243 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.137507 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.138878 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
2.113459 
[1] &quot;75.6&quot;
[1] 2
   60.2 
2.11394 
[1] &quot;60.2&quot;
[1] 3
    50.2 
2.113415 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
2.099126 
[1] &quot;75.6&quot;
[1] 2
    60.2 
2.097845 
[1] &quot;60.2&quot;
[1] 3
    50.2 
1.983783 
[1] &quot;50.2&quot;
[1] 4
[1] 1
    75.6 
2.125295 
[1] &quot;75.6&quot;
[1] 2
[1] 1
[1] 1
     75.6 
0.6192607 
[1] &quot;75.6&quot;
[1] 2
[1] 1
     75.6 
0.5384861 
[1] &quot;75.6&quot;
[1] 2
[1] 1
[1] 1
     75.6 
0.1421203 
[1] &quot;75.6&quot;
[1] 2
     60.2 
0.4311788 
[1] &quot;60.2&quot;
[1] 3
     50.2 
0.4257884 
[1] &quot;50.2&quot;
[1] 4
[1] 1
     75.6 
0.4508783 
[1] &quot;75.6&quot;
[1] 2
[1] 1
[1] 1
[1] 1
       75.6 
-0.02684843 
[1] &quot;75.6&quot;
[1] 2
     60.2 
0.1685653 
[1] &quot;60.2&quot;
[1] 3
[1] 1
       75.6 
-0.02733828 
[1] &quot;75.6&quot;
[1] 2
      60.2 
0.08160645 
[1] &quot;60.2&quot;
[1] 3
     50.2 
0.2677058 
[1] &quot;50.2&quot;
[1] 4
[1] 1
[1] 1
       75.6 
-0.02991028 
[1] &quot;75.6&quot;
[1] 2
     60.2 
0.1458343 
[1] &quot;60.2&quot;
[1] 3</code></pre>
<pre class="r"><code>df_ave_slope &lt;- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time)

# check
i = 36
plot(colnames(mat_Ir), mat_Ir[i,])
points(vec_max_time[i], vec_max_NO_mM[i], col=&#39;red&#39;)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for (i in 1:dim(mat_Ir)[1]){
  plot(colnames(mat_Ir), mat_Ir[i,])
  points(vec_max_time[i], vec_max_NO_mM[i], col=&#39;red&#39;)
}</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-2.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-3.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-4.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-5.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-6.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-7.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-8.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-9.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-10.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-11.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-12.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-13.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-14.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-15.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-16.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-17.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-18.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-19.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-20.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-21.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-22.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-23.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-24.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-25.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-26.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-27.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-28.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-29.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-30.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-31.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-32.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-33.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-34.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-35.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-36.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-37.png" width="1056" style="display: block; margin: auto;" /><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-38.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># get average slope
df_ave_slope$H_mM &lt;- as.numeric(df_ave_slope$H_mM)
df_ave_slope$average_slope &lt;- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours)

# plot
plot(vec_H_mM, df_ave_slope$average_slope)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-39.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># quadratic fit
df_ave_slope$H_mM_2 &lt;- (df_ave_slope$H_mM)^2
fit.aveslope &lt;- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope)
summary(fit.aveslope)</code></pre>
<pre><code>
Call:
lm(formula = average_slope ~ H_mM + H_mM_2, data = df_ave_slope)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.032894 -0.008559 -0.001071  0.013350  0.028753 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  5.238e-02  4.026e-03  13.012 9.27e-15 ***
H_mM        -2.460e-04  4.774e-05  -5.153 1.09e-05 ***
H_mM_2      -5.600e-06  9.512e-07  -5.887 1.21e-06 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.01631 on 34 degrees of freedom
Multiple R-squared:  0.6429,    Adjusted R-squared:  0.6219 
F-statistic:  30.6 on 2 and 34 DF,  p-value: 2.499e-08</code></pre>
<pre class="r"><code>perturbH &lt;- seq(-100, 100, 0.1)
aveslopePredict &lt;- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2))
df_avs_quad &lt;- data.frame(H_mM = perturbH, average_slope = aveslopePredict)
plot(perturbH, aveslopePredict)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-40.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (1) Plot fitted quadratic regression line
ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) +
  geom_point(size=2.5, shape=16, color = &quot;brown&quot;) +
  # stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 1) +
  # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;maroon2&quot;,&quot;deepskyblue4&quot;))+
  xlab(&quot;Amount of pH perturbation (H+ mM) \n&quot;) +
  ylab(&quot;\n Average slope of I_r curve (NO2_mM/Time_hours)&quot;) +
  # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+
  #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family=&quot;serif&quot;, show.legend = FALSE)+
  ggtitle(&quot;Correlation with perturbation and average slope of I_r curve \n&quot;) +
  # geom_abline(slope = 1, intercept=0, show.legend = &quot;y=x&quot;)+
  # regression line
  geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = &quot;maroon2&quot;, size = 1) +
  # show equation
  annotate(&quot;text&quot;,x=0,y=0,label= paste0(&quot;y = &quot;, round(coef(fit.aveslope)[[1]],3), &quot;+&quot;, round(coef(fit.aveslope)[[2]],3),&quot;x+&quot;,round(coef(fit.aveslope)[[3]],3), &quot;x^2&quot;, &quot;,  R^2: &quot;, round(summary(fit.aveslope)$r.squared,3)), color = &quot;maroon2&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-21-41.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="t-max2" class="section level2">
<h2>5.1.4. T max/2</h2>
<pre class="r"><code># Michaelis-Menten function (https://rpubs.com/dstoebel/576374)
y = mat_Ir[i,][2:dim(mat_Ir)[2]]
x = as.numeric(names(y))
mm_fit &lt;- nls(y ~ SSmicmen(x, Vmax, Km)) 
coefficients(mm_fit)[1]</code></pre>
<pre><code>      Vmax 
0.02490277 </code></pre>
<pre class="r"><code>coefficients(mm_fit)[2] # Km</code></pre>
<pre><code>       Km 
-2.200063 </code></pre>
<pre class="r"><code>plot(x, y)
curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-22-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>i=2
y = mat_Ir[i,][2:dim(mat_Ir)[2]]
x = as.numeric(names(y))
mm_fit &lt;- nls(y ~ SSmicmen(x, Vmax, Km)) 
coefficients(mm_fit)[1]</code></pre>
<pre><code>    Vmax 
12.72723 </code></pre>
<pre class="r"><code>coefficients(mm_fit)[2] # Km</code></pre>
<pre><code>      Km 
605.4242 </code></pre>
<pre class="r"><code>plot(x, y)
curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-22-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># this failed
# get T max/2
vec_T_0.5max &lt;- rep(-1, dim(mat_Ir)[1])
for (i in 1:dim(mat_Ir)[1]){
  print(i)
  y = mat_Ir[i,][2:dim(mat_Ir)[2]]
  x = as.numeric(names(y))
  mm_fit &lt;- nls(y ~ SSmicmen(x, Vmax, Km)) 
  # T max/2
  vec_T_0.5max[i] &lt;- coefficients(mm_fit)[2] 
}</code></pre>
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20
[1] 21
[1] 22
[1] 23
[1] 24
[1] 25</code></pre>
<pre><code>Error in nls(y ~ x/(K + x), data = xy, start = list(K = abs(pars[2L]/pars[1L])), : step factor 0.000488281 reduced below &#39;minFactor&#39; of 0.000976562</code></pre>
</div>
<div id="correlation-between-mu" class="section level2">
<h2>6. Correlation between mu</h2>
<pre class="r"><code># From A
df_corr$auc # area under curve</code></pre>
<pre><code> [1]  69.41784  64.66909  57.73548  61.97911  64.61093  52.53267  46.85704
 [8]  33.73979  31.04081  32.11532  39.84882  33.67999  31.87042  38.00511
[15]  33.80227  28.62567  31.05936  36.36113  38.05028  37.18269  42.93867
[22]  57.12834  62.75318  74.54182  92.91547 158.98029 167.31270 176.35983
[29] 187.83132 190.83726 197.64303 208.20693 224.09768 223.91755 222.30927
[36] 221.88493 226.03675</code></pre>
<pre class="r"><code># From I
vec_peakheight # peak height</code></pre>
<pre><code>-94.7069346522151 -89.4454382826476 -84.1839419130801 -78.9224455435126 
       1.69911297        1.45055613        1.06861160        0.08762101 
-73.6609491739451 -68.3994528043776 -63.1379564348101 -57.8764600652426 
       0.08949089        1.17437834        1.74445642        1.30175706 
 -52.614963695675 -47.3534673261075   -42.09197095654 -36.8304745869725 
       1.69965253        1.23199660        0.06916343        0.41928339 
 -31.568978217405 -26.3074818478375   -21.04598547827 -15.7844891087025 
       1.28382145        0.79408193        0.84637617        0.41859861 
 -10.522992739135  -5.2614963695675                 0   5.2614963695675 
       0.40985389        0.13181208        0.07466739        0.06770988 
  10.522992739135  15.7844891087025    21.04598547827  26.3074818478375 
       0.08685623        0.06194423        0.03974985        0.03198263 
  31.568978217405  36.8304745869725    42.09197095654  47.3534673261075 
       0.02861360        0.03150666        0.02952354        0.03158267 
  52.614963695675  57.8764600652426  63.1379564348101  68.3994528043776 
       0.02872813        0.03069779        0.03165052        0.03009534 
 73.6609491739451  78.9224455435126  84.1839419130801  89.4454382826476 
       0.03122439        0.03343508        0.03418849        0.03475314 
 94.7069346522151 
       0.03300129 </code></pre>
<pre class="r"><code>df_peak_NO2$Time_hours # peak time</code></pre>
<pre><code> [1] 75.6 60.2 50.2 36.5 36.5 36.5 30.9 24.5 24.5 24.5  9.7 75.6 24.5 24.5 24.5
[16]  9.7  9.7  9.7  6.4  6.4  9.7 30.9 36.5  6.4  9.7  9.7  3.6 30.9 50.2  0.0
[31] 30.9  9.7  9.7  6.4  9.7  9.7  3.6</code></pre>
<pre class="r"><code>df_corr_no2$auc # AUC</code></pre>
<pre><code> [1] 53.655584 25.787037 20.071994  7.659047  7.461700 18.819089 16.825861
 [8] 19.080237 23.402191 17.916583  5.591933 12.720566 18.583178 17.750760
[15] 15.935698 11.045439  8.333255  4.801414  3.717379  3.377461  3.624896
[22]  3.685825  2.957695  2.840629  2.689948  2.708263  2.682678  2.781512
[29]  2.700864  2.722800  2.770751  2.721494  2.786688  2.829378  2.876138
[36]  2.876483  3.026718</code></pre>
<pre class="r"><code># From I_R
df_corr_aac$aac # area above curve</code></pre>
<pre><code> [1] 118.44468 102.87305  88.42486  80.71749  73.26547  79.83168  70.39797
 [8]  54.58656  54.10241  38.82800  44.62841  36.93125  44.47353  51.42837
[15]  27.88605  30.52460  28.03120  39.01440  25.94055  23.58687  32.16270
[22]  49.04435  56.46889  68.85589  85.31662 155.35548 161.37019 172.79544
[29] 172.33387 181.92186 178.62507 193.33087 198.16378 202.16018 199.81702
[36] 203.08865 202.87635</code></pre>
<pre class="r"><code>df_ave_slope$average_slope # average slope</code></pre>
<pre><code> [1] 0.0094516924 0.0242783670 0.0309008787 0.0371812576 0.0390455127
 [6] 0.0377399478 0.0523710416 0.0636363939 0.0642910711 0.0679611682
[11] 0.0200732321 0.0209500345 0.0664629739 0.0558754408 0.0697393157
[16] 0.0677953410 0.0831010395 0.0652911472 0.0676163777 0.0706951364
[21] 0.0592882189 0.0426071331 0.0420999101 0.0395175943 0.0281123741
[26] 0.0082951714 0.0081912788 0.0071228321 0.0035065110 0.0084818411
[31] 0.0059639989 0.0038962981 0.0004116785 0.0028000881 0.0053327849
[36] 0.0016474692 0.0024224968</code></pre>
<pre class="r"><code># wrap it up
df_mu &lt;- data.frame(A_AUC = df_corr$auc,
                    I_peak_height = vec_peakheight,
                    I_peak_time = df_peak_NO2$Time_hours,
                    I_AUC = df_corr_no2$auc,
                    I_R_AAC = df_corr_aac$aac,
                    I_R_average_slope = df_ave_slope$average_slope
                    )

pairs(df_mu, main = &quot;mu space correlation&quot;,
      pch=21, cex.labels = 2, cex=1.5, cex.main=2, col=&#39;blue&#39;)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-23-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_mu</code></pre>
<pre><code>                      A_AUC I_peak_height I_peak_time     I_AUC   I_R_AAC
-94.7069346522151  69.41784    1.69911297        75.6 53.655584 118.44468
-89.4454382826476  64.66909    1.45055613        60.2 25.787037 102.87305
-84.1839419130801  57.73548    1.06861160        50.2 20.071994  88.42486
-78.9224455435126  61.97911    0.08762101        36.5  7.659047  80.71749
-73.6609491739451  64.61093    0.08949089        36.5  7.461700  73.26547
-68.3994528043776  52.53267    1.17437834        36.5 18.819089  79.83168
-63.1379564348101  46.85704    1.74445642        30.9 16.825861  70.39797
-57.8764600652426  33.73979    1.30175706        24.5 19.080237  54.58656
-52.614963695675   31.04081    1.69965253        24.5 23.402191  54.10241
-47.3534673261075  32.11532    1.23199660        24.5 17.916583  38.82800
-42.09197095654    39.84882    0.06916343         9.7  5.591933  44.62841
-36.8304745869725  33.67999    0.41928339        75.6 12.720566  36.93125
-31.568978217405   31.87042    1.28382145        24.5 18.583178  44.47353
-26.3074818478375  38.00511    0.79408193        24.5 17.750760  51.42837
-21.04598547827    33.80227    0.84637617        24.5 15.935698  27.88605
-15.7844891087025  28.62567    0.41859861         9.7 11.045439  30.52460
-10.522992739135   31.05936    0.40985389         9.7  8.333255  28.03120
-5.2614963695675   36.36113    0.13181208         9.7  4.801414  39.01440
0                  38.05028    0.07466739         6.4  3.717379  25.94055
5.2614963695675    37.18269    0.06770988         6.4  3.377461  23.58687
10.522992739135    42.93867    0.08685623         9.7  3.624896  32.16270
15.7844891087025   57.12834    0.06194423        30.9  3.685825  49.04435
21.04598547827     62.75318    0.03974985        36.5  2.957695  56.46889
26.3074818478375   74.54182    0.03198263         6.4  2.840629  68.85589
31.568978217405    92.91547    0.02861360         9.7  2.689948  85.31662
36.8304745869725  158.98029    0.03150666         9.7  2.708263 155.35548
42.09197095654    167.31270    0.02952354         3.6  2.682678 161.37019
47.3534673261075  176.35983    0.03158267        30.9  2.781512 172.79544
52.614963695675   187.83132    0.02872813        50.2  2.700864 172.33387
57.8764600652426  190.83726    0.03069779         0.0  2.722800 181.92186
63.1379564348101  197.64303    0.03165052        30.9  2.770751 178.62507
68.3994528043776  208.20693    0.03009534         9.7  2.721494 193.33087
73.6609491739451  224.09768    0.03122439         9.7  2.786688 198.16378
78.9224455435126  223.91755    0.03343508         6.4  2.829378 202.16018
84.1839419130801  222.30927    0.03418849         9.7  2.876138 199.81702
89.4454382826476  221.88493    0.03475314         9.7  2.876483 203.08865
94.7069346522151  226.03675    0.03300129         3.6  3.026718 202.87635
                  I_R_average_slope
-94.7069346522151      0.0094516924
-89.4454382826476      0.0242783670
-84.1839419130801      0.0309008787
-78.9224455435126      0.0371812576
-73.6609491739451      0.0390455127
-68.3994528043776      0.0377399478
-63.1379564348101      0.0523710416
-57.8764600652426      0.0636363939
-52.614963695675       0.0642910711
-47.3534673261075      0.0679611682
-42.09197095654        0.0200732321
-36.8304745869725      0.0209500345
-31.568978217405       0.0664629739
-26.3074818478375      0.0558754408
-21.04598547827        0.0697393157
-15.7844891087025      0.0677953410
-10.522992739135       0.0831010395
-5.2614963695675       0.0652911472
0                      0.0676163777
5.2614963695675        0.0706951364
10.522992739135        0.0592882189
15.7844891087025       0.0426071331
21.04598547827         0.0420999101
26.3074818478375       0.0395175943
31.568978217405        0.0281123741
36.8304745869725       0.0082951714
42.09197095654         0.0081912788
47.3534673261075       0.0071228321
52.614963695675        0.0035065110
57.8764600652426       0.0084818411
63.1379564348101       0.0059639989
68.3994528043776       0.0038962981
73.6609491739451       0.0004116785
78.9224455435126       0.0028000881
84.1839419130801       0.0053327849
89.4454382826476       0.0016474692
94.7069346522151       0.0024224968</code></pre>
<pre class="r"><code>scaled.dat &lt;- scale(df_mu)
pairs(scaled.dat, main = &quot;(scaled) mu space correlation&quot;,
      pch=21, cex.labels = 2, cex=1.5, cex.main=2, col=&#39;blue&#39;)</code></pre>
<p><img src="figure/220131_Griess_pH_perturbation3.Rmd/unnamed-chunk-23-2.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="add-ammonia" class="section level2">
<h2>6.1. Add ammonia</h2>
<pre class="r"><code># wrap it up
df_mu_amm &lt;- data.frame(A_AUC = df_corr$auc,
                    I_peak_height = vec_peakheight,
                    I_peak_time = df_peak_NO2$Time_hours,
                    I_AUC = df_corr_no2$auc,
                    I_R_AAC = df_corr_aac$aac,
                    I_R_average_slope = df_ave_slope$average_slope,
                    Amm_AUC = am_corr$auc,
                    Amm_average_slope  = am_ave_slope$average_slope
                    )</code></pre>
<pre><code>Error in data.frame(A_AUC = df_corr$auc, I_peak_height = vec_peakheight, : object &#39;am_corr&#39; not found</code></pre>
<pre class="r"><code>pairs(df_mu_amm, main = &quot;mu space correlation&quot;,
      pch=21, cex.labels = 1.7, cex=1.5, cex.main=2, col=&#39;blue&#39;)</code></pre>
<pre><code>Error in pairs(df_mu_amm, main = &quot;mu space correlation&quot;, pch = 21, cex.labels = 1.7, : object &#39;df_mu_amm&#39; not found</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pracma_2.3.3       plot3D_1.4         colorRamps_2.3     rmarkdown_2.9     
 [5] ggpubr_0.4.0       ggrepel_0.9.1      ape_5.5            openxlsx_4.2.3    
 [9] devtools_2.4.0     usethis_2.0.1      gtools_3.8.2       reshape2_1.4.4    
[13] readxl_1.3.1       magrittr_2.0.1     forcats_0.5.1      stringr_1.4.0     
[17] purrr_0.3.4        readr_1.4.0        tidyr_1.1.3        tibble_3.0.4      
[21] tidyverse_1.3.1    vegan_2.5-7        lattice_0.20-41    permute_0.9-5     
[25] RColorBrewer_1.1-2 ggplot2_3.3.5      dplyr_1.0.5        workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0  ggsignif_0.6.2    ellipsis_0.3.2    rio_0.5.27       
 [5] rprojroot_2.0.2   fs_1.5.0          rstudioapi_0.13   farver_2.1.0     
 [9] remotes_2.4.0     fansi_0.4.2       lubridate_1.7.10  xml2_1.3.2       
[13] splines_4.0.3     cachem_1.0.4      knitr_1.37        pkgload_1.2.1    
[17] jsonlite_1.7.2    broom_0.7.9       cluster_2.1.0     dbplyr_2.1.1     
[21] compiler_4.0.3    httr_1.4.2        backports_1.2.1   assertthat_0.2.1 
[25] Matrix_1.2-18     fastmap_1.1.0     cli_3.0.1         later_1.2.0      
[29] htmltools_0.5.1.1 prettyunits_1.1.1 tools_4.0.3       misc3d_0.9-0     
[33] gtable_0.3.0      glue_1.4.2        Rcpp_1.0.8.3      carData_3.0-4    
[37] cellranger_1.1.0  jquerylib_0.1.4   vctrs_0.3.8       nlme_3.1-149     
[41] xfun_0.29         ps_1.6.0          testthat_3.0.2    rvest_1.0.1      
[45] lifecycle_1.0.0   rstatix_0.7.0     MASS_7.3-53       scales_1.1.1     
[49] hms_1.1.0         promises_1.2.0.1  parallel_4.0.3    yaml_2.2.1       
[53] curl_4.3.2        memoise_2.0.0     sass_0.4.0        stringi_1.5.3    
[57] highr_0.9         desc_1.3.0        pkgbuild_1.2.0    zip_2.1.1        
[61] rlang_0.4.10      pkgconfig_2.0.3   evaluate_0.14     labeling_0.4.2   
[65] tidyselect_1.1.1  processx_3.5.1    plyr_1.8.6        R6_2.5.0         
[69] generics_0.1.0    DBI_1.1.1         pillar_1.6.0      haven_2.4.1      
[73] whisker_0.4       foreign_0.8-80    withr_2.4.2       mgcv_1.8-33      
[77] abind_1.4-5       modelr_0.1.8      crayon_1.4.1      car_3.0-11       
[81] utf8_1.1.4        grid_4.0.3        data.table_1.14.0 callr_3.7.0      
[85] git2r_0.28.0      reprex_2.0.0      digest_0.6.27     httpuv_1.6.0     
[89] munsell_0.5.0     bslib_0.2.5.1     tcltk_4.0.3       sessioninfo_1.1.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
