<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUchicago" />

<meta name="date" content="2022-04-24" />

<title>220421_Griess_antibiotics_and_sterile</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Denit_visualization_R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/Denit_visualization_R">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">220421_Griess_antibiotics_and_sterile</h1>
<h4 class="author">KiseokUchicago</h4>
<h4 class="date">2022-04-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Denit_visualization_R/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210924code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210924code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210924)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  211020_df_full_innate.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  220519_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/120321_soil_buffering_capacity_Nichols-B.xlsx
    Untracked:  data/210911_TOC_sample_plate1.xlsx
    Untracked:  data/210922_Griess_sample_plate1-investigating.xlsx
    Untracked:  data/210922_Griess_sample_plate1.xlsx
    Untracked:  data/210927_TOC_sample_plate1_100ul.xlsx
    Untracked:  data/210927_TOC_sample_plate1_10ul.xlsx
    Untracked:  data/210928_Griess_sample_plate1.xlsx
    Untracked:  data/211005_Griess_sample_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate2.xlsx
    Untracked:  data/211008_Griess_blank_plate0.xlsx
    Untracked:  data/211008_Griess_blank_plate1.xlsx
    Untracked:  data/211017_Griess_plate0.xlsx
    Untracked:  data/211017_Griess_plate1.xlsx
    Untracked:  data/211017_Griess_plate2.xlsx
    Untracked:  data/211017_Griess_plate3.xlsx
    Untracked:  data/211017_Griess_plate4.xlsx
    Untracked:  data/211017_Griess_plate5.xlsx
    Untracked:  data/211017_Griess_plate6.xlsx
    Untracked:  data/211017_Griess_plate7.xlsx
    Untracked:  data/211017_Griess_plate8.xlsx
    Untracked:  data/211019_Griess_plate1.xlsx
    Untracked:  data/211020_Griess_plate0.xlsx
    Untracked:  data/211020_Griess_plate1.xlsx
    Untracked:  data/211020_df_full_innate.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate0.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate1.xlsx
    Untracked:  data/211025_Griess_plate0.xlsx
    Untracked:  data/211025_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate2.xlsx
    Untracked:  data/211028_Griess_plate2_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate3.xlsx
    Untracked:  data/211028_Griess_plate3_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate4.xlsx
    Untracked:  data/211028_Griess_plate4_no3_fit.xlsx
    Untracked:  data/211028_time_table.xlsx
    Untracked:  data/211127_Griess_SUP1.xlsx
    Untracked:  data/211127_Griess_SUP11.xlsx
    Untracked:  data/211127_Griess_SUP3.xlsx
    Untracked:  data/211127_Griess_SUP5.xlsx
    Untracked:  data/211127_Griess_SUP7.xlsx
    Untracked:  data/211127_Griess_SUP9.xlsx
    Untracked:  data/211127_Griess_plate0.xlsx
    Untracked:  data/211127_time_table.xlsx
    Untracked:  data/211128_Ammonia_sample_plate10.xlsx
    Untracked:  data/211128_Ammonia_sample_plate11.xlsx
    Untracked:  data/211128_Griess_plate1.xlsx
    Untracked:  data/211128_Griess_plate10.xlsx
    Untracked:  data/211128_Griess_plate11.xlsx
    Untracked:  data/211128_Griess_plate2.xlsx
    Untracked:  data/211128_Griess_plate3.xlsx
    Untracked:  data/211128_Griess_plate4.xlsx
    Untracked:  data/211128_Griess_plate5.xlsx
    Untracked:  data/211128_Griess_plate6.xlsx
    Untracked:  data/211128_Griess_plate7.xlsx
    Untracked:  data/211128_Griess_plate8.xlsx
    Untracked:  data/211128_Griess_plate9.xlsx
    Untracked:  data/211128_time_table.xlsx
    Untracked:  data/211201_pH_colorimetric.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_table.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_time.xlsx
    Untracked:  data/211208_pH_colorimetric_T3.xlsx
    Untracked:  data/211208_pH_colorimetric_T4.xlsx
    Untracked:  data/211208_pH_colorimetric_T5.xlsx
    Untracked:  data/211208_pH_colorimetric_plate1.xlsx
    Untracked:  data/211208_pH_colorimetric_plate2.xlsx
    Untracked:  data/211208_pH_colorimetric_plate3.xlsx
    Untracked:  data/211208_time_table.xlsx
    Untracked:  data/211209_OD600_T1.xlsx
    Untracked:  data/211209_OD600_T2.xlsx
    Untracked:  data/211209_OD600_T3.xlsx
    Untracked:  data/211209_time_table.xlsx
    Untracked:  data/211210_pH_colorimetric_T0.xlsx
    Untracked:  data/211210_pH_colorimetric_T1.xlsx
    Untracked:  data/211210_pH_colorimetric_T2.xlsx
    Untracked:  data/211210_pH_colorimetric_T3.xlsx
    Untracked:  data/211210_time_table.xlsx
    Untracked:  data/211218_pH_colorimetric_t0.xlsx
    Untracked:  data/211218_pH_colorimetric_t1.xlsx
    Untracked:  data/211218_pH_colorimetric_t2.xlsx
    Untracked:  data/211218_pH_colorimetric_t3.xlsx
    Untracked:  data/211218_pH_colorimetric_t4.xlsx
    Untracked:  data/211218_time_table.xlsx
    Untracked:  data/211220_Griess_plate1.xlsx
    Untracked:  data/211220_Griess_plate10.xlsx
    Untracked:  data/211220_Griess_plate11.xlsx
    Untracked:  data/211220_Griess_plate2.xlsx
    Untracked:  data/211220_Griess_plate3.xlsx
    Untracked:  data/211220_Griess_plate4.xlsx
    Untracked:  data/211220_Griess_plate5.xlsx
    Untracked:  data/211220_Griess_plate6.xlsx
    Untracked:  data/211220_Griess_plate7.xlsx
    Untracked:  data/211220_Griess_plate8.xlsx
    Untracked:  data/211220_Griess_plate9.xlsx
    Untracked:  data/211220_time_table_AU.xlsx
    Untracked:  data/211220_time_table_BN.xlsx
    Untracked:  data/211224_TOC_sample_plate1.xlsx
    Untracked:  data/220103_Griess_T0.xlsx
    Untracked:  data/220103_Griess_T1.xlsx
    Untracked:  data/220103_Griess_T11.xlsx
    Untracked:  data/220103_Griess_T12.xlsx
    Untracked:  data/220103_Griess_T2.xlsx
    Untracked:  data/220103_Griess_T3.xlsx
    Untracked:  data/220103_Griess_T4.xlsx
    Untracked:  data/220103_Griess_T5.xlsx
    Untracked:  data/220103_Griess_T6.xlsx
    Untracked:  data/220103_Griess_T7.xlsx
    Untracked:  data/220103_Griess_T8.xlsx
    Untracked:  data/220103_Griess_T9.xlsx
    Untracked:  data/220103_time_table.xlsx
    Untracked:  data/220106_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-2_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-3_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-4_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-5_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-6_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-7_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-8_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-9_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107_Ammonia_sample_plate1.xlsx
    Untracked:  data/220108_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate2.xlsx
    Untracked:  data/220111_Ammonia_sample_T0.xlsx
    Untracked:  data/220111_Ammonia_sample_T1.xlsx
    Untracked:  data/220111_Ammonia_sample_T2.xlsx
    Untracked:  data/220111_Ammonia_sample_T3.xlsx
    Untracked:  data/220111_TOC_sample_plate1.xlsx
    Untracked:  data/220112_Ammonia_sample_T11.xlsx
    Untracked:  data/220112_Ammonia_sample_T12.xlsx
    Untracked:  data/220112_Ammonia_sample_T4.xlsx
    Untracked:  data/220112_Ammonia_sample_T5.xlsx
    Untracked:  data/220112_Ammonia_sample_T6.xlsx
    Untracked:  data/220112_Ammonia_sample_T7.xlsx
    Untracked:  data/220112_Ammonia_sample_T8.xlsx
    Untracked:  data/220112_Ammonia_sample_T9.xlsx
    Untracked:  data/220113_pH_colorimetric_T0.xlsx
    Untracked:  data/220113_pH_colorimetric_T12.xlsx
    Untracked:  data/220113_pH_colorimetric_T2.xlsx
    Untracked:  data/220113_pH_colorimetric_T3.xlsx
    Untracked:  data/220113_pH_colorimetric_T8.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_using6.66.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_using6.66.xlsx
    Untracked:  data/220125_time_table.xlsx
    Untracked:  data/220128_Ammonia_sample_plate1.xlsx
    Untracked:  data/220128_Ammonia_sample_plate2.xlsx
    Untracked:  data/220128_Ammonia_sample_plate3.xlsx
    Untracked:  data/220128_Ammonia_sample_plate4.xlsx
    Untracked:  data/220128_Ammonia_sample_plate5.xlsx
    Untracked:  data/220128_Ammonia_sample_plate6.xlsx
    Untracked:  data/220128_Griess_plate1.xlsx
    Untracked:  data/220128_Griess_plate2.xlsx
    Untracked:  data/220128_Griess_plate3.xlsx
    Untracked:  data/220128_Griess_plate4.xlsx
    Untracked:  data/220128_Griess_plate5.xlsx
    Untracked:  data/220128_Griess_plate6.xlsx
    Untracked:  data/220128_time_table.xlsx
    Untracked:  data/220131_Ammonia_sample_plate1.xlsx
    Untracked:  data/220131_Ammonia_sample_plate2.xlsx
    Untracked:  data/220131_Ammonia_sample_plate3.xlsx
    Untracked:  data/220131_Ammonia_sample_plate4.xlsx
    Untracked:  data/220131_Ammonia_sample_plate5.xlsx
    Untracked:  data/220131_Ammonia_sample_plate6.xlsx
    Untracked:  data/220131_Griess_plate1.xlsx
    Untracked:  data/220131_Griess_plate2.xlsx
    Untracked:  data/220131_Griess_plate3.xlsx
    Untracked:  data/220131_Griess_plate4.xlsx
    Untracked:  data/220131_Griess_plate5.xlsx
    Untracked:  data/220131_Griess_plate6.xlsx
    Untracked:  data/220131_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220131_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220131_time_table.xlsx
    Untracked:  data/220327_Griess_pastplate9.xlsx
    Untracked:  data/220327_Griess_plate1.xlsx
    Untracked:  data/220327_Griess_plate10.xlsx
    Untracked:  data/220327_Griess_plate11.xlsx
    Untracked:  data/220327_Griess_plate12.xlsx
    Untracked:  data/220327_Griess_plate13.xlsx
    Untracked:  data/220327_Griess_plate14.xlsx
    Untracked:  data/220327_Griess_plate15.xlsx
    Untracked:  data/220327_Griess_plate16.xlsx
    Untracked:  data/220327_Griess_plate2.xlsx
    Untracked:  data/220327_Griess_plate3.xlsx
    Untracked:  data/220327_Griess_plate4.xlsx
    Untracked:  data/220327_Griess_plate5.xlsx
    Untracked:  data/220327_Griess_plate6.xlsx
    Untracked:  data/220327_Griess_plate7.xlsx
    Untracked:  data/220327_Griess_plate8.xlsx
    Untracked:  data/220327_Griess_plate9.xlsx
    Untracked:  data/220327_time_table.xlsx
    Untracked:  data/220329_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH3_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH4_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH6_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH7_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH3_half_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_base_treatment_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_150ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_300ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_60ul_mu_opt_T0.xlsx
    Untracked:  data/220408_pH_colorimetric_pH_limit_mu_opt_water.xlsx
    Untracked:  data/220412_pH_colorimetric_pH_half_mu_opt_WW.xlsx
    Untracked:  data/220414_Ammonia_sample_plate0.xlsx
    Untracked:  data/220418_Ammonia_sample_plate0.xlsx
    Untracked:  data/220419_Ammonia_sample_plate0.xlsx
    Untracked:  data/220420-2_Ammonia_bcf.xlsx
    Untracked:  data/220420_Ammonia_sample_plate1.xlsx
    Untracked:  data/220420_Ammonia_sample_plate10.xlsx
    Untracked:  data/220420_Ammonia_sample_plate11.xlsx
    Untracked:  data/220420_Ammonia_sample_plate12.xlsx
    Untracked:  data/220420_Ammonia_sample_plate13.xlsx
    Untracked:  data/220420_Ammonia_sample_plate14.xlsx
    Untracked:  data/220420_Ammonia_sample_plate15.xlsx
    Untracked:  data/220420_Ammonia_sample_plate16.xlsx
    Untracked:  data/220420_Ammonia_sample_plate2.xlsx
    Untracked:  data/220420_Ammonia_sample_plate3.xlsx
    Untracked:  data/220420_Ammonia_sample_plate4.xlsx
    Untracked:  data/220420_Ammonia_sample_plate5.xlsx
    Untracked:  data/220420_Ammonia_sample_plate6.xlsx
    Untracked:  data/220420_Ammonia_sample_plate7.xlsx
    Untracked:  data/220420_Ammonia_sample_plate8.xlsx
    Untracked:  data/220420_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_df_a_bcf.xlsx
    Untracked:  data/220421_Ammonia_df_pH_range.xlsx
    Untracked:  data/220421_Ammonia_sample_plate1.xlsx
    Untracked:  data/220421_Ammonia_sample_plate10.xlsx
    Untracked:  data/220421_Ammonia_sample_plate11.xlsx
    Untracked:  data/220421_Ammonia_sample_plate2.xlsx
    Untracked:  data/220421_Ammonia_sample_plate3.xlsx
    Untracked:  data/220421_Ammonia_sample_plate4.xlsx
    Untracked:  data/220421_Ammonia_sample_plate5.xlsx
    Untracked:  data/220421_Ammonia_sample_plate6.xlsx
    Untracked:  data/220421_Ammonia_sample_plate7.xlsx
    Untracked:  data/220421_Ammonia_sample_plate8.xlsx
    Untracked:  data/220421_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_sterile.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  data/220421_Griess_plate1.xlsx
    Untracked:  data/220421_Griess_plate10.xlsx
    Untracked:  data/220421_Griess_plate11.xlsx
    Untracked:  data/220421_Griess_plate2.xlsx
    Untracked:  data/220421_Griess_plate3.xlsx
    Untracked:  data/220421_Griess_plate4.xlsx
    Untracked:  data/220421_Griess_plate5.xlsx
    Untracked:  data/220421_Griess_plate6.xlsx
    Untracked:  data/220421_Griess_plate7.xlsx
    Untracked:  data/220421_Griess_plate8.xlsx
    Untracked:  data/220421_Griess_plate9.xlsx
    Untracked:  data/220421_nitrate_reduced_autoclaved_samples.xlsx
    Untracked:  data/220421_time_table.xlsx
    Untracked:  data/220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T4.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T7.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T9.xlsx
    Untracked:  data/220508_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate2.xlsx
    Untracked:  data/220513_TOC_sample_plate3.xlsx
    Untracked:  data/220517_time_table.xlsx
    Untracked:  data/220519_Griess_antibiotics_dose_experiment_remove_blanks.xlsx
    Untracked:  data/220519_Griess_crerar7_extreme.xlsx
    Untracked:  data/220519_Griess_plate1.xlsx
    Untracked:  data/220519_Griess_plate10.xlsx
    Untracked:  data/220519_Griess_plate11.xlsx
    Untracked:  data/220519_Griess_plate2.xlsx
    Untracked:  data/220519_Griess_plate3.xlsx
    Untracked:  data/220519_Griess_plate4.xlsx
    Untracked:  data/220519_Griess_plate5.xlsx
    Untracked:  data/220519_Griess_plate6.xlsx
    Untracked:  data/220519_Griess_plate7.xlsx
    Untracked:  data/220519_Griess_plate8.xlsx
    Untracked:  data/220519_Griess_plate9.xlsx
    Untracked:  data/220519_time_table.xlsx
    Untracked:  data/220520_TOC_sample_plate1.xlsx
    Untracked:  data/220520_TOC_sample_plate2.xlsx
    Untracked:  data/220520_TOC_sample_plate3.xlsx
    Untracked:  data/220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T10_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220526_TOC_sample_plate1.xlsx
    Untracked:  data/220527_TOC_sample_plate1.xlsx
    Untracked:  data/220601_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_control_OD600.xlsx
    Untracked:  data/220602-3_TOC_sample_plate1.xlsx
    Untracked:  data/220602-3_control_OD600.xlsx
    Untracked:  data/220602_TOC_sample_plate1.xlsx
    Untracked:  data/220603_100ul_OD600.xlsx
    Untracked:  data/220603_TOC_sample_plate1.xlsx
    Untracked:  data/220603_TOC_sample_plate2.xlsx
    Untracked:  data/220606_TOC_sample_plate1.xlsx
    Untracked:  data/220606_TOC_sample_plate2.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel_format_edited.xlsx
    Untracked:  data/OD434_OD585_C_WB_T5_220329.xlsx
    Untracked:  data/OD434_OD585_WW_C_T0_220329.xlsx
    Untracked:  data/OD434_OD585_pH3_half_area_220407.xlsx
    Untracked:  data/OD434_OD585_pH_base_treated_T0_220407.xlsx
    Untracked:  data/OD434_OD585_pH_half_150ul_220412.xlsx
    Untracked:  data/SPNa_86437_021122_edit.xlsx
    Untracked:  data/SPNa_86437_021122_edit2.xlsx
    Untracked:  data/TOC_TN_measurement_2_labs_211014.xlsx
    Untracked:  data/dry_weight_curve(9.5.21).xlsx
    Untracked:  data/pH_data(11.17.21)-Midway_etc.xlsx
    Untracked:  data/pH_data(12.6.21)-Midway_incubation_endpoint.xlsx
    Untracked:  data/pH_data(9.14.21).xlsx
    Untracked:  data/pH_data(9.28.21).xlsx
    Untracked:  data/pH_standard_150ul_220407.xlsx
    Untracked:  data/pH_standard_300ul_220407.xlsx
    Untracked:  data/pH_standard_60ul_220407.xlsx
    Untracked:  data/~$220513_TOC_sample_plate1.xlsx
    Untracked:  data/~$SPNa_86437_021122_edit.xlsx
    Untracked:  past_figures/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/220421_Griess_antibiotics_and_sterile.Rmd</code>) and HTML (<code>docs/220421_Griess_antibiotics_and_sterile.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Denit_visualization_R/blob/7905c9343b41b0f3a42aba137993e7c632c5b0e2/analysis/220421_Griess_antibiotics_and_sterile.Rmd" target="_blank">7905c93</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-06-08
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="antibiotics-and-sterile-autoclaved-soil-experiment-griess-assay" class="section level2">
<h2>Antibiotics and sterile (autoclaved soil experiment) (Griess assay)</h2>
<p>Researcher: <strong>Kiseok Lee</strong><br />
Experiment Date: 4/14/22 - 4/18/22 (5 days)<br />
Analysis Date: 4/24/22 Lab: <strong>Seppe Kuehn</strong></p>
<pre class="r"><code># libraries
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(vegan)</code></pre>
<pre><code>Loading required package: permute</code></pre>
<pre><code>Loading required package: lattice</code></pre>
<pre><code>This is vegan 2.5-7</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.0.5</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.1 --</code></pre>
<pre><code>v tibble  3.0.4     v purrr   0.3.4
v tidyr   1.1.3     v stringr 1.4.0
v readr   1.4.0     v forcats 0.5.1</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(magrittr)</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre class="r"><code>library(readxl)
library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(gtools)</code></pre>
<pre><code>
Attaching package: &#39;gtools&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:permute&#39;:

    permute</code></pre>
<pre class="r"><code>library(devtools)</code></pre>
<pre><code>Warning: package &#39;devtools&#39; was built under R version 4.0.5</code></pre>
<pre><code>Loading required package: usethis</code></pre>
<pre><code>Warning: package &#39;usethis&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;devtools&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:permute&#39;:

    check</code></pre>
<pre class="r"><code>library(openxlsx)
library(ape)</code></pre>
<pre><code>Warning: package &#39;ape&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code>library(stringr)
library(tidyr)
library(ggrepel)</code></pre>
<pre><code>Warning: package &#39;ggrepel&#39; was built under R version 4.0.4</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Warning: package &#39;ggpubr&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ape&#39;:

    rotate</code></pre>
<pre class="r"><code>require(gridExtra)</code></pre>
<pre><code>Loading required package: gridExtra</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre class="r"><code># grid.arrange(p_strain, p_ai, nrow=1)

## theme for ggplot
mytheme &lt;- theme_bw() + 
  theme(text = element_text(face=&quot;bold&quot;, colour = &#39;black&#39;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(axis.text.y = element_text(size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(),panel.background=element_blank(),panel.border=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))+
  theme(legend.text=element_text(size=10,face=&quot;bold&quot;, colour = &#39;black&#39;))


mytheme_2d &lt;- theme_bw() + 
  theme(text = element_text(face=&quot;bold&quot;, colour = &#39;black&#39;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(axis.text.y = element_text(size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))+
  theme(legend.text=element_text(size=10,face=&quot;bold&quot;, colour = &#39;black&#39;))


# color collection
my_color_collection &lt;- c(
  &quot;#CBD588&quot;, &quot;#5F7FC7&quot;, &quot;orange&quot;, &quot;#AD6F3B&quot;, &quot;#673770&quot;, 
  &quot;#D14285&quot;, &quot;#652926&quot;, &quot;#C84248&quot;, &quot;#8569D5&quot;, &quot;#5E738F&quot;,
  &quot;#D1A33D&quot;, &quot;#8A7C64&quot;, &quot;#599861&quot;,&quot;#616163&quot;, &quot;#FFCDB2&quot;,
  &quot;#6D9F71&quot;, &quot;#242F40&quot;,
  &quot;#CCA43B&quot;, &quot;#F92A82&quot;, &quot;#ED7B84&quot;, &quot;#7EB77F&quot;, 
  &quot;#DEC4A1&quot;, &quot;#E5D1D0&quot;, &#39;#0E8482&#39;, &#39;#C9DAEA&#39;, &#39;#337357&#39;, 
  &#39;#95C623&#39;, &#39;#E55812&#39;, &#39;#04471C&#39;, &#39;#F2D7EE&#39;, &#39;#D3BCC0&#39;, 
  &#39;#A5668B&#39;, &#39;#69306D&#39;, &#39;#0E103D&#39;, &#39;#1A535C&#39;, &#39;#4ECDC4&#39;, 
  &#39;#F7FFF7&#39;, &#39;#FF6B6B&#39;, &#39;#FFE66D&#39;, &#39;#6699CC&#39;, &#39;#FFF275&#39;, 
  &#39;#FF8C42&#39;, &#39;#FF3C38&#39;, &#39;#A23E48&#39;, &#39;#000000&#39;, &#39;#CF5C36&#39;, 
  &#39;#EEE5E9&#39;, &#39;#7C7C7C&#39;, &#39;#EFC88B&#39;, &#39;#2E5266&#39;, &#39;#6E8898&#39;, 
  &#39;#9FB1BC&#39;, &#39;#D3D0CB&#39;, &#39;#E2C044&#39;, &#39;#5BC0EB&#39;, &#39;#FDE74C&#39;, 
  &#39;#9BC53D&#39;, &#39;#E55934&#39;, &#39;#FA7921&#39;, &quot;#CD9BCD&quot;, &quot;#508578&quot;, &quot;#DA5724&quot;)

# for git push, use this instead of using wflow_git_push()
# git push -u origin master (in the Git app / in the working directory)

# for making pdf file
library(rmarkdown)</code></pre>
<pre><code>Warning: package &#39;rmarkdown&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code># render(&quot;analysis/~~.Rmd&quot;, &quot;pdf_document&quot;)</code></pre>
</div>
<div id="import-data-table-from-python-code" class="section level2">
<h2>1. Import data table from python code</h2>
<p>We are going to use the vcl3 treated standard curve that is fitted with pure nitrate standards</p>
<pre class="r"><code># import file
df_plate1 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate1.xlsx&quot;)
df_plate2 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate2.xlsx&quot;)
df_plate3 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate3.xlsx&quot;)
df_plate4 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate4.xlsx&quot;)
df_plate5 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate5.xlsx&quot;)
df_plate6 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate6.xlsx&quot;)
df_plate7 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate7.xlsx&quot;)
df_plate8 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate8.xlsx&quot;)
df_plate9 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate9.xlsx&quot;)
df_plate10 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate10.xlsx&quot;)
df_plate11 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate11.xlsx&quot;)
# df_plate12 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate12.xlsx&quot;)
# df_plate13 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate13.xlsx&quot;)
# df_plate14 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate14.xlsx&quot;)
# df_plate15 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate15.xlsx&quot;)
# df_plate16 &lt;- openxlsx::read.xlsx(&quot;data/220421_Griess_plate16.xlsx&quot;)

head(df_plate1)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T0   0.00430      0.86470 0.004341071 0.8055666 0.8012255
2     LBA_T0   0.00375      0.89920 0.004056146 0.8407224 0.8366662
3     LBA_T0   0.01160      0.83550 0.008124078 0.7759891 0.7678650
4     LBA_T0   0.01885      0.82070 0.011883516 0.7610589 0.7491754
5     LBA_T0   0.00260      0.84035 0.003460436 0.7808906 0.7774302
6     LBA_T0   0.00100      0.77845 0.002631720 0.7186605 0.7160288</code></pre>
<pre class="r"><code>colnames(df_plate1)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate1)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate2)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T1   0.00145       0.8414 0.002864785 0.7819524 0.7790876
2     LBA_T1   0.00060       0.8387 0.002424559 0.7792226 0.7767980
3     LBA_T1   0.00780       0.7688 0.006154547 0.7090224 0.7028678
4     LBA_T1   0.01720       0.7431 0.011027715 0.6834359 0.6724081
5     LBA_T1   0.00945       0.7717 0.007009659 0.7119170 0.7049074
6     LBA_T1   0.00890       0.7706 0.006724608 0.7108189 0.7040942</code></pre>
<pre class="r"><code>colnames(df_plate2)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate2)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate3)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T2   0.01475      0.86050 0.009757202 0.8013024 0.7915452
2     LBA_T2   0.00160      0.79500 0.002942475 0.7352296 0.7322871
3     LBA_T2   0.09950      0.66995 0.053862716 0.6112481 0.5573854
4     LBA_T2   0.05880      0.72585 0.032641379 0.6663282 0.6336868
5     LBA_T2   0.05515      0.71555 0.030741887 0.6561383 0.6253964
6     LBA_T2   0.06015      0.86780 0.033344083 0.8087162 0.7753721</code></pre>
<pre class="r"><code>colnames(df_plate3)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate3)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate4)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T3   0.02445      0.79685 0.014788959 0.7370848 0.7222959
2     LBA_T3   0.00040      0.80785 0.002320981 0.7481290 0.7458080
3     LBA_T3   0.18255      0.72250 0.097400023 0.6630119 0.5656119
4     LBA_T3   0.09480      0.75955 0.051408267 0.6997995 0.6483913
5     LBA_T3   0.10640      0.73845 0.057467875 0.6788190 0.6213512
6     LBA_T3   0.10190      0.81120 0.055116439 0.7514969 0.6963805</code></pre>
<pre class="r"><code>colnames(df_plate4)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate4)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate5)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T4   0.05970      0.59795 0.03310984 0.5410917 0.5079819
2     LBA_T4   0.00000      0.60820 0.00211383 0.5510262 0.5489124
3     LBA_T4   0.46855      0.64370 0.24982559 0.5855690 0.3357435
4     LBA_T4   0.26555      0.72755 0.14123109 0.6680118 0.5267807
5     LBA_T4   0.33120      0.66180 0.17613115 0.6032628 0.4271317
6     LBA_T4   0.29225      0.64230 0.15540013 0.5842028 0.4288026</code></pre>
<pre class="r"><code>colnames(df_plate5)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate5)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate6)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T5   0.05190      0.57755 0.029051062 0.5213712 0.4923201
2     LBA_T5   0.00200      0.73200 0.003149654 0.6724214 0.6692717
3     LBA_T5   0.49880      0.53510 0.266182033 0.4805523 0.2143703
4     LBA_T5   0.30015      0.66705 0.159598995 0.6084054 0.4488064
5     LBA_T5   0.36560      0.51270 0.194501382 0.4591297 0.2646283
6     LBA_T5   0.37645      0.56970 0.200307428 0.5138008 0.3134934</code></pre>
<pre class="r"><code>colnames(df_plate6)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate6)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate7)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM     NO3_mM
1     LBA_T6   0.07220      0.48415 0.039619996 0.4319404 0.39232036
2     LBA_T6   0.00470      0.69890 0.004548298 0.6397056 0.63515729
3     LBA_T6   0.68060      0.48830 0.365472494 0.4358846 0.07041214
4     LBA_T6   0.36405      0.59190 0.193672416 0.5352361 0.34156367
5     LBA_T6   0.46630      0.41405 0.248610835 0.3657172 0.11710634
6     LBA_T6   0.47110      0.47085 0.251202620 0.4193178 0.16811514</code></pre>
<pre class="r"><code>colnames(df_plate7)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate7)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate8)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T7   0.00360      0.00000 0.003978441 0.0000000 0.0000000
2     LBA_T7   0.00470      0.59710 0.004548298 0.5402687 0.5357204
3     LBA_T7   0.59260      0.31575 0.317196723 0.2740963 0.0000000
4     LBA_T7   0.45125      0.50195 0.240491992 0.4488770 0.2083850
5     LBA_T7   0.28630      0.15655 0.152239669 0.1286372 0.0000000
6     LBA_T7   0.39420      0.21675 0.209818224 0.1832289 0.0000000</code></pre>
<pre class="r"><code>colnames(df_plate8)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate8)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate9)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540     NO2_mM NO2NO3_mM    NO3_mM
1     LBA_T8   0.05570      0.58555 0.03102807 0.5290966 0.4980685
2     LBA_T8   0.03470      0.54075 0.02011058 0.4859685 0.4658579
3     LBA_T8   0.05055      0.53110 0.02834886 0.4767210 0.4483721
4     LBA_T8   0.04435      0.60990 0.02512497 0.5526756 0.5275506
5     LBA_T8   0.04675      0.63915 0.02637272 0.5811299 0.5547572
6     LBA_T8   0.04965      0.58815 0.02788077 0.5316096 0.5037288</code></pre>
<pre class="r"><code>colnames(df_plate9)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate9)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate10)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM  NO2NO3_mM     NO3_mM
1     LBA_T9   0.11950      0.05945 0.064318416 0.04159994 0.00000000
2     LBA_T9   0.00170      0.51765 0.002994269 0.46385684 0.46086257
3     LBA_T9   0.07885      0.03225 0.043086271 0.01743783 0.00000000
4     LBA_T9   0.69515      0.45020 0.373493939 0.39977380 0.02627986
5     LBA_T9   0.00710      0.00000 0.005791808 0.00000000 0.00000000
6     LBA_T9   0.01100      0.00000 0.007813057 0.00000000 0.00000000</code></pre>
<pre class="r"><code>colnames(df_plate10)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate10)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code>head(df_plate11)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1       20
4      Slurry           NaOH              CHL    6               3       20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1    LBA_T10   0.00120      0.00000 0.002735303 0.0000000 0.0000000
2    LBA_T10   0.00155      0.57845 0.002916578 0.5222398 0.5193232
3    LBA_T10   0.01165      0.00000 0.008149997 0.0000000 0.0000000
4    LBA_T10   0.80215      0.43055 0.432834504 0.3812370 0.0000000
5    LBA_T10   0.00805      0.00000 0.006284101 0.0000000 0.0000000
6    LBA_T10   0.00980      0.00000 0.007191062 0.0000000 0.0000000</code></pre>
<pre class="r"><code>colnames(df_plate11)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate11)</code></pre>
<pre><code>[1] 96 17</code></pre>
<pre class="r"><code># head(df_plate16)
# colnames(df_plate16)
# dim(df_plate16)


# X1 to Well
# df_plate1 %&lt;&gt;% rename(Well = X1)
# df_plate2 %&lt;&gt;% rename(Well = X1)
# df_plate3 %&lt;&gt;% rename(Well = X1)
# df_plate4 %&lt;&gt;% rename(Well = X1)
# df_plate5 %&lt;&gt;% rename(Well = X1)
# df_plate6 %&lt;&gt;% rename(Well = X1)
# df_plate7 %&lt;&gt;% rename(Well = X1)
# df_plate8 %&lt;&gt;% rename(Well = X1)
# df_plate9 %&lt;&gt;% rename(Well = X1)
# df_plate10 %&lt;&gt;% rename(Well = X1)
# df_plate11 %&lt;&gt;% rename(Well = X1)
# df_plate12 %&lt;&gt;% rename(Well = X1)
# df_plate13 %&lt;&gt;% rename(Well = X1)
# df_plate14 %&lt;&gt;% rename(Well = X1)
# df_plate15 %&lt;&gt;% rename(Well = X1)
# df_plate16 %&lt;&gt;% rename(Well = X1)


# remove wells that were contaminated during the experiment
df_plate2 %&lt;&gt;% filter(!(Well %in% c(&quot;D02&quot;, &quot;D03&quot;))) # filter defect
df_plate4 %&lt;&gt;% filter(!(Well %in% c(&quot;H05&quot;))) # filter defect
df_plate5 %&lt;&gt;% filter(!(Well %in% c(&quot;B03&quot;, &quot;B04&quot;, &quot;C03&quot;))) # filter defect
df_plate8 %&lt;&gt;% filter(!(Well %in% c(&quot;A01&quot;, &quot;A09&quot;))) # filter defect
df_plate9 %&lt;&gt;% filter(!(Well %in% c(&quot;D11&quot;))) # NA detected below...
df_plate11 %&lt;&gt;% filter(!(Well %in% c(&quot;A12&quot;))) # filter defect

# bind two dataframe
df_p &lt;- rbind(df_plate1, df_plate2, df_plate3, df_plate4, df_plate5, df_plate6, df_plate7, df_plate8, df_plate9, df_plate10, df_plate11)
              # df_plate12, df_plate13, df_plate14, df_plate15, df_plate16)
dim(df_p)</code></pre>
<pre><code>[1] 1046   17</code></pre>
<pre class="r"><code># remove NA
dim(df_p) # 1046</code></pre>
<pre><code>[1] 1046   17</code></pre>
<pre class="r"><code>df_p[is.na(df_p)]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_na &lt;- df_p[rowSums(is.na(df_p)) &gt; 0,]  # D11 in plate9 has NA. Let&#39;s remove from the wells above.
# df_p &lt;- na.omit(df_p)
df_p &lt;- df_p[rowSums(is.na(df_p)) == 0,]
dim(df_p) # 1046</code></pre>
<pre><code>[1] 1046   17</code></pre>
<pre class="r"><code># multiply dilution factor which is 5/2
# df_p %&lt;&gt;% select(-NO2_OD540, -NO2NO3_OD540)
df_p %&lt;&gt;% mutate(NO2_mM = NO2_mM * (5/2), NO2NO3_mM = NO2NO3_mM * (5/2), NO3_mM = NO3_mM * (5/2))

# Get the metadata for time point and left join
Time_table &lt;- openxlsx::read.xlsx(&quot;data/220421_time_table.xlsx&quot;)
# Time_table_BN &lt;- openxlsx::read.xlsx(&quot;data/220421_time_table_BN.xlsx&quot;)
# Time_table &lt;- rbind(Time_table_AU, Time_table_BN)

Time_table %&lt;&gt;% select(-Date) 
Time_table$Time_hours &lt;- round(Time_table$Time_hours, 1)
Time_table$Time_days &lt;- round(Time_table$Time_days, 1)
df_time &lt;- Time_table
dim(df_p)</code></pre>
<pre><code>[1] 1046   17</code></pre>
<pre class="r"><code>df_p &lt;- df_p %&gt;% left_join(Time_table, by=(&quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p)</code></pre>
<pre><code>[1] 1046   20</code></pre>
<pre class="r"><code>colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;       &quot;Time_days&quot;       </code></pre>
<pre class="r"><code># time_point order
df_p$Time_point &lt;-  factor(df_p$Time_point, levels = c(paste0(&quot;LBA_T&quot;,0:10), paste0(&quot;Sterile_T&quot;,0:10)))</code></pre>
</div>
<div id="get-average-and-standard-deviation-moisture-correction-blank-correction" class="section level2">
<h2>2. Get average and standard deviation &amp; Moisture correction &amp; Blank correction</h2>
<pre class="r"><code># plot to see
ggplot(df_p, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Without averaging \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot to see
ggplot(df_p, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Without averaging \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># let&#39;s plot the no nitrate sample&#39;s nitrate, nitrite levels
df_no_nitrite &lt;- df_p %&gt;% filter(Sample_type == &quot;No_Nitrate&quot;)

ggplot(df_no_nitrite, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.5,0.05), limits=c(0, 0.5))+
  ggtitle(&quot;Samples without nitrate addition \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_no_nitrite, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Samples without nitrate addition \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># T8 is weird
# how about blanks?

# let&#39;s plot the blank sample&#39;s nitrate, nitrite levels
df_A_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)
ggplot(df_A_blank, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrate blanks \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_A_blank, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Nitrate blanks \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># let&#39;s plot the Nitrite blank sample&#39;s nitrate, nitrite levels
df_I_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
ggplot(df_I_blank, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrite blanks \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_I_blank, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrite blanks \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># average technical replicate (here only 1 replicate per pH perturbation)
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;       &quot;Time_days&quot;       </code></pre>
<pre class="r"><code>dim(df_p)</code></pre>
<pre><code>[1] 1046   20</code></pre>
<pre class="r"><code># df_p &lt;- df_p %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Concentration_M, Added_ul, Sample_type, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(NO2_mM = mean(NO2_mM), Std_NO2_mM = sd(NO2_mM), NO3_mM = mean(NO3_mM), Std_NO3_mM = sd(NO3_mM)) %&gt;% ungroup()


# mols and molarity unit conversion
# test
df_p$Titration_type %&gt;% length()</code></pre>
<pre><code>[1] 1046</code></pre>
<pre class="r"><code># Setting H_mM column
# df_soil &lt;- df_p %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;))

Added_Volume &lt;- 1.7 # ml
Soil_mg &lt;- 0.85
moisture_percent_1 = 23.96 # LBA - LaBaghWoods for antibiotics
Added_Volume + Soil_mg*(moisture_percent_1/100)</code></pre>
<pre><code>[1] 1.90366</code></pre>
<pre class="r"><code>moisture_percent_2 = 8.12 # Sterile - LaBaghWoods that has been autoclaved 5 times.
Added_Volume + Soil_mg*(moisture_percent_2/100)</code></pre>
<pre><code>[1] 1.76902</code></pre>
<pre class="r"><code># moisture_percent_3 = 9.0 # Crerar7
# Added_Volume + Soil_mg*(moisture_percent_3/100)

df_p$Added_ul &lt;- ifelse(df_p$Titration_type == &quot;NaOH&quot;, -1*df_p$Added_ul, df_p$Added_ul) # HCl is +, NaOH is -
df_p %&lt;&gt;% mutate(H_mol = Concentration_M * Added_ul * 10^(-6)) # Calculate H mol 
df_p$Volume &lt;- ifelse(df_p$Soil == &quot;LaBaghWoods_pH6.66&quot;, Added_Volume + Soil_mg*(moisture_percent_1/100),
                      ifelse(df_p$Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;, Added_Volume + Soil_mg*(moisture_percent_2/100),
                             0))
                             # ifelse(df_p$Soil == &quot;Crerar1_pH8.05&quot;, Added_Volume + Soil_mg*(moisture_percent_3/100),
                                    # 0))) # Calc total volume

# df_p$Volume &lt;- ifelse(df_p$Soil == &quot;Allandale_pH4&quot;, Added_Volume + Soil_mg*(moisture_percent_2/100), df_p$Volume) # Calc total volume
# df_p$Volume %&gt;% unique()
df_p %&lt;&gt;% mutate(H_Molarity = H_mol / (Volume * 10^(-3)))
df_p %&lt;&gt;% mutate(H_mM = H_Molarity * 1000)
# openxlsx::write.xlsx(df_p, &quot;df_p.xlsx&quot;)

# how many levels of H_mM?
df_p %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange(H_mM)</code></pre>
<pre><code>       H_mM
1 -42.02431
2 -31.51823
3 -21.01216
4 -10.50608
5   0.00000
6  10.50608
7  21.01216
8  31.51823
9  42.02431</code></pre>
<pre class="r"><code>df_p %&gt;% filter(Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange(H_mM)</code></pre>
<pre><code>         H_mM
1  -96.098405
2  -90.445557
3  -84.792710
4  -79.139863
5  -73.487015
6  -67.834168
7  -62.181321
8  -56.528473
9  -50.875626
10 -45.222779
11 -39.569931
12 -33.917084
13 -28.264237
14 -22.611389
15 -16.958542
16 -11.305695
17  -5.652847
18   0.000000
19   5.652847
20  11.305695
21  16.958542
22  22.611389
23  28.264237
24  33.917084
25  39.569931
26  45.222779
27  50.875626
28  56.528473
29  62.181321
30  67.834168
31  73.487015
32  79.139863
33  84.792710
34  90.445557</code></pre>
<pre class="r"><code># df_p %&gt;% filter(Soil == &quot;Crerar1_pH8.05&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange()
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;       &quot;Time_days&quot;       
[21] &quot;H_mol&quot;            &quot;Volume&quot;           &quot;H_Molarity&quot;       &quot;H_mM&quot;            </code></pre>
<pre class="r"><code>## Moisture correction
dim(df_p)</code></pre>
<pre><code>[1] 1046   24</code></pre>
<pre class="r"><code># Testing negative samples
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)  # Use Nitrate_Blank instead in the future
df_no3_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input
1   G06             0             2              0
2   G12             0             2              0
3   H05             0             2              0
4   H06             0             2              0
5   H11             0             2              0
6   H12             0             2              0
7   G06             0             2              0
8   G12             0             2              0
9   H05             0             2              0
10  H06             0             2              0
11  H11             0             2              0
12  H12             0             2              0
13  G06             0             2              0
14  G12             0             2              0
15  H05             0             2              0
16  H06             0             2              0
17  H11             0             2              0
18  H12             0             2              0
19  G06             0             2              0
20  G12             0             2              0
21  H06             0             2              0
22  H11             0             2              0
23  H12             0             2              0
24  G06             0             2              0
25  G12             0             2              0
26  H05             0             2              0
27  H06             0             2              0
28  H11             0             2              0
29  H12             0             2              0
30  G06             0             2              0
31  G12             0             2              0
32  H05             0             2              0
33  H06             0             2              0
34  H11             0             2              0
35  H12             0             2              0
36  G06             0             2              0
37  G12             0             2              0
38  H05             0             2              0
39  H06             0             2              0
40  H11             0             2              0
41  H12             0             2              0
42  G06             0             2              0
43  G12             0             2              0
44  H05             0             2              0
45  H06             0             2              0
46  H11             0             2              0
47  H12             0             2              0
48  G06             0             2              0
49  G12             0             2              0
50  H05             0             2              0
51  H06             0             2              0
52  H11             0             2              0
53  H12             0             2              0
54  G06             0             2              0
55  G12             0             2              0
56  H05             0             2              0
57  H06             0             2              0
58  H11             0             2              0
59  H12             0             2              0
60  G06             0             2              0
61  G12             0             2              0
62  H05             0             2              0
63  H06             0             2              0
64  H11             0             2              0
65  H12             0             2              0
                            Soil   Sample_type Titration_type Antibiotics_type
1             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
2  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
3             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
4             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
5  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
6  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
7             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
8  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
9             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
10            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
11 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
12 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
13            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
14 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
15            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
16            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
17 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
18 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
19            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
20 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
21            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
22 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
23 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
24            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
25 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
26            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
27            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
28 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
29 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
30            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
31 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
32            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
33            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
34 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
35 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
36            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
37 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
38            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
39            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
40 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
41 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
42            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
43 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
44            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
45            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
46 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
47 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
48            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
49 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
50            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
51            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
52 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
53 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
54            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
55 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
56            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
57            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
58 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
59 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
60            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
61 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
62            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
63            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
64 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
65 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
   Unit Concentration_M Added_ul  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM
1     0               0       20      LBA_T0   0.00185      0.84865 0.007679903
2     0               0       20  Sterile_T0   0.00380      0.84810 0.010205119
3     0               0       20      LBA_T0   0.00220      0.82835 0.008133116
4     0               0       20      LBA_T0   0.00030      0.84050 0.005672981
5     0               0       20  Sterile_T0   0.00210      0.83720 0.008003625
6     0               0       20  Sterile_T0   0.00215      0.84055 0.008068371
7     0               0       20      LBA_T1   0.00140      0.87535 0.007097221
8     0               0       20  Sterile_T1   0.00260      0.83150 0.008651090
9     0               0       20      LBA_T1   0.00085      0.83735 0.006385084
10    0               0       20      LBA_T1   0.00120      0.86405 0.006838258
11    0               0       20  Sterile_T1   0.00250      0.84320 0.008521595
12    0               0       20  Sterile_T1   0.00305      0.83035 0.009233832
13    0               0       20      LBA_T2   0.00275      0.88520 0.008845335
14    0               0       20  Sterile_T2   0.00145      0.85475 0.007161963
15    0               0       20      LBA_T2   0.00000      0.84705 0.005284575
16    0               0       20      LBA_T2   0.00000      0.87325 0.005284575
17    0               0       20  Sterile_T2   0.00125      0.82580 0.006902999
18    0               0       20  Sterile_T2   0.00205      0.82150 0.007938880
19    0               0       20      LBA_T3   0.00275      0.89955 0.008845335
20    0               0       20  Sterile_T3   0.00390      0.83225 0.010334628
21    0               0       20      LBA_T3   0.00225      0.88295 0.008197862
22    0               0       20  Sterile_T3   0.00205      0.82825 0.007938880
23    0               0       20  Sterile_T3   0.00305      0.83335 0.009233832
24    0               0       20      LBA_T4   0.00000      0.85115 0.005284575
25    0               0       20  Sterile_T4   0.00000      0.85870 0.005284575
26    0               0       20      LBA_T4   0.00000      0.87375 0.005284575
27    0               0       20      LBA_T4   0.00000      0.86850 0.005284575
28    0               0       20  Sterile_T4   0.00000      0.85550 0.005284575
29    0               0       20  Sterile_T4   0.00000      0.85995 0.005284575
30    0               0       20      LBA_T5   0.00000      0.85915 0.005284575
31    0               0       20  Sterile_T5   0.00000      0.87575 0.005284575
32    0               0       20      LBA_T5   0.00170      0.88555 0.007485673
33    0               0       20      LBA_T5   0.00245      0.88010 0.008456848
34    0               0       20  Sterile_T5   0.00195      0.86395 0.007809391
35    0               0       20  Sterile_T5   0.00235      0.87990 0.008327354
36    0               0       20      LBA_T6   0.00255      0.85780 0.008586343
37    0               0       20  Sterile_T6   0.00070      0.86350 0.006190871
38    0               0       20      LBA_T6   0.00340      0.86075 0.009687092
39    0               0       20      LBA_T6   0.00280      0.86860 0.008910084
40    0               0       20  Sterile_T6   0.00285      0.84890 0.008974833
41    0               0       20  Sterile_T6   0.00560      0.85535 0.012536461
42    0               0       20      LBA_T7   0.00465      0.85765 0.011305986
43    0               0       20  Sterile_T7   0.00360      0.84680 0.009946103
44    0               0       20      LBA_T7   0.00455      0.87800 0.011176468
45    0               0       20      LBA_T7   0.00460      0.86985 0.011241227
46    0               0       20  Sterile_T7   0.00435      0.85990 0.010917435
47    0               0       20  Sterile_T7   0.00310      0.85370 0.009298583
48    0               0       20      LBA_T8   0.00030      0.85195 0.005672981
49    0               0       20  Sterile_T8   0.00070      0.86020 0.006190871
50    0               0       20      LBA_T8   0.00045      0.89480 0.005867188
51    0               0       20      LBA_T8   0.00195      0.85455 0.007809391
52    0               0       20  Sterile_T8   0.00045      0.86125 0.005867188
53    0               0       20  Sterile_T8   0.00000      0.82755 0.005284575
54    0               0       20      LBA_T9   0.00140      0.86385 0.007097221
55    0               0       20  Sterile_T9   0.00135      0.86310 0.007032480
56    0               0       20      LBA_T9   0.00320      0.88960 0.009428085
57    0               0       20      LBA_T9   0.00045      0.86290 0.005867188
58    0               0       20  Sterile_T9   0.00195      0.83700 0.007809391
59    0               0       20  Sterile_T9   0.00055      0.84860 0.005996660
60    0               0       20     LBA_T10   0.00105      0.86225 0.006644039
61    0               0       20 Sterile_T10   0.00130      0.86770 0.006967739
62    0               0       20     LBA_T10   0.00075      0.87680 0.006255609
63    0               0       20     LBA_T10   0.00000      0.87560 0.005284575
64    0               0       20 Sterile_T10   0.00100      0.86425 0.006579300
65    0               0       20 Sterile_T10   0.00000      0.86765 0.005284575
   NO2NO3_mM   NO3_mM Time_minutes Time_hours Time_days H_mol  Volume
1   1.973223 1.965543            0        0.0       0.0     0 1.90366
2   1.971831 1.961625            0        0.0       0.0     0 1.76902
3   1.921928 1.913795            0        0.0       0.0     0 1.90366
4   1.952606 1.946933            0        0.0       0.0     0 1.90366
5   1.944267 1.936263            0        0.0       0.0     0 1.76902
6   1.952732 1.944664            0        0.0       0.0     0 1.76902
7   2.040987 2.033889          260        4.3       0.2     0 1.90366
8   1.929875 1.921224          180        3.0       0.1     0 1.76902
9   1.944646 1.938260          260        4.3       0.2     0 1.90366
10  2.012266 2.005428          260        4.3       0.2     0 1.90366
11  1.959432 1.950911          180        3.0       0.1     0 1.76902
12  1.926973 1.917739          180        3.0       0.1     0 1.76902
13  2.066072 2.057226          450        7.5       0.3     0 1.90366
14  1.988675 1.981513          370        6.2       0.3     0 1.76902
15  1.969173 1.963888          450        7.5       0.3     0 1.90366
16  2.035645 2.030360          450        7.5       0.3     0 1.90366
17  1.915498 1.908595          370        6.2       0.3     0 1.76902
18  1.904662 1.896723          370        6.2       0.3     0 1.76902
19  2.102701 2.093855          600       10.0       0.4     0 1.90366
20  1.931767 1.921433          520        8.7       0.4     0 1.76902
21  2.060337 2.052140          600       10.0       0.4     0 1.90366
22  1.921676 1.913737          520        8.7       0.4     0 1.76902
23  1.934544 1.925310          520        8.7       0.4     0 1.76902
24  1.979553 1.974269         1220       20.3       0.8     0 1.90366
25  1.998690 1.993405         1150       19.2       0.8     0 1.76902
26  2.036916 2.031632         1220       20.3       0.8     0 1.90366
27  2.023569 2.018285         1220       20.3       0.8     0 1.90366
28  1.990576 1.985291         1150       19.2       0.8     0 1.76902
29  2.001861 1.996576         1150       19.2       0.8     0 1.76902
30  1.999831 1.994546         1600       26.7       1.1     0 1.90366
31  2.042004 2.036720         1520       25.3       1.1     0 1.76902
32  2.066964 2.059478         1600       26.7       1.1     0 1.90366
33  2.053078 2.044621         1600       26.7       1.1     0 1.90366
34  2.012012 2.004203         1520       25.3       1.1     0 1.76902
35  2.052568 2.044241         1520       25.3       1.1     0 1.76902
36  1.996407 1.987821         1950       32.5       1.4     0 1.90366
37  2.010870 2.004679         1870       31.2       1.3     0 1.76902
38  2.003890 1.994203         1950       32.5       1.4     0 1.90366
39  2.023823 2.014913         1950       32.5       1.4     0 1.90366
40  1.973856 1.964881         1870       31.2       1.3     0 1.76902
41  1.990195 1.977659         1870       31.2       1.3     0 1.76902
42  1.996027 1.984721         2680       44.7       1.9     0 1.90366
43  1.968540 1.958594         2610       43.5       1.8     0 1.76902
44  2.047731 2.036554         2680       44.7       1.9     0 1.90366
45  2.027000 2.015759         2680       44.7       1.9     0 1.90366
46  2.001734 1.990816         2610       43.5       1.8     0 1.76902
47  1.986014 1.976715         2610       43.5       1.8     0 1.76902
48  1.981580 1.975907         3370       56.2       2.3     0 1.90366
49  2.002495 1.996304         3290       54.8       2.3     0 1.76902
50  2.090565 2.084698         3370       56.2       2.3     0 1.90366
51  1.988168 1.980358         3370       56.2       2.3     0 1.90366
52  2.005159 1.999292         3290       54.8       2.3     0 1.76902
53  1.919910 1.914626         3290       54.8       2.3     0 1.76902
54  2.011758 2.004661         4330       72.2       3.0     0 1.90366
55  2.009854 2.002822         4260       71.0       3.0     0 1.76902
56  2.077292 2.067864         4330       72.2       3.0     0 1.90366
57  2.009347 2.003479         4330       72.2       3.0     0 1.90366
58  1.943761 1.935952         4260       71.0       3.0     0 1.76902
59  1.973096 1.967100         4260       71.0       3.0     0 1.76902
60  2.007697 2.001053         5760       96.0       4.0     0 1.90366
61  2.021536 2.014569         5690       94.8       4.0     0 1.76902
62  2.044676 2.038421         5760       96.0       4.0     0 1.90366
63  2.041623 2.036338         5760       96.0       4.0     0 1.90366
64  2.012774 2.006195         5690       94.8       4.0     0 1.76902
65  2.021409 2.016125         5690       94.8       4.0     0 1.76902
   H_Molarity H_mM
1           0    0
2           0    0
3           0    0
4           0    0
5           0    0
6           0    0
7           0    0
8           0    0
9           0    0
10          0    0
11          0    0
12          0    0
13          0    0
14          0    0
15          0    0
16          0    0
17          0    0
18          0    0
19          0    0
20          0    0
21          0    0
22          0    0
23          0    0
24          0    0
25          0    0
26          0    0
27          0    0
28          0    0
29          0    0
30          0    0
31          0    0
32          0    0
33          0    0
34          0    0
35          0    0
36          0    0
37          0    0
38          0    0
39          0    0
40          0    0
41          0    0
42          0    0
43          0    0
44          0    0
45          0    0
46          0    0
47          0    0
48          0    0
49          0    0
50          0    0
51          0    0
52          0    0
53          0    0
54          0    0
55          0    0
56          0    0
57          0    0
58          0    0
59          0    0
60          0    0
61          0    0
62          0    0
63          0    0
64          0    0
65          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input
1   G04             2             0              0
2   G05             2             0              0
3   G10             2             0              0
4   G11             2             0              0
5   H04             2             0              0
6   H10             2             0              0
7   G04             2             0              0
8   G05             2             0              0
9   G10             2             0              0
10  G11             2             0              0
11  H04             2             0              0
12  H10             2             0              0
13  G04             2             0              0
14  G05             2             0              0
15  G10             2             0              0
16  G11             2             0              0
17  H04             2             0              0
18  H10             2             0              0
19  G04             2             0              0
20  G05             2             0              0
21  G10             2             0              0
22  G11             2             0              0
23  H04             2             0              0
24  H10             2             0              0
25  G04             2             0              0
26  G05             2             0              0
27  G10             2             0              0
28  G11             2             0              0
29  H04             2             0              0
30  H10             2             0              0
31  G04             2             0              0
32  G05             2             0              0
33  G10             2             0              0
34  G11             2             0              0
35  H04             2             0              0
36  H10             2             0              0
37  G04             2             0              0
38  G05             2             0              0
39  G10             2             0              0
40  G11             2             0              0
41  H04             2             0              0
42  H10             2             0              0
43  G04             2             0              0
44  G05             2             0              0
45  G10             2             0              0
46  G11             2             0              0
47  H04             2             0              0
48  H10             2             0              0
49  G04             2             0              0
50  G05             2             0              0
51  G10             2             0              0
52  G11             2             0              0
53  H04             2             0              0
54  H10             2             0              0
55  G04             2             0              0
56  G05             2             0              0
57  G10             2             0              0
58  G11             2             0              0
59  H04             2             0              0
60  H10             2             0              0
61  G04             2             0              0
62  G05             2             0              0
63  G10             2             0              0
64  G11             2             0              0
65  H04             2             0              0
66  H10             2             0              0
                            Soil   Sample_type Titration_type Antibiotics_type
1             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
2             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
3  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
4  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
5             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
6  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
7             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
8             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
9  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
10 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
11            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
12 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
13            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
14            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
15 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
16 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
17            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
18 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
19            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
20            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
21 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
22 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
23            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
24 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
25            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
26            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
27 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
28 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
29            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
30 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
31            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
32            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
33 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
34 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
35            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
36 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
37            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
38            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
39 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
40 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
41            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
42 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
43            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
44            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
45 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
46 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
47            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
48 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
49            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
50            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
51 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
52 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
53            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
54 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
55            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
56            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
57 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
58 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
59            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
60 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
61            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
62            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
63 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
64 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
65            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
66 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
   Unit Concentration_M Added_ul  Time_point NO2_OD540 NO2NO3_OD540   NO2_mM
1     0               0       20      LBA_T0   1.28420      0.73540 1.770814
2     0               0       20      LBA_T0   1.25625      0.72680 1.729915
3     0               0       20  Sterile_T0   1.27375      0.73220 1.755508
4     0               0       20  Sterile_T0   1.25275      0.72250 1.724803
5     0               0       20      LBA_T0   1.26850      0.71880 1.747825
6     0               0       20  Sterile_T0   1.28700      0.73115 1.774918
7     0               0       20      LBA_T1   1.27205      0.71720 1.753020
8     0               0       20      LBA_T1   1.28380      0.72505 1.770228
9     0               0       20  Sterile_T1   1.29310      0.73860 1.783864
10    0               0       20  Sterile_T1   1.22090      0.68830 1.678366
11    0               0       20      LBA_T1   1.28835      0.73185 1.776898
12    0               0       20  Sterile_T1   1.25910      0.71855 1.734080
13    0               0       20      LBA_T2   1.27020      0.71075 1.750313
14    0               0       20      LBA_T2   1.27355      0.71650 1.755216
15    0               0       20  Sterile_T2   1.25770      0.70470 1.732034
16    0               0       20  Sterile_T2   1.26390      0.70255 1.741097
17    0               0       20      LBA_T2   1.32125      0.73420 1.825223
18    0               0       20  Sterile_T2   1.22830      0.69255 1.689141
19    0               0       20      LBA_T3   1.33145      0.75875 1.840241
20    0               0       20      LBA_T3   1.36315      0.76305 1.887022
21    0               0       20  Sterile_T3   1.27765      0.72420 1.761219
22    0               0       20  Sterile_T3   1.26775      0.72285 1.746728
23    0               0       20      LBA_T3   1.33220      0.75045 1.841345
24    0               0       20  Sterile_T3   1.27535      0.72070 1.757851
25    0               0       20      LBA_T4   1.41820      0.77100 1.968657
26    0               0       20      LBA_T4   1.37330      0.75800 1.902036
27    0               0       20  Sterile_T4   1.32355      0.72545 1.828608
28    0               0       20  Sterile_T4   1.32680      0.72915 1.833392
29    0               0       20      LBA_T4   1.34280      0.68895 1.856972
30    0               0       20  Sterile_T4   1.39710      0.75735 1.937308
31    0               0       20      LBA_T5   1.33465      0.73815 1.844955
32    0               0       20      LBA_T5   1.35245      0.75000 1.871213
33    0               0       20  Sterile_T5   1.33120      0.73910 1.839872
34    0               0       20  Sterile_T5   1.32280      0.73555 1.827504
35    0               0       20      LBA_T5   1.31110      0.74030 1.810295
36    0               0       20  Sterile_T5   1.32955      0.73990 1.837442
37    0               0       20      LBA_T6   1.31585      0.72080 1.817279
38    0               0       20      LBA_T6   1.33000      0.72715 1.838105
39    0               0       20  Sterile_T6   1.33610      0.73075 1.847093
40    0               0       20  Sterile_T6   1.32390      0.72085 1.829123
41    0               0       20      LBA_T6   1.31795      0.71920 1.820368
42    0               0       20  Sterile_T6   1.35060      0.74060 1.868482
43    0               0       20      LBA_T7   1.32370      0.72175 1.828828
44    0               0       20      LBA_T7   1.36480      0.73930 1.889462
45    0               0       20  Sterile_T7   1.35725      0.73975 1.878303
46    0               0       20  Sterile_T7   1.34255      0.73035 1.856603
47    0               0       20      LBA_T7   1.37120      0.75070 1.898928
48    0               0       20  Sterile_T7   1.34785      0.73880 1.864422
49    0               0       20      LBA_T8   1.43060      0.77035 1.987116
50    0               0       20      LBA_T8   1.48695      0.80045 2.071324
51    0               0       20  Sterile_T8   1.39330      0.74855 1.931670
52    0               0       20  Sterile_T8   1.40160      0.75850 1.943988
53    0               0       20      LBA_T8   1.48665      0.78850 2.070875
54    0               0       20  Sterile_T8   1.42960      0.77200 1.985626
55    0               0       20      LBA_T9   1.40605      0.75530 1.950596
56    0               0       20      LBA_T9   1.40870      0.75905 1.954534
57    0               0       20  Sterile_T9   1.40380      0.74950 1.947255
58    0               0       20  Sterile_T9   1.40415      0.75505 1.947774
59    0               0       20      LBA_T9   1.41595      0.75740 1.965311
60    0               0       20  Sterile_T9   1.39510      0.75050 1.934340
61    0               0       20     LBA_T10   1.42685      0.76600 1.981531
62    0               0       20     LBA_T10   1.45055      0.77990 2.016867
63    0               0       20 Sterile_T10   1.42340      0.76660 1.976395
64    0               0       20 Sterile_T10   1.43185      0.76905 1.988978
65    0               0       20     LBA_T10   1.41760      0.76625 1.967765
66    0               0       20 Sterile_T10   1.42020      0.76380 1.971633
   NO2NO3_mM NO3_mM Time_minutes Time_hours Time_days H_mol  Volume H_Molarity
1   1.689482      0            0        0.0       0.0     0 1.90366          0
2   1.668172      0            0        0.0       0.0     0 1.90366          0
3   1.681549      0            0        0.0       0.0     0 1.76902          0
4   1.657530      0            0        0.0       0.0     0 1.76902          0
5   1.648379      0            0        0.0       0.0     0 1.90366          0
6   1.678947      0            0        0.0       0.0     0 1.76902          0
7   1.644423      0          260        4.3       0.2     0 1.90366          0
8   1.663840      0          260        4.3       0.2     0 1.90366          0
9   1.697420      0          180        3.0       0.1     0 1.76902          0
10  1.573173      0          180        3.0       0.1     0 1.76902          0
11  1.680682      0          260        4.3       0.2     0 1.90366          0
12  1.647761      0          180        3.0       0.1     0 1.76902          0
13  1.628490      0          450        7.5       0.3     0 1.90366          0
14  1.642693      0          450        7.5       0.3     0 1.90366          0
15  1.613561      0          370        6.2       0.3     0 1.76902          0
16  1.608260      0          370        6.2       0.3     0 1.76902          0
17  1.686507      0          450        7.5       0.3     0 1.90366          0
18  1.583628      0          370        6.2       0.3     0 1.76902          0
19  1.747507      0          600       10.0       0.4     0 1.90366          0
20  1.758219      0          600       10.0       0.4     0 1.90366          0
21  1.661736      0          520        8.7       0.4     0 1.76902          0
22  1.658396      0          520        8.7       0.4     0 1.76902          0
23  1.726853      0          600       10.0       0.4     0 1.90366          0
24  1.653077      0          520        8.7       0.4     0 1.76902          0
25  1.778045      0         1220       20.3       0.8     0 1.90366          0
26  1.745639      0         1220       20.3       0.8     0 1.90366          0
27  1.664830      0         1150       19.2       0.8     0 1.76902          0
28  1.673992      0         1150       19.2       0.8     0 1.76902          0
29  1.574771      0         1220       20.3       0.8     0 1.90366          0
30  1.744021      0         1150       19.2       0.8     0 1.76902          0
31  1.696303      0         1600       26.7       1.1     0 1.90366          0
32  1.725734      0         1600       26.7       1.1     0 1.90366          0
33  1.698660      0         1520       25.3       1.1     0 1.76902          0
34  1.689854      0         1520       25.3       1.1     0 1.76902          0
35  1.701638      0         1600       26.7       1.1     0 1.90366          0
36  1.700646      0         1520       25.3       1.1     0 1.76902          0
37  1.653325      0         1950       32.5       1.4     0 1.90366          0
38  1.669039      0         1950       32.5       1.4     0 1.90366          0
39  1.677956      0         1870       31.2       1.3     0 1.76902          0
40  1.653448      0         1870       31.2       1.3     0 1.76902          0
41  1.649368      0         1950       32.5       1.4     0 1.90366          0
42  1.702383      0         1870       31.2       1.3     0 1.76902          0
43  1.655674      0         2680       44.7       1.9     0 1.90366          0
44  1.699157      0         2680       44.7       1.9     0 1.90366          0
45  1.700273      0         2610       43.5       1.8     0 1.76902          0
46  1.676965      0         2610       43.5       1.8     0 1.76902          0
47  1.727475      0         2680       44.7       1.9     0 1.90366          0
48  1.697916      0         2610       43.5       1.8     0 1.76902          0
49  1.776423      0         3370       56.2       2.3     0 1.90366          0
50  1.851742      0         3370       56.2       2.3     0 1.90366          0
51  1.722130      0         3290       54.8       2.3     0 1.76902          0
52  1.746884      0         3290       54.8       2.3     0 1.76902          0
53  1.821790      0         3370       56.2       2.3     0 1.90366          0
54  1.780541      0         3290       54.8       2.3     0 1.76902          0
55  1.738918      0         4330       72.2       3.0     0 1.90366          0
56  1.748254      0         4330       72.2       3.0     0 1.90366          0
57  1.724491      0         4260       71.0       3.0     0 1.76902          0
58  1.738296      0         4260       71.0       3.0     0 1.76902          0
59  1.744145      0         4330       72.2       3.0     0 1.90366          0
60  1.726978      0         4260       71.0       3.0     0 1.76902          0
61  1.765572      0         5760       96.0       4.0     0 1.90366          0
62  1.800275      0         5760       96.0       4.0     0 1.90366          0
63  1.767069      0         5690       94.8       4.0     0 1.76902          0
64  1.773180      0         5690       94.8       4.0     0 1.76902          0
65  1.766196      0         5760       96.0       4.0     0 1.90366          0
66  1.760088      0         5690       94.8       4.0     0 1.76902          0
   H_mM
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
26    0
27    0
28    0
29    0
30    0
31    0
32    0
33    0
34    0
35    0
36    0
37    0
38    0
39    0
40    0
41    0
42    0
43    0
44    0
45    0
46    0
47    0
48    0
49    0
50    0
51    0
52    0
53    0
54    0
55    0
56    0
57    0
58    0
59    0
60    0
61    0
62    0
63    0
64    0
65    0
66    0</code></pre>
<pre class="r"><code># close to zero

# 1. Apply moisture correction factor (correcting for moisture in soil)
soil_spike_ratio = 0.5 # soil weight(0.85g) / spike in volume (1.7ml) 

mcf_1 = (soil_spike_ratio*(moisture_percent_1/100) + 1)
mcf_1</code></pre>
<pre><code>[1] 1.1198</code></pre>
<pre class="r"><code>mcf_2 = (soil_spike_ratio*(moisture_percent_2/100) + 1)
mcf_2</code></pre>
<pre><code>[1] 1.0406</code></pre>
<pre class="r"><code># mcf_3 = (soil_spike_ratio*(moisture_percent_3/100) + 1)
# mcf_3

# apply moisture factor to each soil
dim(df_p)</code></pre>
<pre><code>[1] 1046   24</code></pre>
<pre class="r"><code>df_p %&gt;% select(Soil) %&gt;% unique()</code></pre>
<pre><code>                           Soil
1            LaBaghWoods_pH6.66
7 LaBaghWoods_autoclaved_pH6.66</code></pre>
<pre class="r"><code># this is a special case with no replicates
df_p_mcf_1 &lt;- df_p %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;) %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_1, NO2_mM = NO2_mM * mcf_1)
dim(df_p_mcf_1) #423</code></pre>
<pre><code>[1] 423  24</code></pre>
<pre class="r"><code>df_p_mcf_2 &lt;- df_p %&gt;% filter(Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;) %&gt;%  filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_2, NO2_mM = NO2_mM * mcf_2)
dim(df_p_mcf_2) #426</code></pre>
<pre><code>[1] 426  24</code></pre>
<pre class="r"><code># df_p_mcf_3 &lt;- df_p %&gt;% filter(Soil == &quot;Crerar1_pH8.05&quot;) %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_3, NO2_mM = NO2_mM * mcf_3)
# dim(df_p_mcf_3) #426

df_p_others_mcf &lt;- df_p %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;, &quot;Nitrate_Blank&quot;, &quot;Ammonium_Blank&quot;)) # Use Nitrate_Blank instead in the future
dim(df_p_others_mcf) #197</code></pre>
<pre><code>[1] 197  24</code></pre>
<pre class="r"><code>df_p_mcf &lt;- rbind(df_p_mcf_1, df_p_mcf_2, df_p_others_mcf)
dim(df_p_mcf) # 1046</code></pre>
<pre><code>[1] 1046   24</code></pre>
<pre class="r"><code># plot to see
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hours)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;After averaging with biological replicates \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-3-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. Apply blank correction factor (drying effect during incubation)
# Blank reads
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)
df_no3_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input
1   G06             0             2              0
2   G12             0             2              0
3   H05             0             2              0
4   H06             0             2              0
5   H11             0             2              0
6   H12             0             2              0
7   G06             0             2              0
8   G12             0             2              0
9   H05             0             2              0
10  H06             0             2              0
11  H11             0             2              0
12  H12             0             2              0
13  G06             0             2              0
14  G12             0             2              0
15  H05             0             2              0
16  H06             0             2              0
17  H11             0             2              0
18  H12             0             2              0
19  G06             0             2              0
20  G12             0             2              0
21  H06             0             2              0
22  H11             0             2              0
23  H12             0             2              0
24  G06             0             2              0
25  G12             0             2              0
26  H05             0             2              0
27  H06             0             2              0
28  H11             0             2              0
29  H12             0             2              0
30  G06             0             2              0
31  G12             0             2              0
32  H05             0             2              0
33  H06             0             2              0
34  H11             0             2              0
35  H12             0             2              0
36  G06             0             2              0
37  G12             0             2              0
38  H05             0             2              0
39  H06             0             2              0
40  H11             0             2              0
41  H12             0             2              0
42  G06             0             2              0
43  G12             0             2              0
44  H05             0             2              0
45  H06             0             2              0
46  H11             0             2              0
47  H12             0             2              0
48  G06             0             2              0
49  G12             0             2              0
50  H05             0             2              0
51  H06             0             2              0
52  H11             0             2              0
53  H12             0             2              0
54  G06             0             2              0
55  G12             0             2              0
56  H05             0             2              0
57  H06             0             2              0
58  H11             0             2              0
59  H12             0             2              0
60  G06             0             2              0
61  G12             0             2              0
62  H05             0             2              0
63  H06             0             2              0
64  H11             0             2              0
65  H12             0             2              0
                            Soil   Sample_type Titration_type Antibiotics_type
1             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
2  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
3             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
4             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
5  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
6  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
7             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
8  LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
9             LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
10            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
11 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
12 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
13            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
14 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
15            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
16            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
17 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
18 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
19            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
20 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
21            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
22 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
23 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
24            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
25 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
26            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
27            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
28 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
29 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
30            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
31 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
32            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
33            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
34 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
35 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
36            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
37 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
38            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
39            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
40 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
41 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
42            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
43 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
44            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
45            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
46 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
47 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
48            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
49 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
50            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
51            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
52 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
53 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
54            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
55 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
56            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
57            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
58 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
59 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
60            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
61 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
62            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
63            LaBaghWoods_pH6.66 Nitrate_Blank        Nitrate             None
64 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
65 LaBaghWoods_autoclaved_pH6.66 Nitrate_Blank        Nitrate             None
   Unit Concentration_M Added_ul  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM
1     0               0       20      LBA_T0   0.00185      0.84865 0.007679903
2     0               0       20  Sterile_T0   0.00380      0.84810 0.010205119
3     0               0       20      LBA_T0   0.00220      0.82835 0.008133116
4     0               0       20      LBA_T0   0.00030      0.84050 0.005672981
5     0               0       20  Sterile_T0   0.00210      0.83720 0.008003625
6     0               0       20  Sterile_T0   0.00215      0.84055 0.008068371
7     0               0       20      LBA_T1   0.00140      0.87535 0.007097221
8     0               0       20  Sterile_T1   0.00260      0.83150 0.008651090
9     0               0       20      LBA_T1   0.00085      0.83735 0.006385084
10    0               0       20      LBA_T1   0.00120      0.86405 0.006838258
11    0               0       20  Sterile_T1   0.00250      0.84320 0.008521595
12    0               0       20  Sterile_T1   0.00305      0.83035 0.009233832
13    0               0       20      LBA_T2   0.00275      0.88520 0.008845335
14    0               0       20  Sterile_T2   0.00145      0.85475 0.007161963
15    0               0       20      LBA_T2   0.00000      0.84705 0.005284575
16    0               0       20      LBA_T2   0.00000      0.87325 0.005284575
17    0               0       20  Sterile_T2   0.00125      0.82580 0.006902999
18    0               0       20  Sterile_T2   0.00205      0.82150 0.007938880
19    0               0       20      LBA_T3   0.00275      0.89955 0.008845335
20    0               0       20  Sterile_T3   0.00390      0.83225 0.010334628
21    0               0       20      LBA_T3   0.00225      0.88295 0.008197862
22    0               0       20  Sterile_T3   0.00205      0.82825 0.007938880
23    0               0       20  Sterile_T3   0.00305      0.83335 0.009233832
24    0               0       20      LBA_T4   0.00000      0.85115 0.005284575
25    0               0       20  Sterile_T4   0.00000      0.85870 0.005284575
26    0               0       20      LBA_T4   0.00000      0.87375 0.005284575
27    0               0       20      LBA_T4   0.00000      0.86850 0.005284575
28    0               0       20  Sterile_T4   0.00000      0.85550 0.005284575
29    0               0       20  Sterile_T4   0.00000      0.85995 0.005284575
30    0               0       20      LBA_T5   0.00000      0.85915 0.005284575
31    0               0       20  Sterile_T5   0.00000      0.87575 0.005284575
32    0               0       20      LBA_T5   0.00170      0.88555 0.007485673
33    0               0       20      LBA_T5   0.00245      0.88010 0.008456848
34    0               0       20  Sterile_T5   0.00195      0.86395 0.007809391
35    0               0       20  Sterile_T5   0.00235      0.87990 0.008327354
36    0               0       20      LBA_T6   0.00255      0.85780 0.008586343
37    0               0       20  Sterile_T6   0.00070      0.86350 0.006190871
38    0               0       20      LBA_T6   0.00340      0.86075 0.009687092
39    0               0       20      LBA_T6   0.00280      0.86860 0.008910084
40    0               0       20  Sterile_T6   0.00285      0.84890 0.008974833
41    0               0       20  Sterile_T6   0.00560      0.85535 0.012536461
42    0               0       20      LBA_T7   0.00465      0.85765 0.011305986
43    0               0       20  Sterile_T7   0.00360      0.84680 0.009946103
44    0               0       20      LBA_T7   0.00455      0.87800 0.011176468
45    0               0       20      LBA_T7   0.00460      0.86985 0.011241227
46    0               0       20  Sterile_T7   0.00435      0.85990 0.010917435
47    0               0       20  Sterile_T7   0.00310      0.85370 0.009298583
48    0               0       20      LBA_T8   0.00030      0.85195 0.005672981
49    0               0       20  Sterile_T8   0.00070      0.86020 0.006190871
50    0               0       20      LBA_T8   0.00045      0.89480 0.005867188
51    0               0       20      LBA_T8   0.00195      0.85455 0.007809391
52    0               0       20  Sterile_T8   0.00045      0.86125 0.005867188
53    0               0       20  Sterile_T8   0.00000      0.82755 0.005284575
54    0               0       20      LBA_T9   0.00140      0.86385 0.007097221
55    0               0       20  Sterile_T9   0.00135      0.86310 0.007032480
56    0               0       20      LBA_T9   0.00320      0.88960 0.009428085
57    0               0       20      LBA_T9   0.00045      0.86290 0.005867188
58    0               0       20  Sterile_T9   0.00195      0.83700 0.007809391
59    0               0       20  Sterile_T9   0.00055      0.84860 0.005996660
60    0               0       20     LBA_T10   0.00105      0.86225 0.006644039
61    0               0       20 Sterile_T10   0.00130      0.86770 0.006967739
62    0               0       20     LBA_T10   0.00075      0.87680 0.006255609
63    0               0       20     LBA_T10   0.00000      0.87560 0.005284575
64    0               0       20 Sterile_T10   0.00100      0.86425 0.006579300
65    0               0       20 Sterile_T10   0.00000      0.86765 0.005284575
   NO2NO3_mM   NO3_mM Time_minutes Time_hours Time_days H_mol  Volume
1   1.973223 1.965543            0        0.0       0.0     0 1.90366
2   1.971831 1.961625            0        0.0       0.0     0 1.76902
3   1.921928 1.913795            0        0.0       0.0     0 1.90366
4   1.952606 1.946933            0        0.0       0.0     0 1.90366
5   1.944267 1.936263            0        0.0       0.0     0 1.76902
6   1.952732 1.944664            0        0.0       0.0     0 1.76902
7   2.040987 2.033889          260        4.3       0.2     0 1.90366
8   1.929875 1.921224          180        3.0       0.1     0 1.76902
9   1.944646 1.938260          260        4.3       0.2     0 1.90366
10  2.012266 2.005428          260        4.3       0.2     0 1.90366
11  1.959432 1.950911          180        3.0       0.1     0 1.76902
12  1.926973 1.917739          180        3.0       0.1     0 1.76902
13  2.066072 2.057226          450        7.5       0.3     0 1.90366
14  1.988675 1.981513          370        6.2       0.3     0 1.76902
15  1.969173 1.963888          450        7.5       0.3     0 1.90366
16  2.035645 2.030360          450        7.5       0.3     0 1.90366
17  1.915498 1.908595          370        6.2       0.3     0 1.76902
18  1.904662 1.896723          370        6.2       0.3     0 1.76902
19  2.102701 2.093855          600       10.0       0.4     0 1.90366
20  1.931767 1.921433          520        8.7       0.4     0 1.76902
21  2.060337 2.052140          600       10.0       0.4     0 1.90366
22  1.921676 1.913737          520        8.7       0.4     0 1.76902
23  1.934544 1.925310          520        8.7       0.4     0 1.76902
24  1.979553 1.974269         1220       20.3       0.8     0 1.90366
25  1.998690 1.993405         1150       19.2       0.8     0 1.76902
26  2.036916 2.031632         1220       20.3       0.8     0 1.90366
27  2.023569 2.018285         1220       20.3       0.8     0 1.90366
28  1.990576 1.985291         1150       19.2       0.8     0 1.76902
29  2.001861 1.996576         1150       19.2       0.8     0 1.76902
30  1.999831 1.994546         1600       26.7       1.1     0 1.90366
31  2.042004 2.036720         1520       25.3       1.1     0 1.76902
32  2.066964 2.059478         1600       26.7       1.1     0 1.90366
33  2.053078 2.044621         1600       26.7       1.1     0 1.90366
34  2.012012 2.004203         1520       25.3       1.1     0 1.76902
35  2.052568 2.044241         1520       25.3       1.1     0 1.76902
36  1.996407 1.987821         1950       32.5       1.4     0 1.90366
37  2.010870 2.004679         1870       31.2       1.3     0 1.76902
38  2.003890 1.994203         1950       32.5       1.4     0 1.90366
39  2.023823 2.014913         1950       32.5       1.4     0 1.90366
40  1.973856 1.964881         1870       31.2       1.3     0 1.76902
41  1.990195 1.977659         1870       31.2       1.3     0 1.76902
42  1.996027 1.984721         2680       44.7       1.9     0 1.90366
43  1.968540 1.958594         2610       43.5       1.8     0 1.76902
44  2.047731 2.036554         2680       44.7       1.9     0 1.90366
45  2.027000 2.015759         2680       44.7       1.9     0 1.90366
46  2.001734 1.990816         2610       43.5       1.8     0 1.76902
47  1.986014 1.976715         2610       43.5       1.8     0 1.76902
48  1.981580 1.975907         3370       56.2       2.3     0 1.90366
49  2.002495 1.996304         3290       54.8       2.3     0 1.76902
50  2.090565 2.084698         3370       56.2       2.3     0 1.90366
51  1.988168 1.980358         3370       56.2       2.3     0 1.90366
52  2.005159 1.999292         3290       54.8       2.3     0 1.76902
53  1.919910 1.914626         3290       54.8       2.3     0 1.76902
54  2.011758 2.004661         4330       72.2       3.0     0 1.90366
55  2.009854 2.002822         4260       71.0       3.0     0 1.76902
56  2.077292 2.067864         4330       72.2       3.0     0 1.90366
57  2.009347 2.003479         4330       72.2       3.0     0 1.90366
58  1.943761 1.935952         4260       71.0       3.0     0 1.76902
59  1.973096 1.967100         4260       71.0       3.0     0 1.76902
60  2.007697 2.001053         5760       96.0       4.0     0 1.90366
61  2.021536 2.014569         5690       94.8       4.0     0 1.76902
62  2.044676 2.038421         5760       96.0       4.0     0 1.90366
63  2.041623 2.036338         5760       96.0       4.0     0 1.90366
64  2.012774 2.006195         5690       94.8       4.0     0 1.76902
65  2.021409 2.016125         5690       94.8       4.0     0 1.76902
   H_Molarity H_mM
1           0    0
2           0    0
3           0    0
4           0    0
5           0    0
6           0    0
7           0    0
8           0    0
9           0    0
10          0    0
11          0    0
12          0    0
13          0    0
14          0    0
15          0    0
16          0    0
17          0    0
18          0    0
19          0    0
20          0    0
21          0    0
22          0    0
23          0    0
24          0    0
25          0    0
26          0    0
27          0    0
28          0    0
29          0    0
30          0    0
31          0    0
32          0    0
33          0    0
34          0    0
35          0    0
36          0    0
37          0    0
38          0    0
39          0    0
40          0    0
41          0    0
42          0    0
43          0    0
44          0    0
45          0    0
46          0    0
47          0    0
48          0    0
49          0    0
50          0    0
51          0    0
52          0    0
53          0    0
54          0    0
55          0    0
56          0    0
57          0    0
58          0    0
59          0    0
60          0    0
61          0    0
62          0    0
63          0    0
64          0    0
65          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input
1   G04             2             0              0
2   G05             2             0              0
3   G10             2             0              0
4   G11             2             0              0
5   H04             2             0              0
6   H10             2             0              0
7   G04             2             0              0
8   G05             2             0              0
9   G10             2             0              0
10  G11             2             0              0
11  H04             2             0              0
12  H10             2             0              0
13  G04             2             0              0
14  G05             2             0              0
15  G10             2             0              0
16  G11             2             0              0
17  H04             2             0              0
18  H10             2             0              0
19  G04             2             0              0
20  G05             2             0              0
21  G10             2             0              0
22  G11             2             0              0
23  H04             2             0              0
24  H10             2             0              0
25  G04             2             0              0
26  G05             2             0              0
27  G10             2             0              0
28  G11             2             0              0
29  H04             2             0              0
30  H10             2             0              0
31  G04             2             0              0
32  G05             2             0              0
33  G10             2             0              0
34  G11             2             0              0
35  H04             2             0              0
36  H10             2             0              0
37  G04             2             0              0
38  G05             2             0              0
39  G10             2             0              0
40  G11             2             0              0
41  H04             2             0              0
42  H10             2             0              0
43  G04             2             0              0
44  G05             2             0              0
45  G10             2             0              0
46  G11             2             0              0
47  H04             2             0              0
48  H10             2             0              0
49  G04             2             0              0
50  G05             2             0              0
51  G10             2             0              0
52  G11             2             0              0
53  H04             2             0              0
54  H10             2             0              0
55  G04             2             0              0
56  G05             2             0              0
57  G10             2             0              0
58  G11             2             0              0
59  H04             2             0              0
60  H10             2             0              0
61  G04             2             0              0
62  G05             2             0              0
63  G10             2             0              0
64  G11             2             0              0
65  H04             2             0              0
66  H10             2             0              0
                            Soil   Sample_type Titration_type Antibiotics_type
1             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
2             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
3  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
4  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
5             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
6  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
7             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
8             LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
9  LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
10 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
11            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
12 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
13            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
14            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
15 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
16 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
17            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
18 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
19            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
20            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
21 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
22 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
23            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
24 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
25            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
26            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
27 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
28 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
29            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
30 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
31            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
32            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
33 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
34 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
35            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
36 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
37            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
38            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
39 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
40 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
41            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
42 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
43            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
44            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
45 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
46 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
47            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
48 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
49            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
50            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
51 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
52 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
53            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
54 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
55            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
56            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
57 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
58 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
59            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
60 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
61            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
62            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
63 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
64 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
65            LaBaghWoods_pH6.66 Nitrite_Blank        Nitrite             None
66 LaBaghWoods_autoclaved_pH6.66 Nitrite_Blank        Nitrite             None
   Unit Concentration_M Added_ul  Time_point NO2_OD540 NO2NO3_OD540   NO2_mM
1     0               0       20      LBA_T0   1.28420      0.73540 1.770814
2     0               0       20      LBA_T0   1.25625      0.72680 1.729915
3     0               0       20  Sterile_T0   1.27375      0.73220 1.755508
4     0               0       20  Sterile_T0   1.25275      0.72250 1.724803
5     0               0       20      LBA_T0   1.26850      0.71880 1.747825
6     0               0       20  Sterile_T0   1.28700      0.73115 1.774918
7     0               0       20      LBA_T1   1.27205      0.71720 1.753020
8     0               0       20      LBA_T1   1.28380      0.72505 1.770228
9     0               0       20  Sterile_T1   1.29310      0.73860 1.783864
10    0               0       20  Sterile_T1   1.22090      0.68830 1.678366
11    0               0       20      LBA_T1   1.28835      0.73185 1.776898
12    0               0       20  Sterile_T1   1.25910      0.71855 1.734080
13    0               0       20      LBA_T2   1.27020      0.71075 1.750313
14    0               0       20      LBA_T2   1.27355      0.71650 1.755216
15    0               0       20  Sterile_T2   1.25770      0.70470 1.732034
16    0               0       20  Sterile_T2   1.26390      0.70255 1.741097
17    0               0       20      LBA_T2   1.32125      0.73420 1.825223
18    0               0       20  Sterile_T2   1.22830      0.69255 1.689141
19    0               0       20      LBA_T3   1.33145      0.75875 1.840241
20    0               0       20      LBA_T3   1.36315      0.76305 1.887022
21    0               0       20  Sterile_T3   1.27765      0.72420 1.761219
22    0               0       20  Sterile_T3   1.26775      0.72285 1.746728
23    0               0       20      LBA_T3   1.33220      0.75045 1.841345
24    0               0       20  Sterile_T3   1.27535      0.72070 1.757851
25    0               0       20      LBA_T4   1.41820      0.77100 1.968657
26    0               0       20      LBA_T4   1.37330      0.75800 1.902036
27    0               0       20  Sterile_T4   1.32355      0.72545 1.828608
28    0               0       20  Sterile_T4   1.32680      0.72915 1.833392
29    0               0       20      LBA_T4   1.34280      0.68895 1.856972
30    0               0       20  Sterile_T4   1.39710      0.75735 1.937308
31    0               0       20      LBA_T5   1.33465      0.73815 1.844955
32    0               0       20      LBA_T5   1.35245      0.75000 1.871213
33    0               0       20  Sterile_T5   1.33120      0.73910 1.839872
34    0               0       20  Sterile_T5   1.32280      0.73555 1.827504
35    0               0       20      LBA_T5   1.31110      0.74030 1.810295
36    0               0       20  Sterile_T5   1.32955      0.73990 1.837442
37    0               0       20      LBA_T6   1.31585      0.72080 1.817279
38    0               0       20      LBA_T6   1.33000      0.72715 1.838105
39    0               0       20  Sterile_T6   1.33610      0.73075 1.847093
40    0               0       20  Sterile_T6   1.32390      0.72085 1.829123
41    0               0       20      LBA_T6   1.31795      0.71920 1.820368
42    0               0       20  Sterile_T6   1.35060      0.74060 1.868482
43    0               0       20      LBA_T7   1.32370      0.72175 1.828828
44    0               0       20      LBA_T7   1.36480      0.73930 1.889462
45    0               0       20  Sterile_T7   1.35725      0.73975 1.878303
46    0               0       20  Sterile_T7   1.34255      0.73035 1.856603
47    0               0       20      LBA_T7   1.37120      0.75070 1.898928
48    0               0       20  Sterile_T7   1.34785      0.73880 1.864422
49    0               0       20      LBA_T8   1.43060      0.77035 1.987116
50    0               0       20      LBA_T8   1.48695      0.80045 2.071324
51    0               0       20  Sterile_T8   1.39330      0.74855 1.931670
52    0               0       20  Sterile_T8   1.40160      0.75850 1.943988
53    0               0       20      LBA_T8   1.48665      0.78850 2.070875
54    0               0       20  Sterile_T8   1.42960      0.77200 1.985626
55    0               0       20      LBA_T9   1.40605      0.75530 1.950596
56    0               0       20      LBA_T9   1.40870      0.75905 1.954534
57    0               0       20  Sterile_T9   1.40380      0.74950 1.947255
58    0               0       20  Sterile_T9   1.40415      0.75505 1.947774
59    0               0       20      LBA_T9   1.41595      0.75740 1.965311
60    0               0       20  Sterile_T9   1.39510      0.75050 1.934340
61    0               0       20     LBA_T10   1.42685      0.76600 1.981531
62    0               0       20     LBA_T10   1.45055      0.77990 2.016867
63    0               0       20 Sterile_T10   1.42340      0.76660 1.976395
64    0               0       20 Sterile_T10   1.43185      0.76905 1.988978
65    0               0       20     LBA_T10   1.41760      0.76625 1.967765
66    0               0       20 Sterile_T10   1.42020      0.76380 1.971633
   NO2NO3_mM NO3_mM Time_minutes Time_hours Time_days H_mol  Volume H_Molarity
1   1.689482      0            0        0.0       0.0     0 1.90366          0
2   1.668172      0            0        0.0       0.0     0 1.90366          0
3   1.681549      0            0        0.0       0.0     0 1.76902          0
4   1.657530      0            0        0.0       0.0     0 1.76902          0
5   1.648379      0            0        0.0       0.0     0 1.90366          0
6   1.678947      0            0        0.0       0.0     0 1.76902          0
7   1.644423      0          260        4.3       0.2     0 1.90366          0
8   1.663840      0          260        4.3       0.2     0 1.90366          0
9   1.697420      0          180        3.0       0.1     0 1.76902          0
10  1.573173      0          180        3.0       0.1     0 1.76902          0
11  1.680682      0          260        4.3       0.2     0 1.90366          0
12  1.647761      0          180        3.0       0.1     0 1.76902          0
13  1.628490      0          450        7.5       0.3     0 1.90366          0
14  1.642693      0          450        7.5       0.3     0 1.90366          0
15  1.613561      0          370        6.2       0.3     0 1.76902          0
16  1.608260      0          370        6.2       0.3     0 1.76902          0
17  1.686507      0          450        7.5       0.3     0 1.90366          0
18  1.583628      0          370        6.2       0.3     0 1.76902          0
19  1.747507      0          600       10.0       0.4     0 1.90366          0
20  1.758219      0          600       10.0       0.4     0 1.90366          0
21  1.661736      0          520        8.7       0.4     0 1.76902          0
22  1.658396      0          520        8.7       0.4     0 1.76902          0
23  1.726853      0          600       10.0       0.4     0 1.90366          0
24  1.653077      0          520        8.7       0.4     0 1.76902          0
25  1.778045      0         1220       20.3       0.8     0 1.90366          0
26  1.745639      0         1220       20.3       0.8     0 1.90366          0
27  1.664830      0         1150       19.2       0.8     0 1.76902          0
28  1.673992      0         1150       19.2       0.8     0 1.76902          0
29  1.574771      0         1220       20.3       0.8     0 1.90366          0
30  1.744021      0         1150       19.2       0.8     0 1.76902          0
31  1.696303      0         1600       26.7       1.1     0 1.90366          0
32  1.725734      0         1600       26.7       1.1     0 1.90366          0
33  1.698660      0         1520       25.3       1.1     0 1.76902          0
34  1.689854      0         1520       25.3       1.1     0 1.76902          0
35  1.701638      0         1600       26.7       1.1     0 1.90366          0
36  1.700646      0         1520       25.3       1.1     0 1.76902          0
37  1.653325      0         1950       32.5       1.4     0 1.90366          0
38  1.669039      0         1950       32.5       1.4     0 1.90366          0
39  1.677956      0         1870       31.2       1.3     0 1.76902          0
40  1.653448      0         1870       31.2       1.3     0 1.76902          0
41  1.649368      0         1950       32.5       1.4     0 1.90366          0
42  1.702383      0         1870       31.2       1.3     0 1.76902          0
43  1.655674      0         2680       44.7       1.9     0 1.90366          0
44  1.699157      0         2680       44.7       1.9     0 1.90366          0
45  1.700273      0         2610       43.5       1.8     0 1.76902          0
46  1.676965      0         2610       43.5       1.8     0 1.76902          0
47  1.727475      0         2680       44.7       1.9     0 1.90366          0
48  1.697916      0         2610       43.5       1.8     0 1.76902          0
49  1.776423      0         3370       56.2       2.3     0 1.90366          0
50  1.851742      0         3370       56.2       2.3     0 1.90366          0
51  1.722130      0         3290       54.8       2.3     0 1.76902          0
52  1.746884      0         3290       54.8       2.3     0 1.76902          0
53  1.821790      0         3370       56.2       2.3     0 1.90366          0
54  1.780541      0         3290       54.8       2.3     0 1.76902          0
55  1.738918      0         4330       72.2       3.0     0 1.90366          0
56  1.748254      0         4330       72.2       3.0     0 1.90366          0
57  1.724491      0         4260       71.0       3.0     0 1.76902          0
58  1.738296      0         4260       71.0       3.0     0 1.76902          0
59  1.744145      0         4330       72.2       3.0     0 1.90366          0
60  1.726978      0         4260       71.0       3.0     0 1.76902          0
61  1.765572      0         5760       96.0       4.0     0 1.90366          0
62  1.800275      0         5760       96.0       4.0     0 1.90366          0
63  1.767069      0         5690       94.8       4.0     0 1.76902          0
64  1.773180      0         5690       94.8       4.0     0 1.76902          0
65  1.766196      0         5760       96.0       4.0     0 1.90366          0
66  1.760088      0         5690       94.8       4.0     0 1.76902          0
   H_mM
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
26    0
27    0
28    0
29    0
30    0
31    0
32    0
33    0
34    0
35    0
36    0
37    0
38    0
39    0
40    0
41    0
42    0
43    0
44    0
45    0
46    0
47    0
48    0
49    0
50    0
51    0
52    0
53    0
54    0
55    0
56    0
57    0
58    0
59    0
60    0
61    0
62    0
63    0
64    0
65    0
66    0</code></pre>
<pre class="r"><code># df_aero_blank$Nitrate_input &lt;- as.numeric(as.character(df_aero_blank$Nitrate_input))
df_no2_blank %&lt;&gt;% mutate(Correction_factor_NO2 = (Nitrite_input / NO2_mM)) 
df_no3_blank %&lt;&gt;% mutate(Correction_factor_NO3 = (Nitrate_input / NO3_mM)) 

# average the blanks by time point
cf_no2 &lt;- df_no2_blank %&gt;% select(Soil, Time_point, Correction_factor_NO2) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO2 = mean(Correction_factor_NO2)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>cf_no3 &lt;- df_no3_blank %&gt;% select(Soil, Time_point, Correction_factor_NO3) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO3 = mean(Correction_factor_NO3)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code># left join and multiply the correction factor
# left join to samples
df_sample_mcf &lt;- df_p_mcf %&gt;% filter(!(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;Ammonium_Blank&quot;)))
dim(df_sample_mcf) #1065</code></pre>
<pre><code>[1] 849  24</code></pre>
<pre class="r"><code>df_sample_mcf$Sample_type %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Slurry&quot;     &quot;No_Nitrate&quot;</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_mcf %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_sample_bcf)</code></pre>
<pre><code>[1] 849  26</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_bcf %&gt;% mutate(NO3_mM = NO3_mM * Correction_factor_NO3, NO2_mM = NO2_mM * Correction_factor_NO2)
dim(df_sample_bcf) #849</code></pre>
<pre><code>[1] 849  26</code></pre>
<pre class="r"><code># merge dataframe with blank just for convenience
df_p_blanks_bcf &lt;- df_p_mcf %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;Ammonium_Blank&quot;)) %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p_blanks_bcf) #197 26</code></pre>
<pre><code>[1] 197  26</code></pre>
<pre class="r"><code>dim(df_sample_bcf) # 849 26</code></pre>
<pre><code>[1] 849  26</code></pre>
<pre class="r"><code>df_p_bcf &lt;- rbind(df_sample_bcf, df_p_blanks_bcf)
dim(df_p_bcf) # 1046 26</code></pre>
<pre><code>[1] 1046   26</code></pre>
<pre class="r"><code>## innate nitrate
df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot; &amp; Time_point == &quot;LBA_T0&quot;) %&gt;% select(NO3_mM) %&gt;% unlist() %&gt;% mean()</code></pre>
<pre><code>[1] 0.3212707</code></pre>
<pre class="r"><code>df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot; &amp; Time_point == &quot;Sterile_T0&quot;) %&gt;% select(NO3_mM) %&gt;% unlist() %&gt;% mean()</code></pre>
<pre><code>[1] 0.2333593</code></pre>
<pre class="r"><code># export
df_p_bcf$Time_point</code></pre>
<pre><code>   [1] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
   [7] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
  [13] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
  [19] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
  [25] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
  [31] LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0      LBA_T0     
  [37] LBA_T0      LBA_T0      LBA_T0      LBA_T1      LBA_T1      LBA_T1     
  [43] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1     
  [49] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1     
  [55] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1     
  [61] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1     
  [67] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T1     
  [73] LBA_T1      LBA_T1      LBA_T1      LBA_T1      LBA_T2      LBA_T2     
  [79] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
  [85] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
  [91] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
  [97] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
 [103] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
 [109] LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2      LBA_T2     
 [115] LBA_T2      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [121] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [127] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [133] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [139] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [145] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T3     
 [151] LBA_T3      LBA_T3      LBA_T3      LBA_T3      LBA_T4      LBA_T4     
 [157] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [163] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [169] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [175] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [181] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [187] LBA_T4      LBA_T4      LBA_T4      LBA_T4      LBA_T5      LBA_T5     
 [193] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [199] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [205] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [211] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [217] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [223] LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [229] LBA_T5      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [235] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [241] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [247] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [253] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [259] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [265] LBA_T6      LBA_T6      LBA_T6      LBA_T6      LBA_T7      LBA_T7     
 [271] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [277] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [283] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [289] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [295] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [301] LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [307] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [313] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [319] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [325] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [331] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [337] LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [343] LBA_T8      LBA_T8      LBA_T8      LBA_T9      LBA_T9      LBA_T9     
 [349] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [355] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [361] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [367] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [373] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [379] LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9      LBA_T9     
 [385] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [391] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [397] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [403] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [409] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [415] LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10     LBA_T10    
 [421] LBA_T10     LBA_T10     LBA_T10     Sterile_T0  Sterile_T0  Sterile_T0 
 [427] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [433] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [439] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [445] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [451] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [457] Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0 
 [463] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [469] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [475] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [481] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [487] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [493] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1 
 [499] Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T2  Sterile_T2  Sterile_T2 
 [505] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [511] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [517] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [523] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [529] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [535] Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2 
 [541] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [547] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [553] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [559] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [565] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [571] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3 
 [577] Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T4  Sterile_T4  Sterile_T4 
 [583] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [589] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [595] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [601] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [607] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [613] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4 
 [619] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [625] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [631] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [637] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [643] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [649] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5 
 [655] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T6  Sterile_T6  Sterile_T6 
 [661] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [667] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [673] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [679] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [685] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [691] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6 
 [697] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [703] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [709] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [715] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [721] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [727] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7 
 [733] Sterile_T7  Sterile_T7  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [739] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [745] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [751] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [757] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [763] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8 
 [769] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T9  Sterile_T9 
 [775] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [781] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [787] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [793] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [799] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [805] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9 
 [811] Sterile_T9  Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [817] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [823] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [829] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [835] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [841] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10
 [847] Sterile_T10 Sterile_T10 Sterile_T10 LBA_T0      LBA_T0      LBA_T0     
 [853] LBA_T0      Sterile_T0  Sterile_T0  Sterile_T0  Sterile_T0  LBA_T0     
 [859] LBA_T0      LBA_T0      LBA_T0      LBA_T0      Sterile_T0  Sterile_T0 
 [865] Sterile_T0  Sterile_T0  Sterile_T0  LBA_T1      LBA_T1      LBA_T1     
 [871] LBA_T1      Sterile_T1  Sterile_T1  Sterile_T1  Sterile_T1  LBA_T1     
 [877] LBA_T1      LBA_T1      LBA_T1      LBA_T1      Sterile_T1  Sterile_T1 
 [883] Sterile_T1  Sterile_T1  Sterile_T1  LBA_T2      LBA_T2      LBA_T2     
 [889] LBA_T2      Sterile_T2  Sterile_T2  Sterile_T2  Sterile_T2  LBA_T2     
 [895] LBA_T2      LBA_T2      LBA_T2      LBA_T2      Sterile_T2  Sterile_T2 
 [901] Sterile_T2  Sterile_T2  Sterile_T2  LBA_T3      LBA_T3      LBA_T3     
 [907] LBA_T3      Sterile_T3  Sterile_T3  Sterile_T3  Sterile_T3  LBA_T3     
 [913] LBA_T3      LBA_T3      LBA_T3      Sterile_T3  Sterile_T3  Sterile_T3 
 [919] Sterile_T3  Sterile_T3  LBA_T4      LBA_T4      LBA_T4      LBA_T4     
 [925] Sterile_T4  Sterile_T4  Sterile_T4  Sterile_T4  LBA_T4      LBA_T4     
 [931] LBA_T4      LBA_T4      LBA_T4      Sterile_T4  Sterile_T4  Sterile_T4 
 [937] Sterile_T4  Sterile_T4  LBA_T5      LBA_T5      LBA_T5      LBA_T5     
 [943] Sterile_T5  Sterile_T5  Sterile_T5  Sterile_T5  LBA_T5      LBA_T5     
 [949] LBA_T5      LBA_T5      LBA_T5      Sterile_T5  Sterile_T5  Sterile_T5 
 [955] Sterile_T5  Sterile_T5  LBA_T6      LBA_T6      LBA_T6      LBA_T6     
 [961] Sterile_T6  Sterile_T6  Sterile_T6  Sterile_T6  LBA_T6      LBA_T6     
 [967] LBA_T6      LBA_T6      LBA_T6      Sterile_T6  Sterile_T6  Sterile_T6 
 [973] Sterile_T6  Sterile_T6  LBA_T7      LBA_T7      LBA_T7      LBA_T7     
 [979] Sterile_T7  Sterile_T7  Sterile_T7  Sterile_T7  LBA_T7      LBA_T7     
 [985] LBA_T7      LBA_T7      LBA_T7      Sterile_T7  Sterile_T7  Sterile_T7 
 [991] Sterile_T7  Sterile_T7  LBA_T8      LBA_T8      LBA_T8      LBA_T8     
 [997] Sterile_T8  Sterile_T8  Sterile_T8  Sterile_T8  LBA_T8      LBA_T8     
[1003] LBA_T8      LBA_T8      LBA_T8      Sterile_T8  Sterile_T8  Sterile_T8 
[1009] Sterile_T8  Sterile_T8  LBA_T9      LBA_T9      LBA_T9      LBA_T9     
[1015] Sterile_T9  Sterile_T9  Sterile_T9  Sterile_T9  LBA_T9      LBA_T9     
[1021] LBA_T9      LBA_T9      LBA_T9      Sterile_T9  Sterile_T9  Sterile_T9 
[1027] Sterile_T9  Sterile_T9  LBA_T10     LBA_T10     LBA_T10     LBA_T10    
[1033] Sterile_T10 Sterile_T10 Sterile_T10 Sterile_T10 LBA_T10     LBA_T10    
[1039] LBA_T10     LBA_T10     LBA_T10     Sterile_T10 Sterile_T10 Sterile_T10
[1045] Sterile_T10 Sterile_T10
22 Levels: LBA_T0 LBA_T1 LBA_T2 LBA_T3 LBA_T4 LBA_T5 LBA_T6 LBA_T7 ... Sterile_T10</code></pre>
<pre class="r"><code># removing timepoint T8
df_kyle &lt;- df_p_bcf %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;) %&gt;%  filter(!(Time_point %in% c(&quot;LBA_T8&quot;, &quot;Sterile_T8&quot;)))
df_kyle$Time_point %&gt;% unique()</code></pre>
<pre><code> [1] LBA_T0  LBA_T1  LBA_T2  LBA_T3  LBA_T4  LBA_T5  LBA_T6  LBA_T7  LBA_T9 
[10] LBA_T10
22 Levels: LBA_T0 LBA_T1 LBA_T2 LBA_T3 LBA_T4 LBA_T5 LBA_T6 LBA_T7 ... Sterile_T10</code></pre>
<pre class="r"><code>dim(df_kyle)</code></pre>
<pre><code>[1] 473  26</code></pre>
<pre class="r"><code># write.xlsx(df_kyle, &quot;220421_Griess_antibiotics_experiment_to_kyle.xlsx&quot;)

df_kyle2 &lt;- df_kyle %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;, &quot;HCl&quot;)) 
dim(df_kyle2)</code></pre>
<pre><code>[1] 354  26</code></pre>
<pre class="r"><code>write.xlsx(df_kyle2, &quot;220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx&quot;)</code></pre>
</div>
<div id="sanity-check-moisture-correction-blank-correction-factor" class="section level2">
<h2>3. Sanity check: moisture correction &amp; blank correction factor</h2>
<p>Blank correction<br />
Use the nitrate concentration of the blank and use the ratio.<br />
For example, multiply to nitrate concentration x (2mM / changed [NO3])<br />
</p>
<pre class="r"><code># without any correction
# nitrate blanks
dim(df_no3_blank)</code></pre>
<pre><code>[1] 65 25</code></pre>
<pre class="r"><code>ggplot(df_no3_blank, aes(x=Time_hours, y=NO3_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.5), limits=c(0, 2.5))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrate blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite blanks
dim(df_no2_blank)</code></pre>
<pre><code>[1] 66 25</code></pre>
<pre class="r"><code>ggplot(df_no2_blank, aes(x=Time_hours, y=NO2_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.7,0.5), limits=c(0, 2.7))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrite blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># without any correction
ggplot(df_p, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(Without any correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### What are those points with very high std?
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;       &quot;Time_days&quot;       
[21] &quot;H_mol&quot;            &quot;Volume&quot;           &quot;H_Molarity&quot;       &quot;H_mM&quot;            </code></pre>
<pre class="r"><code># df_p %&gt;% arrange(desc(Std_NO3_mM)) %&gt;% select(Soil, Titration_type, Concentration_M, Sample_type, Time_point, Std_NO3_mM) 
## based on this I corrected the wrongly removed E02 -&lt; removed E01 (filter burst). Eliminated point P11 G11.

# After moisture correction
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># After blank correction (evaporation) 
ggplot(df_p_bcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture + blank correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="what-is-going-on-in-nitrate-spike-plate9" class="section level2">
<h2>3.2. What is going on in nitrate spike plate9?</h2>
<pre class="r"><code>df_T8 &lt;- df_p_mcf %&gt;% filter(Time_point %in% c(&quot;LBA_T8&quot;, &quot;Sterile_T8&quot;))

ggplot(df_T8, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(df_T8)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input               Soil
1  A01             0             2              0 LaBaghWoods_pH6.66
2  A02             0             2              0 LaBaghWoods_pH6.66
3  A03             0             2              0 LaBaghWoods_pH6.66
4  A04             0             2              0 LaBaghWoods_pH6.66
5  A05             0             2              0 LaBaghWoods_pH6.66
6  A06             0             2              0 LaBaghWoods_pH6.66
  Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1      Slurry            HCl              CHL    2               1       20
2      Slurry            HCl              CHL    6               3       20
3      Slurry           NaOH              CHL    2               1      -20
4      Slurry           NaOH              CHL    6               3      -20
5      Slurry            HCl              CHL    0               0       20
6      Slurry            HCl              CHL    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540     NO2_mM NO2NO3_mM   NO3_mM Time_minutes
1     LBA_T8   0.05570      0.58555 0.08686309  1.322741 1.394343         3370
2     LBA_T8   0.03470      0.54075 0.05629956  1.214921 1.304169         3370
3     LBA_T8   0.05055      0.53110 0.07936263  1.191802 1.255218         3370
4     LBA_T8   0.04435      0.60990 0.07033736  1.381689 1.476878         3370
5     LBA_T8   0.04675      0.63915 0.07383044  1.452825 1.553043         3370
6     LBA_T8   0.04965      0.58815 0.07805221  1.329024 1.410189         3370
  Time_hours Time_days  H_mol  Volume  H_Molarity      H_mM
1       56.2       2.3  2e-05 1.90366  0.01050608  10.50608
2       56.2       2.3  6e-05 1.90366  0.03151823  31.51823
3       56.2       2.3 -2e-05 1.90366 -0.01050608 -10.50608
4       56.2       2.3 -6e-05 1.90366 -0.03151823 -31.51823
5       56.2       2.3  0e+00 1.90366  0.00000000   0.00000
6       56.2       2.3  0e+00 1.90366  0.00000000   0.00000</code></pre>
<pre class="r"><code>ggplot(df_T8, aes(x=NO2_OD540, y=NO2_mM, color=Soil, group=Soil)) +
  geom_point(size=4, shape=16, alpha=0.6) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2- (mM) \n&quot;) +
  xlab(&quot;\n NO2_OD540&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) OD540 and estimated nitrite concentration \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_T8, aes(x=NO2NO3_OD540, y=NO2NO3_mM, color=Soil, group=Soil)) +
  geom_point(size=4, shape=16, alpha=0.6) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2NO3_mM (mM) \n&quot;) +
  xlab(&quot;\n NO2NO3_OD540&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) OD540 and estimated nitrite concentration \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_blank3 &lt;- df_T8 %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;, &quot;Nitrate_Blank&quot;, &quot;Ammonium_Blank&quot;))

ggplot(df_blank3, aes(x=NO3_mM, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=5, shape=16) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2- (mM) \n&quot;) +
  xlab(&quot;\n Measured NO3- (mM)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) Blank samples \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colnames(df_T8)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;         &quot;Time_point&quot;      
[13] &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;           &quot;NO2NO3_mM&quot;       
[17] &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;       &quot;Time_days&quot;       
[21] &quot;H_mol&quot;            &quot;Volume&quot;           &quot;H_Molarity&quot;       &quot;H_mM&quot;            </code></pre>
<p>I don’t know. The blanks look okay.</p>
</div>
<div id="nitrogen-dynamics" class="section level2">
<h2>4. Nitrogen dynamics</h2>
</div>
<div id="ph-perturbation-experiment." class="section level2">
<h2>4.1. pH perturbation experiment.</h2>
<ul>
<li>Is there a difference in pH perturbation levels?</li>
</ul>
<pre class="r"><code># pH color
col_pH &lt;- colorRampPalette(c(&quot;gold&quot;,&quot;red&quot;,&quot;purple&quot;))

library(colorRamps)
colorRamps::green2red</code></pre>
<pre><code>function (n) 
rgb.tables(n, red = c(1, 0, 2), green = c(0, 0, 2), blue = c(0, 
    0, 0, 0))
&lt;bytecode: 0x000000002e0d8ce8&gt;
&lt;environment: namespace:colorRamps&gt;</code></pre>
<pre class="r"><code>plot(rep(1,13),col=col_pH(13),pch=19,cex=3)</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>grad_pH &lt;- scale_colour_gradientn(colours = col_pH(100))

# Confer this page (https://stackoverflow.com/questions/21537782/how-to-set-fixed-continuous-colour-values-in-ggplot2)
# myPalette &lt;- colorRampPalette(rev(brewer.pal(11, &quot;Spectral&quot;)))
# sc &lt;- scale_colour_gradientn(colours = myPalette(100), limits=c(1, 8))

# 1. pH perturbation
# plot Nitrate

# only autoclaved soil is relevant for this analysis
df_p_bcf$Soil &lt;-  factor(df_p_bcf$Soil, levels = c(&quot;LaBaghWoods_pH6.66&quot;,&quot;LaBaghWoods_autoclaved_pH6.66&quot;))

df_pH &lt;- df_p_bcf %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;)) 
dim(df_pH) # 393</code></pre>
<pre><code>[1] 783  26</code></pre>
<pre class="r"><code># average the no acid base one.
df_plot &lt;- df_pH %&gt;% select(H_mM, Titration_type, Unit, NO3_mM, NO2_mM, Soil, Time_hours, Time_point, Antibiotics_type)
df_plot %&gt;% filter(H_mM == 0) %&gt;% dim()</code></pre>
<pre><code>[1] 165   9</code></pre>
<pre class="r"><code>df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 55 x 9
    H_mM Titration_type  Unit Soil        Time_hours Time_point Antibiotics_type
   &lt;dbl&gt; &lt;chr&gt;          &lt;dbl&gt; &lt;fct&gt;            &lt;dbl&gt; &lt;fct&gt;      &lt;chr&gt;           
 1     0 HCl                0 LaBaghWood~        0   LBA_T0     CHL             
 2     0 HCl                0 LaBaghWood~        0   LBA_T0     CHL_NYT         
 3     0 HCl                0 LaBaghWood~        0   LBA_T0     None            
 4     0 HCl                0 LaBaghWood~        0   LBA_T0     NYT             
 5     0 HCl                0 LaBaghWood~        4.3 LBA_T1     CHL             
 6     0 HCl                0 LaBaghWood~        4.3 LBA_T1     CHL_NYT         
 7     0 HCl                0 LaBaghWood~        4.3 LBA_T1     None            
 8     0 HCl                0 LaBaghWood~        4.3 LBA_T1     NYT             
 9     0 HCl                0 LaBaghWood~        7.5 LBA_T2     CHL             
10     0 HCl                0 LaBaghWood~        7.5 LBA_T2     CHL_NYT         
# ... with 45 more rows, and 2 more variables: NO3_mM &lt;dbl&gt;, NO2_mM &lt;dbl&gt;</code></pre>
<pre class="r"><code>df_sub &lt;- df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>df_main &lt;- df_plot %&gt;% filter(H_mM != 0)
dim(df_plot)</code></pre>
<pre><code>[1] 783   9</code></pre>
<pre class="r"><code>dim(df_main)</code></pre>
<pre><code>[1] 618   9</code></pre>
<pre class="r"><code>df_plot2 &lt;- rbind(df_main, df_sub)
dim(df_plot2) # 1001</code></pre>
<pre><code>[1] 673   9</code></pre>
<pre class="r"><code># remove antibiotics
df_plot3 &lt;- df_plot2 %&gt;% filter(Antibiotics_type == &quot;None&quot;) %&gt;% filter(!(Time_point %in% c(&quot;LBA_T8&quot;,&quot;Sterile_T8&quot;)))

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot Nitrite
ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;pH perturbation \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot together
# include pH data
df_220422_pH &lt;- read.xlsx(&quot;data/220422_pH_Antibiotics_Sterile_experiment.xlsx&quot;) %&gt;% select(-H_mM)

df_plot3 &lt;- df_plot3 %&gt;% left_join(df_220422_pH, by=c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Titration_type&quot;=&quot;Titration_type&quot;,&quot;Unit&quot;=&quot;Unit&quot;, &quot;Antibiotics_type&quot;=&quot;Antibiotics_type&quot;))
df_plot3$Soil &lt;-  factor(df_plot3$Soil, levels = c(&quot;LaBaghWoods_pH6.66&quot;,&quot;LaBaghWoods_autoclaved_pH6.66&quot;))


p_A &lt;- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))
p_A</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot Nitrite
p_I &lt;- ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))

p_I</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>require(gridExtra)
library(grid)
library(gtable)

legend = gtable_filter(ggplotGrob(p_A), &quot;guide-box&quot;)

grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position=&quot;none&quot;), 
                         p_I+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), nrow=2,
             top = textGrob(&quot;pH perturbation experiment with autoclaved soil&quot;, vjust = 0.5, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             # left = textGrob(&quot;Relative abundance of strains with opt_pH&quot;, vjust = 0.5, rot=90, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             bottom = textGrob(&quot;Time (hr)&quot;, vjust = 0, gp = gpar(fontface=&quot;bold&quot;, cex = 1.5)),
             
             right = legend
             ))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-6-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot nitrate, nitrite, ammonium at the same time</p>
<pre class="r"><code>df_ammo_plot3 &lt;- read.xlsx(&quot;data/220421_Ammonia_sterile.xlsx&quot;)

df_ammo_plot3$Soil &lt;- factor(df_ammo_plot3$Soil, levels=c(&quot;LaBaghWoods_pH6.66&quot;, &quot;LaBaghWoods_autoclaved_pH6.66&quot;))

p_NH4 &lt;- ggplot(df_ammo_plot3, aes(x=Time_hours, y=NH4_mM, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NH4_mM - Std_NH4_mM, ymax=NH4_mM + Std_NH4_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NH4+ (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,4,0.5), limits=c(0, 4))+
  # ggtitle(&quot;pH perturbation \n&quot;) +
  # label
  # geom_text(aes(label = round(NH4_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))

legend = gtable_filter(ggplotGrob(p_A), &quot;guide-box&quot;)

grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position=&quot;none&quot;), 
                         p_I+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()),
                         p_NH4+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank())
                         , nrow=3,
             top = textGrob(&quot;pH perturbation experiment with autoclaved soil&quot;, vjust = 0.5, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             # left = textGrob(&quot;Relative abundance of strains with opt_pH&quot;, vjust = 0.5, rot=90, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             bottom = textGrob(&quot;Time (hr)&quot;, vjust = 0, gp = gpar(fontface=&quot;bold&quot;, cex = 1.5)),
             
             right = legend
             ))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Which are doing nitrate/nitrite reduction?</p>
<pre class="r"><code>colnames(df_plot3)</code></pre>
<pre><code> [1] &quot;H_mM&quot;             &quot;Titration_type&quot;   &quot;Unit&quot;             &quot;NO3_mM&quot;          
 [5] &quot;NO2_mM&quot;           &quot;Soil&quot;             &quot;Time_hours&quot;       &quot;Time_point&quot;      
 [9] &quot;Antibiotics_type&quot; &quot;pH&quot;              </code></pre>
<pre class="r"><code>df_plot3$Soil</code></pre>
<pre><code>  [1] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
  [3] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
  [5] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
  [7] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
  [9] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [11] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [13] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [15] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [17] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [19] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [21] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [23] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [25] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [27] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [29] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [31] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [33] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [35] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [37] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [39] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [41] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [43] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [45] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [47] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [49] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [51] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [53] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [55] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [57] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [59] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [61] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [63] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [65] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [67] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [69] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [71] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [73] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [75] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [77] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [79] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
 [81] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [83] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [85] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [87] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [89] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [91] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [93] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [95] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [97] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
 [99] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[101] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[103] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[105] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[107] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[109] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[111] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[113] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[115] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[117] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[119] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[121] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[123] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[125] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[127] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[129] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[131] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[133] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[135] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[137] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[139] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[141] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[143] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[145] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[147] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[149] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[151] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[153] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[155] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[157] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[159] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[161] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[163] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[165] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[167] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[169] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[171] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[173] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[175] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[177] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[179] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[181] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[183] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[185] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[187] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[189] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[191] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[193] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[195] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[197] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[199] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[201] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[203] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[205] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[207] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[209] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[211] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[213] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[215] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[217] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[219] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[221] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[223] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[225] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[227] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[229] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[231] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[233] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[235] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[237] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[239] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[241] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[243] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[245] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[247] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[249] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[251] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[253] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[255] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[257] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[259] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[261] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[263] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[265] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[267] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[269] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[271] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[273] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[275] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[277] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[279] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[281] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[283] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[285] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[287] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[289] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[291] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[293] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[295] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[297] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[299] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[301] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[303] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[305] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[307] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[309] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[311] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[313] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[315] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[317] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[319] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[321] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[323] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[325] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[327] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[329] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[331] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[333] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[335] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[337] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[339] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[341] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[343] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[345] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[347] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[349] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[351] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[353] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[355] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[357] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[359] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[361] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[363] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[365] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[367] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[369] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[371] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[373] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[375] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[377] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[379] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[381] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[383] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[385] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[387] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[389] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[391] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[393] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[395] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[397] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[399] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[401] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[403] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[405] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[407] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[409] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
[411] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
[413] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
[415] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
[417] LaBaghWoods_pH6.66            LaBaghWoods_pH6.66           
[419] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[421] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[423] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[425] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
[427] LaBaghWoods_autoclaved_pH6.66 LaBaghWoods_autoclaved_pH6.66
Levels: LaBaghWoods_pH6.66 LaBaghWoods_autoclaved_pH6.66</code></pre>
<pre class="r"><code>df_au &lt;- df_plot3 %&gt;% filter(Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;)
df_au_end &lt;- df_au %&gt;% filter(Time_point == &quot;Sterile_T10&quot;)

df_reduced &lt;- df_au_end %&gt;% filter(NO3_mM == 0) %&gt;% mutate(Nitrate_reduced = T)
df_not &lt;- df_au_end %&gt;% filter(NO3_mM != 0) %&gt;% mutate(Nitrate_reduced =F)

df_reduced &lt;- rbind(df_reduced, df_not)

# plot
ggplot(df_reduced, aes(x=Nitrate_reduced, y=pH, color=pH, group=pH)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  ggtitle(&quot;Counting perturbed samples that reduced all nitrate in autoclaved soil&quot;) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;H+ (mM) \n&quot;) +
  xlab(&quot;\n Nitrate completely reduced?&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># write.xlsx(df_reduced, &quot;220421_nitrate_reduced_autoclaved_samples.xlsx&quot;)</code></pre>
<p>No nitrate sample?</p>
<pre class="r"><code>df_nn &lt;- df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot;) %&gt;% filter(!(Time_point %in% c(&quot;LBA_T8&quot;, &quot;Sterile_T8&quot;)))
colnames(df_nn)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot;</code></pre>
<pre class="r"><code>dim(df_nn)</code></pre>
<pre><code>[1] 60 26</code></pre>
<pre class="r"><code>df_nn_ave &lt;- df_nn %&gt;% group_by(Soil, Time_hours, Time_point) %&gt;% summarize(Std_NO3_mM = sd(NO3_mM), Std_NO2_mM = sd(NO2_mM), Ave_NO3_mM = mean(NO3_mM), Ave_NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;, &#39;Time_hours&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO3_mM, color=Soil)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=Ave_NO3_mM - Std_NO3_mM, ymax=Ave_NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+
  ggtitle(&quot;Samples provided with no Nitrate \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot Nitrite
ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO2_mM, color=Soil)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=Ave_NO2_mM - Std_NO2_mM, ymax=Ave_NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+
  ggtitle(&quot;Samples provided with no Nitrate \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p_A &lt;- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO3_mM, color=Soil)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=Ave_NO3_mM - Std_NO3_mM, ymax=Ave_NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+
  # ggtitle(&quot;Samples provided with no Nitrate \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))

# plot Nitrite
p_I &lt;- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO2_mM, color=Soil)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=Ave_NO2_mM - Std_NO2_mM, ymax=Ave_NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+
  # ggtitle(&quot;Samples provided with no Nitrate \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))

grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position=&quot;none&quot;), 
                         p_I+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), nrow=2,
             top = textGrob(&quot;Samples provided with no Nitrate&quot;, vjust = 0.5, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             # left = textGrob(&quot;Relative abundance of strains with opt_pH&quot;, vjust = 0.5, rot=90, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             bottom = textGrob(&quot;Time (hr)&quot;, vjust = 0, gp = gpar(fontface=&quot;bold&quot;, cex = 1.5))
             ))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot ammonia as well

df_a_bcf &lt;- read.xlsx(&quot;data/220421_Ammonia_df_a_bcf.xlsx&quot;)

df_a_bcf$Soil &lt;- factor(df_a_bcf$Soil, levels=c(&quot;LaBaghWoods_pH6.66&quot;, &quot;LaBaghWoods_autoclaved_pH6.66&quot;))
df_a_bcf_nn &lt;- df_a_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot;) %&gt;% filter(!(Time_point %in% c(&quot;LBA_T8&quot;, &quot;Sterile_T8&quot;)))
df_a_bcf_ave &lt;- df_a_bcf_nn %&gt;% group_by(Soil, Time_hours) %&gt;% summarize(Std_NH4_mM = sd(NH4_mM), Ave_NH4_mM = mean(NH4_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p_NH4 &lt;- ggplot(df_a_bcf_ave, aes(x=Time_hours, y=Ave_NH4_mM, color=Soil)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=Ave_NH4_mM - Std_NH4_mM, ymax=Ave_NH4_mM + Std_NH4_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NH4- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.5), limits=c(0, 2.5))+
  # ggtitle(&quot;Samples provided with no Nitrate \n&quot;) +
  # label
  # geom_text(aes(label = round(NH4_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))

p_NH4</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># legend = gtable_filter(ggplotGrob(p_A), &quot;guide-box&quot;)

grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position=&quot;none&quot;), 
                         p_I+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()),
                         p_NH4+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank())
                         , nrow=3,
             top = textGrob(&quot;Samples provided with no Nitrate.&quot;, vjust = 0.5, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             # left = textGrob(&quot;Relative abundance of strains with opt_pH&quot;, vjust = 0.5, rot=90, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
             bottom = textGrob(&quot;Time (hr)&quot;, vjust = 0, gp = gpar(fontface=&quot;bold&quot;, cex = 1.5))
             ))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="lets-see-if-fungi-or-bacteria-matter" class="section level2">
<h2>4.2. Let’s see if fungi or bacteria matter</h2>
<pre class="r"><code># only antibiotics treated samples
df_LBA &lt;- df_p_bcf %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;) %&gt;% filter(!(Time_point == &quot;LBA_T8&quot;))
dim(df_LBA) # 521 -&gt; 473 (removed LBA_T8)</code></pre>
<pre><code>[1] 473  26</code></pre>
<pre class="r"><code>df_effect &lt;- df_LBA %&gt;% filter(H_mM == 0) %&gt;% filter(!(Titration_type %in% c(&quot;Nitrate&quot;, &quot;Nitrite&quot;, &quot;Ammonium&quot;, &quot;No_Nitrate&quot;)))
dim(df_effect) # 263 -&gt; 165 -&gt; 132 -&gt; 120 (removed T8)</code></pre>
<pre><code>[1] 120  26</code></pre>
<pre class="r"><code>df_effect$Titration_type</code></pre>
<pre><code>  [1] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [13] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [25] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [37] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [49] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [61] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [73] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [85] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
 [97] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;
[109] &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot; &quot;HCl&quot;</code></pre>
<pre class="r"><code>df_effect$Antibiotics_type</code></pre>
<pre><code>  [1] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot;
  [8] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;    
 [15] &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;   
 [22] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;    
 [29] &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
 [36] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [43] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;    
 [50] &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot;
 [57] &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
 [64] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot;
 [71] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [78] &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   
 [85] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot;
 [92] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;    
 [99] &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;   
[106] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;    
[113] &quot;NYT&quot;     &quot;NYT&quot;     &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
[120] &quot;None&quot;   </code></pre>
<pre class="r"><code>df_effect %&gt;% filter(Time_point == &quot;LBA_T0&quot;)</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input               Soil
1   A05             0             2              0 LaBaghWoods_pH6.66
2   A06             0             2              0 LaBaghWoods_pH6.66
3   B05             0             2              0 LaBaghWoods_pH6.66
4   C05             0             2              0 LaBaghWoods_pH6.66
5   D04             0             2              0 LaBaghWoods_pH6.66
6   D05             0             2              0 LaBaghWoods_pH6.66
7   E04             0             2              0 LaBaghWoods_pH6.66
8   E05             0             2              0 LaBaghWoods_pH6.66
9   E06             0             2              0 LaBaghWoods_pH6.66
10  F04             0             2              0 LaBaghWoods_pH6.66
11  F05             0             2              0 LaBaghWoods_pH6.66
12  F06             0             2              0 LaBaghWoods_pH6.66
   Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1       Slurry            HCl              CHL    0               0       20
2       Slurry            HCl              CHL    0               0       20
3       Slurry            HCl              CHL    0               0       20
4       Slurry            HCl              NYT    0               0       20
5       Slurry            HCl              NYT    0               0       20
6       Slurry            HCl              NYT    0               0       20
7       Slurry            HCl          CHL_NYT    0               0       20
8       Slurry            HCl          CHL_NYT    0               0       20
9       Slurry            HCl             None    0               0       20
10      Slurry            HCl          CHL_NYT    0               0       20
11      Slurry            HCl             None    0               0       20
12      Slurry            HCl             None    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM   NO3_mM
1      LBA_T0   0.00260      0.84035 0.011075478  1.952227 2.241586
2      LBA_T0   0.00100      0.77845 0.008423088  1.796651 2.064546
3      LBA_T0   0.00345      0.86865 0.012484708  2.023950 2.323038
4      LBA_T0   0.00530      0.85260 0.015552210  1.983226 2.273306
5      LBA_T0   0.00350      0.87175 0.012567607  2.031830 2.332051
6      LBA_T0   0.00490      0.85180 0.014888925  1.981200 2.271566
7      LBA_T0   0.00240      0.86225 0.010743910  2.007697 2.305860
8      LBA_T0   0.00425      0.85805 0.013811136  1.997041 2.290808
9      LBA_T0   0.00520      0.75615 0.015386387  1.741034 1.994127
10     LBA_T0   0.00415      0.86395 0.013645327  2.012012 2.308224
11     LBA_T0   0.00340      0.86825 0.012401810  2.022934 2.321940
12     LBA_T0   0.00460      0.81910 0.014391477  1.898618 2.176771
   Time_minutes Time_hours Time_days H_mol  Volume H_Molarity H_mM
1             0          0         0     0 1.90366          0    0
2             0          0         0     0 1.90366          0    0
3             0          0         0     0 1.90366          0    0
4             0          0         0     0 1.90366          0    0
5             0          0         0     0 1.90366          0    0
6             0          0         0     0 1.90366          0    0
7             0          0         0     0 1.90366          0    0
8             0          0         0     0 1.90366          0    0
9             0          0         0     0 1.90366          0    0
10            0          0         0     0 1.90366          0    0
11            0          0         0     0 1.90366          0    0
12            0          0         0     0 1.90366          0    0
   Correction_factor_NO2 Correction_factor_NO3
1               1.143276              1.029944
2               1.143276              1.029944
3               1.143276              1.029944
4               1.143276              1.029944
5               1.143276              1.029944
6               1.143276              1.029944
7               1.143276              1.029944
8               1.143276              1.029944
9               1.143276              1.029944
10              1.143276              1.029944
11              1.143276              1.029944
12              1.143276              1.029944</code></pre>
<pre class="r"><code># let&#39;s also include autoclaved sample
df_autoclaved &lt;- df_pH %&gt;% filter(H_mM == 0) %&gt;% filter(!(Time_point == &quot;Sterile_T8&quot;))
df_autoclaved$Antibiotics_type &lt;- &quot;Autoclaved&quot;
dim(df_autoclaved) # 33</code></pre>
<pre><code>[1] 162  26</code></pre>
<pre class="r"><code>df_effect &lt;- rbind(df_effect, df_autoclaved)
dim(df_effect) #153 -&gt; 150</code></pre>
<pre><code>[1] 282  26</code></pre>
<pre class="r"><code># this is for kyle
df_kyle2 &lt;- rbind(df_kyle, df_autoclaved)
write.xlsx(df_kyle2, &quot;220421_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx&quot;)

# reorder factor
df_effect$Antibiotics_type &lt;- factor(df_effect$Antibiotics_type, levels = c(&quot;None&quot;, &quot;CHL&quot;, &quot;NYT&quot;, &quot;CHL_NYT&quot;, &quot;Autoclaved&quot;))

# plot
ggplot(df_effect, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  # geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application (Chloramphenicol: CHL, Nystatin: NYT, CHL+NYT: CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # facet_grid(. ~ Soil) +
  # theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))



# plot average and standard deviation
colnames(df_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot;</code></pre>
<pre class="r"><code>dim(df_effect)</code></pre>
<pre><code>[1] 282  26</code></pre>
<pre class="r"><code>df_effect_ave &lt;- df_effect %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Nitrite_input&#39;, &#39;Nitrate_input&#39;, &#39;Soil&#39;, &#39;Titration_type&#39;, &#39;Sample_type&#39;, &#39;Antibiotics_type&#39;, &#39;H_mM&#39;, &#39;Time_point&#39;, &#39;Time_minutes&#39;, &#39;Time_hours&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(df_effect_ave)</code></pre>
<pre><code>[1] 61 15</code></pre>
<pre class="r"><code># nitrate
ggplot(df_effect_ave, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application: 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite
ggplot(df_effect_ave, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application: 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="lets-see-other-ph-perturbed-levels." class="section level2">
<h2>4.3. Let’s see other pH perturbed levels.</h2>
<pre class="r"><code>df_pH_effect &lt;- df_LBA %&gt;% filter(!(Titration_type %in% c(&quot;Nitrate&quot;, &quot;Nitrite&quot;, &quot;Ammonium&quot;, &quot;No_Nitrate&quot;)))
dim(df_pH_effect) # 354</code></pre>
<pre><code>[1] 354  26</code></pre>
<pre class="r"><code>df_pH_effect$Titration_type</code></pre>
<pre><code>  [1] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [11] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [21] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [31] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [41] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [51] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [61] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [71] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [81] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [91] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[101] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[111] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[121] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[131] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[141] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[151] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[161] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[171] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[181] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[191] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[201] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[211] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[221] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[231] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[241] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[251] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[261] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[271] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[281] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[291] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[301] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[311] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[321] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[331] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[341] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[351] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; </code></pre>
<pre class="r"><code>df_pH_effect$Antibiotics_type</code></pre>
<pre><code>  [1] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
  [8] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [15] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [22] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
 [29] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
 [36] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
 [43] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [50] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
 [57] &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot;
 [64] &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   
 [71] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
 [78] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [85] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [92] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
 [99] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
[106] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[113] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[120] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[127] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[134] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[141] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[148] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[155] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[162] &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot;
[169] &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   
[176] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[183] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[190] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[197] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[204] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
[211] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[218] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[225] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[232] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[239] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[246] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[253] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[260] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[267] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[274] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[281] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[288] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;    
[295] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;    
[302] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[309] &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[316] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[323] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[330] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;   
[337] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;   
[344] &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[351] &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   </code></pre>
<pre class="r"><code>df_pH_effect %&gt;% filter(Time_point == &quot;LBA_T0&quot;)</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input               Soil
1   A01             0             2              0 LaBaghWoods_pH6.66
2   A02             0             2              0 LaBaghWoods_pH6.66
3   A03             0             2              0 LaBaghWoods_pH6.66
4   A04             0             2              0 LaBaghWoods_pH6.66
5   A05             0             2              0 LaBaghWoods_pH6.66
6   A06             0             2              0 LaBaghWoods_pH6.66
7   B01             0             2              0 LaBaghWoods_pH6.66
8   B02             0             2              0 LaBaghWoods_pH6.66
9   B03             0             2              0 LaBaghWoods_pH6.66
10  B04             0             2              0 LaBaghWoods_pH6.66
11  B05             0             2              0 LaBaghWoods_pH6.66
12  B06             0             2              0 LaBaghWoods_pH6.66
13  C01             0             2              0 LaBaghWoods_pH6.66
14  C02             0             2              0 LaBaghWoods_pH6.66
15  C03             0             2              0 LaBaghWoods_pH6.66
16  C04             0             2              0 LaBaghWoods_pH6.66
17  C05             0             2              0 LaBaghWoods_pH6.66
18  C06             0             2              0 LaBaghWoods_pH6.66
19  D01             0             2              0 LaBaghWoods_pH6.66
20  D02             0             2              0 LaBaghWoods_pH6.66
21  D03             0             2              0 LaBaghWoods_pH6.66
22  D04             0             2              0 LaBaghWoods_pH6.66
23  D05             0             2              0 LaBaghWoods_pH6.66
24  D06             0             2              0 LaBaghWoods_pH6.66
25  E01             0             2              0 LaBaghWoods_pH6.66
26  E02             0             2              0 LaBaghWoods_pH6.66
27  E03             0             2              0 LaBaghWoods_pH6.66
28  E04             0             2              0 LaBaghWoods_pH6.66
29  E05             0             2              0 LaBaghWoods_pH6.66
30  E06             0             2              0 LaBaghWoods_pH6.66
31  F01             0             2              0 LaBaghWoods_pH6.66
32  F02             0             2              0 LaBaghWoods_pH6.66
33  F03             0             2              0 LaBaghWoods_pH6.66
34  F04             0             2              0 LaBaghWoods_pH6.66
35  F05             0             2              0 LaBaghWoods_pH6.66
36  F06             0             2              0 LaBaghWoods_pH6.66
   Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1       Slurry            HCl              CHL    2               1       20
2       Slurry            HCl              CHL    6               3       20
3       Slurry           NaOH              CHL    2               1      -20
4       Slurry           NaOH              CHL    6               3      -20
5       Slurry            HCl              CHL    0               0       20
6       Slurry            HCl              CHL    0               0       20
7       Slurry            HCl              CHL    4               2       20
8       Slurry            HCl              CHL    8               4       20
9       Slurry           NaOH              CHL    4               2      -20
10      Slurry           NaOH              CHL    8               4      -20
11      Slurry            HCl              CHL    0               0       20
12      Slurry            HCl              NYT    2               1       20
13      Slurry            HCl              NYT    4               2       20
14      Slurry            HCl              NYT    8               4       20
15      Slurry           NaOH              NYT    4               2      -20
16      Slurry           NaOH              NYT    8               4      -20
17      Slurry            HCl              NYT    0               0       20
18      Slurry            HCl             None    2               1       20
19      Slurry            HCl              NYT    6               3       20
20      Slurry           NaOH              NYT    2               1      -20
21      Slurry           NaOH              NYT    6               3      -20
22      Slurry            HCl              NYT    0               0       20
23      Slurry            HCl              NYT    0               0       20
24      Slurry            HCl             None    4               2       20
25      Slurry            HCl             None    6               3       20
26      Slurry           NaOH             None    2               1      -20
27      Slurry           NaOH             None    6               3      -20
28      Slurry            HCl          CHL_NYT    0               0       20
29      Slurry            HCl          CHL_NYT    0               0       20
30      Slurry            HCl             None    0               0       20
31      Slurry            HCl             None    8               4       20
32      Slurry           NaOH             None    4               2      -20
33      Slurry           NaOH             None    8               4      -20
34      Slurry            HCl          CHL_NYT    0               0       20
35      Slurry            HCl             None    0               0       20
36      Slurry            HCl             None    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM   NO3_mM
1      LBA_T0   0.00430      0.86470 0.013894041  2.013917 2.310196
2      LBA_T0   0.00375      0.89920 0.012982108  2.101806 2.412383
3      LBA_T0   0.01160      0.83550 0.026001940  1.939973 2.214007
4      LBA_T0   0.01885      0.82070 0.038034405  1.902647 2.160118
5      LBA_T0   0.00260      0.84035 0.011075478  1.952227 2.241586
6      LBA_T0   0.00100      0.77845 0.008423088  1.796651 2.064546
7      LBA_T0   0.00340      0.88860 0.012401810  2.074741 2.381691
8      LBA_T0   0.00440      0.85365 0.014059851  1.985887 2.277719
9      LBA_T0   0.01925      0.81970 0.038698483  1.900129 2.156616
10     LBA_T0   0.01760      0.84950 0.035959309  1.975375 2.245867
11     LBA_T0   0.00345      0.86865 0.012484708  2.023950 2.323038
12     LBA_T0   0.00305      0.83410 0.011821529  1.936438 2.222704
13     LBA_T0   0.00445      0.78455 0.014142757  1.811904 2.076985
14     LBA_T0   0.00550      0.85645 0.015883861  1.992984 2.284261
15     LBA_T0   0.01645      0.79690 0.034050417  1.842837 2.094727
16     LBA_T0   0.01550      0.84875 0.032473647  1.973476 2.246817
17     LBA_T0   0.00530      0.85260 0.015552210  1.983226 2.273306
18     LBA_T0   0.00440      0.81810 0.014059851  1.896101 2.174166
19     LBA_T0   0.00365      0.84200 0.012816307  1.956398 2.244829
20     LBA_T0   0.01325      0.82415 0.028739706  1.911339 2.178516
21     LBA_T0   0.01705      0.82690 0.035046337  1.918271 2.180830
22     LBA_T0   0.00350      0.87175 0.012567607  2.031830 2.332051
23     LBA_T0   0.00490      0.85180 0.014888925  1.981200 2.271566
24     LBA_T0   0.00260      0.84100 0.011075478  1.953870 2.243481
25     LBA_T0   0.00255      0.83295 0.010992586  1.933534 2.220102
26     LBA_T0   0.01375      0.84455 0.029569409  1.962847 2.237175
27     LBA_T0   0.01485      0.78935 0.031394879  1.823919 2.075299
28     LBA_T0   0.00240      0.86225 0.010743910  2.007697 2.305860
29     LBA_T0   0.00425      0.85805 0.013811136  1.997041 2.290808
30     LBA_T0   0.00520      0.75615 0.015386387  1.741034 1.994127
31     LBA_T0   0.00475      0.85035 0.014640199  1.977527 2.267555
32     LBA_T0   0.01580      0.84415 0.032971561  1.961835 2.232943
33     LBA_T0   0.01680      0.80460 0.034631364  1.862159 2.116488
34     LBA_T0   0.00415      0.86395 0.013645327  2.012012 2.308224
35     LBA_T0   0.00340      0.86825 0.012401810  2.022934 2.321940
36     LBA_T0   0.00460      0.81910 0.014391477  1.898618 2.176771
   Time_minutes Time_hours Time_days  H_mol  Volume  H_Molarity      H_mM
1             0          0         0  2e-05 1.90366  0.01050608  10.50608
2             0          0         0  6e-05 1.90366  0.03151823  31.51823
3             0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
4             0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
5             0          0         0  0e+00 1.90366  0.00000000   0.00000
6             0          0         0  0e+00 1.90366  0.00000000   0.00000
7             0          0         0  4e-05 1.90366  0.02101216  21.01216
8             0          0         0  8e-05 1.90366  0.04202431  42.02431
9             0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
10            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
11            0          0         0  0e+00 1.90366  0.00000000   0.00000
12            0          0         0  2e-05 1.90366  0.01050608  10.50608
13            0          0         0  4e-05 1.90366  0.02101216  21.01216
14            0          0         0  8e-05 1.90366  0.04202431  42.02431
15            0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
16            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
17            0          0         0  0e+00 1.90366  0.00000000   0.00000
18            0          0         0  2e-05 1.90366  0.01050608  10.50608
19            0          0         0  6e-05 1.90366  0.03151823  31.51823
20            0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
21            0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
22            0          0         0  0e+00 1.90366  0.00000000   0.00000
23            0          0         0  0e+00 1.90366  0.00000000   0.00000
24            0          0         0  4e-05 1.90366  0.02101216  21.01216
25            0          0         0  6e-05 1.90366  0.03151823  31.51823
26            0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
27            0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
28            0          0         0  0e+00 1.90366  0.00000000   0.00000
29            0          0         0  0e+00 1.90366  0.00000000   0.00000
30            0          0         0  0e+00 1.90366  0.00000000   0.00000
31            0          0         0  8e-05 1.90366  0.04202431  42.02431
32            0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
33            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
34            0          0         0  0e+00 1.90366  0.00000000   0.00000
35            0          0         0  0e+00 1.90366  0.00000000   0.00000
36            0          0         0  0e+00 1.90366  0.00000000   0.00000
   Correction_factor_NO2 Correction_factor_NO3
1               1.143276              1.029944
2               1.143276              1.029944
3               1.143276              1.029944
4               1.143276              1.029944
5               1.143276              1.029944
6               1.143276              1.029944
7               1.143276              1.029944
8               1.143276              1.029944
9               1.143276              1.029944
10              1.143276              1.029944
11              1.143276              1.029944
12              1.143276              1.029944
13              1.143276              1.029944
14              1.143276              1.029944
15              1.143276              1.029944
16              1.143276              1.029944
17              1.143276              1.029944
18              1.143276              1.029944
19              1.143276              1.029944
20              1.143276              1.029944
21              1.143276              1.029944
22              1.143276              1.029944
23              1.143276              1.029944
24              1.143276              1.029944
25              1.143276              1.029944
26              1.143276              1.029944
27              1.143276              1.029944
28              1.143276              1.029944
29              1.143276              1.029944
30              1.143276              1.029944
31              1.143276              1.029944
32              1.143276              1.029944
33              1.143276              1.029944
34              1.143276              1.029944
35              1.143276              1.029944
36              1.143276              1.029944</code></pre>
<pre class="r"><code># let&#39;s also include autoclaved sample
df_autoclaved &lt;- df_pH %&gt;% filter(H_mM == 0) %&gt;% filter(!(Time_point == &quot;Sterile_T8&quot;))
df_autoclaved$Antibiotics_type &lt;- &quot;Autoclaved&quot;
dim(df_autoclaved) # 33</code></pre>
<pre><code>[1] 162  26</code></pre>
<pre class="r"><code>df_pH_effect &lt;- rbind(df_pH_effect, df_autoclaved)
dim(df_pH_effect) #384</code></pre>
<pre><code>[1] 516  26</code></pre>
<pre class="r"><code># reorder factor
df_pH_effect$Antibiotics_type &lt;- factor(df_pH_effect$Antibiotics_type, levels = c(&quot;None&quot;, &quot;CHL&quot;, &quot;NYT&quot;, &quot;CHL_NYT&quot;, &quot;Autoclaved&quot;))

# plot average and standard deviation
colnames(df_pH_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot;</code></pre>
<pre class="r"><code>dim(df_pH_effect)</code></pre>
<pre><code>[1] 516  26</code></pre>
<pre class="r"><code>df_pH_effect_ave &lt;- df_pH_effect %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Nitrite_input&#39;, &#39;Nitrate_input&#39;, &#39;Soil&#39;, &#39;Titration_type&#39;, &#39;Sample_type&#39;, &#39;Antibiotics_type&#39;, &#39;H_mM&#39;, &#39;Time_point&#39;, &#39;Time_minutes&#39;, &#39;Time_hours&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(df_pH_effect_ave) # 284</code></pre>
<pre><code>[1] 295  15</code></pre>
<pre class="r"><code># nitrate
ggplot(df_pH_effect_ave, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation(H_mM): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_wrap(. ~ H_mM, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite
ggplot(df_pH_effect_ave, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation(H_mM): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_wrap(. ~ H_mM, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot pH in these plots</p>
<pre class="r"><code>df_kyle2 &lt;- read.xlsx(&quot;data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx&quot;)

df_pH_effect &lt;- df_kyle2 %&gt;% filter(!(Titration_type %in% c(&quot;Nitrate&quot;, &quot;Nitrite&quot;, &quot;Ammonium&quot;, &quot;No_Nitrate&quot;)))
dim(df_pH_effect) # 354</code></pre>
<pre><code>[1] 354  27</code></pre>
<pre class="r"><code>df_pH_effect$Titration_type</code></pre>
<pre><code>  [1] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [11] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [21] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [31] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [41] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [51] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [61] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
 [71] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
 [81] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
 [91] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[101] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[111] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[121] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[131] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[141] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[151] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[161] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[171] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[181] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[191] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[201] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[211] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[221] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[231] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[241] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot;
[251] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[261] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[271] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[281] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[291] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[301] &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[311] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; 
[321] &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot; 
[331] &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot; 
[341] &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot; &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot;  &quot;NaOH&quot;
[351] &quot;NaOH&quot; &quot;HCl&quot;  &quot;HCl&quot;  &quot;HCl&quot; </code></pre>
<pre class="r"><code>df_pH_effect$Antibiotics_type</code></pre>
<pre><code>  [1] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
  [8] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [15] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [22] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
 [29] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
 [36] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
 [43] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [50] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
 [57] &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot;
 [64] &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   
 [71] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
 [78] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [85] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
 [92] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
 [99] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
[106] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[113] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[120] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[127] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[134] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[141] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[148] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[155] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[162] &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot;
[169] &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   
[176] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[183] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[190] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[197] &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[204] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;   
[211] &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[218] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[225] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[232] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[239] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[246] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[253] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;     &quot;NYT&quot;    
[260] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;     &quot;NYT&quot;    
[267] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[274] &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot;
[281] &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[288] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;NYT&quot;    
[295] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;NYT&quot;    
[302] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[309] &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[316] &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[323] &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;     &quot;CHL&quot;    
[330] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;   
[337] &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;NYT&quot;     &quot;None&quot;    &quot;None&quot;   
[344] &quot;None&quot;    &quot;None&quot;    &quot;CHL_NYT&quot; &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;    &quot;None&quot;   
[351] &quot;None&quot;    &quot;CHL_NYT&quot; &quot;None&quot;    &quot;None&quot;   </code></pre>
<pre class="r"><code>df_pH_effect %&gt;% filter(Time_point == &quot;LBA_T0&quot;)</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input               Soil
1   A01             0             2              0 LaBaghWoods_pH6.66
2   A02             0             2              0 LaBaghWoods_pH6.66
3   A03             0             2              0 LaBaghWoods_pH6.66
4   A04             0             2              0 LaBaghWoods_pH6.66
5   A05             0             2              0 LaBaghWoods_pH6.66
6   A06             0             2              0 LaBaghWoods_pH6.66
7   B01             0             2              0 LaBaghWoods_pH6.66
8   B02             0             2              0 LaBaghWoods_pH6.66
9   B03             0             2              0 LaBaghWoods_pH6.66
10  B04             0             2              0 LaBaghWoods_pH6.66
11  B05             0             2              0 LaBaghWoods_pH6.66
12  B06             0             2              0 LaBaghWoods_pH6.66
13  C01             0             2              0 LaBaghWoods_pH6.66
14  C02             0             2              0 LaBaghWoods_pH6.66
15  C03             0             2              0 LaBaghWoods_pH6.66
16  C04             0             2              0 LaBaghWoods_pH6.66
17  C05             0             2              0 LaBaghWoods_pH6.66
18  C06             0             2              0 LaBaghWoods_pH6.66
19  D01             0             2              0 LaBaghWoods_pH6.66
20  D02             0             2              0 LaBaghWoods_pH6.66
21  D03             0             2              0 LaBaghWoods_pH6.66
22  D04             0             2              0 LaBaghWoods_pH6.66
23  D05             0             2              0 LaBaghWoods_pH6.66
24  D06             0             2              0 LaBaghWoods_pH6.66
25  E01             0             2              0 LaBaghWoods_pH6.66
26  E02             0             2              0 LaBaghWoods_pH6.66
27  E03             0             2              0 LaBaghWoods_pH6.66
28  E04             0             2              0 LaBaghWoods_pH6.66
29  E05             0             2              0 LaBaghWoods_pH6.66
30  E06             0             2              0 LaBaghWoods_pH6.66
31  F01             0             2              0 LaBaghWoods_pH6.66
32  F02             0             2              0 LaBaghWoods_pH6.66
33  F03             0             2              0 LaBaghWoods_pH6.66
34  F04             0             2              0 LaBaghWoods_pH6.66
35  F05             0             2              0 LaBaghWoods_pH6.66
36  F06             0             2              0 LaBaghWoods_pH6.66
   Sample_type Titration_type Antibiotics_type Unit Concentration_M Added_ul
1       Slurry            HCl              CHL    2               1       20
2       Slurry            HCl              CHL    6               3       20
3       Slurry           NaOH              CHL    2               1      -20
4       Slurry           NaOH              CHL    6               3      -20
5       Slurry            HCl              CHL    0               0       20
6       Slurry            HCl              CHL    0               0       20
7       Slurry            HCl              CHL    4               2       20
8       Slurry            HCl              CHL    8               4       20
9       Slurry           NaOH              CHL    4               2      -20
10      Slurry           NaOH              CHL    8               4      -20
11      Slurry            HCl              CHL    0               0       20
12      Slurry            HCl              NYT    2               1       20
13      Slurry            HCl              NYT    4               2       20
14      Slurry            HCl              NYT    8               4       20
15      Slurry           NaOH              NYT    4               2      -20
16      Slurry           NaOH              NYT    8               4      -20
17      Slurry            HCl              NYT    0               0       20
18      Slurry            HCl             None    2               1       20
19      Slurry            HCl              NYT    6               3       20
20      Slurry           NaOH              NYT    2               1      -20
21      Slurry           NaOH              NYT    6               3      -20
22      Slurry            HCl              NYT    0               0       20
23      Slurry            HCl              NYT    0               0       20
24      Slurry            HCl             None    4               2       20
25      Slurry            HCl             None    6               3       20
26      Slurry           NaOH             None    2               1      -20
27      Slurry           NaOH             None    6               3      -20
28      Slurry            HCl          CHL_NYT    0               0       20
29      Slurry            HCl          CHL_NYT    0               0       20
30      Slurry            HCl             None    0               0       20
31      Slurry            HCl             None    8               4       20
32      Slurry           NaOH             None    4               2      -20
33      Slurry           NaOH             None    8               4      -20
34      Slurry            HCl          CHL_NYT    0               0       20
35      Slurry            HCl             None    0               0       20
36      Slurry            HCl             None    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM   NO3_mM
1      LBA_T0   0.00430      0.86470 0.013894041  2.013917 2.310196
2      LBA_T0   0.00375      0.89920 0.012982108  2.101806 2.412383
3      LBA_T0   0.01160      0.83550 0.026001940  1.939973 2.214007
4      LBA_T0   0.01885      0.82070 0.038034405  1.902647 2.160118
5      LBA_T0   0.00260      0.84035 0.011075478  1.952227 2.241586
6      LBA_T0   0.00100      0.77845 0.008423088  1.796651 2.064546
7      LBA_T0   0.00340      0.88860 0.012401810  2.074741 2.381691
8      LBA_T0   0.00440      0.85365 0.014059851  1.985887 2.277719
9      LBA_T0   0.01925      0.81970 0.038698483  1.900129 2.156616
10     LBA_T0   0.01760      0.84950 0.035959309  1.975375 2.245867
11     LBA_T0   0.00345      0.86865 0.012484708  2.023950 2.323038
12     LBA_T0   0.00305      0.83410 0.011821529  1.936438 2.222704
13     LBA_T0   0.00445      0.78455 0.014142757  1.811904 2.076985
14     LBA_T0   0.00550      0.85645 0.015883861  1.992984 2.284261
15     LBA_T0   0.01645      0.79690 0.034050417  1.842837 2.094727
16     LBA_T0   0.01550      0.84875 0.032473647  1.973476 2.246817
17     LBA_T0   0.00530      0.85260 0.015552210  1.983226 2.273306
18     LBA_T0   0.00440      0.81810 0.014059851  1.896101 2.174166
19     LBA_T0   0.00365      0.84200 0.012816307  1.956398 2.244829
20     LBA_T0   0.01325      0.82415 0.028739706  1.911339 2.178516
21     LBA_T0   0.01705      0.82690 0.035046337  1.918271 2.180830
22     LBA_T0   0.00350      0.87175 0.012567607  2.031830 2.332051
23     LBA_T0   0.00490      0.85180 0.014888925  1.981200 2.271566
24     LBA_T0   0.00260      0.84100 0.011075478  1.953870 2.243481
25     LBA_T0   0.00255      0.83295 0.010992586  1.933534 2.220102
26     LBA_T0   0.01375      0.84455 0.029569409  1.962847 2.237175
27     LBA_T0   0.01485      0.78935 0.031394879  1.823919 2.075299
28     LBA_T0   0.00240      0.86225 0.010743910  2.007697 2.305860
29     LBA_T0   0.00425      0.85805 0.013811136  1.997041 2.290808
30     LBA_T0   0.00520      0.75615 0.015386387  1.741034 1.994127
31     LBA_T0   0.00475      0.85035 0.014640199  1.977527 2.267555
32     LBA_T0   0.01580      0.84415 0.032971561  1.961835 2.232943
33     LBA_T0   0.01680      0.80460 0.034631364  1.862159 2.116488
34     LBA_T0   0.00415      0.86395 0.013645327  2.012012 2.308224
35     LBA_T0   0.00340      0.86825 0.012401810  2.022934 2.321940
36     LBA_T0   0.00460      0.81910 0.014391477  1.898618 2.176771
   Time_minutes Time_hours Time_days  H_mol  Volume  H_Molarity      H_mM
1             0          0         0  2e-05 1.90366  0.01050608  10.50608
2             0          0         0  6e-05 1.90366  0.03151823  31.51823
3             0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
4             0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
5             0          0         0  0e+00 1.90366  0.00000000   0.00000
6             0          0         0  0e+00 1.90366  0.00000000   0.00000
7             0          0         0  4e-05 1.90366  0.02101216  21.01216
8             0          0         0  8e-05 1.90366  0.04202431  42.02431
9             0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
10            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
11            0          0         0  0e+00 1.90366  0.00000000   0.00000
12            0          0         0  2e-05 1.90366  0.01050608  10.50608
13            0          0         0  4e-05 1.90366  0.02101216  21.01216
14            0          0         0  8e-05 1.90366  0.04202431  42.02431
15            0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
16            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
17            0          0         0  0e+00 1.90366  0.00000000   0.00000
18            0          0         0  2e-05 1.90366  0.01050608  10.50608
19            0          0         0  6e-05 1.90366  0.03151823  31.51823
20            0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
21            0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
22            0          0         0  0e+00 1.90366  0.00000000   0.00000
23            0          0         0  0e+00 1.90366  0.00000000   0.00000
24            0          0         0  4e-05 1.90366  0.02101216  21.01216
25            0          0         0  6e-05 1.90366  0.03151823  31.51823
26            0          0         0 -2e-05 1.90366 -0.01050608 -10.50608
27            0          0         0 -6e-05 1.90366 -0.03151823 -31.51823
28            0          0         0  0e+00 1.90366  0.00000000   0.00000
29            0          0         0  0e+00 1.90366  0.00000000   0.00000
30            0          0         0  0e+00 1.90366  0.00000000   0.00000
31            0          0         0  8e-05 1.90366  0.04202431  42.02431
32            0          0         0 -4e-05 1.90366 -0.02101216 -21.01216
33            0          0         0 -8e-05 1.90366 -0.04202431 -42.02431
34            0          0         0  0e+00 1.90366  0.00000000   0.00000
35            0          0         0  0e+00 1.90366  0.00000000   0.00000
36            0          0         0  0e+00 1.90366  0.00000000   0.00000
   Correction_factor_NO2 Correction_factor_NO3    pH
1               1.143276              1.029944 5.781
2               1.143276              1.029944 4.027
3               1.143276              1.029944 8.159
4               1.143276              1.029944 8.520
5               1.143276              1.029944 7.625
6               1.143276              1.029944 7.625
7               1.143276              1.029944 4.628
8               1.143276              1.029944 3.438
9               1.143276              1.029944 8.375
10              1.143276              1.029944 8.592
11              1.143276              1.029944 7.625
12              1.143276              1.029944 6.574
13              1.143276              1.029944 4.640
14              1.143276              1.029944 3.474
15              1.143276              1.029944 8.387
16              1.143276              1.029944 8.688
17              1.143276              1.029944 7.803
18              1.143276              1.029944 6.514
19              1.143276              1.029944 3.979
20              1.143276              1.029944 8.111
21              1.143276              1.029944 8.508
22              1.143276              1.029944 7.803
23              1.143276              1.029944 7.803
24              1.143276              1.029944 4.916
25              1.143276              1.029944 3.967
26              1.143276              1.029944 8.195
27              1.143276              1.029944 8.532
28              1.143276              1.029944 7.727
29              1.143276              1.029944 7.727
30              1.143276              1.029944 7.631
31              1.143276              1.029944 3.955
32              1.143276              1.029944 8.399
33              1.143276              1.029944 8.664
34              1.143276              1.029944 7.727
35              1.143276              1.029944 7.631
36              1.143276              1.029944 7.631</code></pre>
<pre class="r"><code># let&#39;s also include autoclaved sample
df_autoclaved &lt;- df_pH %&gt;% filter(H_mM == 0) %&gt;% filter(!(Time_point == &quot;Sterile_T8&quot;))
df_autoclaved$Antibiotics_type &lt;- &quot;Autoclaved&quot;
dim(df_autoclaved) # 33</code></pre>
<pre><code>[1] 162  26</code></pre>
<pre class="r"><code>df_autoclaved$pH &lt;- 7.7 # any number between 7.6, 7.8 to be classified as neutral soil.

colnames(df_pH_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot; &quot;pH&quot;                   </code></pre>
<pre class="r"><code>colnames(df_autoclaved)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot; &quot;pH&quot;                   </code></pre>
<pre class="r"><code>df_pH_effect &lt;- rbind(df_pH_effect, df_autoclaved)
dim(df_pH_effect) #384</code></pre>
<pre><code>[1] 516  27</code></pre>
<pre class="r"><code># reorder factor
df_pH_effect$Antibiotics_type &lt;- factor(df_pH_effect$Antibiotics_type, levels = c(&quot;None&quot;, &quot;CHL&quot;, &quot;NYT&quot;, &quot;CHL_NYT&quot;, &quot;Autoclaved&quot;))

# plot average and standard deviation
colnames(df_pH_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Unit&quot;                 
[10] &quot;Concentration_M&quot;       &quot;Added_ul&quot;              &quot;Time_point&quot;           
[13] &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;          &quot;NO2_mM&quot;               
[16] &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;                &quot;Time_minutes&quot;         
[19] &quot;Time_hours&quot;            &quot;Time_days&quot;             &quot;H_mol&quot;                
[22] &quot;Volume&quot;                &quot;H_Molarity&quot;            &quot;H_mM&quot;                 
[25] &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot; &quot;pH&quot;                   </code></pre>
<pre class="r"><code>dim(df_pH_effect)</code></pre>
<pre><code>[1] 516  27</code></pre>
<pre class="r"><code>df_pH_effect_ave &lt;- df_pH_effect %&gt;% group_by(Soil, Titration_type, Unit, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days, pH) %&gt;% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Sample_type&#39;, &#39;Antibiotics_type&#39;, &#39;H_mM&#39;, &#39;Time_point&#39;, &#39;Time_minutes&#39;, &#39;Time_hours&#39;, &#39;Time_days&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(df_pH_effect_ave) # 284</code></pre>
<pre><code>[1] 295  15</code></pre>
<pre class="r"><code># label with pH range
df_pH_label &lt;- df_pH_effect_ave %&gt;% group_by(Titration_type, Unit) %&gt;% summarize(pH_min = min(pH), pH_max = max(pH))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Titration_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code># df_pH_label$pH_max &lt;- as.character()

df_pH_label %&lt;&gt;% unite(col=&#39;pH_range&#39;, c(&quot;pH_min&quot;,&quot;pH_max&quot;), sep=&quot;-&quot;, remove=F)

df_pH_range &lt;- df_pH_effect_ave %&gt;% left_join(df_pH_label, by =c(&quot;Titration_type&quot;=&quot;Titration_type&quot;, &quot;Unit&quot;=&quot;Unit&quot;))


# nitrate
ggplot(df_pH_range, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation (pH at day2): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_wrap(. ~ pH_range, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite
ggplot(df_pH_range, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation (pH at day2): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_wrap(. ~ pH_range, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220421_Griess_antibiotics_and_sterile.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ## (Temporary) Filter the T8 for now. -->
<!-- ```{r} -->
<!-- df_plot2 <- df_plot2 %>% filter(!(Time_point %in% c("WB_T5", "C_T5")))  -->
<!-- dim(df_plot2) -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- ``` -->
<!-- ## 4.1.1. Nitrite peakheight and H_mM -->
<!-- ```{r echo=F} -->
<!-- colnames(df_plot2) -->
<!-- df_plot2 %>% group_by(H_mM) %>% summarize(Max_NO2_mM = max(NO2_mM)) %>% ungroup() -->
<!-- # library(ggpmisc) -->
<!-- # x[ggpmisc:::find_peaks(df_plot2$NO2_mM)] -->
<!-- # get global peaks -->
<!-- df_plot2 -->
<!-- mat_H_time <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO2_mM") -->
<!-- mat_H_time <- tibble::column_to_rownames(mat_H_time, var = "H_mM") -->
<!-- mat_H_time <- as.matrix(mat_H_time) -->
<!-- class(mat_H_time) -->
<!-- ## get rid of NA -->
<!-- # Substitute NA to average of NO2_mM of before and after. -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- mat_H_time <- na_to_average(mat_H_time) -->
<!-- mat_H_time[is.na(mat_H_time)] # no more NA -->
<!-- # maximum peak value -->
<!-- apply(mat_H_time, 1, max) -->
<!-- ## plot with peak height and H_mM -->
<!-- vec_peakheight <- apply(mat_H_time, 1, max) -->
<!-- vec_H_mM <- names(vec_peakheight) -->
<!-- plot(vec_H_mM, vec_peakheight) -->
<!-- df_peakheight <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_peakheight) -->
<!-- df_peakheight$H_mM <- as.numeric(df_peakheight$H_mM) -->
<!-- ## plot -->
<!-- # linear regression -->
<!-- fit.peakheight <- lm(NO2_mM ~ H_mM, df_peakheight) -->
<!-- summary(fit.peakheight) -->
<!-- # Plot H_mM with peak time -->
<!-- ## 1. linear regression -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_abline(slope = coef(fit.peakheight)[[2]], intercept = coef(fit.peakheight)[[1]], -->
<!--               color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=1, label= paste0("y = ", round(coef(fit.peakheight)[[1]],3),"+",round(coef(fit.peakheight)[[2]],3),"x", ",  R^2: ", round(summary(fit.peakheight)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # 2. quadratic regression -->
<!-- df_peakheight$H_mM_2 <- (df_peakheight$H_mM)^2 -->
<!-- quad.peakheight <- lm(NO2_mM ~ H_mM + H_mM_2, df_peakheight) -->
<!-- summary(quad.peakheight) -->
<!-- perturbValues <- seq(-100, 100, 0.1) -->
<!-- peakPredict <- predict(quad.peakheight,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) -->
<!-- df_quad <- data.frame(H_mM = perturbValues, NO2_mM = peakPredict) -->
<!-- plot(perturbValues, peakPredict) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak height (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak height (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   geom_line(data = df_quad, aes(x = H_mM, y = NO2_mM), color = "blue", size = 1) + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=1, label= paste0("y = ", round(coef(quad.peakheight)[[1]],3), "+", round(coef(quad.peakheight)[[2]],3),"x+",round(coef(quad.peakheight)[[3]],3), "x^2", ",  R^2: ", round(summary(quad.peakheight)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 4.1.1.2. PCA on the nitrate, nitrite time readings. -->
<!-- Before PCA, we need to make the time point the same. In PCA, it wouldn't matter, right? yes. The value of the column or row is not incorporated in the SVD process. -->
<!-- ```{r} -->
<!-- # Data only with autoclave and antibiotic treated samples -->
<!-- df_PCA <- df_pH_effect_ave %>% select(-Std_NO2_mM, -Std_NO3_mM)  # remove standard deviation -->
<!-- # Remove soil name in time_point string -->
<!-- df_PCA$Time_point <- str_replace(df_PCA$Time_point, ".*_","") -->
<!-- any(is.na(df_PCA)) # no NAs. That's good -->
<!-- dim(df_PCA) # 284 15 -->
<!-- # remove NA -->
<!-- df_PCA[is.na(df_PCA)] -->
<!-- df_na <- df_PCA[rowSums(is.na(df_PCA)) > 0,] -->
<!-- # df_PCA <- na.omit(df_PCA) -->
<!-- # df_PCA <- df_PCA[rowSums(is.na(df_PCA)) == 0,] -->
<!-- # dim(df_PCA) # 1046 -->
<!-- paste0("T",00:10) -->
<!-- # sprintf("%02d", 0:10) -->
<!-- df_PCA$Time_point <- factor(df_PCA$Time_point, levels = paste0("T",00:10)) -->
<!-- dim(df_PCA) -->
<!-- ``` -->
<!-- ## (1) Plot PCA with nitrate + nitrite -->
<!-- ```{r} -->
<!-- # function -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- ## make the above values to matrices -->
<!-- mat_NO2 <- dcast(df_PCA, H_mM + Titration_type + Antibiotics_type ~ Time_point, value.var = "NO2_mM") -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- # mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") -->
<!-- dim(mat_NO2) -->
<!-- colnames(mat_NO2) -->
<!-- col_NO2 <- mat_NO2[,1:3] -->
<!-- mat_NO2 <- as.matrix(mat_NO2[,4:ncol(mat_NO2)])  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- mat_NO2 <- na_to_average(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- # col_NO2[24,] -->
<!-- # mat_NO2[24,] -->
<!-- # need to remove row 24 -->
<!-- mat_NO3 <- dcast(df_PCA, H_mM + Titration_type + Antibiotics_type ~ Time_point, value.var = "NO3_mM") -->
<!-- dim(mat_NO3) -->
<!-- col_NO3 <- mat_NO3[,1:3] -->
<!-- mat_NO3 <- as.matrix(mat_NO3[,4:ncol(mat_NO3)])  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- # col_NO3[24,] -->
<!-- # mat_NO3[24,] -->
<!-- # merge 2 matrix -->
<!-- col_NO3 -->
<!-- col_NO2 -->
<!-- identical(col_NO3, col_NO2) # True -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO2NO3 <- cbind(mat_NO2, mat_NO3) -->
<!-- dim(mat_NO2NO3) -->
<!-- # put in the row names -->
<!-- col_NO2$H_mM <- round(col_NO2$H_mM, 2) -->
<!-- class(col_NO2) -->
<!-- vec_sample = unite(col_NO2, "Sample", Titration_type, H_mM,Antibiotics_type, sep=":") -->
<!-- vec_sample = vec_sample$Sample -->
<!-- length(vec_sample) -->
<!-- dim(mat_NO2NO3) -->
<!-- rownames(mat_NO2NO3) = vec_sample -->
<!-- rownames(mat_NO2NO3) -->
<!-- mat_NO2NO3 -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # remove row 24 -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- dim(df_out) -->
<!-- df_out$new <- rownames(df_out) -->
<!-- identical(vec_sample, df_out$new) -->
<!-- df_out <- separate(df_out, new, c("Titration_type","H_mM","Antibiotics_type"), sep=":", remove=T) -->
<!-- df_out$H_mM <- as.numeric(df_out$H_mM) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # make histogram for PC1 -->
<!-- ggplot(df_out, aes(x=PC1)) + -->
<!--   geom_freqpoly(binwidth = 0.5, aes(color=Antibiotics_type), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC1") + -->
<!--   ggtitle(paste0("Marginal distribution of PC1"))+ -->
<!--   mytheme_2d -->
<!-- # make histogram for PC2 -->
<!-- ggplot(df_out, aes(x=PC2)) + -->
<!--   geom_freqpoly(binwidth = 0.85, aes(color=Soil), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC2") + -->
<!--   ggtitle(paste0("Marginal distribution of PC2"))+ -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- # library("plot3D") -->
<!-- # rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- # scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), -->
<!-- #           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!-- #           ylab ="PC2", zlab = "PC3", -->
<!-- #           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- Again with pH values -->
<!-- ```{r} -->
<!-- # function -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- df_PCA$Titration_type %>% unique() -->
<!-- ## make the above values to matrices -->
<!-- mat_NO2 <- dcast(df_PCA, pH + Antibiotics_type ~ Time_point, value.var = "NO2_mM") -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- # mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") -->
<!-- dim(mat_NO2) -->
<!-- colnames(mat_NO2) -->
<!-- col_NO2 <- mat_NO2[,1:2] -->
<!-- mat_NO2 <- as.matrix(mat_NO2[,3:ncol(mat_NO2)])  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- mat_NO2 <- na_to_average(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- # col_NO2[24,] -->
<!-- # mat_NO2[24,] -->
<!-- # need to remove row 24 -->
<!-- mat_NO3 <- dcast(df_PCA, pH + Antibiotics_type ~ Time_point, value.var = "NO3_mM") -->
<!-- dim(mat_NO3) -->
<!-- col_NO3 <- mat_NO3[,1:2] -->
<!-- mat_NO3 <- as.matrix(mat_NO3[,3:ncol(mat_NO3)])  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- # col_NO3[24,] -->
<!-- # mat_NO3[24,] -->
<!-- # merge 2 matrix -->
<!-- col_NO3 -->
<!-- col_NO2 -->
<!-- identical(col_NO3, col_NO2) # True -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO2NO3 <- cbind(mat_NO2, mat_NO3) -->
<!-- dim(mat_NO2NO3) -->
<!-- # put in the row names -->
<!-- col_NO2$pH <- round(col_NO2$pH, 2) -->
<!-- class(col_NO2) -->
<!-- vec_sample = unite(col_NO2, "Sample", pH,Antibiotics_type, sep=":") -->
<!-- vec_sample = vec_sample$Sample -->
<!-- length(vec_sample) -->
<!-- dim(mat_NO2NO3) -->
<!-- rownames(mat_NO2NO3) = vec_sample -->
<!-- rownames(mat_NO2NO3) -->
<!-- mat_NO2NO3 -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # remove row 24 -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- dim(df_out) -->
<!-- df_out$new <- rownames(df_out) -->
<!-- identical(vec_sample, df_out$new) -->
<!-- df_out <- separate(df_out, new, c("pH","Antibiotics_type"), sep=":", remove=T) -->
<!-- df_out$pH <- as.numeric(df_out$pH) -->
<!-- df_out$Antibiotics_type <- factor(df_out$Antibiotics_type, levels = c("None", "CHL", "NYT", "CHL_NYT", "Autoclaved")) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # make histogram for PC1 -->
<!-- ggplot(df_out, aes(x=PC1)) + -->
<!--   geom_freqpoly(binwidth = 0.5, aes(color=pH), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC1") + -->
<!--   ggtitle(paste0("Marginal distribution of PC1"))+ -->
<!--   mytheme_2d -->
<!-- # make histogram for PC2 -->
<!-- ggplot(df_out, aes(x=PC2)) + -->
<!--   geom_freqpoly(binwidth = 0.85, aes(color=pH), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC2") + -->
<!--   ggtitle(paste0("Marginal distribution of PC2"))+ -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot pH and PC1 -->
<!-- ggplot(df_out,aes(x=PC1,y=pH, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("pH \n")) + -->
<!--   ggtitle("pH and PC1 \n") + -->
<!--   # ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- Let's plot delta I / delta A -->
<!-- ```{r} -->
<!-- # compute delta I -->
<!-- mat_NO2 -->
<!-- dim(mat_NO2[,1]) -->
<!-- length(mat_NO2[,1]) -->
<!-- mat_delI <- matrix(nrow=nrow(mat_NO2), ncol=ncol(mat_NO2)-1) -->
<!-- for (i in 1:(ncol(mat_NO2)-1)){ -->
<!--   print(i) -->
<!--   # print(mat_NO2[,i+1, drop=F] - mat_NO2[,i, drop=F] ) -->
<!--   mat_delI[,i] <- mat_NO2[,i+1, drop=F] - mat_NO2[,i, drop=F] -->
<!-- } -->
<!-- # compute delta A -->
<!-- mat_NO3 -->
<!-- mat_delA <- matrix(nrow=nrow(mat_NO3), ncol=ncol(mat_NO3)-1) -->
<!-- for (i in 1:(ncol(mat_NO3)-1)){ -->
<!--   print(i) -->
<!--   # print(mat_NO3[,i+1, drop=F] - mat_NO3[,i, drop=F] ) -->
<!--   mat_delA[,i] <- mat_NO3[,i+1, drop=F] - mat_NO3[,i, drop=F] -->
<!-- } -->
<!-- col_NO2 -->
<!-- abs(mat_delI / mat_delA) -->
<!-- # confirming -->
<!-- mat_delI[2,3] / mat_delA[2,3] -->
<!-- mat_delI[28,8] / mat_delA[28,8] -->
<!-- # histogram -->
<!-- hist( abs(mat_delI / mat_delA)) -->
<!-- # remove Inf? -->
<!-- # let's plot -->
<!-- df_ratio_delI_delA <- cbind(col_NO2, abs(mat_delI / mat_delA)) -->
<!-- df_ratio_delI_delA <- melt(df_ratio_delI_delA, id.vars = c("Antibiotics_type","pH"), variable.name = "Time", value.name = "ratio_delI_delA") -->
<!-- df_delA <- cbind(col_NO2, abs(mat_delA)) -->
<!-- df_delA <- melt(df_delA, id.vars = c("Antibiotics_type","pH"), variable.name = "Time", value.name = "delA") -->
<!-- # left join -->
<!-- df_merge_ratio_delA <- df_ratio_delI_delA %>% left_join(df_delA, by = c("Antibiotics_type"="Antibiotics_type","pH"="pH","Time"="Time")) -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I delta A ratio \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   scale_y_continuous(limits=c(0, 1)) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I / delta A \n") + -->
<!--   # ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   scale_y_continuous(limits=c(0, 1)) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I / delta A \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## (1-1) How about an NMDS? -->
<!-- ```{r} -->
<!-- ``` -->
<!-- <!-- (1-2) PCA nitrate -->
<p>–&gt;</p>
<!-- <!-- ```{r echo =F} -->
<p>–&gt; <!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) --> –&gt; <!-- <!-- any(is.na(mat_NO3)) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO3) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- dim(df_out) --> –&gt; <!-- <!-- df_out$H_mM <- rownames(df_out) --> –&gt; <!-- <!-- df_out$H_mM <- as.numeric(df_out$H_mM) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrate dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrate dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), --> –&gt; <!-- <!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- (2) PCA with only nitrite dynamics -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) --> –&gt; <!-- <!-- any(is.na(mat_NO2)) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO2) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- df_out$H_mM <- rownames(df_out) --> –&gt; <!-- <!-- df_out$H_mM <- as.numeric(df_out$H_mM) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrite dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrite dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), --> –&gt; <!-- <!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- (3) PCA with both nitrate and nitrite dynamics -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) --> –&gt; <!-- <!-- dim(mat_NO2) --> –&gt; <!-- <!-- dim(mat_NO3) --> –&gt; <!-- <!-- mat_NO23 <- cbind(mat_NO3, mat_NO2) --> –&gt; <!-- <!-- any(is.na(mat_NO23)) --> –&gt; <!-- <!-- dim(mat_NO23) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO23) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- df_out$H_mM <- rownames(df_out) --> –&gt; <!-- <!-- df_out$H_mM <- as.numeric(df_out$H_mM) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrite + Nitrate dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("Principle component analysis (Nitrite + Nitrate dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), --> –&gt; <!-- <!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 4.1.1.3. (using inferred pH) PCA on the nitrate, nitrite time readings.  -->
<p>–&gt;</p>
<!-- <!-- (1) (using inferred pH) First plot PCA with only the nitrate -->
<p>–&gt; <!-- <!-- ```{r echo =F} --> –&gt; <!-- <!-- ## make the above values to matrices --> –&gt; <!-- <!-- df_plot2$pH <- (6.5 / (1+ exp(1*(0.07*(df_plot2$H_mM+125))-10))) + 2.5 --> –&gt; <!-- <!-- mat_NO2 <- dcast(df_plot2, pH ~ Time_hours, value.var = "NO2_mM") --> –&gt; <!-- <!-- mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "pH") --> –&gt; <!-- <!-- mat_NO2 <- as.matrix(mat_NO2)  --> –&gt; <!-- <!-- class(mat_NO2) --> –&gt; <!-- <!-- dim(mat_NO2) --> –&gt; <!-- <!-- mat_NO2[is.na(mat_NO2)] --> –&gt; <!-- <!-- mat_NO2 <- na_to_average(mat_NO2) --> –&gt; <!-- <!-- mat_NO2[is.na(mat_NO2)] # no more NA --> –&gt;</p>
<!-- <!-- mat_NO3 <- dcast(df_plot2, pH ~ Time_hours, value.var = "NO3_mM") -->
<p>–&gt; <!-- <!-- mat_NO3 <- tibble::column_to_rownames(mat_NO3, var = "pH") --> –&gt; <!-- <!-- mat_NO3 <- as.matrix(mat_NO3)  --> –&gt; <!-- <!-- class(mat_NO3) --> –&gt; <!-- <!-- dim(mat_NO3) --> –&gt; <!-- <!-- mat_NO3[is.na(mat_NO3)] --> –&gt; <!-- <!-- mat_NO3 <- na_to_average(mat_NO3) --> –&gt; <!-- <!-- mat_NO3[is.na(mat_NO3)] # no more NA --> –&gt;</p>
<!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<p>–&gt; <!-- <!-- any(is.na(mat_NO3)) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO3) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- df_out$pH <- rownames(df_out) --> –&gt; <!-- <!-- df_out$pH <- as.numeric(df_out$pH) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrate dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrate dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), --> –&gt; <!-- <!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- (2) (using inferred pH) PCA with only nitrite dynamics -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) --> –&gt; <!-- <!-- any(is.na(mat_NO2)) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO2) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- df_out$pH <- rownames(df_out) --> –&gt; <!-- <!-- df_out$pH <- as.numeric(df_out$pH) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), --> –&gt; <!-- <!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- (3) (using inferred pH) PCA with both nitrate and nitrite dynamics -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) --> –&gt; <!-- <!-- dim(mat_NO2) --> –&gt; <!-- <!-- dim(mat_NO3) --> –&gt; <!-- <!-- mat_NO23 <- cbind(mat_NO3, mat_NO2) --> –&gt; <!-- <!-- any(is.na(mat_NO23)) --> –&gt; <!-- <!-- dim(mat_NO23) --> –&gt;</p>
<!-- <!-- # PCA with prcomp -->
<p>–&gt; <!-- <!-- pc <- prcomp(mat_NO23) --> –&gt; <!-- <!-- eigs <- pc$sdev^2 --> –&gt; <!-- <!-- # variance explained --> –&gt; <!-- <!-- eigs[1] / sum(eigs) # pc1 --> –&gt;</p>
<!-- <!-- eigs[2] / sum(eigs) # pc2 -->
<p>–&gt;</p>
<!-- <!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<p>–&gt; <!-- <!-- df_out <- as.data.frame(pc$x) --> –&gt; <!-- <!-- df_out --> –&gt;</p>
<!-- <!-- head(df_out) -->
<p>–&gt; <!-- <!-- df_out$pH <- rownames(df_out) --> –&gt; <!-- <!-- df_out$pH <- as.numeric(df_out$pH) --> –&gt;</p>
<!-- <!-- # plot PCA with timepoint labels -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2))+ --> –&gt; <!-- <!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Label with mM -->
<p>–&gt; <!-- <!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ --> –&gt; <!-- <!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ --> –&gt; <!-- <!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + --> –&gt; <!-- <!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n") + --> –&gt; <!-- <!--   ggrepel::geom_label_repel() + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<p>–&gt; <!-- <!-- library("plot3D") --> –&gt; <!-- <!-- rownames(df_out) <- as.numeric(rownames(df_out)) --> –&gt; <!-- <!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), --> –&gt; <!-- <!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", --> –&gt; <!-- <!--           ylab ="PC2", zlab = "PC3", --> –&gt; <!-- <!--           theta = 15, phi = 50) --> –&gt; <!-- <!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- Let's take a closer look at the basic pH curves -->
<p>–&gt; <!-- <!-- ```{r echo=F} --> –&gt; <!-- <!-- df_plot3 <- df_plot2 --> –&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -70, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (H_mM > -70) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## 60 -->
<p>–&gt;</p>
<!-- <!-- df_plot3 <- df_plot2 -->
<p>–&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -60, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (H_mM > -60) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## 50 -->
<p>–&gt; <!-- <!-- df_plot3 <- df_plot2 --> –&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -50, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (H_mM > -50) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## -->
<p>–&gt;</p>
<!-- <!-- df_plot3 <- df_plot2 -->
<p>–&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH < 8.9, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (inferred pH > 8.9) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## 3 clusters - first cluster -->
<p>–&gt; <!-- <!-- df_plot3 <- df_plot2 --> –&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH < 8.9, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (inferred pH > 8.9) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## 3 clusters - second cluster -->
<p>–&gt; <!-- <!-- df_plot3 <- df_plot2 --> –&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH > 8.9 | df_plot3$pH < 4, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (4 < inferred pH < 8.9) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## 3 clusters - third cluster -->
<p>–&gt; <!-- <!-- df_plot3 <- df_plot2 --> –&gt; <!-- <!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH > 4, NA, df_plot3$NO3_mM) --> –&gt;</p>
<!-- <!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.2)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = rev(col_pH(100))) + --> –&gt; <!-- <!--   # scale_color_manual(values=grad_pH) + --> –&gt; <!-- <!--   ylab("NO3- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   scale_y_continuous(breaks = seq(0,2.5,0.25), limits=c(0, 2.5))+ --> –&gt; <!-- <!--   ggtitle("pH perturbation (inferred pH < 4) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   facet_grid(. ~ Soil) + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 4.1.2. Peak time -->
<p>–&gt; <!-- <!-- ```{r echo=F} --> –&gt; <!-- <!-- # maximum peak time --> –&gt; <!-- <!-- vec_peaktime <- apply(mat_H_time, 1, which.max) --> –&gt; <!-- <!-- vec_hr <- df_time$Time_hours --> –&gt;</p>
<!-- <!-- vec_hr[vec_peaktime] -->
<p>–&gt; <!-- <!-- names(vec_peaktime) --> –&gt;</p>
<!-- <!-- plot(names(vec_peaktime), vec_hr[vec_peaktime]) -->
<p>–&gt;</p>
<!-- <!-- df_peak_NO2 <- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime]) -->
<p>–&gt; <!-- <!-- df_peak_NO2$H_mM <- as.numeric(df_peak_NO2$H_mM) --> –&gt;</p>
<!-- <!-- # 1. linear regression -->
<p>–&gt; <!-- <!-- fit.no2peak <- lm(Time_hours ~ H_mM, df_peak_NO2) --> –&gt; <!-- <!-- summary(fit.no2peak) --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x", ",  R^2: ", round(summary(fit.no2peak)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 2. quadratic regression -->
<p>–&gt;</p>
<!-- <!-- df_peak_NO2$H_mM_2 <- (df_peak_NO2$H_mM)^2  -->
<p>–&gt; <!-- <!-- quadraticModel <- lm(Time_hours ~ H_mM + H_mM_2, df_peak_NO2) --> –&gt; <!-- <!-- summary(quadraticModel) --> –&gt;</p>
<!-- <!-- perturbValues <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- peakPredict <- predict(quadraticModel,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) --> –&gt; <!-- <!-- df_quad <- data.frame(H_mM = perturbValues, Time_hours = peakPredict) --> –&gt; <!-- <!-- plot(perturbValues, peakPredict) --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               # color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + --> –&gt; <!-- <!--   geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = "blue", size = 1) + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               # color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   annotate("text",x=0,y=30, label= paste0("y = ", round(coef(quadraticModel)[[1]],3), "+", round(coef(quadraticModel)[[2]],3),"x+",round(coef(quadraticModel)[[3]],3), "x^2", ",  R^2: ", round(summary(quadraticModel)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 4.1.3. Get rid of noise by peak threshold -->
<p>–&gt; <!-- <!-- ```{r echo=F} --> –&gt; <!-- <!-- # maximum peak value --> –&gt; <!-- <!-- apply(mat_H_time, 1, max) %>% length() --> –&gt; <!-- <!-- vec_max <- apply(mat_H_time, 1, max) --> –&gt; <!-- <!-- hist(vec_max, breaks = 50, xlab = "NO2 mM at peak", main = "Histogram of peak NO2 concentration") --> –&gt; <!-- <!-- sort(vec_max) --> –&gt; <!-- <!-- min_peak_NO2_mM <- 0.05 --> –&gt;</p>
<!-- <!-- vec_max[vec_max > min_peak_NO2_mM] -->
<p>–&gt; <!-- <!-- # maximum peak time --> –&gt; <!-- <!-- vec_peaktime <- apply(mat_H_time, 1, which.max) --> –&gt; <!-- <!-- vec_hr <- df_time$Time_hours --> –&gt;</p>
<!-- <!-- vec_hr[vec_peaktime] -->
<p>–&gt; <!-- <!-- names(vec_peaktime) --> –&gt;</p>
<!-- <!-- plot(names(vec_peaktime), vec_hr[vec_peaktime]) -->
<p>–&gt;</p>
<!-- <!-- df_peak_NO2 <- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime]) -->
<p>–&gt; <!-- <!-- df_peak_NO2$H_mM <- as.numeric(df_peak_NO2$H_mM) --> –&gt;</p>
<!-- <!-- df_peak_NO2_threshold <- df_peak_NO2[vec_max > min_peak_NO2_mM, ] -->
<p>–&gt; <!-- <!-- dim(df_peak_NO2_threshold) --> –&gt; <!-- <!-- df_threshold <- na.omit(df_peak_NO2_threshold) --> –&gt; <!-- <!-- dim(df_threshold) --> –&gt;</p>
<!-- <!-- # 1. linear regression -->
<p>–&gt; <!-- <!-- fit.no2peak <- lm(Time_hours ~ H_mM, df_threshold) --> –&gt; <!-- <!-- summary(fit.no2peak) --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   annotate("text",x=-40,y=40, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x", ",  R^2: ", round(summary(fit.no2peak)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # 2. quadratic regression -->
<p>–&gt;</p>
<!-- <!-- df_threshold$H_mM_2 <- (df_threshold$H_mM)^2  -->
<p>–&gt; <!-- <!-- quadraticModel_th <- lm(Time_hours ~ H_mM + H_mM_2, df_threshold) --> –&gt; <!-- <!-- summary(quadraticModel_th) --> –&gt;</p>
<!-- <!-- perturbValues <- seq(-100, 10, 0.1) -->
<p>–&gt; <!-- <!-- peakPredict <- predict(quadraticModel_th,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) --> –&gt; <!-- <!-- df_quad <- data.frame(H_mM = perturbValues, Time_hours = peakPredict) --> –&gt; <!-- <!-- plot(perturbValues, peakPredict) --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               # color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- # Plot H_mM with peak time -->
<p>–&gt; <!-- <!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "blue") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n NO2 (Nitrite) peak time (hr)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + --> –&gt; <!-- <!--   geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = "blue", size = 1) + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], --> –&gt; <!-- <!--               # color = "maroon2") + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + --> –&gt; <!-- <!--   annotate("text",x=-45,y=40, label= paste0("y = ", round(coef(quadraticModel_th)[[1]],3), "+", round(coef(quadraticModel_th)[[2]],3),"x+",round(coef(quadraticModel_th)[[3]],3), "x^2", ",  R^2: ", round(summary(quadraticModel_th)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt; <!-- <!-- ``` --> –&gt;</p>
<!-- <!-- ## 4.2. Area under curve calculation -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- df_plot2$Time_hours --> –&gt; <!-- <!-- require(pracma) --> –&gt; <!-- <!-- colnames(df_plot2) --> –&gt; <!-- <!-- df_corr <- df_plot2 %>% select(H_mM, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM) %>% summarize(auc = trapz(Time_hours, NO3_mM)) %>% ungroup() --> –&gt;</p>
<!-- <!-- plot(df_corr$H_mM, df_corr$auc) -->
<p>–&gt;</p>
<!-- <!-- df_corr$H_mM_2 <- (df_corr$H_mM)^2 -->
<p>–&gt; <!-- <!-- fit.no3 <- lm(auc ~ H_mM + H_mM_2, df_corr) --> –&gt; <!-- <!-- summary(fit.no3) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aucPredict <- predict(fit.no3,list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_auc_quad <- data.frame(H_mM = perturbH, Time_hours = aucPredict) --> –&gt; <!-- <!-- plot(perturbH, aucPredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted linear regression line -->
<p>–&gt; <!-- <!-- ggplot(df_corr, aes(x=H_mM, y=auc)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Area under curve (NO3-)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and area under nitrate curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=0,y=100, label= paste0("y = ", round(coef(fit.no3)[[1]],3), "+", round(coef(fit.no3)[[2]],3),"x+",round(coef(fit.no3)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no3)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ## 220213 Use the inferred pH values for x axis -->
<p>–&gt; <!-- <!-- # y <- (6.5 / (1+ exp(1*(0.07*(x+125))-10))) + 2.5 --> –&gt; <!-- <!-- df_corr$inferred_pH <-  (6.5 / (1+ exp(1*(0.07*(df_corr$H_mM+125))-10))) + 2.5 --> –&gt;</p>
<!-- <!-- plot(df_corr$inferred_pH, df_corr$auc) -->
<p>–&gt;</p>
<!-- <!-- df_corr$inferred_pH_2 <- (df_corr$inferred_pH)^2 -->
<p>–&gt; <!-- <!-- fit.no3 <- lm(auc ~ inferred_pH + inferred_pH_2, df_corr) --> –&gt; <!-- <!-- summary(fit.no3) --> –&gt;</p>
<!-- <!-- perturbpH <- seq(2, 11, 0.01) -->
<p>–&gt; <!-- <!-- aucPredict <- predict(fit.no3,list(inferred_pH=perturbpH, inferred_pH_2=perturbpH^2)) --> –&gt; <!-- <!-- df_auc_quad <- data.frame(H_mM = perturbpH, auc = aucPredict) --> –&gt; <!-- <!-- plot(perturbpH, aucPredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted linear regression line -->
<p>–&gt; <!-- <!-- ggplot(df_corr, aes(x=inferred_pH, y=auc)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("inferred pH \n") + --> –&gt; <!-- <!--   ylab("\n Area under curve (NO3-)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbed pH and area under nitrate curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = auc), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=7,y=100, label= paste0("y = ", round(coef(fit.no3)[[1]],3), "+", round(coef(fit.no3)[[2]],3),"x+",round(coef(fit.no3)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no3)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ## NO2 area under curve -->
<p>–&gt; <!-- <!-- colnames(df_plot2) --> –&gt; <!-- <!-- df_corr_no2 <- df_plot2 %>% select(H_mM, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM) %>% summarize(auc = trapz(Time_hours, NO2_mM)) %>% ungroup() --> –&gt;</p>
<!-- <!-- plot(df_corr_no2$H_mM, df_corr_no2$auc) -->
<p>–&gt;</p>
<!-- <!-- fit.no2 <- lm(auc ~ H_mM, df_corr_no2) -->
<p>–&gt; <!-- <!-- summary(fit.no2) --> –&gt;</p>
<!-- <!-- df_corr_no2$H_mM_2 <- (df_corr_no2$H_mM)^2  -->
<p>–&gt; <!-- <!-- fit.no2.quad <- lm(auc ~ H_mM + H_mM_2, df_corr_no2) --> –&gt; <!-- <!-- summary(fit.no2.quad) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aucPredict <- predict(fit.no2.quad,list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_auc_quad <- data.frame(H_mM = perturbH, Time_hours = aucPredict) --> –&gt; <!-- <!-- plot(perturbH, aucPredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted linear regression line -->
<p>–&gt; <!-- <!-- ggplot(df_corr_no2, aes(x=H_mM, y=auc)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=21, color = "brown") + --> –&gt; <!-- <!--   # geom_line(size=1)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Area under curve (NO2-)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and area under nitrite curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=0,y=20, label= paste0("y = ", round(coef(fit.no2.quad)[[1]],3), "+", round(coef(fit.no2.quad)[[2]],3),"x+",round(coef(fit.no2.quad)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no2.quad)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 4.3. 1/AUC -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- df_plot2$Time_hours --> –&gt; <!-- <!-- require(pracma) --> –&gt; <!-- <!-- colnames(df_plot2) --> –&gt; <!-- <!-- dim(df_plot2) --> –&gt; <!-- <!-- head(df_plot2) --> –&gt; <!-- <!-- df_corr <- df_plot2 %>% select(H_mM, Soil, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM, Soil) %>% summarize(auc = trapz(Time_hours, NO3_mM)) %>% ungroup() --> –&gt; <!-- <!-- df_corr$auc_inv <- 1/(df_corr$auc) --> –&gt; <!-- <!-- plot(df_corr$H_mM, df_corr$auc_inv) --> –&gt;</p>
<!-- <!-- # remove the outlier 1 point -->
<p>–&gt; <!-- <!-- df_corr %>% filter(auc_inv>1) --> –&gt;</p>
<!-- <!-- dim(df_corr) # 94 -->
<p>–&gt; <!-- <!-- df_corr %<>% filter(!(auc_inv>1)) --> –&gt; <!-- <!-- dim(df_corr) # 93 --> –&gt;</p>
<!-- <!-- # (1) Plot fitted linear regression line -->
<p>–&gt; <!-- <!-- ggplot(df_corr, aes(x=H_mM, y=auc_inv, color=Soil, group=Soil)) + --> –&gt; <!-- <!--   geom_point(size=3, shape=21) + --> –&gt; <!-- <!--   geom_line() + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n 1/AUC of Nitrate dynamics") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Unimodal distribution of 3 soils \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ggplot(df_corr, aes(x=H_mM, y=auc_inv, color=Soil, group=Soil)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n 1/AUC of Nitrate dynamics") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Unimodal distribution of 3 soils \n") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<p>–&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5. Get Ir dynamics (Reduction rate of the nitrite) -->
<p>–&gt;</p>
<!-- <!-- ## 5.1. I p -->
<p>–&gt; <!-- <!-- ```{r echo=F} --> –&gt;</p>
<!-- <!-- df_plot2 -->
<p>–&gt;</p>
<!-- <!-- # Nitrite produced I_produced = A - A(t) -->
<p>–&gt; <!-- <!-- df_plot2 %>% filter(Time_hours == 0) --> –&gt; <!-- <!-- df_plot2 %>% group_by(H_mM, Soil, Time_hours) --> –&gt;</p>
<!-- <!-- mat_NO3 <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO3_mM") -->
<p>–&gt; <!-- <!-- mat_NO3 <- tibble::column_to_rownames(mat_NO3, var = "H_mM") --> –&gt; <!-- <!-- mat_NO3 <- as.matrix(mat_NO3)  --> –&gt; <!-- <!-- class(mat_NO3) --> –&gt; <!-- <!-- dim(mat_NO3) --> –&gt;</p>
<!-- <!-- # NA -->
<p>–&gt; <!-- <!-- mat_NO3 <- na_to_average(mat_NO3) --> –&gt; <!-- <!-- mat_NO3[is.na(mat_NO3)] # no more NA --> –&gt;</p>
<!-- <!-- mat_NO3[,2:dim(mat_NO3)[2]] -->
<p>–&gt;</p>
<!-- <!-- mat_NO3[,1] - mat_NO3[,1] -->
<p>–&gt;</p>
<!-- <!-- mat_NO3_consumed <- mat_NO3 -->
<p>–&gt; <!-- <!-- for (j in 1:dim(mat_NO3_consumed)[2]){ --> –&gt; <!-- <!--   mat_NO3_consumed[,j] = mat_NO3[,1] - mat_NO3_consumed[,j] --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- mat_NO3_consumed -->
<p>–&gt;</p>
<!-- <!-- # make negative values into 0. -->
<p>–&gt; <!-- <!-- mat_NO3_consumed[mat_NO3_consumed<0] <- 0 --> –&gt; <!-- <!-- mat_NO3_consumed --> –&gt;</p>
<!-- <!-- # This is NO2 produced -->
<p>–&gt; <!-- <!-- mat_NO2_produced = mat_NO3_consumed --> –&gt; <!-- <!-- df_Ip <- melt(mat_NO2_produced)  --> –&gt; <!-- <!-- names(df_Ip) <- c("H_mM","Time_hours","NO2_mM") --> –&gt;</p>
<!-- <!-- # plot I p = Nitrite produced (accumulative) = Nitrate reduced  -->
<p>–&gt; <!-- <!-- ggplot(df_Ip, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_p (Nitrite produced) = A0 - A(t) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5.1. I r -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # plot I r = Nitrite reduced (accumulative) = I p - I --> –&gt; <!-- <!-- mat_NO2_produced[is.na(mat_NO2_produced)]  --> –&gt;</p>
<!-- <!-- mat_NO2_produced -->
<p>–&gt; <!-- <!-- dim(mat_NO2_produced) --> –&gt;</p>
<!-- <!-- mat_NO2 <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO2_mM") -->
<p>–&gt; <!-- <!-- mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") --> –&gt; <!-- <!-- mat_NO2 <- as.matrix(mat_NO2)  --> –&gt; <!-- <!-- class(mat_NO2) --> –&gt; <!-- <!-- dim(mat_NO2) --> –&gt;</p>
<!-- <!-- # Substitute NA to average of NO2_mM of before and after. -->
<p>–&gt; <!-- <!-- mat_NO2[is.na(mat_NO2)]  --> –&gt; <!-- <!-- for (i in 1:dim(mat_NO2)[1]){ --> –&gt; <!-- <!--   # print(i) --> –&gt; <!-- <!--   for (j in 1:dim(mat_NO2)[2]){ --> –&gt; <!-- <!--     # print(j) --> –&gt; <!-- <!--     if (is.na(mat_NO2[i,j])){ --> –&gt; <!-- <!--       mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 --> –&gt; <!-- <!--     } --> –&gt; <!-- <!--   } --> –&gt; <!-- <!-- } --> –&gt; <!-- <!-- mat_NO2[is.na(mat_NO2)] # no more NA --> –&gt;</p>
<!-- <!-- mat_Ir <- mat_NO2_produced - mat_NO2 -->
<p>–&gt;</p>
<!-- <!-- df_Ir <- melt(mat_Ir)  -->
<p>–&gt; <!-- <!-- names(df_Ir) <- c("H_mM","Time_hours","NO2_mM") --> –&gt;</p>
<!-- <!-- # plot I r = reduced Nitrite (accumulative) -->
<p>–&gt; <!-- <!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5.1.1. AAC (area above curve) -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- df_Ir$Time_hours --> –&gt; <!-- <!-- require(pracma) --> –&gt; <!-- <!-- colnames(df_Ir) --> –&gt;</p>
<!-- <!-- # Let's get the area above curve -->
<p>–&gt; <!-- <!-- # what is the total area? --> –&gt; <!-- <!-- time_max = max(df_Ir$Time_hours) --> –&gt; <!-- <!-- upper_limit_mean = mean(df_Ir %>% filter(Time_hours == time_max) %>% filter(NO2_mM > 1.8) %>% select(NO2_mM) %>% unlist()) --> –&gt; <!-- <!-- total_area = time_max * upper_limit_mean --> –&gt;</p>
<!-- <!-- # correlation -->
<p>–&gt;</p>
<!-- <!-- df_corr_aac <- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(aac = total_area - trapz(Time_hours, NO2_mM)) %>% ungroup() -->
<p>–&gt;</p>
<!-- <!-- is.na(df_corr_aac) ## no NA -->
<p>–&gt;</p>
<!-- <!-- plot(df_corr$H_mM, df_corr_aac$aac) -->
<p>–&gt;</p>
<!-- <!-- df_corr_aac$H_mM_2 <- (df_corr_aac$H_mM)^2 -->
<p>–&gt; <!-- <!-- fit.aac <- lm(aac ~ H_mM + H_mM_2, df_corr_aac) --> –&gt; <!-- <!-- summary(fit.aac) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aacPredict <- predict(fit.aac,list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_aac_quad <- data.frame(H_mM = perturbH, Time_hours = aacPredict) --> –&gt; <!-- <!-- plot(perturbH, aacPredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted quadratic regression line -->
<p>–&gt; <!-- <!-- ggplot(df_corr_aac, aes(x=H_mM, y=aac)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Area above I_r curve (NO2-)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and area under nitrate curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_aac_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=0,y=100, label= paste0("y = ", round(coef(fit.aac)[[1]],3), "+", round(coef(fit.aac)[[2]],3),"x+",round(coef(fit.aac)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aac)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5.1.2. Average slope (Max NO2_mM / Time) -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- df_Ir_av_slope <- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(average_slope = max(NO2_mM) trapz(Time_hours, NO2_mM)) %>% ungroup() --> –&gt;</p>
<!-- <!-- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(max(NO2_mM)) -->
<p>–&gt;</p>
<!-- <!-- ## get peaks -->
<p>–&gt; <!-- <!-- # https://stackoverflow.com/questions/55342232/how-to-find-local-maximum-in-r-from-graph  --> –&gt; <!-- <!-- # plot I r = reduced Nitrite (accumulative) --> –&gt; <!-- <!-- library(ggpmisc) --> –&gt; <!-- <!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   stat_peaks(col = "green", size = 4) + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## plot in a different way -->
<p>–&gt; <!-- <!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 15 & H_mM > -15) --> –&gt; <!-- <!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # stat_peaks(col = "green", size = 4) + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## plot in a different way -->
<p>–&gt; <!-- <!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 30 & H_mM > -30) --> –&gt; <!-- <!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # stat_peaks(col = "green", size = 4) + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## plot in a different way -->
<p>–&gt; <!-- <!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 50 & H_mM >-50) --> –&gt; <!-- <!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # stat_peaks(col = "green", size = 4) + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## plot in a different way -->
<p>–&gt; <!-- <!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 70 & H_mM > -70) --> –&gt; <!-- <!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16) + --> –&gt; <!-- <!--   geom_line(size=1.5)+ --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ --> –&gt; <!-- <!--   scale_colour_gradientn(colours = col_pH(100)) + --> –&gt; <!-- <!--   # scale_color_manual(values=col_pH(4)) + --> –&gt; <!-- <!--   ylab("NO2- (mM) \n") + --> –&gt; <!-- <!--   xlab("\n Time (hr)") + --> –&gt; <!-- <!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + --> –&gt; <!-- <!--   # stat_peaks(col = "green", size = 4) + --> –&gt; <!-- <!--   # label --> –&gt; <!-- <!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   mytheme_2d + --> –&gt; <!-- <!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) --> –&gt;</p>
<!-- <!-- ## get peaks coordinates -->
<p>–&gt; <!-- <!-- mat_Ir --> –&gt; <!-- <!-- x = colnames(mat_Ir) --> –&gt; <!-- <!-- y = mat_Ir[1,] --> –&gt; <!-- <!-- x[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)] --> –&gt; <!-- <!-- y[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)] --> –&gt;</p>
<!-- <!-- plot(x, y) -->
<p>–&gt;</p>
<!-- <!-- for (i in 1:dim(mat_Ir)[1]){ -->
<p>–&gt; <!-- <!--   print(i) --> –&gt; <!-- <!--   x = colnames(mat_Ir) --> –&gt; <!-- <!--   y = mat_Ir[i,] --> –&gt; <!-- <!--   x_max = x[ggpmisc:::find_peaks(y, span = 3)] --> –&gt; <!-- <!--   y_max = y[ggpmisc:::find_peaks(y, span = 3)] --> –&gt; <!-- <!--   # print(x_max) --> –&gt; <!-- <!--   print(y_max) --> –&gt; <!-- <!--   plot(x, y) --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- ## Max -->
<p>–&gt; <!-- <!-- vec_H_mM <- rownames(mat_Ir) --> –&gt; <!-- <!-- vec_max_NO_mM <- rep(-1, dim(mat_Ir)[1]) --> –&gt; <!-- <!-- vec_max_time <- rep(-1, dim(mat_Ir)[1]) --> –&gt;</p>
<!-- <!-- for (i in 1:dim(mat_Ir)[1]){ -->
<p>–&gt; <!-- <!--   y = mat_Ir[i,] --> –&gt; <!-- <!--   max_j = max(y) --> –&gt; <!-- <!--   # save --> –&gt; <!-- <!--   vec_max_NO_mM[i] <- max_j --> –&gt; <!-- <!--   vec_max_time[i] <- which.max(y) --> –&gt; <!-- <!--   iter_j <- which.max(y) -1 --> –&gt; <!-- <!--   # get the more early points if the difference with max is not big (10%) --> –&gt; <!-- <!--   for (k in 1:iter_j){ --> –&gt; <!-- <!--     print(k) --> –&gt; <!-- <!--     if (abs(max_j - y[which.max(y)-k])/max_j < 0.1){ --> –&gt; <!-- <!--       max_j = y[which.max(y)-k] --> –&gt; <!-- <!--       # print(max_j) --> –&gt; <!-- <!--       vec_max_NO_mM[i] <- max_j --> –&gt; <!-- <!--       print(names(max_j)) --> –&gt; <!-- <!--       vec_max_time[i] <- names(max_j) --> –&gt; <!-- <!--     } else{ break } --> –&gt; <!-- <!--   } --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- df_ave_slope <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time) -->
<p>–&gt;</p>
<!-- <!-- # check -->
<p>–&gt; <!-- <!-- i = 3 --> –&gt; <!-- <!-- plot(colnames(mat_Ir), mat_Ir[i,]) --> –&gt; <!-- <!-- points(vec_max_time[i], vec_max_NO_mM[i], col='red') --> –&gt;</p>
<!-- <!-- for (i in 1:dim(mat_Ir)[1]){ -->
<p>–&gt; <!-- <!--   plot(colnames(mat_Ir), mat_Ir[i,]) --> –&gt; <!-- <!--   points(vec_max_time[i], vec_max_NO_mM[i], col='red') --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- # get average slope -->
<p>–&gt; <!-- <!-- df_ave_slope$H_mM <- as.numeric(df_ave_slope$H_mM) --> –&gt; <!-- <!-- df_ave_slope$average_slope <- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours) --> –&gt;</p>
<!-- <!-- # plot -->
<p>–&gt; <!-- <!-- plot(vec_H_mM, df_ave_slope$average_slope) --> –&gt;</p>
<!-- <!-- # quadratic fit -->
<p>–&gt; <!-- <!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 --> –&gt; <!-- <!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) --> –&gt; <!-- <!-- summary(fit.aveslope) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) --> –&gt; <!-- <!-- plot(perturbH, aveslopePredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted quadratic regression line -->
<p>–&gt; <!-- <!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ## restrict the samples with maximum over 1.5 mM NO2 -->
<p>–&gt; <!-- <!-- df_ave_slope %<>% filter(NO2_mM > 1.5) --> –&gt;</p>
<!-- <!-- # quadratic fit -->
<p>–&gt; <!-- <!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 --> –&gt; <!-- <!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) --> –&gt; <!-- <!-- summary(fit.aveslope) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) --> –&gt; <!-- <!-- plot(perturbH, aveslopePredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted quadratic regression line -->
<p>–&gt; <!-- <!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   # geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   # annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5.1.3. Got a new idea. Start from the end -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- ## Max --> –&gt; <!-- <!-- vec_H_mM <- rownames(mat_Ir) --> –&gt; <!-- <!-- vec_max_NO_mM <- rep(-1, dim(mat_Ir)[1]) --> –&gt; <!-- <!-- vec_max_time <- rep(-1, dim(mat_Ir)[1]) --> –&gt;</p>
<!-- <!-- for (i in 1:dim(mat_Ir)[1]){ -->
<p>–&gt; <!-- <!--   x = colnames(mat_Ir) --> –&gt; <!-- <!--   y = mat_Ir[i,] --> –&gt; <!-- <!--   max_j = y[dim(mat_Ir)[2]] --> –&gt; <!-- <!--   # save --> –&gt; <!-- <!--   vec_max_NO_mM[i] <- max_j --> –&gt; <!-- <!--   vec_max_time[i] <- x[dim(mat_Ir)[2]] --> –&gt; <!-- <!--   iter_j <- dim(mat_Ir)[2] -1 --> –&gt; <!-- <!--   # get the more early points if the difference with max is not big (10%) --> –&gt; <!-- <!--   for (k in 1:iter_j){ --> –&gt; <!-- <!--     print(k) --> –&gt; <!-- <!--     if ( (y[dim(mat_Ir)[2]-k] > max_j) || (abs(max_j - y[dim(mat_Ir)[2]-k])/max_j < 0.1) ){ --> –&gt; <!-- <!--       max_j = y[dim(mat_Ir)[2] - k] --> –&gt; <!-- <!--       print(max_j) --> –&gt; <!-- <!--       vec_max_NO_mM[i] <- max_j --> –&gt; <!-- <!--       print(names(max_j)) --> –&gt; <!-- <!--       vec_max_time[i] <- names(max_j) --> –&gt; <!-- <!--     } else{ break } --> –&gt; <!-- <!--   } --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- df_ave_slope <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time) -->
<p>–&gt;</p>
<!-- <!-- # check -->
<p>–&gt; <!-- <!-- i = 36 --> –&gt; <!-- <!-- plot(colnames(mat_Ir), mat_Ir[i,]) --> –&gt; <!-- <!-- points(vec_max_time[i], vec_max_NO_mM[i], col='red') --> –&gt;</p>
<!-- <!-- for (i in 1:dim(mat_Ir)[1]){ -->
<p>–&gt; <!-- <!--   plot(colnames(mat_Ir), mat_Ir[i,]) --> –&gt; <!-- <!--   points(vec_max_time[i], vec_max_NO_mM[i], col='red') --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- # get average slope -->
<p>–&gt; <!-- <!-- df_ave_slope$H_mM <- as.numeric(df_ave_slope$H_mM) --> –&gt; <!-- <!-- df_ave_slope$average_slope <- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours) --> –&gt;</p>
<!-- <!-- # plot -->
<p>–&gt; <!-- <!-- plot(vec_H_mM, df_ave_slope$average_slope) --> –&gt;</p>
<!-- <!-- # quadratic fit -->
<p>–&gt; <!-- <!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 --> –&gt; <!-- <!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) --> –&gt; <!-- <!-- summary(fit.aveslope) --> –&gt;</p>
<!-- <!-- perturbH <- seq(-100, 100, 0.1) -->
<p>–&gt; <!-- <!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) --> –&gt; <!-- <!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) --> –&gt; <!-- <!-- plot(perturbH, aveslopePredict) --> –&gt;</p>
<!-- <!-- # (1) Plot fitted quadratic regression line -->
<p>–&gt; <!-- <!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + --> –&gt; <!-- <!--   geom_point(size=2.5, shape=16, color = "brown") + --> –&gt; <!-- <!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + --> –&gt; <!-- <!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ --> –&gt; <!-- <!--   # scale_color_brewer(palette='Set2') + --> –&gt; <!-- <!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ --> –&gt; <!-- <!--   xlab("Amount of pH perturbation (H+ mM) \n") + --> –&gt; <!-- <!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + --> –&gt; <!-- <!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ --> –&gt; <!-- <!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ --> –&gt; <!-- <!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ --> –&gt; <!-- <!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + --> –&gt; <!-- <!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ --> –&gt; <!-- <!--   # regression line --> –&gt; <!-- <!--   geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + --> –&gt; <!-- <!--   # show equation --> –&gt; <!-- <!--   annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + --> –&gt; <!-- <!--   mytheme_2d --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 5.1.4. T max/2 -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # Michaelis-Menten function (https://rpubs.com/dstoebel/576374) --> –&gt; <!-- <!-- y = mat_Ir[i,][2:dim(mat_Ir)[2]] --> –&gt; <!-- <!-- x = as.numeric(names(y)) --> –&gt; <!-- <!-- mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  --> –&gt; <!-- <!-- coefficients(mm_fit)[1] --> –&gt; <!-- <!-- coefficients(mm_fit)[2] # Km --> –&gt; <!-- <!-- plot(x, y) --> –&gt; <!-- <!-- curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE) --> –&gt;</p>
<!-- <!-- i=2 -->
<p>–&gt; <!-- <!-- y = mat_Ir[i,][2:dim(mat_Ir)[2]] --> –&gt; <!-- <!-- x = as.numeric(names(y)) --> –&gt; <!-- <!-- mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  --> –&gt; <!-- <!-- coefficients(mm_fit)[1] --> –&gt; <!-- <!-- coefficients(mm_fit)[2] # Km --> –&gt; <!-- <!-- plot(x, y) --> –&gt; <!-- <!-- curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE) --> –&gt;</p>
<!-- <!-- # this failed -->
<p>–&gt; <!-- <!-- # get T max/2 --> –&gt; <!-- <!-- vec_T_0.5max <- rep(-1, dim(mat_Ir)[1]) --> –&gt; <!-- <!-- for (i in 1:dim(mat_Ir)[1]){ --> –&gt; <!-- <!--   print(i) --> –&gt; <!-- <!--   y = mat_Ir[i,][2:dim(mat_Ir)[2]] --> –&gt; <!-- <!--   x = as.numeric(names(y)) --> –&gt; <!-- <!--   mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  --> –&gt; <!-- <!--   # T max/2 --> –&gt; <!-- <!--   vec_T_0.5max[i] <- coefficients(mm_fit)[2]  --> –&gt; <!-- <!-- } --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 6. Correlation between mu -->
<p>–&gt;</p>
<!-- <!-- ```{r echo=F} -->
<p>–&gt; <!-- <!-- # From A --> –&gt; <!-- <!-- df_corr$auc # area under curve --> –&gt;</p>
<!-- <!-- # From I -->
<p>–&gt; <!-- <!-- vec_peakheight # peak height --> –&gt; <!-- <!-- df_peak_NO2$Time_hours # peak time --> –&gt; <!-- <!-- df_corr_no2$auc # AUC --> –&gt;</p>
<!-- <!-- # From I_R -->
<p>–&gt; <!-- <!-- df_corr_aac$aac # area above curve --> –&gt; <!-- <!-- df_ave_slope$average_slope # average slope --> –&gt;</p>
<!-- <!-- # wrap it up -->
<p>–&gt; <!-- <!-- df_mu <- data.frame(A_AUC = df_corr$auc, --> –&gt; <!-- <!--                     I_peak_height = vec_peakheight, --> –&gt; <!-- <!--                     I_peak_time = df_peak_NO2$Time_hours, --> –&gt; <!-- <!--                     I_AUC = df_corr_no2$auc, --> –&gt; <!-- <!--                     I_R_AAC = df_corr_aac$aac, --> –&gt; <!-- <!--                     I_R_average_slope = df_ave_slope$average_slope --> –&gt; <!-- <!--                     ) --> –&gt;</p>
<!-- <!-- pairs(df_mu, main = "mu space correlation", -->
<p>–&gt; <!-- <!--       pch=21, cex.labels = 2, cex=1.5, cex.main=2, col='blue') --> –&gt;</p>
<!-- <!-- df_mu -->
<p>–&gt;</p>
<!-- <!-- scaled.dat <- scale(df_mu) -->
<p>–&gt; <!-- <!-- pairs(scaled.dat, main = "(scaled) mu space correlation", --> –&gt; <!-- <!--       pch=21, cex.labels = 2, cex=1.5, cex.main=2, col='blue') --> –&gt;</p>
<!-- <!-- ``` -->
<p>–&gt;</p>
<!-- <!-- ## 6.1. Add ammonia -->
<p>–&gt; <!-- <!-- ```{r echo=F} --> –&gt; <!-- <!-- # wrap it up --> –&gt; <!-- <!-- df_mu_amm <- data.frame(A_AUC = df_corr$auc, --> –&gt; <!-- <!--                     I_peak_height = vec_peakheight, --> –&gt; <!-- <!--                     I_peak_time = df_peak_NO2$Time_hours, --> –&gt; <!-- <!--                     I_AUC = df_corr_no2$auc, --> –&gt; <!-- <!--                     I_R_AAC = df_corr_aac$aac, --> –&gt; <!-- <!--                     I_R_average_slope = df_ave_slope$average_slope, --> –&gt; <!-- <!--                     Amm_AUC = am_corr$auc, --> –&gt; <!-- <!--                     Amm_average_slope  = am_ave_slope$average_slope --> –&gt; <!-- <!--                     ) --> –&gt;</p>
<!-- <!-- pairs(df_mu_amm, main = "mu space correlation", -->
<p>–&gt; <!-- <!--       pch=21, cex.labels = 1.7, cex=1.5, cex.main=2, col='blue') --> –&gt; <!-- <!-- ``` --> –&gt;</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gtable_0.3.0       colorRamps_2.3     rmarkdown_2.9      gridExtra_2.3     
 [5] ggpubr_0.4.0       ggrepel_0.9.1      ape_5.5            openxlsx_4.2.3    
 [9] devtools_2.4.0     usethis_2.0.1      gtools_3.8.2       reshape2_1.4.4    
[13] readxl_1.3.1       magrittr_2.0.1     forcats_0.5.1      stringr_1.4.0     
[17] purrr_0.3.4        readr_1.4.0        tidyr_1.1.3        tibble_3.0.4      
[21] tidyverse_1.3.1    vegan_2.5-7        lattice_0.20-41    permute_0.9-5     
[25] RColorBrewer_1.1-2 ggplot2_3.3.5      dplyr_1.0.5        workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0  ggsignif_0.6.2    ellipsis_0.3.2    rio_0.5.27       
 [5] rprojroot_2.0.2   fs_1.5.0          rstudioapi_0.13   farver_2.1.0     
 [9] remotes_2.4.0     fansi_0.4.2       lubridate_1.7.10  xml2_1.3.2       
[13] splines_4.0.3     cachem_1.0.4      knitr_1.37        pkgload_1.2.1    
[17] jsonlite_1.7.2    broom_0.7.9       cluster_2.1.0     dbplyr_2.1.1     
[21] compiler_4.0.3    httr_1.4.2        backports_1.2.1   assertthat_0.2.1 
[25] Matrix_1.2-18     fastmap_1.1.0     cli_3.0.1         later_1.2.0      
[29] htmltools_0.5.1.1 prettyunits_1.1.1 tools_4.0.3       glue_1.4.2       
[33] Rcpp_1.0.8.3      carData_3.0-4     cellranger_1.1.0  jquerylib_0.1.4  
[37] vctrs_0.3.8       nlme_3.1-149      xfun_0.29         ps_1.6.0         
[41] testthat_3.0.2    rvest_1.0.1       lifecycle_1.0.0   rstatix_0.7.0    
[45] MASS_7.3-53       scales_1.1.1      hms_1.1.0         promises_1.2.0.1 
[49] parallel_4.0.3    yaml_2.2.1        curl_4.3.2        memoise_2.0.0    
[53] sass_0.4.0        stringi_1.5.3     highr_0.9         desc_1.3.0       
[57] pkgbuild_1.2.0    zip_2.1.1         rlang_0.4.10      pkgconfig_2.0.3  
[61] evaluate_0.14     labeling_0.4.2    tidyselect_1.1.1  processx_3.5.1   
[65] plyr_1.8.6        R6_2.5.0          generics_0.1.0    DBI_1.1.1        
[69] pillar_1.6.0      haven_2.4.1       whisker_0.4       foreign_0.8-80   
[73] withr_2.4.2       mgcv_1.8-33       abind_1.4-5       modelr_0.1.8     
[77] crayon_1.4.1      car_3.0-11        utf8_1.1.4        data.table_1.14.0
[81] callr_3.7.0       git2r_0.28.0      reprex_2.0.0      digest_0.6.27    
[85] httpuv_1.6.0      munsell_0.5.0     bslib_0.2.5.1     sessioninfo_1.1.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
