<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUchicago" />

<meta name="date" content="2022-05-23" />

<title>220519_Griess_antibiotics_dose_effect</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Denit_visualization_R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/Denit_visualization_R">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">220519_Griess_antibiotics_dose_effect</h1>
<h4 class="author">KiseokUchicago</h4>
<h4 class="date">2022-05-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Denit_visualization_R/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210924code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210924code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210924)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoDenitvisualizationRtree7905c9343b41b0f3a42aba137993e7c632c5b0e2targetblank7905c93a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/Denit_visualization_R/tree/7905c9343b41b0f3a42aba137993e7c632c5b0e2" target="_blank">7905c93</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  211020_df_full_innate.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  220421_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  220519_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx
    Untracked:  220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/120321_soil_buffering_capacity_Nichols-B.xlsx
    Untracked:  data/210911_TOC_sample_plate1.xlsx
    Untracked:  data/210922_Griess_sample_plate1-investigating.xlsx
    Untracked:  data/210922_Griess_sample_plate1.xlsx
    Untracked:  data/210927_TOC_sample_plate1_100ul.xlsx
    Untracked:  data/210927_TOC_sample_plate1_10ul.xlsx
    Untracked:  data/210928_Griess_sample_plate1.xlsx
    Untracked:  data/211005_Griess_sample_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate1.xlsx
    Untracked:  data/211007_Griess_blank_plate2.xlsx
    Untracked:  data/211008_Griess_blank_plate0.xlsx
    Untracked:  data/211008_Griess_blank_plate1.xlsx
    Untracked:  data/211017_Griess_plate0.xlsx
    Untracked:  data/211017_Griess_plate1.xlsx
    Untracked:  data/211017_Griess_plate2.xlsx
    Untracked:  data/211017_Griess_plate3.xlsx
    Untracked:  data/211017_Griess_plate4.xlsx
    Untracked:  data/211017_Griess_plate5.xlsx
    Untracked:  data/211017_Griess_plate6.xlsx
    Untracked:  data/211017_Griess_plate7.xlsx
    Untracked:  data/211017_Griess_plate8.xlsx
    Untracked:  data/211019_Griess_plate1.xlsx
    Untracked:  data/211020_Griess_plate0.xlsx
    Untracked:  data/211020_Griess_plate1.xlsx
    Untracked:  data/211020_df_full_innate.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate0.xlsx
    Untracked:  data/211025_Griess_no3_fit_plate1.xlsx
    Untracked:  data/211025_Griess_plate0.xlsx
    Untracked:  data/211025_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1.xlsx
    Untracked:  data/211028_Griess_plate1_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate2.xlsx
    Untracked:  data/211028_Griess_plate2_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate3.xlsx
    Untracked:  data/211028_Griess_plate3_no3_fit.xlsx
    Untracked:  data/211028_Griess_plate4.xlsx
    Untracked:  data/211028_Griess_plate4_no3_fit.xlsx
    Untracked:  data/211028_time_table.xlsx
    Untracked:  data/211127_Griess_SUP1.xlsx
    Untracked:  data/211127_Griess_SUP11.xlsx
    Untracked:  data/211127_Griess_SUP3.xlsx
    Untracked:  data/211127_Griess_SUP5.xlsx
    Untracked:  data/211127_Griess_SUP7.xlsx
    Untracked:  data/211127_Griess_SUP9.xlsx
    Untracked:  data/211127_Griess_plate0.xlsx
    Untracked:  data/211127_time_table.xlsx
    Untracked:  data/211128_Ammonia_sample_plate10.xlsx
    Untracked:  data/211128_Ammonia_sample_plate11.xlsx
    Untracked:  data/211128_Griess_plate1.xlsx
    Untracked:  data/211128_Griess_plate10.xlsx
    Untracked:  data/211128_Griess_plate11.xlsx
    Untracked:  data/211128_Griess_plate2.xlsx
    Untracked:  data/211128_Griess_plate3.xlsx
    Untracked:  data/211128_Griess_plate4.xlsx
    Untracked:  data/211128_Griess_plate5.xlsx
    Untracked:  data/211128_Griess_plate6.xlsx
    Untracked:  data/211128_Griess_plate7.xlsx
    Untracked:  data/211128_Griess_plate8.xlsx
    Untracked:  data/211128_Griess_plate9.xlsx
    Untracked:  data/211128_time_table.xlsx
    Untracked:  data/211201_pH_colorimetric.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_table.xlsx
    Untracked:  data/211203_slurry_vs_water_drying_time.xlsx
    Untracked:  data/211208_pH_colorimetric_T3.xlsx
    Untracked:  data/211208_pH_colorimetric_T4.xlsx
    Untracked:  data/211208_pH_colorimetric_T5.xlsx
    Untracked:  data/211208_pH_colorimetric_plate1.xlsx
    Untracked:  data/211208_pH_colorimetric_plate2.xlsx
    Untracked:  data/211208_pH_colorimetric_plate3.xlsx
    Untracked:  data/211208_time_table.xlsx
    Untracked:  data/211209_OD600_T1.xlsx
    Untracked:  data/211209_OD600_T2.xlsx
    Untracked:  data/211209_OD600_T3.xlsx
    Untracked:  data/211209_time_table.xlsx
    Untracked:  data/211210_pH_colorimetric_T0.xlsx
    Untracked:  data/211210_pH_colorimetric_T1.xlsx
    Untracked:  data/211210_pH_colorimetric_T2.xlsx
    Untracked:  data/211210_pH_colorimetric_T3.xlsx
    Untracked:  data/211210_time_table.xlsx
    Untracked:  data/211218_pH_colorimetric_t0.xlsx
    Untracked:  data/211218_pH_colorimetric_t1.xlsx
    Untracked:  data/211218_pH_colorimetric_t2.xlsx
    Untracked:  data/211218_pH_colorimetric_t3.xlsx
    Untracked:  data/211218_pH_colorimetric_t4.xlsx
    Untracked:  data/211218_time_table.xlsx
    Untracked:  data/211220_Griess_plate1.xlsx
    Untracked:  data/211220_Griess_plate10.xlsx
    Untracked:  data/211220_Griess_plate11.xlsx
    Untracked:  data/211220_Griess_plate2.xlsx
    Untracked:  data/211220_Griess_plate3.xlsx
    Untracked:  data/211220_Griess_plate4.xlsx
    Untracked:  data/211220_Griess_plate5.xlsx
    Untracked:  data/211220_Griess_plate6.xlsx
    Untracked:  data/211220_Griess_plate7.xlsx
    Untracked:  data/211220_Griess_plate8.xlsx
    Untracked:  data/211220_Griess_plate9.xlsx
    Untracked:  data/211220_time_table_AU.xlsx
    Untracked:  data/211220_time_table_BN.xlsx
    Untracked:  data/211224_TOC_sample_plate1.xlsx
    Untracked:  data/220103_Griess_T0.xlsx
    Untracked:  data/220103_Griess_T1.xlsx
    Untracked:  data/220103_Griess_T11.xlsx
    Untracked:  data/220103_Griess_T12.xlsx
    Untracked:  data/220103_Griess_T2.xlsx
    Untracked:  data/220103_Griess_T3.xlsx
    Untracked:  data/220103_Griess_T4.xlsx
    Untracked:  data/220103_Griess_T5.xlsx
    Untracked:  data/220103_Griess_T6.xlsx
    Untracked:  data/220103_Griess_T7.xlsx
    Untracked:  data/220103_Griess_T8.xlsx
    Untracked:  data/220103_Griess_T9.xlsx
    Untracked:  data/220103_time_table.xlsx
    Untracked:  data/220106_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-2_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-3_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-4_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-5_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-6_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-7_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-8_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107-9_Ammonia_sample_plate1.xlsx
    Untracked:  data/220107_Ammonia_sample_plate1.xlsx
    Untracked:  data/220108_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate1.xlsx
    Untracked:  data/220110_Ammonia_sample_plate2.xlsx
    Untracked:  data/220111_Ammonia_sample_T0.xlsx
    Untracked:  data/220111_Ammonia_sample_T1.xlsx
    Untracked:  data/220111_Ammonia_sample_T2.xlsx
    Untracked:  data/220111_Ammonia_sample_T3.xlsx
    Untracked:  data/220111_TOC_sample_plate1.xlsx
    Untracked:  data/220112_Ammonia_sample_T11.xlsx
    Untracked:  data/220112_Ammonia_sample_T12.xlsx
    Untracked:  data/220112_Ammonia_sample_T4.xlsx
    Untracked:  data/220112_Ammonia_sample_T5.xlsx
    Untracked:  data/220112_Ammonia_sample_T6.xlsx
    Untracked:  data/220112_Ammonia_sample_T7.xlsx
    Untracked:  data/220112_Ammonia_sample_T8.xlsx
    Untracked:  data/220112_Ammonia_sample_T9.xlsx
    Untracked:  data/220113_pH_colorimetric_T0.xlsx
    Untracked:  data/220113_pH_colorimetric_T12.xlsx
    Untracked:  data/220113_pH_colorimetric_T2.xlsx
    Untracked:  data/220113_pH_colorimetric_T3.xlsx
    Untracked:  data/220113_pH_colorimetric_T8.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH1_using6.66.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_mu_opt0.099.xlsx
    Untracked:  data/220125_pH_colorimetric_pH2_using6.66.xlsx
    Untracked:  data/220125_time_table.xlsx
    Untracked:  data/220128_Ammonia_sample_plate1.xlsx
    Untracked:  data/220128_Ammonia_sample_plate2.xlsx
    Untracked:  data/220128_Ammonia_sample_plate3.xlsx
    Untracked:  data/220128_Ammonia_sample_plate4.xlsx
    Untracked:  data/220128_Ammonia_sample_plate5.xlsx
    Untracked:  data/220128_Ammonia_sample_plate6.xlsx
    Untracked:  data/220128_Griess_plate1.xlsx
    Untracked:  data/220128_Griess_plate2.xlsx
    Untracked:  data/220128_Griess_plate3.xlsx
    Untracked:  data/220128_Griess_plate4.xlsx
    Untracked:  data/220128_Griess_plate5.xlsx
    Untracked:  data/220128_Griess_plate6.xlsx
    Untracked:  data/220128_time_table.xlsx
    Untracked:  data/220131_Ammonia_sample_plate1.xlsx
    Untracked:  data/220131_Ammonia_sample_plate2.xlsx
    Untracked:  data/220131_Ammonia_sample_plate3.xlsx
    Untracked:  data/220131_Ammonia_sample_plate4.xlsx
    Untracked:  data/220131_Ammonia_sample_plate5.xlsx
    Untracked:  data/220131_Ammonia_sample_plate6.xlsx
    Untracked:  data/220131_Griess_plate1.xlsx
    Untracked:  data/220131_Griess_plate2.xlsx
    Untracked:  data/220131_Griess_plate3.xlsx
    Untracked:  data/220131_Griess_plate4.xlsx
    Untracked:  data/220131_Griess_plate5.xlsx
    Untracked:  data/220131_Griess_plate6.xlsx
    Untracked:  data/220131_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220131_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220131_time_table.xlsx
    Untracked:  data/220327_Griess_pastplate9.xlsx
    Untracked:  data/220327_Griess_plate1.xlsx
    Untracked:  data/220327_Griess_plate10.xlsx
    Untracked:  data/220327_Griess_plate11.xlsx
    Untracked:  data/220327_Griess_plate12.xlsx
    Untracked:  data/220327_Griess_plate13.xlsx
    Untracked:  data/220327_Griess_plate14.xlsx
    Untracked:  data/220327_Griess_plate15.xlsx
    Untracked:  data/220327_Griess_plate16.xlsx
    Untracked:  data/220327_Griess_plate2.xlsx
    Untracked:  data/220327_Griess_plate3.xlsx
    Untracked:  data/220327_Griess_plate4.xlsx
    Untracked:  data/220327_Griess_plate5.xlsx
    Untracked:  data/220327_Griess_plate6.xlsx
    Untracked:  data/220327_Griess_plate7.xlsx
    Untracked:  data/220327_Griess_plate8.xlsx
    Untracked:  data/220327_Griess_plate9.xlsx
    Untracked:  data/220327_time_table.xlsx
    Untracked:  data/220329_pH_colorimetric_pH1_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH2_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH3_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH4_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH6_mu_opt_T0.xlsx
    Untracked:  data/220329_pH_colorimetric_pH7_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH3_half_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_base_treatment_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_150ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_300ul_mu_opt_T0.xlsx
    Untracked:  data/220407_pH_colorimetric_pH_standard_60ul_mu_opt_T0.xlsx
    Untracked:  data/220408_pH_colorimetric_pH_limit_mu_opt_water.xlsx
    Untracked:  data/220412_pH_colorimetric_pH_half_mu_opt_WW.xlsx
    Untracked:  data/220414_Ammonia_sample_plate0.xlsx
    Untracked:  data/220418_Ammonia_sample_plate0.xlsx
    Untracked:  data/220419_Ammonia_sample_plate0.xlsx
    Untracked:  data/220420-2_Ammonia_bcf.xlsx
    Untracked:  data/220420_Ammonia_sample_plate1.xlsx
    Untracked:  data/220420_Ammonia_sample_plate10.xlsx
    Untracked:  data/220420_Ammonia_sample_plate11.xlsx
    Untracked:  data/220420_Ammonia_sample_plate12.xlsx
    Untracked:  data/220420_Ammonia_sample_plate13.xlsx
    Untracked:  data/220420_Ammonia_sample_plate14.xlsx
    Untracked:  data/220420_Ammonia_sample_plate15.xlsx
    Untracked:  data/220420_Ammonia_sample_plate16.xlsx
    Untracked:  data/220420_Ammonia_sample_plate2.xlsx
    Untracked:  data/220420_Ammonia_sample_plate3.xlsx
    Untracked:  data/220420_Ammonia_sample_plate4.xlsx
    Untracked:  data/220420_Ammonia_sample_plate5.xlsx
    Untracked:  data/220420_Ammonia_sample_plate6.xlsx
    Untracked:  data/220420_Ammonia_sample_plate7.xlsx
    Untracked:  data/220420_Ammonia_sample_plate8.xlsx
    Untracked:  data/220420_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_df_a_bcf.xlsx
    Untracked:  data/220421_Ammonia_df_pH_range.xlsx
    Untracked:  data/220421_Ammonia_sample_plate1.xlsx
    Untracked:  data/220421_Ammonia_sample_plate10.xlsx
    Untracked:  data/220421_Ammonia_sample_plate11.xlsx
    Untracked:  data/220421_Ammonia_sample_plate2.xlsx
    Untracked:  data/220421_Ammonia_sample_plate3.xlsx
    Untracked:  data/220421_Ammonia_sample_plate4.xlsx
    Untracked:  data/220421_Ammonia_sample_plate5.xlsx
    Untracked:  data/220421_Ammonia_sample_plate6.xlsx
    Untracked:  data/220421_Ammonia_sample_plate7.xlsx
    Untracked:  data/220421_Ammonia_sample_plate8.xlsx
    Untracked:  data/220421_Ammonia_sample_plate9.xlsx
    Untracked:  data/220421_Ammonia_sterile.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks.xlsx
    Untracked:  data/220421_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx
    Untracked:  data/220421_Griess_plate1.xlsx
    Untracked:  data/220421_Griess_plate10.xlsx
    Untracked:  data/220421_Griess_plate11.xlsx
    Untracked:  data/220421_Griess_plate2.xlsx
    Untracked:  data/220421_Griess_plate3.xlsx
    Untracked:  data/220421_Griess_plate4.xlsx
    Untracked:  data/220421_Griess_plate5.xlsx
    Untracked:  data/220421_Griess_plate6.xlsx
    Untracked:  data/220421_Griess_plate7.xlsx
    Untracked:  data/220421_Griess_plate8.xlsx
    Untracked:  data/220421_Griess_plate9.xlsx
    Untracked:  data/220421_nitrate_reduced_autoclaved_samples.xlsx
    Untracked:  data/220421_time_table.xlsx
    Untracked:  data/220422_pH_Antibiotics_Sterile_experiment.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T4_mu_opt_T4.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T7_mu_opt_T7.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T0.xlsx
    Untracked:  data/220422_pH_colorimetric_LBA_Sterile_T9_mu_opt_T9.xlsx
    Untracked:  data/220508_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate1.xlsx
    Untracked:  data/220513_TOC_sample_plate2.xlsx
    Untracked:  data/220513_TOC_sample_plate3.xlsx
    Untracked:  data/220517_time_table.xlsx
    Untracked:  data/220519_Griess_antibiotics_dose_experiment_remove_blanks.xlsx
    Untracked:  data/220519_Griess_crerar7_extreme.xlsx
    Untracked:  data/220519_Griess_plate1.xlsx
    Untracked:  data/220519_Griess_plate10.xlsx
    Untracked:  data/220519_Griess_plate11.xlsx
    Untracked:  data/220519_Griess_plate2.xlsx
    Untracked:  data/220519_Griess_plate3.xlsx
    Untracked:  data/220519_Griess_plate4.xlsx
    Untracked:  data/220519_Griess_plate5.xlsx
    Untracked:  data/220519_Griess_plate6.xlsx
    Untracked:  data/220519_Griess_plate7.xlsx
    Untracked:  data/220519_Griess_plate8.xlsx
    Untracked:  data/220519_Griess_plate9.xlsx
    Untracked:  data/220519_time_table.xlsx
    Untracked:  data/220520_TOC_sample_plate1.xlsx
    Untracked:  data/220520_TOC_sample_plate2.xlsx
    Untracked:  data/220520_TOC_sample_plate3.xlsx
    Untracked:  data/220521_pH_Antibiotics_dose_effect_experiment.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T0_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T10_mu_opt_T0.xlsx
    Untracked:  data/220521_pH_colorimetric_Barneveld2_Sterile_T7_mu_opt_T0.xlsx
    Untracked:  data/220526_TOC_sample_plate1.xlsx
    Untracked:  data/220527_TOC_sample_plate1.xlsx
    Untracked:  data/220601_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_TOC_sample_plate1.xlsx
    Untracked:  data/220602-2_control_OD600.xlsx
    Untracked:  data/220602-3_TOC_sample_plate1.xlsx
    Untracked:  data/220602-3_control_OD600.xlsx
    Untracked:  data/220602_TOC_sample_plate1.xlsx
    Untracked:  data/220603_100ul_OD600.xlsx
    Untracked:  data/220603_TOC_sample_plate1.xlsx
    Untracked:  data/220603_TOC_sample_plate2.xlsx
    Untracked:  data/220606_TOC_sample_plate1.xlsx
    Untracked:  data/220606_TOC_sample_plate2.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel.xlsx
    Untracked:  data/ForKiseok_from_Dr_Cuhel_format_edited.xlsx
    Untracked:  data/OD434_OD585_C_WB_T5_220329.xlsx
    Untracked:  data/OD434_OD585_WW_C_T0_220329.xlsx
    Untracked:  data/OD434_OD585_pH3_half_area_220407.xlsx
    Untracked:  data/OD434_OD585_pH_base_treated_T0_220407.xlsx
    Untracked:  data/OD434_OD585_pH_half_150ul_220412.xlsx
    Untracked:  data/SPNa_86437_021122_edit.xlsx
    Untracked:  data/SPNa_86437_021122_edit2.xlsx
    Untracked:  data/TOC_TN_measurement_2_labs_211014.xlsx
    Untracked:  data/dry_weight_curve(9.5.21).xlsx
    Untracked:  data/pH_data(11.17.21)-Midway_etc.xlsx
    Untracked:  data/pH_data(12.6.21)-Midway_incubation_endpoint.xlsx
    Untracked:  data/pH_data(9.14.21).xlsx
    Untracked:  data/pH_data(9.28.21).xlsx
    Untracked:  data/pH_standard_150ul_220407.xlsx
    Untracked:  data/pH_standard_300ul_220407.xlsx
    Untracked:  data/pH_standard_60ul_220407.xlsx
    Untracked:  data/~$220513_TOC_sample_plate1.xlsx
    Untracked:  data/~$SPNa_86437_021122_edit.xlsx
    Untracked:  past_figures/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/220519_Griess_antibiotics_dose_effect.Rmd</code>) and HTML (<code>docs/220519_Griess_antibiotics_dose_effect.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Denit_visualization_R/blob/7905c9343b41b0f3a42aba137993e7c632c5b0e2/analysis/220519_Griess_antibiotics_dose_effect.Rmd" target="_blank">7905c93</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2022-06-08
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="antibiotics-and-dose-effect-griess-assay" class="section level2">
<h2>Antibiotics and dose effect (Griess assay)</h2>
<p>Researcher: <strong>Kiseok Lee</strong><br />
Experiment Date: 5/8/22 - 5/12/22 (5 days)<br />
Analysis Date: 5/23/22 Lab: <strong>Seppe Kuehn</strong></p>
<pre class="r"><code># libraries
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(vegan)</code></pre>
<pre><code>Loading required package: permute</code></pre>
<pre><code>Loading required package: lattice</code></pre>
<pre><code>This is vegan 2.5-7</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.0.5</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.1 --</code></pre>
<pre><code>v tibble  3.0.4     v purrr   0.3.4
v tidyr   1.1.3     v stringr 1.4.0
v readr   1.4.0     v forcats 0.5.1</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(magrittr)</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre class="r"><code>library(readxl)
library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(gtools)</code></pre>
<pre><code>
Attaching package: &#39;gtools&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:permute&#39;:

    permute</code></pre>
<pre class="r"><code>library(devtools)</code></pre>
<pre><code>Warning: package &#39;devtools&#39; was built under R version 4.0.5</code></pre>
<pre><code>Loading required package: usethis</code></pre>
<pre><code>Warning: package &#39;usethis&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;devtools&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:permute&#39;:

    check</code></pre>
<pre class="r"><code>library(openxlsx)
library(ape)</code></pre>
<pre><code>Warning: package &#39;ape&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code>library(stringr)
library(tidyr)
library(ggrepel)</code></pre>
<pre><code>Warning: package &#39;ggrepel&#39; was built under R version 4.0.4</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Warning: package &#39;ggpubr&#39; was built under R version 4.0.5</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ape&#39;:

    rotate</code></pre>
<pre class="r"><code>require(gridExtra)</code></pre>
<pre><code>Loading required package: gridExtra</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre class="r"><code># grid.arrange(p_strain, p_ai, nrow=1)

## theme for ggplot
mytheme &lt;- theme_bw() + 
  theme(text = element_text(face=&quot;bold&quot;, colour = &#39;black&#39;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(axis.text.y = element_text(size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(),panel.background=element_blank(),panel.border=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))+
  theme(legend.text=element_text(size=10,face=&quot;bold&quot;, colour = &#39;black&#39;))


mytheme_2d &lt;- theme_bw() + 
  theme(text = element_text(face=&quot;bold&quot;, colour = &#39;black&#39;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5,face=&quot;bold&quot;, colour = &#39;black&#39;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  theme(axis.text.y = element_text(size=13,face=&quot;bold&quot;, colour = &#39;black&#39;))+
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))+
  theme(legend.text=element_text(size=10,face=&quot;bold&quot;, colour = &#39;black&#39;))


# color collection
my_color_collection &lt;- c(
  &quot;#CBD588&quot;, &quot;#5F7FC7&quot;, &quot;orange&quot;, &quot;#AD6F3B&quot;, &quot;#673770&quot;, 
  &quot;#D14285&quot;, &quot;#652926&quot;, &quot;#C84248&quot;, &quot;#8569D5&quot;, &quot;#5E738F&quot;,
  &quot;#D1A33D&quot;, &quot;#8A7C64&quot;, &quot;#599861&quot;,&quot;#616163&quot;, &quot;#FFCDB2&quot;,
  &quot;#6D9F71&quot;, &quot;#242F40&quot;,
  &quot;#CCA43B&quot;, &quot;#F92A82&quot;, &quot;#ED7B84&quot;, &quot;#7EB77F&quot;, 
  &quot;#DEC4A1&quot;, &quot;#E5D1D0&quot;, &#39;#0E8482&#39;, &#39;#C9DAEA&#39;, &#39;#337357&#39;, 
  &#39;#95C623&#39;, &#39;#E55812&#39;, &#39;#04471C&#39;, &#39;#F2D7EE&#39;, &#39;#D3BCC0&#39;, 
  &#39;#A5668B&#39;, &#39;#69306D&#39;, &#39;#0E103D&#39;, &#39;#1A535C&#39;, &#39;#4ECDC4&#39;, 
  &#39;#F7FFF7&#39;, &#39;#FF6B6B&#39;, &#39;#FFE66D&#39;, &#39;#6699CC&#39;, &#39;#FFF275&#39;, 
  &#39;#FF8C42&#39;, &#39;#FF3C38&#39;, &#39;#A23E48&#39;, &#39;#000000&#39;, &#39;#CF5C36&#39;, 
  &#39;#EEE5E9&#39;, &#39;#7C7C7C&#39;, &#39;#EFC88B&#39;, &#39;#2E5266&#39;, &#39;#6E8898&#39;, 
  &#39;#9FB1BC&#39;, &#39;#D3D0CB&#39;, &#39;#E2C044&#39;, &#39;#5BC0EB&#39;, &#39;#FDE74C&#39;, 
  &#39;#9BC53D&#39;, &#39;#E55934&#39;, &#39;#FA7921&#39;, &quot;#CD9BCD&quot;, &quot;#508578&quot;, &quot;#DA5724&quot;)

# for git push, use this instead of using wflow_git_push()
# git push -u origin master (in the Git app / in the working directory)

# for making pdf file
library(rmarkdown)</code></pre>
<pre><code>Warning: package &#39;rmarkdown&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code># render(&quot;analysis/~~.Rmd&quot;, &quot;pdf_document&quot;)</code></pre>
</div>
<div id="import-data-table-from-python-code" class="section level2">
<h2>1. Import data table from python code</h2>
<p>We are going to use the vcl3 treated standard curve that is fitted with pure nitrate standards</p>
<pre class="r"><code># import file
df_plate1 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate1.xlsx&quot;)
df_plate2 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate2.xlsx&quot;)
df_plate3 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate3.xlsx&quot;)
df_plate4 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate4.xlsx&quot;)
df_plate5 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate5.xlsx&quot;)
df_plate6 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate6.xlsx&quot;)
df_plate7 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate7.xlsx&quot;)
df_plate8 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate8.xlsx&quot;)
df_plate9 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate9.xlsx&quot;)
df_plate10 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate10.xlsx&quot;)
df_plate11 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate11.xlsx&quot;)
# df_plate12 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate12.xlsx&quot;)
# df_plate13 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate13.xlsx&quot;)
# df_plate14 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate14.xlsx&quot;)
# df_plate15 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate15.xlsx&quot;)
# df_plate16 &lt;- openxlsx::read.xlsx(&quot;data/220519_Griess_plate16.xlsx&quot;)

head(df_plate1)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T0   0.00225      0.77785 0.003328244 0.7006403 0.6973120
2         T0   0.00155      0.81620 0.002982502 0.7382703 0.7352878
3         T0   0.00625      0.79515 0.005304284 0.7175891 0.7122848
4         T0   0.00000           NA 0.002216999        NA        NA
5         T0   0.00595      0.78330 0.005156059 0.7059750 0.7008189
6         T0   0.00035      0.74375 0.002389846 0.6673581 0.6649683</code></pre>
<pre class="r"><code>colnames(df_plate1)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate1)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate2)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T1   0.00130      0.77915 0.002859028 0.7019124 0.6990533
2         T1   0.00060      0.86555 0.002513312 0.7870108 0.7844974
3         T1   0.00625      0.78815 0.005304284 0.7107260 0.7054217
4         T1   0.00115      0.78290 0.002784944 0.7055833 0.7027984
5         T1   0.00835      0.80305 0.006341956 0.7253431 0.7190012
6         T1   0.00000      0.76610 0.002216999 0.6891533 0.6869363</code></pre>
<pre class="r"><code>colnames(df_plate2)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate2)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate3)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T2   0.00000      0.73940 0.002216999 0.6631243 0.6609073
2         T2   0.03575      0.83020 0.019897054 0.7520609 0.7321638
3         T2   0.02915      0.72475 0.016629245 0.6488854 0.6322561
4         T2   0.00240      0.80440 0.003402335 0.7266691 0.7232668
5         T2   0.08225      0.76105 0.042969329 0.6842225 0.6412531
6         T2   0.00135      0.77510 0.002883722 0.6979501 0.6950663</code></pre>
<pre class="r"><code>colnames(df_plate3)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate3)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate4)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T3   0.00000      0.69630 0.002216999 0.6213196 0.6191026
2         T3   0.10995      0.76130 0.056754576 0.6844665 0.6277119
3         T3   0.08555      0.75005 0.044609997 0.6734945 0.6288845
4         T3   0.00545      0.74295 0.004909026 0.6665793 0.6616702
5         T3   0.20220      0.76330 0.102888295 0.6864189 0.5835306
6         T3   0.00165      0.76965 0.003031893 0.6926218 0.6895899</code></pre>
<pre class="r"><code>colnames(df_plate4)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate4)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate5)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T4   0.00030      0.75610 0.002365153 0.6793928 0.6770276
2         T4   0.30925      0.68480 0.156864143 0.6102089 0.4533448
3         T4   0.24325      0.76685 0.123529596 0.6898859 0.5663563
4         T4   0.04470      0.70580 0.024331159 0.6305118 0.6061806
5         T4   0.49995      0.75155 0.254229048 0.6749564 0.4207274
6         T4   0.00010      0.76725 0.002266383 0.6902767 0.6880103</code></pre>
<pre class="r"><code>colnames(df_plate5)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate5)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate6)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T5   0.00135      0.69530 0.002883722 0.6203527 0.6174690
2         T5   0.44725      0.56600 0.227163789 0.4964834 0.2693196
3         T5   0.33535      0.67675 0.170097203 0.6024423 0.4323451
4         T5   0.09495      0.65085 0.049285811 0.5775145 0.5282287
5         T5   0.35150      0.59560 0.178300016 0.5246421 0.3463421
6         T5   0.00385      0.74120 0.004118585 0.6648759 0.6607573</code></pre>
<pre class="r"><code>colnames(df_plate6)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate6)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate7)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM     NO3_mM
1         T6   0.00305      0.70795 0.003723402 0.6325938 0.62887044
2         T6   0.58560      0.50175 0.298480858 0.4357558 0.13727496
3         T6   0.49165      0.75095 0.249958244 0.6743716 0.42441338
4         T6   0.14610      0.61335 0.074791657 0.5415835 0.46679188
5         T6   0.94605      0.65575 0.488460187 0.5822235 0.09376335
6         T6   0.00170      0.72830 0.003056588 0.6523330 0.64927641</code></pre>
<pre class="r"><code>colnames(df_plate7)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate7)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate8)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T7   0.00165      0.64065 0.003031893 0.5677225 0.5646906
2         T7   0.71035      0.36455 0.363533189 0.3078278 0.0000000
3         T7   0.66660      0.71390 0.340637155 0.6383592 0.2977221
4         T7   0.19865      0.54575 0.101106514 0.4772857 0.3761792
5         T7   1.03050      0.54405 0.533889859 0.4756765 0.0000000
6         T7   0.00235      0.67545 0.003377638 0.6011889 0.5978113</code></pre>
<pre class="r"><code>colnames(df_plate8)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate8)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate9)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T8   0.00185      0.58835 0.003130675 0.5177344 0.5146037
2         T8   0.38725      0.19860 0.196497744 0.1561272 0.0000000
3         T8   0.93585      0.69525 0.482997514 0.6203044 0.1373068
4         T8   0.29925      0.46280 0.151801664 0.3991991 0.2473974
5         T8   0.96705      0.48035 0.499723323 0.4156469 0.0000000
6         T8   0.00135      0.66435 0.002883722 0.5904963 0.5876126</code></pre>
<pre class="r"><code>colnames(df_plate9)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate9)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate10)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM    NO3_mM
1         T9   0.00385      0.52125 0.004118585 0.4541303 0.4500117
2         T9   0.00195      0.00000 0.003180067 0.0000000 0.0000000
3         T9   1.03915      0.55265 0.538563651 0.4838211 0.0000000
4         T9   0.43665      0.32345 0.221734700 0.2699545 0.0482198
5         T9   0.51555      0.25995 0.262264400 0.2118343 0.0000000
6         T9   0.00390      0.61395 0.004143284 0.5421569 0.5380137</code></pre>
<pre class="r"><code>colnames(df_plate10)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate10)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code>head(df_plate11)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil Sample_type
1  A01             0             2              0 Barneveld2      Slurry
2  A02             0             2              0 Barneveld2      Slurry
3  A03             0             2              0 Barneveld2      Slurry
4  A04             0             2              0 Barneveld2      Slurry
5  A05             0             2              0 Barneveld2      Slurry
6  A06             0             2              0 Barneveld2      Slurry
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1            HCl              CHL      200    8               4       20
2            HCl              CHL      200    0               0       20
3           NaOH              CHL      200    8               4       20
4            HCl              CHL     1000    4               2       20
5           NaOH              CHL     1000    4               2       20
6            HCl              CHL     2500    8               4       20
  Time_point NO2_OD540 NO2NO3_OD540      NO2_mM  NO2NO3_mM    NO3_mM
1        T10   0.00540      0.40885 0.004884323 0.34887886 0.3439945
2        T10   0.00370      0.00000 0.004044486 0.00000000 0.0000000
3        T10   0.90720      0.46375 0.467681626 0.40008845 0.0000000
4        T10   0.30135      0.14860 0.152864433 0.11104264 0.0000000
5        T10   0.20335      0.09780 0.103465603 0.06552209 0.0000000
6        T10   0.00255      0.50740 0.003476426 0.44107469 0.4375983</code></pre>
<pre class="r"><code>colnames(df_plate11)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;          </code></pre>
<pre class="r"><code>dim(df_plate11)</code></pre>
<pre><code>[1] 96 18</code></pre>
<pre class="r"><code># head(df_plate16)
# colnames(df_plate16)
# dim(df_plate16)


# X1 to Well
# df_plate1 %&lt;&gt;% rename(Well = X1)
# df_plate2 %&lt;&gt;% rename(Well = X1)
# df_plate3 %&lt;&gt;% rename(Well = X1)
# df_plate4 %&lt;&gt;% rename(Well = X1)
# df_plate5 %&lt;&gt;% rename(Well = X1)
# df_plate6 %&lt;&gt;% rename(Well = X1)
# df_plate7 %&lt;&gt;% rename(Well = X1)
# df_plate8 %&lt;&gt;% rename(Well = X1)
# df_plate9 %&lt;&gt;% rename(Well = X1)
# df_plate10 %&lt;&gt;% rename(Well = X1)
# df_plate11 %&lt;&gt;% rename(Well = X1)
# df_plate12 %&lt;&gt;% rename(Well = X1)
# df_plate13 %&lt;&gt;% rename(Well = X1)
# df_plate14 %&lt;&gt;% rename(Well = X1)
# df_plate15 %&lt;&gt;% rename(Well = X1)
# df_plate16 %&lt;&gt;% rename(Well = X1)


# remove wells that were contaminated during the experiment
df_plate1 %&lt;&gt;% filter(!(Well %in% c(&quot;A04&quot;))) # NA detected below... don&#39;t know the reason
# df_plate4 %&lt;&gt;% filter(!(Well %in% c(&quot;H05&quot;))) # filter defect
df_plate6 %&lt;&gt;% filter(!(Well %in% c(&quot;G05&quot;))) # Too low.... why???
# df_plate8 %&lt;&gt;% filter(!(Well %in% c(&quot;A01&quot;, &quot;A09&quot;))) # filter defect
# df_plate9 %&lt;&gt;% filter(!(Well %in% c(&quot;D11&quot;))) # NA detected below...
# df_plate11 %&lt;&gt;% filter(!(Well %in% c(&quot;A12&quot;))) # filter defect

# bind two dataframe
df_p &lt;- rbind(df_plate1, df_plate2, df_plate3, df_plate4, df_plate5, df_plate6, df_plate7, df_plate8, df_plate9, df_plate10, df_plate11)
              # df_plate12, df_plate13, df_plate14, df_plate15, df_plate16)
dim(df_p)</code></pre>
<pre><code>[1] 1054   18</code></pre>
<pre class="r"><code># remove NA
dim(df_p) # 1056</code></pre>
<pre><code>[1] 1054   18</code></pre>
<pre class="r"><code>df_p[is.na(df_p)]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_na &lt;- df_p[rowSums(is.na(df_p)) &gt; 0,]  # A04 in plate1 has NA. Let&#39;s remove from the wells above.
# df_p &lt;- na.omit(df_p)
df_p &lt;- df_p[rowSums(is.na(df_p)) == 0,]
dim(df_p) # 1055</code></pre>
<pre><code>[1] 1054   18</code></pre>
<pre class="r"><code># multiply dilution factor which is 5/2
# df_p %&lt;&gt;% select(-NO2_OD540, -NO2NO3_OD540)
df_p %&lt;&gt;% mutate(NO2_mM = NO2_mM * (5/2), NO2NO3_mM = NO2NO3_mM * (5/2), NO3_mM = NO3_mM * (5/2))

# Get the metadata for time point and left join
Time_table &lt;- openxlsx::read.xlsx(&quot;data/220519_time_table.xlsx&quot;)
# Time_table_BN &lt;- openxlsx::read.xlsx(&quot;data/220519_time_table_BN.xlsx&quot;)
# Time_table &lt;- rbind(Time_table_AU, Time_table_BN)

Time_table %&lt;&gt;% select(-Date) 
Time_table$Time_hours &lt;- round(Time_table$Time_hours, 1)
Time_table$Time_days &lt;- round(Time_table$Time_days, 1)
df_time &lt;- Time_table
dim(df_p)</code></pre>
<pre><code>[1] 1054   18</code></pre>
<pre class="r"><code>df_p &lt;- df_p %&gt;% left_join(Time_table, by=(&quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p)</code></pre>
<pre><code>[1] 1054   21</code></pre>
<pre class="r"><code>colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;      
[21] &quot;Time_days&quot;       </code></pre>
<pre class="r"><code># time_point order
# df_p$Time_point &lt;-  factor(df_p$Time_point, levels = c(paste0(&quot;LBA_T&quot;,0:10), paste0(&quot;Sterile_T&quot;,0:10)))
df_p$Time_point &lt;-  factor(df_p$Time_point, levels = c(paste0(&quot;T&quot;,0:10)))</code></pre>
</div>
<div id="get-average-and-standard-deviation-moisture-correction-blank-correction" class="section level2">
<h2>2. Get average and standard deviation &amp; Moisture correction &amp; Blank correction</h2>
<pre class="r"><code># plot to see
ggplot(df_p, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Without averaging \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot to see
ggplot(df_p, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Without averaging \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># let&#39;s plot the no nitrate sample&#39;s nitrate, nitrite levels
df_no_nitrite &lt;- df_p %&gt;% filter(Sample_type == &quot;No_Nitrate&quot;)

ggplot(df_no_nitrite, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  # scale_y_continuous(breaks = seq(0,0.5,0.05), limits=c(0, 0.5))+
  ggtitle(&quot;Samples without nitrate addition \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_no_nitrite, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Samples without nitrate addition \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># T8 is weird
# how about blanks?

# let&#39;s plot the blank sample&#39;s nitrate, nitrite levels
df_A_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)
ggplot(df_A_blank, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrate blanks \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_A_blank, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Nitrate blanks \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># let&#39;s plot the Nitrite blank sample&#39;s nitrate, nitrite levels
df_I_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
ggplot(df_I_blank, aes(x=Time_point, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrite blanks \n&quot;) +
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_I_blank, aes(x=Time_point, y=NO2_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time_point&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.1), limits=c(0, 2.5))+
  ggtitle(&quot;Nitrite blanks \n&quot;) +
  mytheme_2d+
  theme(axis.text.x = element_text(hjust = 0.8, vjust=0.8, size=13, family=&quot;serif&quot;, angle = 40))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Important ###
# Due to the nitrate decrease in time, something is wrong. Therefore, I would omit the evaporation correction????


# average technical replicate (here only 1 replicate per pH perturbation)
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;      
[21] &quot;Time_days&quot;       </code></pre>
<pre class="r"><code>dim(df_p)</code></pre>
<pre><code>[1] 1054   21</code></pre>
<pre class="r"><code># df_p &lt;- df_p %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Concentration_M, Added_ul, Sample_type, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(NO2_mM = mean(NO2_mM), Std_NO2_mM = sd(NO2_mM), NO3_mM = mean(NO3_mM), Std_NO3_mM = sd(NO3_mM)) %&gt;% ungroup()


# mols and molarity unit conversion
# test
df_p$Titration_type %&gt;% length()</code></pre>
<pre><code>[1] 1054</code></pre>
<pre class="r"><code># Setting H_mM column
# df_soil &lt;- df_p %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;))

Added_Volume &lt;- 1.7 # ml
Soil_mg &lt;- 0.85
moisture_percent_1 = 20.15 # Barneveld2
Added_Volume + Soil_mg*(moisture_percent_1/100)</code></pre>
<pre><code>[1] 1.871275</code></pre>
<pre class="r"><code># moisture_percent_2 = 8.12 # Sterile - LaBaghWoods that has been autoclaved 5 times.
# Added_Volume + Soil_mg*(moisture_percent_2/100)
# moisture_percent_3 = 9.0 # Crerar7
# Added_Volume + Soil_mg*(moisture_percent_3/100)

df_p$Added_ul &lt;- ifelse(df_p$Titration_type == &quot;NaOH&quot;, -1*df_p$Added_ul, df_p$Added_ul) # HCl is +, NaOH is -
df_p %&lt;&gt;% mutate(H_mol = Concentration_M * Added_ul * 10^(-6)) # Calculate H mol 
df_p$Volume &lt;- ifelse(df_p$Soil == &quot;Barneveld2&quot;, Added_Volume + Soil_mg*(moisture_percent_1/100),
                      ifelse(df_p$Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;, Added_Volume + Soil_mg*(moisture_percent_2/100),
                             0))
                             # ifelse(df_p$Soil == &quot;Crerar1_pH8.05&quot;, Added_Volume + Soil_mg*(moisture_percent_3/100),
                                    # 0))) # Calc total volume

# df_p$Volume &lt;- ifelse(df_p$Soil == &quot;Allandale_pH4&quot;, Added_Volume + Soil_mg*(moisture_percent_2/100), df_p$Volume) # Calc total volume
# df_p$Volume %&gt;% unique()
df_p %&lt;&gt;% mutate(H_Molarity = H_mol / (Volume * 10^(-3)))
df_p %&lt;&gt;% mutate(H_mM = H_Molarity * 1000)
# openxlsx::write.xlsx(df_p, &quot;df_p.xlsx&quot;)

# how many levels of H_mM?
df_p %&gt;% filter(Soil == &quot;Barneveld2&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange(H_mM)</code></pre>
<pre><code>      H_mM
1 -42.7516
2 -21.3758
3   0.0000
4  21.3758
5  42.7516</code></pre>
<pre class="r"><code># df_p %&gt;% filter(Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange(H_mM)
# df_p %&gt;% filter(Soil == &quot;Crerar1_pH8.05&quot;)%&gt;% select(H_mM) %&gt;% unique() %&gt;% arrange()
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;      
[21] &quot;Time_days&quot;        &quot;H_mol&quot;            &quot;Volume&quot;           &quot;H_Molarity&quot;      
[25] &quot;H_mM&quot;            </code></pre>
<pre class="r"><code>## Moisture correction
dim(df_p)</code></pre>
<pre><code>[1] 1054   25</code></pre>
<pre class="r"><code># Testing negative samples
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)  # Use Nitrate_Blank instead in the future
df_no3_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input       Soil   Sample_type
1   G06             0             2              0 Barneveld2 Nitrate_Blank
2   G12             0             2              0 Barneveld2 Nitrate_Blank
3   H06             0             2              0 Barneveld2 Nitrate_Blank
4   H12             0             2              0 Barneveld2 Nitrate_Blank
5   G06             0             2              0 Barneveld2 Nitrate_Blank
6   G12             0             2              0 Barneveld2 Nitrate_Blank
7   H06             0             2              0 Barneveld2 Nitrate_Blank
8   H12             0             2              0 Barneveld2 Nitrate_Blank
9   G06             0             2              0 Barneveld2 Nitrate_Blank
10  G12             0             2              0 Barneveld2 Nitrate_Blank
11  H06             0             2              0 Barneveld2 Nitrate_Blank
12  H12             0             2              0 Barneveld2 Nitrate_Blank
13  G06             0             2              0 Barneveld2 Nitrate_Blank
14  G12             0             2              0 Barneveld2 Nitrate_Blank
15  H06             0             2              0 Barneveld2 Nitrate_Blank
16  H12             0             2              0 Barneveld2 Nitrate_Blank
17  G06             0             2              0 Barneveld2 Nitrate_Blank
18  G12             0             2              0 Barneveld2 Nitrate_Blank
19  H06             0             2              0 Barneveld2 Nitrate_Blank
20  H12             0             2              0 Barneveld2 Nitrate_Blank
21  G06             0             2              0 Barneveld2 Nitrate_Blank
22  G12             0             2              0 Barneveld2 Nitrate_Blank
23  H06             0             2              0 Barneveld2 Nitrate_Blank
24  H12             0             2              0 Barneveld2 Nitrate_Blank
25  G06             0             2              0 Barneveld2 Nitrate_Blank
26  G12             0             2              0 Barneveld2 Nitrate_Blank
27  H06             0             2              0 Barneveld2 Nitrate_Blank
28  H12             0             2              0 Barneveld2 Nitrate_Blank
29  G06             0             2              0 Barneveld2 Nitrate_Blank
30  G12             0             2              0 Barneveld2 Nitrate_Blank
31  H06             0             2              0 Barneveld2 Nitrate_Blank
32  H12             0             2              0 Barneveld2 Nitrate_Blank
33  G06             0             2              0 Barneveld2 Nitrate_Blank
34  G12             0             2              0 Barneveld2 Nitrate_Blank
35  H06             0             2              0 Barneveld2 Nitrate_Blank
36  H12             0             2              0 Barneveld2 Nitrate_Blank
37  G06             0             2              0 Barneveld2 Nitrate_Blank
38  G12             0             2              0 Barneveld2 Nitrate_Blank
39  H06             0             2              0 Barneveld2 Nitrate_Blank
40  H12             0             2              0 Barneveld2 Nitrate_Blank
41  G06             0             2              0 Barneveld2 Nitrate_Blank
42  G12             0             2              0 Barneveld2 Nitrate_Blank
43  H06             0             2              0 Barneveld2 Nitrate_Blank
44  H12             0             2              0 Barneveld2 Nitrate_Blank
   Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1         Nitrate             None        0    0               0       20
2         Nitrate             None        0    0               0       20
3         Nitrate             None        0    0               0       20
4         Nitrate             None        0    0               0       20
5         Nitrate             None        0    0               0       20
6         Nitrate             None        0    0               0       20
7         Nitrate             None        0    0               0       20
8         Nitrate             None        0    0               0       20
9         Nitrate             None        0    0               0       20
10        Nitrate             None        0    0               0       20
11        Nitrate             None        0    0               0       20
12        Nitrate             None        0    0               0       20
13        Nitrate             None        0    0               0       20
14        Nitrate             None        0    0               0       20
15        Nitrate             None        0    0               0       20
16        Nitrate             None        0    0               0       20
17        Nitrate             None        0    0               0       20
18        Nitrate             None        0    0               0       20
19        Nitrate             None        0    0               0       20
20        Nitrate             None        0    0               0       20
21        Nitrate             None        0    0               0       20
22        Nitrate             None        0    0               0       20
23        Nitrate             None        0    0               0       20
24        Nitrate             None        0    0               0       20
25        Nitrate             None        0    0               0       20
26        Nitrate             None        0    0               0       20
27        Nitrate             None        0    0               0       20
28        Nitrate             None        0    0               0       20
29        Nitrate             None        0    0               0       20
30        Nitrate             None        0    0               0       20
31        Nitrate             None        0    0               0       20
32        Nitrate             None        0    0               0       20
33        Nitrate             None        0    0               0       20
34        Nitrate             None        0    0               0       20
35        Nitrate             None        0    0               0       20
36        Nitrate             None        0    0               0       20
37        Nitrate             None        0    0               0       20
38        Nitrate             None        0    0               0       20
39        Nitrate             None        0    0               0       20
40        Nitrate             None        0    0               0       20
41        Nitrate             None        0    0               0       20
42        Nitrate             None        0    0               0       20
43        Nitrate             None        0    0               0       20
44        Nitrate             None        0    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM   NO3_mM
1          T0   0.00380      0.89365 0.010234713  2.037315 2.027080
2          T0   0.00120      0.84155 0.007024096  1.908156 1.901131
3          T0   0.00300      0.87300 0.009246760  1.986000 1.976754
4          T0   0.00105      0.83005 0.006838888  1.879782 1.872944
5          T1   0.00245      0.86005 0.008567579  1.953902 1.945334
6          T1   0.00160      0.85800 0.007517994  1.948827 1.941309
7          T1   0.00080      0.88425 0.006530214  2.013936 2.007406
8          T1   0.00300      0.87745 0.009246760  1.997045 1.987798
9          T2   0.00145      0.87245 0.007332780  1.984636 1.977303
10         T2   0.00215      0.88570 0.008197129  2.017540 2.009343
11         T2   0.00170      0.84995 0.007641471  1.928911 1.921270
12         T2   0.00055      0.84820 0.006221546  1.924585 1.918363
13         T3   0.00120      0.79690 0.007024096  1.798265 1.791241
14         T3   0.00205      0.86060 0.008073648  1.955264 1.947190
15         T3   0.00325      0.89995 0.009555488  2.053002 2.043446
16         T3   0.00315      0.86515 0.009431996  1.966536 1.957104
17         T4   0.00000      0.85090 0.005542497  1.931260 1.925718
18         T4   0.00240      0.83085 0.008505836  1.881755 1.873249
19         T4   0.00380      0.86510 0.010234713  1.966412 1.956177
20         T4   0.00510      0.83900 0.011840270  1.901860 1.890020
21         T5   0.00185      0.80020 0.007826688  1.806362 1.798535
22         T5   0.00220      0.78715 0.008258870  1.774365 1.766106
23         T5   0.00230      0.82905 0.008382353  1.877317 1.868935
24         T5   0.00205      0.80780 0.008073648  1.825024 1.816951
25         T6   0.00105      0.80925 0.006838888  1.828587 1.821748
26         T6   0.00085      0.76585 0.006591948  1.722273 1.715681
27         T6   0.00100      0.80355 0.006777153  1.814586 1.807808
28         T6   0.00080      0.79530 0.006530214  1.794341 1.787810
29         T7   0.00195      0.73035 0.007950167  1.635812 1.627862
30         T7   0.00155      0.76855 0.007456256  1.728867 1.721411
31         T7   0.00325      0.77405 0.009555488  1.742308 1.732752
32         T7   0.00215      0.73680 0.008197129  1.651488 1.643291
33         T8   0.00025      0.76445 0.005851152  1.718855 1.713003
34         T8   0.00030      0.81550 0.005912884  1.843954 1.838041
35         T8   0.00265      0.74330 0.008814550  1.667300 1.658485
36         T8   0.00105      0.67075 0.006838888  1.491648 1.484810
37         T9   0.00270      0.70630 0.008876294  1.577490 1.568613
38         T9   0.00125      0.68100 0.007085832  1.516354 1.509268
39         T9   0.00315      0.67430 0.009431996  1.500201 1.490769
40         T9   0.00270      0.65215 0.008876294  1.446909 1.438032
41        T10   0.00235      0.65500 0.008444094  1.453756 1.445312
42        T10   0.00270      0.61840 0.008876294  1.366028 1.357152
43        T10   0.00350      0.59825 0.009864223  1.317922 1.308058
44        T10   0.00475      0.61735 0.011407988  1.363518 1.352110
   Time_minutes Time_hours Time_days H_mol   Volume H_Molarity H_mM
1             0        0.0       0.0     0 1.871275          0    0
2             0        0.0       0.0     0 1.871275          0    0
3             0        0.0       0.0     0 1.871275          0    0
4             0        0.0       0.0     0 1.871275          0    0
5           220        3.7       0.2     0 1.871275          0    0
6           220        3.7       0.2     0 1.871275          0    0
7           220        3.7       0.2     0 1.871275          0    0
8           220        3.7       0.2     0 1.871275          0    0
9           400        6.7       0.3     0 1.871275          0    0
10          400        6.7       0.3     0 1.871275          0    0
11          400        6.7       0.3     0 1.871275          0    0
12          400        6.7       0.3     0 1.871275          0    0
13          575        9.6       0.4     0 1.871275          0    0
14          575        9.6       0.4     0 1.871275          0    0
15          575        9.6       0.4     0 1.871275          0    0
16          575        9.6       0.4     0 1.871275          0    0
17          980       16.3       0.7     0 1.871275          0    0
18          980       16.3       0.7     0 1.871275          0    0
19          980       16.3       0.7     0 1.871275          0    0
20          980       16.3       0.7     0 1.871275          0    0
21         1365       22.8       0.9     0 1.871275          0    0
22         1365       22.8       0.9     0 1.871275          0    0
23         1365       22.8       0.9     0 1.871275          0    0
24         1365       22.8       0.9     0 1.871275          0    0
25         1732       28.9       1.2     0 1.871275          0    0
26         1732       28.9       1.2     0 1.871275          0    0
27         1732       28.9       1.2     0 1.871275          0    0
28         1732       28.9       1.2     0 1.871275          0    0
29         2300       38.3       1.6     0 1.871275          0    0
30         2300       38.3       1.6     0 1.871275          0    0
31         2300       38.3       1.6     0 1.871275          0    0
32         2300       38.3       1.6     0 1.871275          0    0
33         2970       49.5       2.1     0 1.871275          0    0
34         2970       49.5       2.1     0 1.871275          0    0
35         2970       49.5       2.1     0 1.871275          0    0
36         2970       49.5       2.1     0 1.871275          0    0
37         4460       74.3       3.1     0 1.871275          0    0
38         4460       74.3       3.1     0 1.871275          0    0
39         4460       74.3       3.1     0 1.871275          0    0
40         4460       74.3       3.1     0 1.871275          0    0
41         6160      102.7       4.3     0 1.871275          0    0
42         6160      102.7       4.3     0 1.871275          0    0
43         6160      102.7       4.3     0 1.871275          0    0
44         6160      102.7       4.3     0 1.871275          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank # 33</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input       Soil   Sample_type
1   G05             2             0              0 Barneveld2 Nitrite_Blank
2   G11             2             0              0 Barneveld2 Nitrite_Blank
3   H05             2             0              0 Barneveld2 Nitrite_Blank
4   H11             2             0              0 Barneveld2 Nitrite_Blank
5   G05             2             0              0 Barneveld2 Nitrite_Blank
6   G11             2             0              0 Barneveld2 Nitrite_Blank
7   H05             2             0              0 Barneveld2 Nitrite_Blank
8   H11             2             0              0 Barneveld2 Nitrite_Blank
9   G05             2             0              0 Barneveld2 Nitrite_Blank
10  G11             2             0              0 Barneveld2 Nitrite_Blank
11  H05             2             0              0 Barneveld2 Nitrite_Blank
12  H11             2             0              0 Barneveld2 Nitrite_Blank
13  G05             2             0              0 Barneveld2 Nitrite_Blank
14  G11             2             0              0 Barneveld2 Nitrite_Blank
15  H05             2             0              0 Barneveld2 Nitrite_Blank
16  H11             2             0              0 Barneveld2 Nitrite_Blank
17  G05             2             0              0 Barneveld2 Nitrite_Blank
18  G11             2             0              0 Barneveld2 Nitrite_Blank
19  H05             2             0              0 Barneveld2 Nitrite_Blank
20  H11             2             0              0 Barneveld2 Nitrite_Blank
21  G11             2             0              0 Barneveld2 Nitrite_Blank
22  H05             2             0              0 Barneveld2 Nitrite_Blank
23  H11             2             0              0 Barneveld2 Nitrite_Blank
24  G05             2             0              0 Barneveld2 Nitrite_Blank
25  G11             2             0              0 Barneveld2 Nitrite_Blank
26  H05             2             0              0 Barneveld2 Nitrite_Blank
27  H11             2             0              0 Barneveld2 Nitrite_Blank
28  G05             2             0              0 Barneveld2 Nitrite_Blank
29  G11             2             0              0 Barneveld2 Nitrite_Blank
30  H05             2             0              0 Barneveld2 Nitrite_Blank
31  H11             2             0              0 Barneveld2 Nitrite_Blank
32  G05             2             0              0 Barneveld2 Nitrite_Blank
33  G11             2             0              0 Barneveld2 Nitrite_Blank
34  H05             2             0              0 Barneveld2 Nitrite_Blank
35  H11             2             0              0 Barneveld2 Nitrite_Blank
36  G05             2             0              0 Barneveld2 Nitrite_Blank
37  G11             2             0              0 Barneveld2 Nitrite_Blank
38  H05             2             0              0 Barneveld2 Nitrite_Blank
39  H11             2             0              0 Barneveld2 Nitrite_Blank
40  G05             2             0              0 Barneveld2 Nitrite_Blank
41  G11             2             0              0 Barneveld2 Nitrite_Blank
42  H05             2             0              0 Barneveld2 Nitrite_Blank
43  H11             2             0              0 Barneveld2 Nitrite_Blank
   Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1         Nitrite             None        0    0               0       20
2         Nitrite             None        0    0               0       20
3         Nitrite             None        0    0               0       20
4         Nitrite             None        0    0               0       20
5         Nitrite             None        0    0               0       20
6         Nitrite             None        0    0               0       20
7         Nitrite             None        0    0               0       20
8         Nitrite             None        0    0               0       20
9         Nitrite             None        0    0               0       20
10        Nitrite             None        0    0               0       20
11        Nitrite             None        0    0               0       20
12        Nitrite             None        0    0               0       20
13        Nitrite             None        0    0               0       20
14        Nitrite             None        0    0               0       20
15        Nitrite             None        0    0               0       20
16        Nitrite             None        0    0               0       20
17        Nitrite             None        0    0               0       20
18        Nitrite             None        0    0               0       20
19        Nitrite             None        0    0               0       20
20        Nitrite             None        0    0               0       20
21        Nitrite             None        0    0               0       20
22        Nitrite             None        0    0               0       20
23        Nitrite             None        0    0               0       20
24        Nitrite             None        0    0               0       20
25        Nitrite             None        0    0               0       20
26        Nitrite             None        0    0               0       20
27        Nitrite             None        0    0               0       20
28        Nitrite             None        0    0               0       20
29        Nitrite             None        0    0               0       20
30        Nitrite             None        0    0               0       20
31        Nitrite             None        0    0               0       20
32        Nitrite             None        0    0               0       20
33        Nitrite             None        0    0               0       20
34        Nitrite             None        0    0               0       20
35        Nitrite             None        0    0               0       20
36        Nitrite             None        0    0               0       20
37        Nitrite             None        0    0               0       20
38        Nitrite             None        0    0               0       20
39        Nitrite             None        0    0               0       20
40        Nitrite             None        0    0               0       20
41        Nitrite             None        0    0               0       20
42        Nitrite             None        0    0               0       20
43        Nitrite             None        0    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540   NO2_mM NO2NO3_mM    NO3_mM Time_minutes
1          T0   1.41350      0.77720 1.861654  1.750011 0.0000000            0
2          T0   1.32880      0.73255 1.743389  1.641157 0.0000000            0
3          T0   1.39225      0.76990 1.831887  1.732165 0.0000000            0
4          T0   1.36215      0.76205 1.789833  1.712996 0.0000000            0
5          T1   1.44475      0.77915 1.905547  1.754781 0.0000000          220
6          T1   1.42840      0.77420 1.882565  1.742675 0.0000000          220
7          T1   1.45255      0.78945 1.916525  1.780000 0.0000000          220
8          T1   1.44835      0.79525 1.910613  1.794218 0.0000000          220
9          T2   1.42290      0.77825 1.874842  1.752579 0.0000000          400
10         T2   1.40030      0.77645 1.843156  1.748176 0.0000000          400
11         T2   1.38015      0.75945 1.814966  1.706652 0.0000000          400
12         T2   1.38400      0.76415 1.820348  1.718122 0.0000000          400
13         T3   1.43895      0.78000 1.897390  1.756861 0.0000000          575
14         T3   1.46330      0.79140 1.931669  1.784779 0.0000000          575
15         T3   1.42025      0.76970 1.871123  1.731677 0.0000000          575
16         T3   1.50105      0.81135 1.984984  1.833749 0.0000000          575
17         T4   1.41710      0.77300 1.866703  1.739741 0.0000000          980
18         T4   1.44010      0.78000 1.899007  1.756861 0.0000000          980
19         T4   1.44025      0.78015 1.899218  1.757228 0.0000000          980
20         T4   1.42870      0.77765 1.882986  1.751111 0.0000000          980
21         T5   1.43415      0.76975 1.890643  1.731799 0.0000000         1365
22         T5   1.46285      0.78230 1.931035  1.762490 0.0000000         1365
23         T5   1.41575      0.78475 1.864810  1.768488 0.0000000         1365
24         T6   1.44800      0.78375 1.910120  1.766039 0.0000000         1732
25         T6   1.44170      0.77325 1.901257  1.740352 0.0000000         1732
26         T6   1.45700      0.77915 1.922792  1.754781 0.0000000         1732
27         T6   1.46050      0.77735 1.927723  1.750378 0.0000000         1732
28         T7   1.33270      0.71170 1.748812  1.590567 0.0000000         2300
29         T7   1.44135      0.76585 1.900765  1.722273 0.0000000         2300
30         T7   1.46940      0.78025 1.940270  1.757472 0.0000000         2300
31         T7   1.41385      0.75080 1.862145  1.685564 0.0000000         2300
32         T8   1.56635      0.81720 2.077707  1.848136 0.0000000         2970
33         T8   1.45110      0.88585 1.914484  2.017913 0.1034296         2970
34         T8   1.52040      0.78130 2.012394  1.760042 0.0000000         2970
35         T8   1.51060      0.77360 1.998505  1.741208 0.0000000         2970
36         T9   1.50165      0.78385 1.985833  1.766284 0.0000000         4460
37         T9   1.55115      0.80510 2.056067  1.818392 0.0000000         4460
38         T9   1.52895      0.80420 2.024523  1.816182 0.0000000         4460
39         T9   1.52765      0.79790 2.022678  1.800718 0.0000000         4460
40        T10   1.68605      0.88410 2.249358  2.013563 0.0000000         6160
41        T10   1.68135      0.87885 2.242576  2.000521 0.0000000         6160
42        T10   1.54770      0.80200 2.051160  1.810780 0.0000000         6160
43        T10   1.60060      0.82440 2.126597  1.865860 0.0000000         6160
   Time_hours Time_days H_mol   Volume H_Molarity H_mM
1         0.0       0.0     0 1.871275          0    0
2         0.0       0.0     0 1.871275          0    0
3         0.0       0.0     0 1.871275          0    0
4         0.0       0.0     0 1.871275          0    0
5         3.7       0.2     0 1.871275          0    0
6         3.7       0.2     0 1.871275          0    0
7         3.7       0.2     0 1.871275          0    0
8         3.7       0.2     0 1.871275          0    0
9         6.7       0.3     0 1.871275          0    0
10        6.7       0.3     0 1.871275          0    0
11        6.7       0.3     0 1.871275          0    0
12        6.7       0.3     0 1.871275          0    0
13        9.6       0.4     0 1.871275          0    0
14        9.6       0.4     0 1.871275          0    0
15        9.6       0.4     0 1.871275          0    0
16        9.6       0.4     0 1.871275          0    0
17       16.3       0.7     0 1.871275          0    0
18       16.3       0.7     0 1.871275          0    0
19       16.3       0.7     0 1.871275          0    0
20       16.3       0.7     0 1.871275          0    0
21       22.8       0.9     0 1.871275          0    0
22       22.8       0.9     0 1.871275          0    0
23       22.8       0.9     0 1.871275          0    0
24       28.9       1.2     0 1.871275          0    0
25       28.9       1.2     0 1.871275          0    0
26       28.9       1.2     0 1.871275          0    0
27       28.9       1.2     0 1.871275          0    0
28       38.3       1.6     0 1.871275          0    0
29       38.3       1.6     0 1.871275          0    0
30       38.3       1.6     0 1.871275          0    0
31       38.3       1.6     0 1.871275          0    0
32       49.5       2.1     0 1.871275          0    0
33       49.5       2.1     0 1.871275          0    0
34       49.5       2.1     0 1.871275          0    0
35       49.5       2.1     0 1.871275          0    0
36       74.3       3.1     0 1.871275          0    0
37       74.3       3.1     0 1.871275          0    0
38       74.3       3.1     0 1.871275          0    0
39       74.3       3.1     0 1.871275          0    0
40      102.7       4.3     0 1.871275          0    0
41      102.7       4.3     0 1.871275          0    0
42      102.7       4.3     0 1.871275          0    0
43      102.7       4.3     0 1.871275          0    0</code></pre>
<pre class="r"><code># close to zero

# 1. Apply moisture correction factor (correcting for moisture in soil)
soil_spike_ratio = 0.5 # soil weight(0.85g) / spike in volume (1.7ml) 

mcf_1 = (soil_spike_ratio*(moisture_percent_1/100) + 1)
mcf_1</code></pre>
<pre><code>[1] 1.10075</code></pre>
<pre class="r"><code># mcf_2 = (soil_spike_ratio*(moisture_percent_2/100) + 1)
# mcf_2
# mcf_3 = (soil_spike_ratio*(moisture_percent_3/100) + 1)
# mcf_3

# apply moisture factor to each soil
dim(df_p)</code></pre>
<pre><code>[1] 1054   25</code></pre>
<pre class="r"><code>df_p %&gt;% select(Soil) %&gt;% unique()</code></pre>
<pre><code>        Soil
1 Barneveld2</code></pre>
<pre class="r"><code># this is a special case with no replicates
df_p_mcf_1 &lt;- df_p %&gt;% filter(Soil == &quot;Barneveld2&quot;) %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_1, NO2_mM = NO2_mM * mcf_1)
dim(df_p_mcf_1) #923</code></pre>
<pre><code>[1] 923  25</code></pre>
<pre class="r"><code># df_p_mcf_2 &lt;- df_p %&gt;% filter(Soil == &quot;LaBaghWoods_autoclaved_pH6.66&quot;) %&gt;%  filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_2, NO2_mM = NO2_mM * mcf_2)
# dim(df_p_mcf_2) #426
# df_p_mcf_3 &lt;- df_p %&gt;% filter(Soil == &quot;Crerar1_pH8.05&quot;) %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;,&quot;No_Nitrate&quot;)) %&gt;% mutate(NO3_mM = NO3_mM * mcf_3, NO2_mM = NO2_mM * mcf_3)
# dim(df_p_mcf_3) #426

df_p_others_mcf &lt;- df_p %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;, &quot;Nitrate_Blank&quot;, &quot;Ammonium_Blank&quot;)) # Use Nitrate_Blank instead in the future
dim(df_p_others_mcf) #132</code></pre>
<pre><code>[1] 131  25</code></pre>
<pre class="r"><code>df_p_mcf &lt;- rbind(df_p_mcf_1, df_p_others_mcf)
dim(df_p_mcf) # 1055</code></pre>
<pre><code>[1] 1054   25</code></pre>
<pre class="r"><code># plot to see
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hours)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;After averaging with biological replicates \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-3-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. Apply blank correction factor (drying effect during incubation)
# Blank reads
df_no3_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrate_Blank&quot;)
df_no3_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input       Soil   Sample_type
1   G06             0             2              0 Barneveld2 Nitrate_Blank
2   G12             0             2              0 Barneveld2 Nitrate_Blank
3   H06             0             2              0 Barneveld2 Nitrate_Blank
4   H12             0             2              0 Barneveld2 Nitrate_Blank
5   G06             0             2              0 Barneveld2 Nitrate_Blank
6   G12             0             2              0 Barneveld2 Nitrate_Blank
7   H06             0             2              0 Barneveld2 Nitrate_Blank
8   H12             0             2              0 Barneveld2 Nitrate_Blank
9   G06             0             2              0 Barneveld2 Nitrate_Blank
10  G12             0             2              0 Barneveld2 Nitrate_Blank
11  H06             0             2              0 Barneveld2 Nitrate_Blank
12  H12             0             2              0 Barneveld2 Nitrate_Blank
13  G06             0             2              0 Barneveld2 Nitrate_Blank
14  G12             0             2              0 Barneveld2 Nitrate_Blank
15  H06             0             2              0 Barneveld2 Nitrate_Blank
16  H12             0             2              0 Barneveld2 Nitrate_Blank
17  G06             0             2              0 Barneveld2 Nitrate_Blank
18  G12             0             2              0 Barneveld2 Nitrate_Blank
19  H06             0             2              0 Barneveld2 Nitrate_Blank
20  H12             0             2              0 Barneveld2 Nitrate_Blank
21  G06             0             2              0 Barneveld2 Nitrate_Blank
22  G12             0             2              0 Barneveld2 Nitrate_Blank
23  H06             0             2              0 Barneveld2 Nitrate_Blank
24  H12             0             2              0 Barneveld2 Nitrate_Blank
25  G06             0             2              0 Barneveld2 Nitrate_Blank
26  G12             0             2              0 Barneveld2 Nitrate_Blank
27  H06             0             2              0 Barneveld2 Nitrate_Blank
28  H12             0             2              0 Barneveld2 Nitrate_Blank
29  G06             0             2              0 Barneveld2 Nitrate_Blank
30  G12             0             2              0 Barneveld2 Nitrate_Blank
31  H06             0             2              0 Barneveld2 Nitrate_Blank
32  H12             0             2              0 Barneveld2 Nitrate_Blank
33  G06             0             2              0 Barneveld2 Nitrate_Blank
34  G12             0             2              0 Barneveld2 Nitrate_Blank
35  H06             0             2              0 Barneveld2 Nitrate_Blank
36  H12             0             2              0 Barneveld2 Nitrate_Blank
37  G06             0             2              0 Barneveld2 Nitrate_Blank
38  G12             0             2              0 Barneveld2 Nitrate_Blank
39  H06             0             2              0 Barneveld2 Nitrate_Blank
40  H12             0             2              0 Barneveld2 Nitrate_Blank
41  G06             0             2              0 Barneveld2 Nitrate_Blank
42  G12             0             2              0 Barneveld2 Nitrate_Blank
43  H06             0             2              0 Barneveld2 Nitrate_Blank
44  H12             0             2              0 Barneveld2 Nitrate_Blank
   Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1         Nitrate             None        0    0               0       20
2         Nitrate             None        0    0               0       20
3         Nitrate             None        0    0               0       20
4         Nitrate             None        0    0               0       20
5         Nitrate             None        0    0               0       20
6         Nitrate             None        0    0               0       20
7         Nitrate             None        0    0               0       20
8         Nitrate             None        0    0               0       20
9         Nitrate             None        0    0               0       20
10        Nitrate             None        0    0               0       20
11        Nitrate             None        0    0               0       20
12        Nitrate             None        0    0               0       20
13        Nitrate             None        0    0               0       20
14        Nitrate             None        0    0               0       20
15        Nitrate             None        0    0               0       20
16        Nitrate             None        0    0               0       20
17        Nitrate             None        0    0               0       20
18        Nitrate             None        0    0               0       20
19        Nitrate             None        0    0               0       20
20        Nitrate             None        0    0               0       20
21        Nitrate             None        0    0               0       20
22        Nitrate             None        0    0               0       20
23        Nitrate             None        0    0               0       20
24        Nitrate             None        0    0               0       20
25        Nitrate             None        0    0               0       20
26        Nitrate             None        0    0               0       20
27        Nitrate             None        0    0               0       20
28        Nitrate             None        0    0               0       20
29        Nitrate             None        0    0               0       20
30        Nitrate             None        0    0               0       20
31        Nitrate             None        0    0               0       20
32        Nitrate             None        0    0               0       20
33        Nitrate             None        0    0               0       20
34        Nitrate             None        0    0               0       20
35        Nitrate             None        0    0               0       20
36        Nitrate             None        0    0               0       20
37        Nitrate             None        0    0               0       20
38        Nitrate             None        0    0               0       20
39        Nitrate             None        0    0               0       20
40        Nitrate             None        0    0               0       20
41        Nitrate             None        0    0               0       20
42        Nitrate             None        0    0               0       20
43        Nitrate             None        0    0               0       20
44        Nitrate             None        0    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540      NO2_mM NO2NO3_mM   NO3_mM
1          T0   0.00380      0.89365 0.010234713  2.037315 2.027080
2          T0   0.00120      0.84155 0.007024096  1.908156 1.901131
3          T0   0.00300      0.87300 0.009246760  1.986000 1.976754
4          T0   0.00105      0.83005 0.006838888  1.879782 1.872944
5          T1   0.00245      0.86005 0.008567579  1.953902 1.945334
6          T1   0.00160      0.85800 0.007517994  1.948827 1.941309
7          T1   0.00080      0.88425 0.006530214  2.013936 2.007406
8          T1   0.00300      0.87745 0.009246760  1.997045 1.987798
9          T2   0.00145      0.87245 0.007332780  1.984636 1.977303
10         T2   0.00215      0.88570 0.008197129  2.017540 2.009343
11         T2   0.00170      0.84995 0.007641471  1.928911 1.921270
12         T2   0.00055      0.84820 0.006221546  1.924585 1.918363
13         T3   0.00120      0.79690 0.007024096  1.798265 1.791241
14         T3   0.00205      0.86060 0.008073648  1.955264 1.947190
15         T3   0.00325      0.89995 0.009555488  2.053002 2.043446
16         T3   0.00315      0.86515 0.009431996  1.966536 1.957104
17         T4   0.00000      0.85090 0.005542497  1.931260 1.925718
18         T4   0.00240      0.83085 0.008505836  1.881755 1.873249
19         T4   0.00380      0.86510 0.010234713  1.966412 1.956177
20         T4   0.00510      0.83900 0.011840270  1.901860 1.890020
21         T5   0.00185      0.80020 0.007826688  1.806362 1.798535
22         T5   0.00220      0.78715 0.008258870  1.774365 1.766106
23         T5   0.00230      0.82905 0.008382353  1.877317 1.868935
24         T5   0.00205      0.80780 0.008073648  1.825024 1.816951
25         T6   0.00105      0.80925 0.006838888  1.828587 1.821748
26         T6   0.00085      0.76585 0.006591948  1.722273 1.715681
27         T6   0.00100      0.80355 0.006777153  1.814586 1.807808
28         T6   0.00080      0.79530 0.006530214  1.794341 1.787810
29         T7   0.00195      0.73035 0.007950167  1.635812 1.627862
30         T7   0.00155      0.76855 0.007456256  1.728867 1.721411
31         T7   0.00325      0.77405 0.009555488  1.742308 1.732752
32         T7   0.00215      0.73680 0.008197129  1.651488 1.643291
33         T8   0.00025      0.76445 0.005851152  1.718855 1.713003
34         T8   0.00030      0.81550 0.005912884  1.843954 1.838041
35         T8   0.00265      0.74330 0.008814550  1.667300 1.658485
36         T8   0.00105      0.67075 0.006838888  1.491648 1.484810
37         T9   0.00270      0.70630 0.008876294  1.577490 1.568613
38         T9   0.00125      0.68100 0.007085832  1.516354 1.509268
39         T9   0.00315      0.67430 0.009431996  1.500201 1.490769
40         T9   0.00270      0.65215 0.008876294  1.446909 1.438032
41        T10   0.00235      0.65500 0.008444094  1.453756 1.445312
42        T10   0.00270      0.61840 0.008876294  1.366028 1.357152
43        T10   0.00350      0.59825 0.009864223  1.317922 1.308058
44        T10   0.00475      0.61735 0.011407988  1.363518 1.352110
   Time_minutes Time_hours Time_days H_mol   Volume H_Molarity H_mM
1             0        0.0       0.0     0 1.871275          0    0
2             0        0.0       0.0     0 1.871275          0    0
3             0        0.0       0.0     0 1.871275          0    0
4             0        0.0       0.0     0 1.871275          0    0
5           220        3.7       0.2     0 1.871275          0    0
6           220        3.7       0.2     0 1.871275          0    0
7           220        3.7       0.2     0 1.871275          0    0
8           220        3.7       0.2     0 1.871275          0    0
9           400        6.7       0.3     0 1.871275          0    0
10          400        6.7       0.3     0 1.871275          0    0
11          400        6.7       0.3     0 1.871275          0    0
12          400        6.7       0.3     0 1.871275          0    0
13          575        9.6       0.4     0 1.871275          0    0
14          575        9.6       0.4     0 1.871275          0    0
15          575        9.6       0.4     0 1.871275          0    0
16          575        9.6       0.4     0 1.871275          0    0
17          980       16.3       0.7     0 1.871275          0    0
18          980       16.3       0.7     0 1.871275          0    0
19          980       16.3       0.7     0 1.871275          0    0
20          980       16.3       0.7     0 1.871275          0    0
21         1365       22.8       0.9     0 1.871275          0    0
22         1365       22.8       0.9     0 1.871275          0    0
23         1365       22.8       0.9     0 1.871275          0    0
24         1365       22.8       0.9     0 1.871275          0    0
25         1732       28.9       1.2     0 1.871275          0    0
26         1732       28.9       1.2     0 1.871275          0    0
27         1732       28.9       1.2     0 1.871275          0    0
28         1732       28.9       1.2     0 1.871275          0    0
29         2300       38.3       1.6     0 1.871275          0    0
30         2300       38.3       1.6     0 1.871275          0    0
31         2300       38.3       1.6     0 1.871275          0    0
32         2300       38.3       1.6     0 1.871275          0    0
33         2970       49.5       2.1     0 1.871275          0    0
34         2970       49.5       2.1     0 1.871275          0    0
35         2970       49.5       2.1     0 1.871275          0    0
36         2970       49.5       2.1     0 1.871275          0    0
37         4460       74.3       3.1     0 1.871275          0    0
38         4460       74.3       3.1     0 1.871275          0    0
39         4460       74.3       3.1     0 1.871275          0    0
40         4460       74.3       3.1     0 1.871275          0    0
41         6160      102.7       4.3     0 1.871275          0    0
42         6160      102.7       4.3     0 1.871275          0    0
43         6160      102.7       4.3     0 1.871275          0    0
44         6160      102.7       4.3     0 1.871275          0    0</code></pre>
<pre class="r"><code>df_no2_blank &lt;- df_p %&gt;% filter(Sample_type == &quot;Nitrite_Blank&quot;)
df_no2_blank</code></pre>
<pre><code>   Well Nitrite_input Nitrate_input Ammonium_input       Soil   Sample_type
1   G05             2             0              0 Barneveld2 Nitrite_Blank
2   G11             2             0              0 Barneveld2 Nitrite_Blank
3   H05             2             0              0 Barneveld2 Nitrite_Blank
4   H11             2             0              0 Barneveld2 Nitrite_Blank
5   G05             2             0              0 Barneveld2 Nitrite_Blank
6   G11             2             0              0 Barneveld2 Nitrite_Blank
7   H05             2             0              0 Barneveld2 Nitrite_Blank
8   H11             2             0              0 Barneveld2 Nitrite_Blank
9   G05             2             0              0 Barneveld2 Nitrite_Blank
10  G11             2             0              0 Barneveld2 Nitrite_Blank
11  H05             2             0              0 Barneveld2 Nitrite_Blank
12  H11             2             0              0 Barneveld2 Nitrite_Blank
13  G05             2             0              0 Barneveld2 Nitrite_Blank
14  G11             2             0              0 Barneveld2 Nitrite_Blank
15  H05             2             0              0 Barneveld2 Nitrite_Blank
16  H11             2             0              0 Barneveld2 Nitrite_Blank
17  G05             2             0              0 Barneveld2 Nitrite_Blank
18  G11             2             0              0 Barneveld2 Nitrite_Blank
19  H05             2             0              0 Barneveld2 Nitrite_Blank
20  H11             2             0              0 Barneveld2 Nitrite_Blank
21  G11             2             0              0 Barneveld2 Nitrite_Blank
22  H05             2             0              0 Barneveld2 Nitrite_Blank
23  H11             2             0              0 Barneveld2 Nitrite_Blank
24  G05             2             0              0 Barneveld2 Nitrite_Blank
25  G11             2             0              0 Barneveld2 Nitrite_Blank
26  H05             2             0              0 Barneveld2 Nitrite_Blank
27  H11             2             0              0 Barneveld2 Nitrite_Blank
28  G05             2             0              0 Barneveld2 Nitrite_Blank
29  G11             2             0              0 Barneveld2 Nitrite_Blank
30  H05             2             0              0 Barneveld2 Nitrite_Blank
31  H11             2             0              0 Barneveld2 Nitrite_Blank
32  G05             2             0              0 Barneveld2 Nitrite_Blank
33  G11             2             0              0 Barneveld2 Nitrite_Blank
34  H05             2             0              0 Barneveld2 Nitrite_Blank
35  H11             2             0              0 Barneveld2 Nitrite_Blank
36  G05             2             0              0 Barneveld2 Nitrite_Blank
37  G11             2             0              0 Barneveld2 Nitrite_Blank
38  H05             2             0              0 Barneveld2 Nitrite_Blank
39  H11             2             0              0 Barneveld2 Nitrite_Blank
40  G05             2             0              0 Barneveld2 Nitrite_Blank
41  G11             2             0              0 Barneveld2 Nitrite_Blank
42  H05             2             0              0 Barneveld2 Nitrite_Blank
43  H11             2             0              0 Barneveld2 Nitrite_Blank
   Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1         Nitrite             None        0    0               0       20
2         Nitrite             None        0    0               0       20
3         Nitrite             None        0    0               0       20
4         Nitrite             None        0    0               0       20
5         Nitrite             None        0    0               0       20
6         Nitrite             None        0    0               0       20
7         Nitrite             None        0    0               0       20
8         Nitrite             None        0    0               0       20
9         Nitrite             None        0    0               0       20
10        Nitrite             None        0    0               0       20
11        Nitrite             None        0    0               0       20
12        Nitrite             None        0    0               0       20
13        Nitrite             None        0    0               0       20
14        Nitrite             None        0    0               0       20
15        Nitrite             None        0    0               0       20
16        Nitrite             None        0    0               0       20
17        Nitrite             None        0    0               0       20
18        Nitrite             None        0    0               0       20
19        Nitrite             None        0    0               0       20
20        Nitrite             None        0    0               0       20
21        Nitrite             None        0    0               0       20
22        Nitrite             None        0    0               0       20
23        Nitrite             None        0    0               0       20
24        Nitrite             None        0    0               0       20
25        Nitrite             None        0    0               0       20
26        Nitrite             None        0    0               0       20
27        Nitrite             None        0    0               0       20
28        Nitrite             None        0    0               0       20
29        Nitrite             None        0    0               0       20
30        Nitrite             None        0    0               0       20
31        Nitrite             None        0    0               0       20
32        Nitrite             None        0    0               0       20
33        Nitrite             None        0    0               0       20
34        Nitrite             None        0    0               0       20
35        Nitrite             None        0    0               0       20
36        Nitrite             None        0    0               0       20
37        Nitrite             None        0    0               0       20
38        Nitrite             None        0    0               0       20
39        Nitrite             None        0    0               0       20
40        Nitrite             None        0    0               0       20
41        Nitrite             None        0    0               0       20
42        Nitrite             None        0    0               0       20
43        Nitrite             None        0    0               0       20
   Time_point NO2_OD540 NO2NO3_OD540   NO2_mM NO2NO3_mM    NO3_mM Time_minutes
1          T0   1.41350      0.77720 1.861654  1.750011 0.0000000            0
2          T0   1.32880      0.73255 1.743389  1.641157 0.0000000            0
3          T0   1.39225      0.76990 1.831887  1.732165 0.0000000            0
4          T0   1.36215      0.76205 1.789833  1.712996 0.0000000            0
5          T1   1.44475      0.77915 1.905547  1.754781 0.0000000          220
6          T1   1.42840      0.77420 1.882565  1.742675 0.0000000          220
7          T1   1.45255      0.78945 1.916525  1.780000 0.0000000          220
8          T1   1.44835      0.79525 1.910613  1.794218 0.0000000          220
9          T2   1.42290      0.77825 1.874842  1.752579 0.0000000          400
10         T2   1.40030      0.77645 1.843156  1.748176 0.0000000          400
11         T2   1.38015      0.75945 1.814966  1.706652 0.0000000          400
12         T2   1.38400      0.76415 1.820348  1.718122 0.0000000          400
13         T3   1.43895      0.78000 1.897390  1.756861 0.0000000          575
14         T3   1.46330      0.79140 1.931669  1.784779 0.0000000          575
15         T3   1.42025      0.76970 1.871123  1.731677 0.0000000          575
16         T3   1.50105      0.81135 1.984984  1.833749 0.0000000          575
17         T4   1.41710      0.77300 1.866703  1.739741 0.0000000          980
18         T4   1.44010      0.78000 1.899007  1.756861 0.0000000          980
19         T4   1.44025      0.78015 1.899218  1.757228 0.0000000          980
20         T4   1.42870      0.77765 1.882986  1.751111 0.0000000          980
21         T5   1.43415      0.76975 1.890643  1.731799 0.0000000         1365
22         T5   1.46285      0.78230 1.931035  1.762490 0.0000000         1365
23         T5   1.41575      0.78475 1.864810  1.768488 0.0000000         1365
24         T6   1.44800      0.78375 1.910120  1.766039 0.0000000         1732
25         T6   1.44170      0.77325 1.901257  1.740352 0.0000000         1732
26         T6   1.45700      0.77915 1.922792  1.754781 0.0000000         1732
27         T6   1.46050      0.77735 1.927723  1.750378 0.0000000         1732
28         T7   1.33270      0.71170 1.748812  1.590567 0.0000000         2300
29         T7   1.44135      0.76585 1.900765  1.722273 0.0000000         2300
30         T7   1.46940      0.78025 1.940270  1.757472 0.0000000         2300
31         T7   1.41385      0.75080 1.862145  1.685564 0.0000000         2300
32         T8   1.56635      0.81720 2.077707  1.848136 0.0000000         2970
33         T8   1.45110      0.88585 1.914484  2.017913 0.1034296         2970
34         T8   1.52040      0.78130 2.012394  1.760042 0.0000000         2970
35         T8   1.51060      0.77360 1.998505  1.741208 0.0000000         2970
36         T9   1.50165      0.78385 1.985833  1.766284 0.0000000         4460
37         T9   1.55115      0.80510 2.056067  1.818392 0.0000000         4460
38         T9   1.52895      0.80420 2.024523  1.816182 0.0000000         4460
39         T9   1.52765      0.79790 2.022678  1.800718 0.0000000         4460
40        T10   1.68605      0.88410 2.249358  2.013563 0.0000000         6160
41        T10   1.68135      0.87885 2.242576  2.000521 0.0000000         6160
42        T10   1.54770      0.80200 2.051160  1.810780 0.0000000         6160
43        T10   1.60060      0.82440 2.126597  1.865860 0.0000000         6160
   Time_hours Time_days H_mol   Volume H_Molarity H_mM
1         0.0       0.0     0 1.871275          0    0
2         0.0       0.0     0 1.871275          0    0
3         0.0       0.0     0 1.871275          0    0
4         0.0       0.0     0 1.871275          0    0
5         3.7       0.2     0 1.871275          0    0
6         3.7       0.2     0 1.871275          0    0
7         3.7       0.2     0 1.871275          0    0
8         3.7       0.2     0 1.871275          0    0
9         6.7       0.3     0 1.871275          0    0
10        6.7       0.3     0 1.871275          0    0
11        6.7       0.3     0 1.871275          0    0
12        6.7       0.3     0 1.871275          0    0
13        9.6       0.4     0 1.871275          0    0
14        9.6       0.4     0 1.871275          0    0
15        9.6       0.4     0 1.871275          0    0
16        9.6       0.4     0 1.871275          0    0
17       16.3       0.7     0 1.871275          0    0
18       16.3       0.7     0 1.871275          0    0
19       16.3       0.7     0 1.871275          0    0
20       16.3       0.7     0 1.871275          0    0
21       22.8       0.9     0 1.871275          0    0
22       22.8       0.9     0 1.871275          0    0
23       22.8       0.9     0 1.871275          0    0
24       28.9       1.2     0 1.871275          0    0
25       28.9       1.2     0 1.871275          0    0
26       28.9       1.2     0 1.871275          0    0
27       28.9       1.2     0 1.871275          0    0
28       38.3       1.6     0 1.871275          0    0
29       38.3       1.6     0 1.871275          0    0
30       38.3       1.6     0 1.871275          0    0
31       38.3       1.6     0 1.871275          0    0
32       49.5       2.1     0 1.871275          0    0
33       49.5       2.1     0 1.871275          0    0
34       49.5       2.1     0 1.871275          0    0
35       49.5       2.1     0 1.871275          0    0
36       74.3       3.1     0 1.871275          0    0
37       74.3       3.1     0 1.871275          0    0
38       74.3       3.1     0 1.871275          0    0
39       74.3       3.1     0 1.871275          0    0
40      102.7       4.3     0 1.871275          0    0
41      102.7       4.3     0 1.871275          0    0
42      102.7       4.3     0 1.871275          0    0
43      102.7       4.3     0 1.871275          0    0</code></pre>
<pre class="r"><code># df_aero_blank$Nitrate_input &lt;- as.numeric(as.character(df_aero_blank$Nitrate_input))
df_no2_blank %&lt;&gt;% mutate(Correction_factor_NO2 = (Nitrite_input / NO2_mM)) 
df_no3_blank %&lt;&gt;% mutate(Correction_factor_NO3 = (Nitrate_input / NO3_mM)) 

# average the blanks by time point
cf_no2 &lt;- df_no2_blank %&gt;% select(Soil, Time_point, Correction_factor_NO2) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO2 = mean(Correction_factor_NO2)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>cf_no3 &lt;- df_no3_blank %&gt;% select(Soil, Time_point, Correction_factor_NO3) %&gt;% group_by(Soil, Time_point) %&gt;% summarize(Correction_factor_NO3 = mean(Correction_factor_NO3)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Soil&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code># why is nitrate decreasing????


# left join and multiply the correction factor
# left join to samples
df_sample_mcf &lt;- df_p_mcf %&gt;% filter(!(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;Ammonium_Blank&quot;)))
dim(df_sample_mcf) #1065</code></pre>
<pre><code>[1] 923  25</code></pre>
<pre class="r"><code>df_sample_mcf$Sample_type %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Slurry&quot;</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_mcf %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_sample_bcf)</code></pre>
<pre><code>[1] 923  27</code></pre>
<pre class="r"><code>df_sample_bcf &lt;- df_sample_bcf %&gt;% mutate(NO3_mM = NO3_mM * Correction_factor_NO3, NO2_mM = NO2_mM * Correction_factor_NO2)
dim(df_sample_bcf) #849</code></pre>
<pre><code>[1] 923  27</code></pre>
<pre class="r"><code># merge dataframe with blank just for convenience
df_p_blanks_bcf &lt;- df_p_mcf %&gt;% filter(Sample_type %in% c(&quot;Nitrite_Blank&quot;,&quot;Nitrate_Blank&quot;,&quot;Ammonium_Blank&quot;)) %&gt;% left_join(cf_no2, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;)) %&gt;% left_join(cf_no3, by = c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Time_point&quot;=&quot;Time_point&quot;))
dim(df_p_blanks_bcf) #197 26</code></pre>
<pre><code>[1] 131  27</code></pre>
<pre class="r"><code>dim(df_sample_bcf) # 849 26</code></pre>
<pre><code>[1] 923  27</code></pre>
<pre class="r"><code>df_p_bcf &lt;- rbind(df_sample_bcf, df_p_blanks_bcf)
dim(df_p_bcf) # 1055</code></pre>
<pre><code>[1] 1054   27</code></pre>
<pre class="r"><code>## innate nitrate
df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot; &amp; Time_point == &quot;LBA_T0&quot;) %&gt;% select(NO3_mM) %&gt;% unlist() %&gt;% mean()</code></pre>
<pre><code>[1] NaN</code></pre>
<pre class="r"><code>df_p_bcf %&gt;% filter(Titration_type == &quot;No_Nitrate&quot; &amp; Time_point == &quot;Sterile_T0&quot;) %&gt;% select(NO3_mM) %&gt;% unlist() %&gt;% mean()</code></pre>
<pre><code>[1] NaN</code></pre>
<pre class="r"><code># export
df_p_bcf$Time_point</code></pre>
<pre><code>   [1] T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0 
  [19] T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0 
  [37] T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0 
  [55] T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0 
  [73] T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T1  T1  T1  T1  T1  T1  T1 
  [91] T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1 
 [109] T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1 
 [127] T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1 
 [145] T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1 
 [163] T1  T1  T1  T1  T1  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2 
 [181] T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2 
 [199] T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2 
 [217] T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2 
 [235] T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T2  T3 
 [253] T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3 
 [271] T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3 
 [289] T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3 
 [307] T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3 
 [325] T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T4  T4  T4  T4  T4  T4  T4 
 [343] T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4 
 [361] T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4 
 [379] T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4 
 [397] T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4 
 [415] T4  T4  T4  T4  T4  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5 
 [433] T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5 
 [451] T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5 
 [469] T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5 
 [487] T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T5  T6 
 [505] T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6 
 [523] T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6 
 [541] T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6 
 [559] T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6 
 [577] T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T7  T7  T7  T7  T7  T7  T7 
 [595] T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7 
 [613] T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7 
 [631] T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7 
 [649] T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T7 
 [667] T7  T7  T7  T7  T7  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8 
 [685] T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8 
 [703] T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8 
 [721] T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8 
 [739] T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T8  T9 
 [757] T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9 
 [775] T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9 
 [793] T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9 
 [811] T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9 
 [829] T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T10 T10 T10 T10 T10 T10 T10
 [847] T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10
 [865] T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10
 [883] T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10
 [901] T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10 T10
 [919] T10 T10 T10 T10 T10 T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T0  T1 
 [937] T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T1  T2  T2  T2  T2  T2  T2  T2 
 [955] T2  T2  T2  T2  T2  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T3  T4 
 [973] T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T4  T5  T5  T5  T5  T5  T5  T5 
 [991] T5  T5  T5  T5  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T6  T7  T7 
[1009] T7  T7  T7  T7  T7  T7  T7  T7  T7  T7  T8  T8  T8  T8  T8  T8  T8  T8 
[1027] T8  T8  T8  T8  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T9  T10 T10
[1045] T10 T10 T10 T10 T10 T10 T10 T10 T10 T10
Levels: T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10</code></pre>
<pre class="r"><code># removing timepoint T8
df_kyle &lt;- df_p_bcf
df_kyle$Time_point %&gt;% unique()</code></pre>
<pre><code> [1] T0  T1  T2  T3  T4  T5  T6  T7  T8  T9  T10
Levels: T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10</code></pre>
<pre class="r"><code>dim(df_kyle)</code></pre>
<pre><code>[1] 1054   27</code></pre>
<pre class="r"><code># write.xlsx(df_kyle, &quot;220519_Griess_antibiotics_experiment_to_kyle.xlsx&quot;)

df_kyle2 &lt;- df_kyle %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;, &quot;HCl&quot;))
dim(df_kyle2)</code></pre>
<pre><code>[1] 923  27</code></pre>
<pre class="r"><code># write.xlsx(df_kyle2, &quot;220519_Griess_antibiotics_dose_experiment_remove_blanks.xlsx&quot;)</code></pre>
</div>
<div id="sanity-check-moisture-correction-blank-correction-factor" class="section level2">
<h2>3. Sanity check: moisture correction &amp; blank correction factor</h2>
<p>Blank correction<br />
Use the nitrate concentration of the blank and use the ratio.<br />
For example, multiply to nitrate concentration x (2mM / changed [NO3])<br />
</p>
<pre class="r"><code># without any correction
# nitrate blanks
dim(df_no3_blank)</code></pre>
<pre><code>[1] 44 26</code></pre>
<pre class="r"><code>ggplot(df_no3_blank, aes(x=Time_hours, y=NO3_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.5,0.5), limits=c(0, 2.5))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrate blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite blanks
dim(df_no2_blank)</code></pre>
<pre><code>[1] 43 26</code></pre>
<pre class="r"><code>ggplot(df_no2_blank, aes(x=Time_hours, y=NO2_mM, color=Soil, group=Soil)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  scale_y_continuous(breaks = seq(0,2.7,0.5), limits=c(0, 2.7))+
  # scale_x_continuous(breaks = seq(0,2.1,0.25), limits=c(0, 2.1))+
  ggtitle(&quot;Nitrite blank \n&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># without any correction
ggplot(df_p, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(Without any correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### What are those points with very high std?
colnames(df_p)</code></pre>
<pre><code> [1] &quot;Well&quot;             &quot;Nitrite_input&quot;    &quot;Nitrate_input&quot;    &quot;Ammonium_input&quot;  
 [5] &quot;Soil&quot;             &quot;Sample_type&quot;      &quot;Titration_type&quot;   &quot;Antibiotics_type&quot;
 [9] &quot;Dose_ppm&quot;         &quot;Unit&quot;             &quot;Concentration_M&quot;  &quot;Added_ul&quot;        
[13] &quot;Time_point&quot;       &quot;NO2_OD540&quot;        &quot;NO2NO3_OD540&quot;     &quot;NO2_mM&quot;          
[17] &quot;NO2NO3_mM&quot;        &quot;NO3_mM&quot;           &quot;Time_minutes&quot;     &quot;Time_hours&quot;      
[21] &quot;Time_days&quot;        &quot;H_mol&quot;            &quot;Volume&quot;           &quot;H_Molarity&quot;      
[25] &quot;H_mM&quot;            </code></pre>
<pre class="r"><code># df_p %&gt;% arrange(desc(Std_NO3_mM)) %&gt;% select(Soil, Titration_type, Concentration_M, Sample_type, Time_point, Std_NO3_mM) 
## based on this I corrected the wrongly removed E02 -&lt; removed E01 (filter burst). Eliminated point P11 G11.

# After moisture correction
ggplot(df_p_mcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># After blank correction (evaporation) 
ggplot(df_p_bcf, aes(x=Time_hours, y=NO3_mM, color=Sample_type, group=Sample_type)) +
  geom_point(size=2.5, shape=21) +
  # geom_line(size=1)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_color_brewer(palette=&#39;Set2&#39;) +
  # scale_color_manual(values = c(&quot;deepskyblue4&quot;,&quot;maroon2&quot;))+
  ylab(&quot;Measured NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  # scale_x_continuous(breaks = seq(0,5.1,1), limits=c(0, 5.1))+
  ggtitle(&quot;(After moisture + blank correction) All samples \n&quot;) +
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## what??
df_p_bcf %&gt;% filter(NO2_mM &gt; 2)</code></pre>
<pre><code>  Well Nitrite_input Nitrate_input Ammonium_input       Soil   Sample_type
1  G05             2             0              0 Barneveld2 Nitrite_Blank
2  H05             2             0              0 Barneveld2 Nitrite_Blank
3  G11             2             0              0 Barneveld2 Nitrite_Blank
4  H05             2             0              0 Barneveld2 Nitrite_Blank
5  H11             2             0              0 Barneveld2 Nitrite_Blank
6  G05             2             0              0 Barneveld2 Nitrite_Blank
7  G11             2             0              0 Barneveld2 Nitrite_Blank
8  H05             2             0              0 Barneveld2 Nitrite_Blank
9  H11             2             0              0 Barneveld2 Nitrite_Blank
  Titration_type Antibiotics_type Dose_ppm Unit Concentration_M Added_ul
1        Nitrite             None        0    0               0       20
2        Nitrite             None        0    0               0       20
3        Nitrite             None        0    0               0       20
4        Nitrite             None        0    0               0       20
5        Nitrite             None        0    0               0       20
6        Nitrite             None        0    0               0       20
7        Nitrite             None        0    0               0       20
8        Nitrite             None        0    0               0       20
9        Nitrite             None        0    0               0       20
  Time_point NO2_OD540 NO2NO3_OD540   NO2_mM NO2NO3_mM NO3_mM Time_minutes
1         T8   1.56635      0.81720 2.077707  1.848136      0         2970
2         T8   1.52040      0.78130 2.012394  1.760042      0         2970
3         T9   1.55115      0.80510 2.056067  1.818392      0         4460
4         T9   1.52895      0.80420 2.024523  1.816182      0         4460
5         T9   1.52765      0.79790 2.022678  1.800718      0         4460
6        T10   1.68605      0.88410 2.249358  2.013563      0         6160
7        T10   1.68135      0.87885 2.242576  2.000521      0         6160
8        T10   1.54770      0.80200 2.051160  1.810780      0         6160
9        T10   1.60060      0.82440 2.126597  1.865860      0         6160
  Time_hours Time_days H_mol   Volume H_Molarity H_mM Correction_factor_NO2
1       49.5       2.1     0 1.871275          0    0             1.0004642
2       49.5       2.1     0 1.871275          0    0             1.0004642
3       74.3       3.1     0 1.871275          0    0             0.9891350
4       74.3       3.1     0 1.871275          0    0             0.9891350
5       74.3       3.1     0 1.871275          0    0             0.9891350
6      102.7       4.3     0 1.871275          0    0             0.9241254
7      102.7       4.3     0 1.871275          0    0             0.9241254
8      102.7       4.3     0 1.871275          0    0             0.9241254
9      102.7       4.3     0 1.871275          0    0             0.9241254
  Correction_factor_NO3
1              1.202137
2              1.202137
3              1.333134
4              1.333134
5              1.333134
6              1.466403
7              1.466403
8              1.466403
9              1.466403</code></pre>
</div>
<div id="nitrogen-dynamics" class="section level2">
<h2>4. Nitrogen dynamics</h2>
</div>
<div id="ph-perturbation-experiment." class="section level2">
<h2>4.1. pH perturbation experiment.</h2>
<ul>
<li>Is there a difference in pH perturbation levels?</li>
</ul>
<pre class="r"><code># pH color
col_pH &lt;- colorRampPalette(c(&quot;gold&quot;,&quot;red&quot;,&quot;purple&quot;))

library(colorRamps)
colorRamps::green2red</code></pre>
<pre><code>function (n) 
rgb.tables(n, red = c(1, 0, 2), green = c(0, 0, 2), blue = c(0, 
    0, 0, 0))
&lt;bytecode: 0x0000000024900b78&gt;
&lt;environment: namespace:colorRamps&gt;</code></pre>
<pre class="r"><code>plot(rep(1,13),col=col_pH(13),pch=19,cex=3)</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>grad_pH &lt;- scale_colour_gradientn(colours = col_pH(100))

# Confer this page (https://stackoverflow.com/questions/21537782/how-to-set-fixed-continuous-colour-values-in-ggplot2)
# myPalette &lt;- colorRampPalette(rev(brewer.pal(11, &quot;Spectral&quot;)))
# sc &lt;- scale_colour_gradientn(colours = myPalette(100), limits=c(1, 8))

# 1. pH perturbation
# plot Nitrate

# only autoclaved soil is relevant for this analysis
df_p_bcf$Soil &lt;-  factor(df_p_bcf$Soil, levels = c(&quot;Barneveld2&quot;))

df_pH &lt;- df_p_bcf %&gt;% filter(Titration_type %in% c(&quot;NaOH&quot;,&quot;HCl&quot;)) 
dim(df_pH) # 393</code></pre>
<pre><code>[1] 923  27</code></pre>
<pre class="r"><code># average the no acid base one.
df_plot &lt;- df_pH %&gt;% select(H_mM, Titration_type, Unit, NO3_mM, NO2_mM, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm)
df_plot %&gt;% filter(H_mM == 0) %&gt;% dim()</code></pre>
<pre><code>[1] 220  10</code></pre>
<pre class="r"><code>df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;, &#39;Antibiotics_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 187 x 10
    H_mM Titration_type  Unit Soil       Time_hours Time_point Antibiotics_type
   &lt;dbl&gt; &lt;chr&gt;          &lt;dbl&gt; &lt;fct&gt;           &lt;dbl&gt; &lt;fct&gt;      &lt;chr&gt;           
 1     0 HCl                0 Barneveld2          0 T0         CHL             
 2     0 HCl                0 Barneveld2          0 T0         CHL             
 3     0 HCl                0 Barneveld2          0 T0         CHL             
 4     0 HCl                0 Barneveld2          0 T0         CHX             
 5     0 HCl                0 Barneveld2          0 T0         CHX             
 6     0 HCl                0 Barneveld2          0 T0         CHX             
 7     0 HCl                0 Barneveld2          0 T0         KAN             
 8     0 HCl                0 Barneveld2          0 T0         KAN             
 9     0 HCl                0 Barneveld2          0 T0         KAN             
10     0 HCl                0 Barneveld2          0 T0         None            
# ... with 177 more rows, and 3 more variables: Dose_ppm &lt;dbl&gt;, NO3_mM &lt;dbl&gt;,
#   NO2_mM &lt;dbl&gt;</code></pre>
<pre class="r"><code>df_sub &lt;- df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;, &#39;Antibiotics_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>df_main &lt;- df_plot %&gt;% filter(H_mM != 0)
dim(df_plot)</code></pre>
<pre><code>[1] 923  10</code></pre>
<pre class="r"><code>dim(df_main)</code></pre>
<pre><code>[1] 703  10</code></pre>
<pre class="r"><code>df_plot2 &lt;- rbind(df_main, df_sub)
dim(df_plot2) # 890</code></pre>
<pre><code>[1] 890  10</code></pre>
<pre class="r"><code># include pH data of the endpoint
# df_pH_T7_10 &lt;- df_220521_pH %&gt;% filter(Time_point %in% c(&quot;T7&quot;,&quot;T10&quot;))
# 
# df_plot2 &lt;- df_plot2 %&gt;% left_join(df_pH_T7_10, by=c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Titration_type&quot;=&quot;Titration_type&quot;,&quot;Unit&quot;=&quot;Unit&quot;, &quot;Antibiotics_type&quot;=&quot;Antibiotics_type&quot;, &quot;Dose_ppm&quot;=&quot;Dose_ppm&quot;))

# plot with H_mM
ggplot(df_plot2, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = (col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot Nitrite
ggplot(df_plot2, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.5)+
  # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  scale_colour_gradientn(colours = (col_pH(100))) +
  # scale_color_manual(values=col_pH(4)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_grid(. ~ Soil) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 17))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># make dose (2500 and 5000)
df_plot3 &lt;- df_plot2

df_plot3$Dose_ppm &lt;- ifelse(df_plot3$Dose_ppm == 2500, 5000, df_plot3$Dose_ppm)
df_plot3$Dose_ppm &lt;- ifelse(df_plot3$Dose_ppm == 5000, &quot;2500 or 5000&quot;, df_plot3$Dose_ppm)
df_plot3$Dose_ppm &lt;- factor(df_plot3$Dose_ppm, levels = c(200, 1000, &quot;2500 or 5000&quot;))
df_plot3$Antibiotics_type &lt;- factor(df_plot3$Antibiotics_type, levels = c(&quot;CHL&quot;, &quot;KAN&quot;, &quot;STM&quot;,&quot;NYT&quot;,&quot;CHX&quot;,&quot;STM_CHX&quot;))
df_plot3 %&lt;&gt;% filter(Dose_ppm != 0)

ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Different antibiotics and dose effect&quot;)+
  facet_grid(Dose_ppm ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Different antibiotics and dose effect&quot;)+
  facet_grid(Dose_ppm ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dose effect - NO3-
ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect of antibiotics&quot;)+
  facet_grid(H_mM ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dose effect - NO2-
ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect of antibiotics&quot;)+
  facet_grid(H_mM ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># color is antibiotics, pH is slots
# ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
#   geom_point(size=2.5, shape=16) +
#   geom_line(size=1.2)+
#   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
#   # scale_colour_gradientn(colours = rev(col_pH(100))) +
#   # scale_color_manual(values=grad_pH) +
#   ylab(&quot;NO3- (mM)&quot;) +
#   xlab(&quot;Time (hr)&quot;) +
#   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
#   # label
#   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
#   mytheme_2d +
#   ggtitle(&quot;H_mM&quot;)+
#   facet_grid(Dose_ppm ~ pH) +
#   theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
#   theme(strip.text.x = element_text(size = 13)) +
#   theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))

#NO2
ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;H_mM&quot;)+
  facet_grid(Dose_ppm ~ H_mM) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># compare only with varying dose level
df_dose &lt;- df_plot2 %&gt;% filter(H_mM == 0, Dose_ppm != 0)
df_nodose &lt;- df_plot2 %&gt;% filter(Dose_ppm == 0)
df_nodose_CHL &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;CHL&quot;)
df_nodose_KAN &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;KAN&quot;)
df_nodose_STM &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;STM&quot;)
df_nodose_NYT &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;NYT&quot;)
df_nodose_CHX &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;CHX&quot;)
df_nodose_STM_CHX &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;STM_CHX&quot;)

df_dose &lt;- rbind(df_dose, df_nodose_CHL, df_nodose_KAN, df_nodose_STM, df_nodose_NYT, df_nodose_CHX, df_nodose_STM_CHX)

df_dose$Dose_ppm &lt;- factor(df_dose$Dose_ppm)
df_dose$Antibiotics_type &lt;- factor(df_dose$Antibiotics_type, levels = c(&quot;CHL&quot;, &quot;KAN&quot;, &quot;STM&quot;,&quot;NYT&quot;,&quot;CHX&quot;,&quot;STM_CHX&quot;))

ggplot(df_dose, aes(x=Time_hours, y=NO3_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect&quot;) +
  facet_wrap(. ~ Antibiotics_type, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))

# NO2
ggplot(df_dose, aes(x=Time_hours, y=NO2_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect&quot;) +
  facet_wrap(. ~ Antibiotics_type, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-5-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))






# p_A &lt;- ggplot(df_plot5, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) +
#   geom_point(size=2.5, shape=16) +
#   geom_line(size=1.2)+
#   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
#   scale_colour_gradientn(colours = rev(col_pH(100))) +
#   # scale_color_manual(values=grad_pH) +
#   ylab(&quot;NO3- (mM) \n&quot;) +
#   xlab(&quot;\n Time (hr)&quot;) +
#   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
#   # label
#   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
#   mytheme_2d +
#   facet_grid(Dose_ppm ~ Antibiotics_type) +
#   theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
#   theme(strip.text.x = element_text(size = 17))
# p_A
# 
# # plot Nitrite
# p_I &lt;- ggplot(df_plot3, aes(x=Time_hours, y=NO2_mM, color=pH, group=pH)) +
#   geom_point(size=2.5, shape=16) +
#   geom_line(size=1.5)+
#   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
#   scale_colour_gradientn(colours = rev(col_pH(100))) +
#   # scale_color_manual(values=col_pH(4)) +
#   ylab(&quot;NO2- (mM) \n&quot;) +
#   xlab(&quot;\n Time (hr)&quot;) +
#   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
#   # label
#   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
#   mytheme_2d +
#   facet_grid(. ~ Soil) +
#   theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
#   theme(strip.text.x = element_text(size = 17))
# 
# p_I

# require(gridExtra)
# library(grid)
# library(gtable)
# 
# legend = gtable_filter(ggplotGrob(p_A), &quot;guide-box&quot;)
# 
# grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position=&quot;none&quot;), 
#                          p_I+xlab(NULL)+theme(legend.position=&quot;none&quot;)+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), nrow=2,
#              top = textGrob(&quot;pH perturbation experiment with autoclaved soil&quot;, vjust = 0.5, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
#              # left = textGrob(&quot;Relative abundance of strains with opt_pH&quot;, vjust = 0.5, rot=90, gp = gpar(fontface = &quot;bold&quot;, cex = 1.5)),
#              bottom = textGrob(&quot;Time (hr)&quot;, vjust = 0, gp = gpar(fontface=&quot;bold&quot;, cex = 1.5)),
#              
#              right = legend
#              ))</code></pre>
</div>
<div id="plot-with-ph" class="section level1">
<h1>plot with pH</h1>
<pre class="r"><code># plot together with pH data
df_220521_pH &lt;- read.xlsx(&quot;data/220521_pH_Antibiotics_dose_effect_experiment.xlsx&quot;) %&gt;% select(-H_mM)
df_220521_pH$Dose_ppm</code></pre>
<pre><code>  [1]  200  200  200  200  200  200  200  200  200  200 1000 1000 1000 1000 1000
 [16] 1000 1000 1000 1000 1000 1000 2500 2500 2500 2500 2500 2500 2500 2500 2500
 [31] 2500 2500 2500 2500  200  200  200  200  200  200  200  200  200  200  200
 [46]  200  200  200  200 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000
 [61] 1000 1000 1000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000
 [76] 5000 5000  200  200  200  200  200  200  200  200  200  200  200 1000 1000
 [91] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 5000 5000 5000
[106] 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000    0    0    0  200
[121]  200  200  200  200  200  200  200  200  200 1000 1000 1000 1000 1000 1000
[136] 1000 1000 1000 1000 1000 1000 1000 1000  200  200  200  200  200  200  200
[151]  200  200  200  200  200 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000
[166] 1000 1000 1000 1000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000
[181] 5000 5000 5000 5000  200  200  200  200  200  200  200  200  200  200  200
[196] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000</code></pre>
<pre class="r"><code>df_220521_pH$Time_point</code></pre>
<pre><code>  [1] &quot;T0&quot;  &quot;T10&quot; &quot;T0&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T0&quot;  &quot;T10&quot;
 [13] &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; 
 [25] &quot;T10&quot; &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
 [37] &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
 [49] &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; 
 [61] &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot; 
 [73] &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
 [85] &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; 
 [97] &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; 
[109] &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot; 
[121] &quot;T0&quot;  &quot;T10&quot; &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; 
[133] &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot; 
[145] &quot;T10&quot; &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot; 
[157] &quot;T10&quot; &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
[169] &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
[181] &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot;
[193] &quot;T7&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T0&quot; 
[205] &quot;T10&quot; &quot;T7&quot;  &quot;T0&quot;  &quot;T10&quot; &quot;T7&quot; </code></pre>
<pre class="r"><code>df_220521_pH %&gt;% filter(Antibiotics_type == &quot;KAN&quot;) %&gt;% arrange(pH) %&gt;% filter(pH &gt; 8, Dose_ppm == 5000)</code></pre>
<pre><code>        Soil Titration_type Unit Antibiotics_type Dose_ppm Time_point       pH
1 Barneveld2            HCl    0              KAN     5000         T7 8.099099
2 Barneveld2            HCl    0              KAN     5000        T10 8.123123
3 Barneveld2           NaOH    4              KAN     5000         T0 8.243243
4 Barneveld2           NaOH    4              KAN     5000         T7 8.459459
5 Barneveld2           NaOH    8              KAN     5000         T7 8.651652
6 Barneveld2           NaOH    8              KAN     5000        T10 8.759760
7 Barneveld2           NaOH    8              KAN     5000         T0 8.807808
8 Barneveld2           NaOH    4              KAN     5000        T10 9.516517</code></pre>
<pre class="r"><code>df_plot &lt;- df_pH %&gt;% select(H_mM, Titration_type, Unit, NO3_mM, NO2_mM, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm)
df_plot %&gt;% filter(H_mM == 0) %&gt;% dim()</code></pre>
<pre><code>[1] 220  10</code></pre>
<pre class="r"><code>df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;, &#39;Antibiotics_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 187 x 10
    H_mM Titration_type  Unit Soil       Time_hours Time_point Antibiotics_type
   &lt;dbl&gt; &lt;chr&gt;          &lt;dbl&gt; &lt;fct&gt;           &lt;dbl&gt; &lt;fct&gt;      &lt;chr&gt;           
 1     0 HCl                0 Barneveld2          0 T0         CHL             
 2     0 HCl                0 Barneveld2          0 T0         CHL             
 3     0 HCl                0 Barneveld2          0 T0         CHL             
 4     0 HCl                0 Barneveld2          0 T0         CHX             
 5     0 HCl                0 Barneveld2          0 T0         CHX             
 6     0 HCl                0 Barneveld2          0 T0         CHX             
 7     0 HCl                0 Barneveld2          0 T0         KAN             
 8     0 HCl                0 Barneveld2          0 T0         KAN             
 9     0 HCl                0 Barneveld2          0 T0         KAN             
10     0 HCl                0 Barneveld2          0 T0         None            
# ... with 177 more rows, and 3 more variables: Dose_ppm &lt;dbl&gt;, NO3_mM &lt;dbl&gt;,
#   NO2_mM &lt;dbl&gt;</code></pre>
<pre class="r"><code>df_sub &lt;- df_plot %&gt;% filter(H_mM == 0) %&gt;% group_by(H_mM, Titration_type, Unit, Soil, Time_hours, Time_point, Antibiotics_type, Dose_ppm) %&gt;% summarize(NO3_mM = mean(NO3_mM), NO2_mM = mean(NO2_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;H_mM&#39;, &#39;Titration_type&#39;, &#39;Unit&#39;, &#39;Soil&#39;, &#39;Time_hours&#39;, &#39;Time_point&#39;, &#39;Antibiotics_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>df_main &lt;- df_plot %&gt;% filter(H_mM != 0)
dim(df_plot)</code></pre>
<pre><code>[1] 923  10</code></pre>
<pre class="r"><code>dim(df_main)</code></pre>
<pre><code>[1] 703  10</code></pre>
<pre class="r"><code>df_plot2 &lt;- rbind(df_main, df_sub)
dim(df_plot2) # 890</code></pre>
<pre><code>[1] 890  10</code></pre>
<pre class="r"><code>colnames(df_plot2)</code></pre>
<pre><code> [1] &quot;H_mM&quot;             &quot;Titration_type&quot;   &quot;Unit&quot;             &quot;NO3_mM&quot;          
 [5] &quot;NO2_mM&quot;           &quot;Soil&quot;             &quot;Time_hours&quot;       &quot;Time_point&quot;      
 [9] &quot;Antibiotics_type&quot; &quot;Dose_ppm&quot;        </code></pre>
<pre class="r"><code>df_plot2$Dose_ppm</code></pre>
<pre><code>  [1]  200  200 1000 2500 5000 5000  200  200 1000  200  200 1000 1000 2500 5000
 [16] 5000  200  200 1000 2500  200  200 1000 1000 5000 5000  200 2500  200  200
 [31] 1000 1000 1000 5000 5000  200  200 1000 5000 5000  200  200  200 1000 1000
 [46]  200  200 5000 5000  200  200 1000 1000 1000  200  200 1000 1000 1000 1000
 [61] 1000 1000 1000  200  200 1000 1000 2500 5000 5000  200  200 1000  200  200
 [76] 1000 1000 2500 5000 5000  200  200 1000 2500  200  200 1000 1000 5000 5000
 [91]  200 2500  200  200 1000 1000 1000 5000 5000  200  200 1000 5000 5000  200
[106]  200  200 1000 1000  200  200 5000 5000  200  200 1000 1000 1000  200  200
[121] 1000 1000 1000 1000 1000 1000 1000  200  200 1000 1000 2500 5000 5000  200
[136]  200 1000  200  200 1000 1000 2500 5000 5000  200  200 1000 2500  200  200
[151] 1000 1000 5000 5000  200 2500  200  200 1000 1000 1000 5000 5000  200  200
[166] 1000 5000 5000  200  200  200 1000 1000  200  200 5000 5000  200  200 1000
[181] 1000 1000  200  200 1000 1000 1000 1000 1000 1000 1000  200  200 1000 1000
[196] 2500 5000 5000  200  200 1000  200  200 1000 1000 2500 5000 5000  200  200
[211] 1000 2500  200  200 1000 1000 5000 5000  200 2500  200  200 1000 1000 1000
[226] 5000 5000  200  200 1000 5000 5000  200  200  200 1000 1000  200  200 5000
[241] 5000  200  200 1000 1000 1000  200  200 1000 1000 1000 1000 1000 1000 1000
[256]  200  200 1000 1000 2500 5000 5000  200  200 1000  200  200 1000 1000 2500
[271] 5000 5000  200  200 1000 2500  200  200 1000 1000 5000 5000  200 2500  200
[286]  200 1000 1000 1000 5000 5000  200  200 1000 5000 5000  200  200  200 1000
[301] 1000  200  200 5000 5000  200  200 1000 1000 1000  200  200 1000 1000 1000
[316] 1000 1000 1000 1000  200  200 1000 1000 2500 5000 5000  200  200 1000  200
[331]  200 1000 1000 2500 5000 5000  200  200 1000 2500  200  200 1000 1000 5000
[346] 5000  200 2500  200  200 1000 1000 1000 5000 5000  200  200 1000 5000 5000
[361]  200  200  200 1000 1000  200  200 5000 5000  200  200 1000 1000 1000  200
[376]  200 1000 1000 1000 1000 1000 1000 1000  200  200 1000 1000 2500 5000 5000
[391]  200  200 1000  200  200 1000 1000 2500 5000 5000  200  200 1000 2500  200
[406]  200 1000 1000 5000 5000  200 2500  200  200 1000 1000 1000 5000 5000  200
[421]  200 1000 5000 5000  200  200  200 1000 1000  200  200 5000 5000  200  200
[436] 1000 1000 1000  200  200 1000 1000 1000 1000 1000 1000 1000  200  200 1000
[451] 1000 2500 5000 5000  200  200 1000  200  200 1000 1000 2500 5000 5000  200
[466]  200 1000 2500  200  200 1000 1000 5000 5000  200 2500  200  200 1000 1000
[481] 1000 5000 5000  200  200 1000 5000 5000  200  200  200 1000 1000  200  200
[496] 5000 5000  200  200 1000 1000 1000  200  200 1000 1000 1000 1000 1000 1000
[511] 1000  200  200 1000 1000 2500 5000 5000  200  200 1000  200  200 1000 1000
[526] 2500 5000 5000  200  200 1000 2500  200  200 1000 1000 5000 5000  200 2500
[541]  200  200 1000 1000 1000 5000 5000  200  200 1000 5000 5000  200  200  200
[556] 1000 1000  200  200 5000 5000  200  200 1000 1000 1000  200  200 1000 1000
[571] 1000 1000 1000 1000 1000  200  200 1000 1000 2500 5000 5000  200  200 1000
[586]  200  200 1000 1000 2500 5000 5000  200  200 1000 2500  200  200 1000 1000
[601] 5000 5000  200 2500  200  200 1000 1000 1000 5000 5000  200  200 1000 5000
[616] 5000  200  200  200 1000 1000  200  200 5000 5000  200  200 1000 1000 1000
[631]  200  200 1000 1000 1000 1000 1000 1000 1000  200  200 1000 1000 2500 5000
[646] 5000  200  200 1000  200  200 1000 1000 2500 5000 5000  200  200 1000 2500
[661]  200  200 1000 1000 5000 5000  200 2500  200  200 1000 1000 1000 5000 5000
[676]  200  200 1000 5000 5000  200  200  200 1000 1000  200  200 5000 5000  200
[691]  200 1000 1000 1000  200  200 1000 1000 1000 1000 1000 1000 1000  200 1000
[706] 2500  200 1000 5000  200 1000 5000    0  200 1000  200 1000 5000  200 1000
[721]  200 1000 2500  200 1000 5000  200 1000 5000    0  200 1000  200 1000 5000
[736]  200 1000  200 1000 2500  200 1000 5000  200 1000 5000    0  200 1000  200
[751] 1000 5000  200 1000  200 1000 2500  200 1000 5000  200 1000 5000    0  200
[766] 1000  200 1000 5000  200 1000  200 1000 2500  200 1000 5000  200 1000 5000
[781]    0  200 1000  200 1000 5000  200 1000  200 1000 2500  200 1000 5000  200
[796] 1000 5000    0  200 1000  200 1000 5000  200 1000  200 1000 2500  200 1000
[811] 5000  200 1000 5000    0  200 1000  200 1000 5000  200 1000  200 1000 2500
[826]  200 1000 5000  200 1000 5000    0  200 1000  200 1000 5000  200 1000  200
[841] 1000 2500  200 1000 5000  200 1000 5000    0  200 1000  200 1000 5000  200
[856] 1000  200 1000 2500  200 1000 5000  200 1000 5000    0  200 1000  200 1000
[871] 5000  200 1000  200 1000 2500  200 1000 5000  200 1000 5000    0  200 1000
[886]  200 1000 5000  200 1000</code></pre>
<pre class="r"><code>df_220521_pH$Dose_ppm</code></pre>
<pre><code>  [1]  200  200  200  200  200  200  200  200  200  200 1000 1000 1000 1000 1000
 [16] 1000 1000 1000 1000 1000 1000 2500 2500 2500 2500 2500 2500 2500 2500 2500
 [31] 2500 2500 2500 2500  200  200  200  200  200  200  200  200  200  200  200
 [46]  200  200  200  200 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000
 [61] 1000 1000 1000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000
 [76] 5000 5000  200  200  200  200  200  200  200  200  200  200  200 1000 1000
 [91] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 5000 5000 5000
[106] 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000    0    0    0  200
[121]  200  200  200  200  200  200  200  200  200 1000 1000 1000 1000 1000 1000
[136] 1000 1000 1000 1000 1000 1000 1000 1000  200  200  200  200  200  200  200
[151]  200  200  200  200  200 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000
[166] 1000 1000 1000 1000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000 5000
[181] 5000 5000 5000 5000  200  200  200  200  200  200  200  200  200  200  200
[196] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000</code></pre>
<pre class="r"><code>df_220521_pH %&lt;&gt;% filter(Time_point %in% c(&quot;T7&quot;)) %&gt;% select(-Time_point)

df_plot2 %&lt;&gt;% left_join(df_220521_pH, by=c(&quot;Soil&quot;=&quot;Soil&quot;, &quot;Titration_type&quot;=&quot;Titration_type&quot;,&quot;Unit&quot;=&quot;Unit&quot;, &quot;Antibiotics_type&quot;=&quot;Antibiotics_type&quot;, &quot;Dose_ppm&quot;=&quot;Dose_ppm&quot;))
colnames(df_plot2)</code></pre>
<pre><code> [1] &quot;H_mM&quot;             &quot;Titration_type&quot;   &quot;Unit&quot;             &quot;NO3_mM&quot;          
 [5] &quot;NO2_mM&quot;           &quot;Soil&quot;             &quot;Time_hours&quot;       &quot;Time_point&quot;      
 [9] &quot;Antibiotics_type&quot; &quot;Dose_ppm&quot;         &quot;pH&quot;              </code></pre>
<pre class="r"><code>df_pH_label &lt;- df_220521_pH %&gt;% group_by(Titration_type, Unit) %&gt;% summarize(pH_min = round(min(pH), 2), pH_max = round(max(pH), 2))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Titration_type&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>df_pH_label$pH_ave &lt;- (df_pH_label$pH_max + df_pH_label$pH_min)/2
df_pH_label %&lt;&gt;% unite(col=&#39;pH_range&#39;, c(&quot;pH_min&quot;,&quot;pH_max&quot;), sep=&quot; - &quot;, remove=F)
df_pH_label</code></pre>
<pre><code># A tibble: 5 x 6
# Groups:   Titration_type [2]
  Titration_type  Unit pH_range    pH_min pH_max pH_ave
  &lt;chr&gt;          &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
1 HCl                0 7.39 - 8.1    7.39   8.1    7.74
2 HCl                4 5.84 - 7.69   5.84   7.69   6.76
3 HCl                8 4.15 - 5.54   4.15   5.54   4.85
4 NaOH               4 8.21 - 8.52   8.21   8.52   8.36
5 NaOH               8 8.36 - 8.66   8.36   8.66   8.51</code></pre>
<pre class="r"><code>df_pH_range &lt;- df_plot2 %&gt;% left_join(df_pH_label, by =c(&quot;Titration_type&quot;=&quot;Titration_type&quot;, &quot;Unit&quot;=&quot;Unit&quot;))

# plot average and standard deviation
# label with pH range

# df_pH_label$pH_max &lt;- as.character()

# make dose (2500 and 5000)
df_pH_range2 &lt;- df_pH_range

df_pH_range2$Dose_ppm &lt;- ifelse(df_pH_range2$Dose_ppm == 2500, 5000, df_pH_range2$Dose_ppm)
df_pH_range2$Dose_ppm &lt;- ifelse(df_pH_range2$Dose_ppm == 5000, &quot;2500 or 5000&quot;, df_pH_range2$Dose_ppm)
df_pH_range2$Dose_ppm &lt;- factor(df_pH_range2$Dose_ppm, levels = c(200, 1000, &quot;2500 or 5000&quot;))
df_pH_range2$Antibiotics_type &lt;- factor(df_pH_range2$Antibiotics_type, levels = c(&quot;CHL&quot;, &quot;KAN&quot;, &quot;STM&quot;,&quot;NYT&quot;,&quot;CHX&quot;,&quot;STM_CHX&quot;))
df_pH_range2 %&lt;&gt;% filter(Dose_ppm != 0)

df_pH_range2 %&gt;% filter(Antibiotics_type == &quot;KAN&quot;) %&gt;% arrange(pH) %&gt;% filter(pH &gt; 8, Dose_ppm == &quot;2500 or 5000&quot;)</code></pre>
<pre><code>       H_mM Titration_type Unit    NO3_mM      NO2_mM       Soil Time_hours
1    0.0000            HCl    0 1.9696440 0.014210626 Barneveld2        0.0
2    0.0000            HCl    0 1.9754917 0.027980982 Barneveld2        3.7
3    0.0000            HCl    0 1.8071523 0.118844783 Barneveld2        6.7
4    0.0000            HCl    0 1.6315764 0.233697292 Barneveld2        9.6
5    0.0000            HCl    0 1.2143007 0.384648600 Barneveld2       16.3
6    0.0000            HCl    0 0.8403810 0.383492321 Barneveld2       22.8
7    0.0000            HCl    0 0.5258345 0.348148555 Barneveld2       28.9
8    0.0000            HCl    0 0.0000000 0.223209403 Barneveld2       38.3
9    0.0000            HCl    0 0.0000000 0.007939327 Barneveld2       49.5
10   0.0000            HCl    0 0.0000000 0.008857732 Barneveld2       74.3
11   0.0000            HCl    0 0.0000000 0.012044290 Barneveld2      102.7
12 -21.3758           NaOH    4 2.0968735 0.018277035 Barneveld2        0.0
13 -21.3758           NaOH    4 2.0241569 0.021906897 Barneveld2        3.7
14 -21.3758           NaOH    4 1.6790563 0.172928630 Barneveld2        6.7
15 -21.3758           NaOH    4 1.4784268 0.423320955 Barneveld2        9.6
16 -21.3758           NaOH    4 0.7927255 1.080036364 Barneveld2       16.3
17 -21.3758           NaOH    4 0.1440239 1.267548832 Barneveld2       22.8
18 -21.3758           NaOH    4 0.0000000 1.062589684 Barneveld2       28.9
19 -21.3758           NaOH    4 0.0000000 0.162467406 Barneveld2       38.3
20 -21.3758           NaOH    4 0.0000000 0.013447285 Barneveld2       49.5
21 -21.3758           NaOH    4 0.0000000 0.015514084 Barneveld2       74.3
22 -21.3758           NaOH    4 0.0000000 0.015625408 Barneveld2      102.7
23 -42.7516           NaOH    8 1.8108942 0.016319025 Barneveld2        0.0
24 -42.7516           NaOH    8 1.9674902 0.018263433 Barneveld2        3.7
25 -42.7516           NaOH    8 1.7209372 0.050535363 Barneveld2        6.7
26 -42.7516           NaOH    8 1.7814127 0.119873816 Barneveld2        9.6
27 -42.7516           NaOH    8 1.5250502 0.356571079 Barneveld2       16.3
28 -42.7516           NaOH    8 1.3673383 0.589332113 Barneveld2       22.8
29 -42.7516           NaOH    8 1.1756624 0.864509753 Barneveld2       28.9
30 -42.7516           NaOH    8 0.3625187 1.466313306 Barneveld2       38.3
31 -42.7516           NaOH    8 0.0000000 0.437379542 Barneveld2       49.5
32 -42.7516           NaOH    8 0.0000000 0.020356716 Barneveld2       74.3
33 -42.7516           NaOH    8 0.0000000 0.020904328 Barneveld2      102.7
   Time_point Antibiotics_type     Dose_ppm       pH    pH_range pH_min pH_max
1          T0              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
2          T1              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
3          T2              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
4          T3              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
5          T4              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
6          T5              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
7          T6              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
8          T7              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
9          T8              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
10         T9              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
11        T10              KAN 2500 or 5000 8.099099  7.39 - 8.1   7.39   8.10
12         T0              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
13         T1              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
14         T2              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
15         T3              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
16         T4              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
17         T5              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
18         T6              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
19         T7              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
20         T8              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
21         T9              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
22        T10              KAN 2500 or 5000 8.459459 8.21 - 8.52   8.21   8.52
23         T0              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
24         T1              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
25         T2              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
26         T3              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
27         T4              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
28         T5              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
29         T6              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
30         T7              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
31         T8              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
32         T9              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
33        T10              KAN 2500 or 5000 8.651652 8.36 - 8.66   8.36   8.66
   pH_ave
1   7.745
2   7.745
3   7.745
4   7.745
5   7.745
6   7.745
7   7.745
8   7.745
9   7.745
10  7.745
11  7.745
12  8.365
13  8.365
14  8.365
15  8.365
16  8.365
17  8.365
18  8.365
19  8.365
20  8.365
21  8.365
22  8.365
23  8.510
24  8.510
25  8.510
26  8.510
27  8.510
28  8.510
29  8.510
30  8.510
31  8.510
32  8.510
33  8.510</code></pre>
<pre class="r"><code>ggplot(df_pH_range2, aes(x=Time_hours, y=NO3_mM, color=pH_ave, group=pH_ave)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Different antibiotics and dose effect&quot;)+
  facet_grid(Dose_ppm ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_pH_range2, aes(x=Time_hours, y=NO2_mM, color=pH_ave, group=pH_ave)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Different antibiotics and dose effect&quot;)+
  facet_grid(Dose_ppm ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dose effect - NO3-
ggplot(df_pH_range2, aes(x=Time_hours, y=NO3_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect of antibiotics&quot;)+
  facet_grid(pH_range ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dose effect - NO2-
ggplot(df_pH_range2, aes(x=Time_hours, y=NO2_mM, color=Dose_ppm, group=Dose_ppm)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Dose effect of antibiotics&quot;)+
  facet_grid(pH_range ~ Antibiotics_type) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># color is antibiotics, pH is slots
ggplot(df_pH_range2, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;pH&quot;)+
  facet_grid(Dose_ppm ~ pH_range) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#NO2
ggplot(df_pH_range2, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;pH&quot;)+
  facet_grid(Dose_ppm ~ pH_range) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-6-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># compare only with varying dose level
df_dose &lt;- df_plot2 %&gt;% filter(H_mM == 0, Dose_ppm != 0)
df_nodose &lt;- df_plot2 %&gt;% filter(Dose_ppm == 0)
df_nodose_CHL &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;CHL&quot;)
df_nodose_KAN &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;KAN&quot;)
df_nodose_STM &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;STM&quot;)
df_nodose_NYT &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;NYT&quot;)
df_nodose_CHX &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;CHX&quot;)
df_nodose_STM_CHX &lt;- df_nodose %&gt;% mutate(Antibiotics_type = &quot;STM_CHX&quot;)

df_dose &lt;- rbind(df_dose, df_nodose_CHL, df_nodose_KAN, df_nodose_STM, df_nodose_NYT, df_nodose_CHX, df_nodose_STM_CHX)

df_dose$Dose_ppm &lt;- factor(df_dose$Dose_ppm)
df_dose$Antibiotics_type &lt;- factor(df_dose$Antibiotics_type, levels = c(&quot;CHL&quot;, &quot;KAN&quot;, &quot;STM&quot;,&quot;NYT&quot;,&quot;CHX&quot;,&quot;STM_CHX&quot;))</code></pre>
<div id="stm-chx-stm_chx" class="section level2">
<h2>STM, CHX, STM_CHX</h2>
<pre class="r"><code># fungal denitrification
df_pH_range3 &lt;- df_pH_range2 %&gt;% filter(Antibiotics_type %in% c(&quot;STM&quot;,&quot;CHX&quot;,&quot;STM_CHX&quot;)) %&gt;% filter(Dose_ppm %in% c(200, 1000))
ggplot(df_pH_range3, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Effect of fungicide cycloheximide&quot;)+
  facet_grid(Dose_ppm ~ pH_range) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_pH_range3, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = rev(col_pH(100))) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO2- (mM)&quot;) +
  xlab(&quot;Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  ggtitle(&quot;Effect of fungicide cycloheximide&quot;)+
  facet_grid(Dose_ppm ~ pH_range) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))+
  theme(strip.text.x = element_text(size = 13)) +
  theme(strip.text.y = element_text(size = 13, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Plot nitrate, nitrite, ammonium at the same time -->
<!-- ```{r} -->
<!-- df_ammo_plot3 <- read.xlsx("data/220519_Ammonia_sterile.xlsx") -->
<!-- df_ammo_plot3$Soil <- factor(df_ammo_plot3$Soil, levels=c("LaBaghWoods_pH6.66", "LaBaghWoods_autoclaved_pH6.66")) -->
<!-- p_NH4 <- ggplot(df_ammo_plot3, aes(x=Time_hours, y=NH4_mM, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NH4_mM - Std_NH4_mM, ymax=NH4_mM + Std_NH4_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NH4+ (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,4,0.5), limits=c(0, 4))+ -->
<!--   # ggtitle("pH perturbation \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NH4_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- legend = gtable_filter(ggplotGrob(p_A), "guide-box") -->
<!-- grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position="none"),  -->
<!--                          p_I+xlab(NULL)+theme(legend.position="none")+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), -->
<!--                          p_NH4+xlab(NULL)+theme(legend.position="none")+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()) -->
<!--                          , nrow=3, -->
<!--              top = textGrob("pH perturbation experiment with autoclaved soil", vjust = 0.5, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              # left = textGrob("Relative abundance of strains with opt_pH", vjust = 0.5, rot=90, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              bottom = textGrob("Time (hr)", vjust = 0, gp = gpar(fontface="bold", cex = 1.5)), -->
<!--              right = legend -->
<!--              )) -->
<!-- ``` -->
<!-- Which are doing nitrate/nitrite reduction? -->
<!-- ```{r} -->
<!-- colnames(df_plot3) -->
<!-- df_plot3$Soil -->
<!-- df_au <- df_plot3 %>% filter(Soil == "LaBaghWoods_autoclaved_pH6.66") -->
<!-- df_au_end <- df_au %>% filter(Time_point == "Sterile_T10") -->
<!-- df_reduced <- df_au_end %>% filter(NO3_mM == 0) %>% mutate(Nitrate_reduced = T) -->
<!-- df_not <- df_au_end %>% filter(NO3_mM != 0) %>% mutate(Nitrate_reduced =F) -->
<!-- df_reduced <- rbind(df_reduced, df_not) -->
<!-- # plot -->
<!-- ggplot(df_reduced, aes(x=Nitrate_reduced, y=pH, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   ggtitle("Counting perturbed samples that reduced all nitrate in autoclaved soil") + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("H+ (mM) \n") + -->
<!--   xlab("\n Nitrate completely reduced?") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d -->
<!-- # write.xlsx(df_reduced, "220519_nitrate_reduced_autoclaved_samples.xlsx") -->
<!-- ``` -->
<!-- No nitrate sample? -->
<!-- ```{r} -->
<!-- df_nn <- df_p_bcf %>% filter(Titration_type == "No_Nitrate") %>% filter(!(Time_point %in% c("LBA_T8", "Sterile_T8"))) -->
<!-- colnames(df_nn) -->
<!-- dim(df_nn) -->
<!-- df_nn_ave <- df_nn %>% group_by(Soil, Time_hours, Time_point) %>% summarize(Std_NO3_mM = sd(NO3_mM), Std_NO2_mM = sd(NO2_mM), Ave_NO3_mM = mean(NO3_mM), Ave_NO2_mM = mean(NO2_mM)) %>% ungroup() -->
<!-- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO3_mM, color=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=Ave_NO3_mM - Std_NO3_mM, ymax=Ave_NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+ -->
<!--   ggtitle("Samples provided with no Nitrate \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- # plot Nitrite -->
<!-- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO2_mM, color=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=Ave_NO2_mM - Std_NO2_mM, ymax=Ave_NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+ -->
<!--   ggtitle("Samples provided with no Nitrate \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- p_A <- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO3_mM, color=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=Ave_NO3_mM - Std_NO3_mM, ymax=Ave_NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+ -->
<!--   # ggtitle("Samples provided with no Nitrate \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- # plot Nitrite -->
<!-- p_I <- ggplot(df_nn_ave, aes(x=Time_hours, y=Ave_NO2_mM, color=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=Ave_NO2_mM - Std_NO2_mM, ymax=Ave_NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2,0.5), limits=c(0, 2))+ -->
<!--   # ggtitle("Samples provided with no Nitrate \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position="none"),  -->
<!--                          p_I+xlab(NULL)+theme(legend.position="none")+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), nrow=2, -->
<!--              top = textGrob("Samples provided with no Nitrate", vjust = 0.5, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              # left = textGrob("Relative abundance of strains with opt_pH", vjust = 0.5, rot=90, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              bottom = textGrob("Time (hr)", vjust = 0, gp = gpar(fontface="bold", cex = 1.5)) -->
<!--              )) -->
<!-- # plot ammonia as well -->
<!-- df_a_bcf <- read.xlsx("data/220519_Ammonia_df_a_bcf.xlsx") -->
<!-- df_a_bcf$Soil <- factor(df_a_bcf$Soil, levels=c("LaBaghWoods_pH6.66", "LaBaghWoods_autoclaved_pH6.66")) -->
<!-- df_a_bcf_nn <- df_a_bcf %>% filter(Titration_type == "No_Nitrate") %>% filter(!(Time_point %in% c("LBA_T8", "Sterile_T8"))) -->
<!-- df_a_bcf_ave <- df_a_bcf_nn %>% group_by(Soil, Time_hours) %>% summarize(Std_NH4_mM = sd(NH4_mM), Ave_NH4_mM = mean(NH4_mM)) %>% ungroup() -->
<!-- p_NH4 <- ggplot(df_a_bcf_ave, aes(x=Time_hours, y=Ave_NH4_mM, color=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=Ave_NH4_mM - Std_NH4_mM, ymax=Ave_NH4_mM + Std_NH4_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NH4- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2.5,0.5), limits=c(0, 2.5))+ -->
<!--   # ggtitle("Samples provided with no Nitrate \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NH4_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1))+ -->
<!--   theme(strip.text.x = element_text(size = 17)) -->
<!-- p_NH4 -->
<!-- # legend = gtable_filter(ggplotGrob(p_A), "guide-box") -->
<!-- grid.arrange(arrangeGrob(p_A+xlab(NULL)+theme(legend.position="none"),  -->
<!--                          p_I+xlab(NULL)+theme(legend.position="none")+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()), -->
<!--                          p_NH4+xlab(NULL)+theme(legend.position="none")+theme(strip.background = element_blank(), strip.text = element_blank(), strip.text.x = element_blank()) -->
<!--                          , nrow=3, -->
<!--              top = textGrob("Samples provided with no Nitrate.", vjust = 0.5, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              # left = textGrob("Relative abundance of strains with opt_pH", vjust = 0.5, rot=90, gp = gpar(fontface = "bold", cex = 1.5)), -->
<!--              bottom = textGrob("Time (hr)", vjust = 0, gp = gpar(fontface="bold", cex = 1.5)) -->
<!--              )) -->
<!-- ``` -->
</div>
<div id="lets-see-if-fungi-or-bacteria-matter" class="section level2">
<h2>4.2. Let’s see if fungi or bacteria matter</h2>
<pre class="r"><code># only antibiotics treated samples
df_LBA &lt;- df_p_bcf %&gt;% filter(Soil == &quot;LaBaghWoods_pH6.66&quot;) %&gt;% filter(!(Time_point == &quot;LBA_T8&quot;))
dim(df_LBA) # 521 -&gt; 473 (removed LBA_T8)</code></pre>
<pre><code>[1]  0 27</code></pre>
<pre class="r"><code>df_effect &lt;- df_LBA %&gt;% filter(H_mM == 0) %&gt;% filter(!(Titration_type %in% c(&quot;Nitrate&quot;, &quot;Nitrite&quot;, &quot;Ammonium&quot;, &quot;No_Nitrate&quot;)))
dim(df_effect) # 263 -&gt; 165 -&gt; 132 -&gt; 120 (removed T8)</code></pre>
<pre><code>[1]  0 27</code></pre>
<pre class="r"><code>df_effect$Titration_type</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_effect$Antibiotics_type</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_effect %&gt;% filter(Time_point == &quot;LBA_T0&quot;)</code></pre>
<pre><code> [1] Well                  Nitrite_input         Nitrate_input        
 [4] Ammonium_input        Soil                  Sample_type          
 [7] Titration_type        Antibiotics_type      Dose_ppm             
[10] Unit                  Concentration_M       Added_ul             
[13] Time_point            NO2_OD540             NO2NO3_OD540         
[16] NO2_mM                NO2NO3_mM             NO3_mM               
[19] Time_minutes          Time_hours            Time_days            
[22] H_mol                 Volume                H_Molarity           
[25] H_mM                  Correction_factor_NO2 Correction_factor_NO3
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># let&#39;s also include autoclaved sample
df_autoclaved &lt;- df_pH %&gt;% filter(H_mM == 0) %&gt;% filter(!(Time_point == &quot;Sterile_T8&quot;))
df_autoclaved$Antibiotics_type &lt;- &quot;Autoclaved&quot;
dim(df_autoclaved) # 33</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code>df_effect &lt;- rbind(df_effect, df_autoclaved)
dim(df_effect) #153 -&gt; 150</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code># this is for kyle
df_kyle2 &lt;- rbind(df_kyle, df_autoclaved)
# write.xlsx(df_kyle2, &quot;220519_Griess_antibiotics_experiment_to_kyle_w_autoclaved.xlsx&quot;)

# reorder factor
df_effect$Antibiotics_type &lt;- factor(df_effect$Antibiotics_type, levels = c(&quot;None&quot;, &quot;CHL&quot;, &quot;NYT&quot;, &quot;CHL_NYT&quot;, &quot;Autoclaved&quot;))

# plot
ggplot(df_effect, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  # geom_line(size=1.2)+
  # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  # scale_color_manual(values=grad_pH) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application (Chloramphenicol: CHL, Nystatin: NYT, CHL+NYT: CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # facet_grid(. ~ Soil) +
  # theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))



# plot average and standard deviation
colnames(df_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Dose_ppm&quot;             
[10] &quot;Unit&quot;                  &quot;Concentration_M&quot;       &quot;Added_ul&quot;             
[13] &quot;Time_point&quot;            &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;         
[16] &quot;NO2_mM&quot;                &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;               
[19] &quot;Time_minutes&quot;          &quot;Time_hours&quot;            &quot;Time_days&quot;            
[22] &quot;H_mol&quot;                 &quot;Volume&quot;                &quot;H_Molarity&quot;           
[25] &quot;H_mM&quot;                  &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot;</code></pre>
<pre class="r"><code>dim(df_effect)</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code>df_effect_ave &lt;- df_effect %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Nitrite_input&#39;, &#39;Nitrate_input&#39;, &#39;Soil&#39;, &#39;Titration_type&#39;, &#39;Sample_type&#39;, &#39;Antibiotics_type&#39;, &#39;H_mM&#39;, &#39;Time_point&#39;, &#39;Time_minutes&#39;, &#39;Time_hours&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(df_effect_ave)</code></pre>
<pre><code>[1] 11 15</code></pre>
<pre class="r"><code># nitrate
ggplot(df_effect_ave, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application: 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite
ggplot(df_effect_ave, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application: 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="lets-see-other-ph-perturbed-levels." class="section level2">
<h2>4.3. Let’s see other pH perturbed levels.</h2>
<pre class="r"><code>df_pH_effect &lt;- df_LBA %&gt;% filter(!(Titration_type %in% c(&quot;Nitrate&quot;, &quot;Nitrite&quot;, &quot;Ammonium&quot;, &quot;No_Nitrate&quot;)))
dim(df_pH_effect) # 354</code></pre>
<pre><code>[1]  0 27</code></pre>
<pre class="r"><code>df_pH_effect$Titration_type</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_pH_effect$Antibiotics_type</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>df_pH_effect %&gt;% filter(Time_point == &quot;LBA_T0&quot;)</code></pre>
<pre><code> [1] Well                  Nitrite_input         Nitrate_input        
 [4] Ammonium_input        Soil                  Sample_type          
 [7] Titration_type        Antibiotics_type      Dose_ppm             
[10] Unit                  Concentration_M       Added_ul             
[13] Time_point            NO2_OD540             NO2NO3_OD540         
[16] NO2_mM                NO2NO3_mM             NO3_mM               
[19] Time_minutes          Time_hours            Time_days            
[22] H_mol                 Volume                H_Molarity           
[25] H_mM                  Correction_factor_NO2 Correction_factor_NO3
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># let&#39;s also include autoclaved sample
df_autoclaved &lt;- df_pH %&gt;% filter(H_mM == 0) %&gt;% filter(!(Time_point == &quot;Sterile_T8&quot;))
df_autoclaved$Antibiotics_type &lt;- &quot;Autoclaved&quot;
dim(df_autoclaved) # 33</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code>df_pH_effect &lt;- rbind(df_pH_effect, df_autoclaved)
dim(df_pH_effect) #384</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code># reorder factor
df_pH_effect$Antibiotics_type &lt;- factor(df_pH_effect$Antibiotics_type, levels = c(&quot;None&quot;, &quot;CHL&quot;, &quot;NYT&quot;, &quot;CHL_NYT&quot;, &quot;Autoclaved&quot;))

# plot average and standard deviation
colnames(df_pH_effect)</code></pre>
<pre><code> [1] &quot;Well&quot;                  &quot;Nitrite_input&quot;         &quot;Nitrate_input&quot;        
 [4] &quot;Ammonium_input&quot;        &quot;Soil&quot;                  &quot;Sample_type&quot;          
 [7] &quot;Titration_type&quot;        &quot;Antibiotics_type&quot;      &quot;Dose_ppm&quot;             
[10] &quot;Unit&quot;                  &quot;Concentration_M&quot;       &quot;Added_ul&quot;             
[13] &quot;Time_point&quot;            &quot;NO2_OD540&quot;             &quot;NO2NO3_OD540&quot;         
[16] &quot;NO2_mM&quot;                &quot;NO2NO3_mM&quot;             &quot;NO3_mM&quot;               
[19] &quot;Time_minutes&quot;          &quot;Time_hours&quot;            &quot;Time_days&quot;            
[22] &quot;H_mol&quot;                 &quot;Volume&quot;                &quot;H_Molarity&quot;           
[25] &quot;H_mM&quot;                  &quot;Correction_factor_NO2&quot; &quot;Correction_factor_NO3&quot;</code></pre>
<pre class="r"><code>dim(df_pH_effect)</code></pre>
<pre><code>[1] 220  27</code></pre>
<pre class="r"><code>df_pH_effect_ave &lt;- df_pH_effect %&gt;% group_by(Nitrite_input, Nitrate_input, Soil, Titration_type, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days) %&gt;% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %&gt;% ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Nitrite_input&#39;, &#39;Nitrate_input&#39;, &#39;Soil&#39;, &#39;Titration_type&#39;, &#39;Sample_type&#39;, &#39;Antibiotics_type&#39;, &#39;H_mM&#39;, &#39;Time_point&#39;, &#39;Time_minutes&#39;, &#39;Time_hours&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(df_pH_effect_ave) # 284</code></pre>
<pre><code>[1] 11 15</code></pre>
<pre class="r"><code># nitrate
ggplot(df_pH_effect_ave, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO3- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation(H_mM): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d +
  facet_wrap(. ~ H_mM, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nitrite
ggplot(df_pH_effect_ave, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) +
  geom_point(size=2.5, shape=16) +
  geom_line(size=1.2)+
  geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+
  # scale_colour_gradientn(colours = col_pH(100)) +
  scale_color_manual(values=c(&quot;Black&quot;,&quot;Red&quot;,&quot;Blue&quot;, &quot;Purple&quot;, &quot;Brown&quot;)) +
  ylab(&quot;NO2- (mM) \n&quot;) +
  xlab(&quot;\n Time (hr)&quot;) +
  # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+
  ggtitle(&quot;Antibiotics application x pH perturbation(H_mM): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n&quot;) +
  # label
  # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family=&quot;serif&quot;, show.legend = FALSE)+
  mytheme_2d+
  facet_wrap(. ~ H_mM, nrow=2) +
  theme(strip.background = element_rect(colour=&quot;black&quot;, fill=&quot;white&quot;, size=0.1))</code></pre>
<p><img src="figure/220519_Griess_antibiotics_dose_effect.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Plot pH in these plots -->
<!-- ```{r} -->
<!-- df_kyle2 <- read.xlsx("data/220519_Griess_antibiotics_experiment_to_kyle_remove_blanks_add_pH.xlsx") -->
<!-- df_pH_effect <- df_kyle2 %>% filter(!(Titration_type %in% c("Nitrate", "Nitrite", "Ammonium", "No_Nitrate"))) -->
<!-- dim(df_pH_effect) # 354 -->
<!-- df_pH_effect$Titration_type -->
<!-- df_pH_effect$Antibiotics_type -->
<!-- df_pH_effect %>% filter(Time_point == "LBA_T0") -->
<!-- # let's also include autoclaved sample -->
<!-- df_autoclaved <- df_pH %>% filter(H_mM == 0) %>% filter(!(Time_point == "Sterile_T8")) -->
<!-- df_autoclaved$Antibiotics_type <- "Autoclaved" -->
<!-- dim(df_autoclaved) # 33 -->
<!-- df_autoclaved$pH <- 7.7 # any number between 7.6, 7.8 to be classified as neutral soil. -->
<!-- colnames(df_pH_effect) -->
<!-- colnames(df_autoclaved) -->
<!-- df_pH_effect <- rbind(df_pH_effect, df_autoclaved) -->
<!-- dim(df_pH_effect) #384 -->
<!-- # reorder factor -->
<!-- df_pH_effect$Antibiotics_type <- factor(df_pH_effect$Antibiotics_type, levels = c("None", "CHL", "NYT", "CHL_NYT", "Autoclaved")) -->
<!-- # plot average and standard deviation -->
<!-- colnames(df_pH_effect) -->
<!-- dim(df_pH_effect) -->
<!-- df_pH_effect_ave <- df_pH_effect %>% group_by(Soil, Titration_type, Unit, Sample_type, Antibiotics_type, H_mM, Time_point, Time_minutes, Time_hours, Time_days, pH) %>% summarise(Std_NO2_mM = sd(NO2_mM), Std_NO3_mM = sd(NO3_mM), NO2_mM = mean(NO2_mM), NO3_mM = mean(NO3_mM)) %>% ungroup() -->
<!-- dim(df_pH_effect_ave) # 284 -->
<!-- # label with pH range -->
<!-- df_pH_label <- df_pH_effect_ave %>% group_by(Titration_type, Unit) %>% summarize(pH_min = min(pH), pH_max = max(pH)) -->
<!-- # df_pH_label$pH_max <- as.character() -->
<!-- df_pH_label %<>% unite(col='pH_range', c("pH_min","pH_max"), sep="-", remove=F) -->
<!-- df_pH_range <- df_pH_effect_ave %>% left_join(df_pH_label, by =c("Titration_type"="Titration_type", "Unit"="Unit")) -->
<!-- # nitrate -->
<!-- ggplot(df_pH_range, aes(x=Time_hours, y=NO3_mM, color=Antibiotics_type, group=Antibiotics_type)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   scale_color_manual(values=c("Black","Red","Blue", "Purple", "Brown")) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("Antibiotics application x pH perturbation (pH at day2): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_wrap(. ~ pH_range, nrow=2) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- # nitrite -->
<!-- ggplot(df_pH_range, aes(x=Time_hours, y=NO2_mM, color=Antibiotics_type, group=Antibiotics_type)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   # scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   scale_color_manual(values=c("Black","Red","Blue", "Purple", "Brown")) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("Antibiotics application x pH perturbation (pH at day2): 200ppm of Chloramphenicol(CHL), Nystatin(NYT), CHL+NYT(CHL_NYT) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d+ -->
<!--   facet_wrap(. ~ pH_range, nrow=2) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ``` -->
<!-- ## (Temporary) Filter the T8 for now. -->
<!-- ```{r} -->
<!-- df_plot2 <- df_plot2 %>% filter(!(Time_point %in% c("WB_T5", "C_T5")))  -->
<!-- dim(df_plot2) -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- ``` -->
<!-- ## 4.1.1. Nitrite peakheight and H_mM -->
<!-- ```{r echo=F} -->
<!-- colnames(df_plot2) -->
<!-- df_plot2 %>% group_by(H_mM) %>% summarize(Max_NO2_mM = max(NO2_mM)) %>% ungroup() -->
<!-- # library(ggpmisc) -->
<!-- # x[ggpmisc:::find_peaks(df_plot2$NO2_mM)] -->
<!-- # get global peaks -->
<!-- df_plot2 -->
<!-- mat_H_time <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO2_mM") -->
<!-- mat_H_time <- tibble::column_to_rownames(mat_H_time, var = "H_mM") -->
<!-- mat_H_time <- as.matrix(mat_H_time) -->
<!-- class(mat_H_time) -->
<!-- ## get rid of NA -->
<!-- # Substitute NA to average of NO2_mM of before and after. -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- mat_H_time <- na_to_average(mat_H_time) -->
<!-- mat_H_time[is.na(mat_H_time)] # no more NA -->
<!-- # maximum peak value -->
<!-- apply(mat_H_time, 1, max) -->
<!-- ## plot with peak height and H_mM -->
<!-- vec_peakheight <- apply(mat_H_time, 1, max) -->
<!-- vec_H_mM <- names(vec_peakheight) -->
<!-- plot(vec_H_mM, vec_peakheight) -->
<!-- df_peakheight <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_peakheight) -->
<!-- df_peakheight$H_mM <- as.numeric(df_peakheight$H_mM) -->
<!-- ## plot -->
<!-- # linear regression -->
<!-- fit.peakheight <- lm(NO2_mM ~ H_mM, df_peakheight) -->
<!-- summary(fit.peakheight) -->
<!-- # Plot H_mM with peak time -->
<!-- ## 1. linear regression -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_abline(slope = coef(fit.peakheight)[[2]], intercept = coef(fit.peakheight)[[1]], -->
<!--               color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=1, label= paste0("y = ", round(coef(fit.peakheight)[[1]],3),"+",round(coef(fit.peakheight)[[2]],3),"x", ",  R^2: ", round(summary(fit.peakheight)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # 2. quadratic regression -->
<!-- df_peakheight$H_mM_2 <- (df_peakheight$H_mM)^2 -->
<!-- quad.peakheight <- lm(NO2_mM ~ H_mM + H_mM_2, df_peakheight) -->
<!-- summary(quad.peakheight) -->
<!-- perturbValues <- seq(-100, 100, 0.1) -->
<!-- peakPredict <- predict(quad.peakheight,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) -->
<!-- df_quad <- data.frame(H_mM = perturbValues, NO2_mM = peakPredict) -->
<!-- plot(perturbValues, peakPredict) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak height (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peakheight, aes(x=H_mM, y=NO2_mM)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak height (mM)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peak height \n") + -->
<!--   geom_line(data = df_quad, aes(x = H_mM, y = NO2_mM), color = "blue", size = 1) + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=1, label= paste0("y = ", round(coef(quad.peakheight)[[1]],3), "+", round(coef(quad.peakheight)[[2]],3),"x+",round(coef(quad.peakheight)[[3]],3), "x^2", ",  R^2: ", round(summary(quad.peakheight)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 4.1.1.2. PCA on the nitrate, nitrite time readings. -->
<!-- Before PCA, we need to make the time point the same. In PCA, it wouldn't matter, right? yes. The value of the column or row is not incorporated in the SVD process. -->
<!-- ```{r} -->
<!-- # Data only with autoclave and antibiotic treated samples -->
<!-- df_PCA <- df_pH_effect_ave %>% select(-Std_NO2_mM, -Std_NO3_mM)  # remove standard deviation -->
<!-- # Remove soil name in time_point string -->
<!-- df_PCA$Time_point <- str_replace(df_PCA$Time_point, ".*_","") -->
<!-- any(is.na(df_PCA)) # no NAs. That's good -->
<!-- dim(df_PCA) # 284 15 -->
<!-- # remove NA -->
<!-- df_PCA[is.na(df_PCA)] -->
<!-- df_na <- df_PCA[rowSums(is.na(df_PCA)) > 0,] -->
<!-- # df_PCA <- na.omit(df_PCA) -->
<!-- # df_PCA <- df_PCA[rowSums(is.na(df_PCA)) == 0,] -->
<!-- # dim(df_PCA) # 1046 -->
<!-- paste0("T",00:10) -->
<!-- # sprintf("%02d", 0:10) -->
<!-- df_PCA$Time_point <- factor(df_PCA$Time_point, levels = paste0("T",00:10)) -->
<!-- dim(df_PCA) -->
<!-- ``` -->
<!-- ## (1) Plot PCA with nitrate + nitrite -->
<!-- ```{r} -->
<!-- # function -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- ## make the above values to matrices -->
<!-- mat_NO2 <- dcast(df_PCA, H_mM + Titration_type + Antibiotics_type ~ Time_point, value.var = "NO2_mM") -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- # mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") -->
<!-- dim(mat_NO2) -->
<!-- colnames(mat_NO2) -->
<!-- col_NO2 <- mat_NO2[,1:3] -->
<!-- mat_NO2 <- as.matrix(mat_NO2[,4:ncol(mat_NO2)])  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- mat_NO2 <- na_to_average(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- # col_NO2[24,] -->
<!-- # mat_NO2[24,] -->
<!-- # need to remove row 24 -->
<!-- mat_NO3 <- dcast(df_PCA, H_mM + Titration_type + Antibiotics_type ~ Time_point, value.var = "NO3_mM") -->
<!-- dim(mat_NO3) -->
<!-- col_NO3 <- mat_NO3[,1:3] -->
<!-- mat_NO3 <- as.matrix(mat_NO3[,4:ncol(mat_NO3)])  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- # col_NO3[24,] -->
<!-- # mat_NO3[24,] -->
<!-- # merge 2 matrix -->
<!-- col_NO3 -->
<!-- col_NO2 -->
<!-- identical(col_NO3, col_NO2) # True -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO2NO3 <- cbind(mat_NO2, mat_NO3) -->
<!-- dim(mat_NO2NO3) -->
<!-- # put in the row names -->
<!-- col_NO2$H_mM <- round(col_NO2$H_mM, 2) -->
<!-- class(col_NO2) -->
<!-- vec_sample = unite(col_NO2, "Sample", Titration_type, H_mM,Antibiotics_type, sep=":") -->
<!-- vec_sample = vec_sample$Sample -->
<!-- length(vec_sample) -->
<!-- dim(mat_NO2NO3) -->
<!-- rownames(mat_NO2NO3) = vec_sample -->
<!-- rownames(mat_NO2NO3) -->
<!-- mat_NO2NO3 -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # remove row 24 -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- dim(df_out) -->
<!-- df_out$new <- rownames(df_out) -->
<!-- identical(vec_sample, df_out$new) -->
<!-- df_out <- separate(df_out, new, c("Titration_type","H_mM","Antibiotics_type"), sep=":", remove=T) -->
<!-- df_out$H_mM <- as.numeric(df_out$H_mM) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # make histogram for PC1 -->
<!-- ggplot(df_out, aes(x=PC1)) + -->
<!--   geom_freqpoly(binwidth = 0.5, aes(color=Antibiotics_type), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC1") + -->
<!--   ggtitle(paste0("Marginal distribution of PC1"))+ -->
<!--   mytheme_2d -->
<!-- # make histogram for PC2 -->
<!-- ggplot(df_out, aes(x=PC2)) + -->
<!--   geom_freqpoly(binwidth = 0.85, aes(color=Soil), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC2") + -->
<!--   ggtitle(paste0("Marginal distribution of PC2"))+ -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- # library("plot3D") -->
<!-- # rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- # scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), -->
<!-- #           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!-- #           ylab ="PC2", zlab = "PC3", -->
<!-- #           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- Again with pH values -->
<!-- ```{r} -->
<!-- # function -->
<!-- na_to_average <- function(mat_NO2){ -->
<!--   for (i in 1:dim(mat_NO2)[1]){ -->
<!--     # print(i) -->
<!--     for (j in 1:dim(mat_NO2)[2]){ -->
<!--       # print(j) -->
<!--       if (is.na(mat_NO2[i,j])){ -->
<!--         mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--       } -->
<!--     } -->
<!--   } -->
<!--   return(mat_NO2) -->
<!-- } -->
<!-- df_PCA$Titration_type %>% unique() -->
<!-- ## make the above values to matrices -->
<!-- mat_NO2 <- dcast(df_PCA, pH + Antibiotics_type ~ Time_point, value.var = "NO2_mM") -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- # mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") -->
<!-- dim(mat_NO2) -->
<!-- colnames(mat_NO2) -->
<!-- col_NO2 <- mat_NO2[,1:2] -->
<!-- mat_NO2 <- as.matrix(mat_NO2[,3:ncol(mat_NO2)])  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- mat_NO2 <- na_to_average(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- # col_NO2[24,] -->
<!-- # mat_NO2[24,] -->
<!-- # need to remove row 24 -->
<!-- mat_NO3 <- dcast(df_PCA, pH + Antibiotics_type ~ Time_point, value.var = "NO3_mM") -->
<!-- dim(mat_NO3) -->
<!-- col_NO3 <- mat_NO3[,1:2] -->
<!-- mat_NO3 <- as.matrix(mat_NO3[,3:ncol(mat_NO3)])  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- # col_NO3[24,] -->
<!-- # mat_NO3[24,] -->
<!-- # merge 2 matrix -->
<!-- col_NO3 -->
<!-- col_NO2 -->
<!-- identical(col_NO3, col_NO2) # True -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO2NO3 <- cbind(mat_NO2, mat_NO3) -->
<!-- dim(mat_NO2NO3) -->
<!-- # put in the row names -->
<!-- col_NO2$pH <- round(col_NO2$pH, 2) -->
<!-- class(col_NO2) -->
<!-- vec_sample = unite(col_NO2, "Sample", pH,Antibiotics_type, sep=":") -->
<!-- vec_sample = vec_sample$Sample -->
<!-- length(vec_sample) -->
<!-- dim(mat_NO2NO3) -->
<!-- rownames(mat_NO2NO3) = vec_sample -->
<!-- rownames(mat_NO2NO3) -->
<!-- mat_NO2NO3 -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # remove row 24 -->
<!-- any(is.na(mat_NO2NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- dim(df_out) -->
<!-- df_out$new <- rownames(df_out) -->
<!-- identical(vec_sample, df_out$new) -->
<!-- df_out <- separate(df_out, new, c("pH","Antibiotics_type"), sep=":", remove=T) -->
<!-- df_out$pH <- as.numeric(df_out$pH) -->
<!-- df_out$Antibiotics_type <- factor(df_out$Antibiotics_type, levels = c("None", "CHL", "NYT", "CHL_NYT", "Autoclaved")) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # make histogram for PC1 -->
<!-- ggplot(df_out, aes(x=PC1)) + -->
<!--   geom_freqpoly(binwidth = 0.5, aes(color=pH), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC1") + -->
<!--   ggtitle(paste0("Marginal distribution of PC1"))+ -->
<!--   mytheme_2d -->
<!-- # make histogram for PC2 -->
<!-- ggplot(df_out, aes(x=PC2)) + -->
<!--   geom_freqpoly(binwidth = 0.85, aes(color=pH), size=2) + -->
<!--   scale_fill_brewer(palette='Set2') + -->
<!--   # scale_x_continuous(limits=c(-2.5, 2.5)) + -->
<!--   ylab("Freq") + -->
<!--   xlab("PC2") + -->
<!--   ggtitle(paste0("Marginal distribution of PC2"))+ -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate + Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot pH and PC1 -->
<!-- ggplot(df_out,aes(x=PC1,y=pH, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("pH \n")) + -->
<!--   ggtitle("pH and PC1 \n") + -->
<!--   # ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- Let's plot delta I / delta A -->
<!-- ```{r} -->
<!-- # compute delta I -->
<!-- mat_NO2 -->
<!-- dim(mat_NO2[,1]) -->
<!-- length(mat_NO2[,1]) -->
<!-- mat_delI <- matrix(nrow=nrow(mat_NO2), ncol=ncol(mat_NO2)-1) -->
<!-- for (i in 1:(ncol(mat_NO2)-1)){ -->
<!--   print(i) -->
<!--   # print(mat_NO2[,i+1, drop=F] - mat_NO2[,i, drop=F] ) -->
<!--   mat_delI[,i] <- mat_NO2[,i+1, drop=F] - mat_NO2[,i, drop=F] -->
<!-- } -->
<!-- # compute delta A -->
<!-- mat_NO3 -->
<!-- mat_delA <- matrix(nrow=nrow(mat_NO3), ncol=ncol(mat_NO3)-1) -->
<!-- for (i in 1:(ncol(mat_NO3)-1)){ -->
<!--   print(i) -->
<!--   # print(mat_NO3[,i+1, drop=F] - mat_NO3[,i, drop=F] ) -->
<!--   mat_delA[,i] <- mat_NO3[,i+1, drop=F] - mat_NO3[,i, drop=F] -->
<!-- } -->
<!-- col_NO2 -->
<!-- abs(mat_delI / mat_delA) -->
<!-- # confirming -->
<!-- mat_delI[2,3] / mat_delA[2,3] -->
<!-- mat_delI[28,8] / mat_delA[28,8] -->
<!-- # histogram -->
<!-- hist( abs(mat_delI / mat_delA)) -->
<!-- # remove Inf? -->
<!-- # let's plot -->
<!-- df_ratio_delI_delA <- cbind(col_NO2, abs(mat_delI / mat_delA)) -->
<!-- df_ratio_delI_delA <- melt(df_ratio_delI_delA, id.vars = c("Antibiotics_type","pH"), variable.name = "Time", value.name = "ratio_delI_delA") -->
<!-- df_delA <- cbind(col_NO2, abs(mat_delA)) -->
<!-- df_delA <- melt(df_delA, id.vars = c("Antibiotics_type","pH"), variable.name = "Time", value.name = "delA") -->
<!-- # left join -->
<!-- df_merge_ratio_delA <- df_ratio_delI_delA %>% left_join(df_delA, by = c("Antibiotics_type"="Antibiotics_type","pH"="pH","Time"="Time")) -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I delta A ratio \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   scale_y_continuous(limits=c(0, 1)) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I / delta A \n") + -->
<!--   # ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # plot -->
<!-- ggplot(df_merge_ratio_delA,aes(x=delA, y=ratio_delI_delA, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH, shape = Antibiotics_type))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   scale_y_continuous(limits=c(0, 1)) + -->
<!--   xlab(paste0('\n delta A'))+ -->
<!--   ylab(paste0("delta I / delta A \n")) + -->
<!--   ggtitle("delta I / delta A \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- (1-2) PCA nitrate -->
<!-- ```{r echo =F} -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- dim(df_out) -->
<!-- df_out$H_mM <- rownames(df_out) -->
<!-- df_out$H_mM <- as.numeric(df_out$H_mM) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrate dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), -->
<!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- (2) PCA with only nitrite dynamics -->
<!-- ```{r echo=F} -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- df_out$H_mM <- rownames(df_out) -->
<!-- df_out$H_mM <- as.numeric(df_out$H_mM) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), -->
<!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- (3) PCA with both nitrate and nitrite dynamics -->
<!-- ```{r echo=F} -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO23 <- cbind(mat_NO3, mat_NO2) -->
<!-- any(is.na(mat_NO23)) -->
<!-- dim(mat_NO23) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO23) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- df_out$H_mM <- rownames(df_out) -->
<!-- df_out$H_mM <- as.numeric(df_out$H_mM) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrite + Nitrate dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(H_mM, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = H_mM))+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("Principle component analysis (Nitrite + Nitrate dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (col_pH(100)), -->
<!--           main = "Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- ## 4.1.1.3. (using inferred pH) PCA on the nitrate, nitrite time readings.  -->
<!-- (1) (using inferred pH) First plot PCA with only the nitrate -->
<!-- ```{r echo =F} -->
<!-- ## make the above values to matrices -->
<!-- df_plot2$pH <- (6.5 / (1+ exp(1*(0.07*(df_plot2$H_mM+125))-10))) + 2.5 -->
<!-- mat_NO2 <- dcast(df_plot2, pH ~ Time_hours, value.var = "NO2_mM") -->
<!-- mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "pH") -->
<!-- mat_NO2 <- as.matrix(mat_NO2)  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] -->
<!-- mat_NO2 <- na_to_average(mat_NO2) -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- mat_NO3 <- dcast(df_plot2, pH ~ Time_hours, value.var = "NO3_mM") -->
<!-- mat_NO3 <- tibble::column_to_rownames(mat_NO3, var = "pH") -->
<!-- mat_NO3 <- as.matrix(mat_NO3)  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO3)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO3) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- df_out$pH <- rownames(df_out) -->
<!-- df_out$pH <- as.numeric(df_out$pH) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrate dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrate dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), -->
<!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- (2) (using inferred pH) PCA with only nitrite dynamics -->
<!-- ```{r echo=F} -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- any(is.na(mat_NO2)) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO2) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- df_out$pH <- rownames(df_out) -->
<!-- df_out$pH <- as.numeric(df_out$pH) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), -->
<!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- (3) (using inferred pH) PCA with both nitrate and nitrite dynamics -->
<!-- ```{r echo=F} -->
<!-- # First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html) -->
<!-- dim(mat_NO2) -->
<!-- dim(mat_NO3) -->
<!-- mat_NO23 <- cbind(mat_NO3, mat_NO2) -->
<!-- any(is.na(mat_NO23)) -->
<!-- dim(mat_NO23) -->
<!-- # PCA with prcomp -->
<!-- pc <- prcomp(mat_NO23) -->
<!-- eigs <- pc$sdev^2 -->
<!-- # variance explained -->
<!-- eigs[1] / sum(eigs) # pc1 -->
<!-- eigs[2] / sum(eigs) # pc2 -->
<!-- # First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA) -->
<!-- df_out <- as.data.frame(pc$x) -->
<!-- df_out -->
<!-- head(df_out) -->
<!-- df_out$pH <- rownames(df_out) -->
<!-- df_out$pH <- as.numeric(df_out$pH) -->
<!-- # plot PCA with timepoint labels -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2))+ -->
<!--   geom_point(size = 4, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n") + -->
<!--   mytheme_2d -->
<!-- # Label with mM -->
<!-- ggplot(df_out,aes(x=PC1,y=PC2, label=sprintf("%0.1f", round(pH, digits = 1))))+ -->
<!--   geom_point(size = 5, alpha=0.9, aes(col = pH))+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   xlab(paste0('\n PC1', " (explained variance: ",round(eigs[1] / sum(eigs),3)*100,"%)"))+ -->
<!--   ylab(paste0("PC2"," (explained variance: ",round(eigs[2] / sum(eigs),3)*100,"%) \n")) + -->
<!--   ggtitle("[Using inferred pH] Principle component analysis (Nitrite + Nitrate dynamics) \n") + -->
<!--   ggrepel::geom_label_repel() + -->
<!--   mytheme_2d -->
<!-- # 3D: http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization -->
<!-- library("plot3D") -->
<!-- rownames(df_out) <- as.numeric(rownames(df_out)) -->
<!-- scatter3D(df_out$PC1, df_out$PC2, df_out$PC3, bty = "b2", pch = 16, cex = 2,col = (rev(col_pH(100))), -->
<!--           main = "[Using inferred pH] Principle component analysis (Nitrate dynamics)", xlab = "PC1", -->
<!--           ylab ="PC2", zlab = "PC3", -->
<!--           theta = 15, phi = 50) -->
<!-- ## is this correct? I don't think so.. doesn't agree with the 2d plot.... let's come back to this some time later.. -->
<!-- ``` -->
<!-- Let's take a closer look at the basic pH curves -->
<!-- ```{r echo=F} -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -70, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (H_mM > -70) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## 60 -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -60, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (H_mM > -60) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## 50 -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$H_mM > -50, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (H_mM > -50) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH < 8.9, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (inferred pH > 8.9) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## 3 clusters - first cluster -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH < 8.9, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (inferred pH > 8.9) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## 3 clusters - second cluster -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH > 8.9 | df_plot3$pH < 4, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("pH perturbation (4 < inferred pH < 8.9) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## 3 clusters - third cluster -->
<!-- df_plot3 <- df_plot2 -->
<!-- df_plot3$NO3_mM <- ifelse(df_plot3$pH > 4, NA, df_plot3$NO3_mM) -->
<!-- ggplot(df_plot3, aes(x=Time_hours, y=NO3_mM, color=pH, group=pH)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.2)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - Std_NO3_mM, ymax=NO3_mM + Std_NO3_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = rev(col_pH(100))) + -->
<!--   # scale_color_manual(values=grad_pH) + -->
<!--   ylab("NO3- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   scale_y_continuous(breaks = seq(0,2.5,0.25), limits=c(0, 2.5))+ -->
<!--   ggtitle("pH perturbation (inferred pH < 4) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO3_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   facet_grid(. ~ Soil) + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ``` -->
<!-- ## 4.1.2. Peak time -->
<!-- ```{r echo=F} -->
<!-- # maximum peak time -->
<!-- vec_peaktime <- apply(mat_H_time, 1, which.max) -->
<!-- vec_hr <- df_time$Time_hours -->
<!-- vec_hr[vec_peaktime] -->
<!-- names(vec_peaktime) -->
<!-- plot(names(vec_peaktime), vec_hr[vec_peaktime]) -->
<!-- df_peak_NO2 <- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime]) -->
<!-- df_peak_NO2$H_mM <- as.numeric(df_peak_NO2$H_mM) -->
<!-- # 1. linear regression -->
<!-- fit.no2peak <- lm(Time_hours ~ H_mM, df_peak_NO2) -->
<!-- summary(fit.no2peak) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x", ",  R^2: ", round(summary(fit.no2peak)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # 2. quadratic regression -->
<!-- df_peak_NO2$H_mM_2 <- (df_peak_NO2$H_mM)^2  -->
<!-- quadraticModel <- lm(Time_hours ~ H_mM + H_mM_2, df_peak_NO2) -->
<!-- summary(quadraticModel) -->
<!-- perturbValues <- seq(-100, 100, 0.1) -->
<!-- peakPredict <- predict(quadraticModel,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) -->
<!-- df_quad <- data.frame(H_mM = perturbValues, Time_hours = peakPredict) -->
<!-- plot(perturbValues, peakPredict) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_peak_NO2, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks \n") + -->
<!--   geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = "blue", size = 1) + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=0,y=30, label= paste0("y = ", round(coef(quadraticModel)[[1]],3), "+", round(coef(quadraticModel)[[2]],3),"x+",round(coef(quadraticModel)[[3]],3), "x^2", ",  R^2: ", round(summary(quadraticModel)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 4.1.3. Get rid of noise by peak threshold -->
<!-- ```{r echo=F} -->
<!-- # maximum peak value -->
<!-- apply(mat_H_time, 1, max) %>% length() -->
<!-- vec_max <- apply(mat_H_time, 1, max) -->
<!-- hist(vec_max, breaks = 50, xlab = "NO2 mM at peak", main = "Histogram of peak NO2 concentration") -->
<!-- sort(vec_max) -->
<!-- min_peak_NO2_mM <- 0.05 -->
<!-- vec_max[vec_max > min_peak_NO2_mM] -->
<!-- # maximum peak time -->
<!-- vec_peaktime <- apply(mat_H_time, 1, which.max) -->
<!-- vec_hr <- df_time$Time_hours -->
<!-- vec_hr[vec_peaktime] -->
<!-- names(vec_peaktime) -->
<!-- plot(names(vec_peaktime), vec_hr[vec_peaktime]) -->
<!-- df_peak_NO2 <- data.frame(H_mM = names(vec_peaktime), Time_hours = vec_hr[vec_peaktime]) -->
<!-- df_peak_NO2$H_mM <- as.numeric(df_peak_NO2$H_mM) -->
<!-- df_peak_NO2_threshold <- df_peak_NO2[vec_max > min_peak_NO2_mM, ] -->
<!-- dim(df_peak_NO2_threshold) -->
<!-- df_threshold <- na.omit(df_peak_NO2_threshold) -->
<!-- dim(df_threshold) -->
<!-- # 1. linear regression -->
<!-- fit.no2peak <- lm(Time_hours ~ H_mM, df_threshold) -->
<!-- summary(fit.no2peak) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_abline(slope = coef(fit.no2peak)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=-40,y=40, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x", ",  R^2: ", round(summary(fit.no2peak)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # 2. quadratic regression -->
<!-- df_threshold$H_mM_2 <- (df_threshold$H_mM)^2  -->
<!-- quadraticModel_th <- lm(Time_hours ~ H_mM + H_mM_2, df_threshold) -->
<!-- summary(quadraticModel_th) -->
<!-- perturbValues <- seq(-100, 10, 0.1) -->
<!-- peakPredict <- predict(quadraticModel_th,list(H_mM=perturbValues, H_mM_2=perturbValues^2)) -->
<!-- df_quad <- data.frame(H_mM = perturbValues, Time_hours = peakPredict) -->
<!-- plot(perturbValues, peakPredict) -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   # annotate("text",x=0,y=30, label= paste0("y = ", round(coef(fit.no2peak)[[1]],3),"+",round(coef(fit.no2peak)[[2]],3),"x"), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- # Plot H_mM with peak time -->
<!-- ggplot(df_threshold, aes(x=H_mM, y=Time_hours)) + -->
<!--   geom_point(size=2.5, shape=16, color = "blue") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_line(size=1)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n NO2 (Nitrite) peak time (hr)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with pH perturbation and nitrite peaks (NO2_mM > 0.02) \n") + -->
<!--   geom_line(data = df_quad, aes(x = H_mM, y = Time_hours), color = "blue", size = 1) + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_abline(slope = coef(fit.no2peak_quad)[[2]], intercept = coef(fit.no2peak)[[1]], -->
<!--               # color = "maroon2") + -->
<!--   # show equation -->
<!--   # stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) + -->
<!--   annotate("text",x=-45,y=40, label= paste0("y = ", round(coef(quadraticModel_th)[[1]],3), "+", round(coef(quadraticModel_th)[[2]],3),"x+",round(coef(quadraticModel_th)[[3]],3), "x^2", ",  R^2: ", round(summary(quadraticModel_th)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 4.2. Area under curve calculation -->
<!-- ```{r echo=F} -->
<!-- df_plot2$Time_hours -->
<!-- require(pracma) -->
<!-- colnames(df_plot2) -->
<!-- df_corr <- df_plot2 %>% select(H_mM, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM) %>% summarize(auc = trapz(Time_hours, NO3_mM)) %>% ungroup() -->
<!-- plot(df_corr$H_mM, df_corr$auc) -->
<!-- df_corr$H_mM_2 <- (df_corr$H_mM)^2 -->
<!-- fit.no3 <- lm(auc ~ H_mM + H_mM_2, df_corr) -->
<!-- summary(fit.no3) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aucPredict <- predict(fit.no3,list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_auc_quad <- data.frame(H_mM = perturbH, Time_hours = aucPredict) -->
<!-- plot(perturbH, aucPredict) -->
<!-- # (1) Plot fitted linear regression line -->
<!-- ggplot(df_corr, aes(x=H_mM, y=auc)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Area under curve (NO3-)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and area under nitrate curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=0,y=100, label= paste0("y = ", round(coef(fit.no3)[[1]],3), "+", round(coef(fit.no3)[[2]],3),"x+",round(coef(fit.no3)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no3)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ## 220213 Use the inferred pH values for x axis -->
<!-- # y <- (6.5 / (1+ exp(1*(0.07*(x+125))-10))) + 2.5 -->
<!-- df_corr$inferred_pH <-  (6.5 / (1+ exp(1*(0.07*(df_corr$H_mM+125))-10))) + 2.5 -->
<!-- plot(df_corr$inferred_pH, df_corr$auc) -->
<!-- df_corr$inferred_pH_2 <- (df_corr$inferred_pH)^2 -->
<!-- fit.no3 <- lm(auc ~ inferred_pH + inferred_pH_2, df_corr) -->
<!-- summary(fit.no3) -->
<!-- perturbpH <- seq(2, 11, 0.01) -->
<!-- aucPredict <- predict(fit.no3,list(inferred_pH=perturbpH, inferred_pH_2=perturbpH^2)) -->
<!-- df_auc_quad <- data.frame(H_mM = perturbpH, auc = aucPredict) -->
<!-- plot(perturbpH, aucPredict) -->
<!-- # (1) Plot fitted linear regression line -->
<!-- ggplot(df_corr, aes(x=inferred_pH, y=auc)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("inferred pH \n") + -->
<!--   ylab("\n Area under curve (NO3-)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbed pH and area under nitrate curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = auc), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=7,y=100, label= paste0("y = ", round(coef(fit.no3)[[1]],3), "+", round(coef(fit.no3)[[2]],3),"x+",round(coef(fit.no3)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no3)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ## NO2 area under curve -->
<!-- colnames(df_plot2) -->
<!-- df_corr_no2 <- df_plot2 %>% select(H_mM, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM) %>% summarize(auc = trapz(Time_hours, NO2_mM)) %>% ungroup() -->
<!-- plot(df_corr_no2$H_mM, df_corr_no2$auc) -->
<!-- fit.no2 <- lm(auc ~ H_mM, df_corr_no2) -->
<!-- summary(fit.no2) -->
<!-- df_corr_no2$H_mM_2 <- (df_corr_no2$H_mM)^2  -->
<!-- fit.no2.quad <- lm(auc ~ H_mM + H_mM_2, df_corr_no2) -->
<!-- summary(fit.no2.quad) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aucPredict <- predict(fit.no2.quad,list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_auc_quad <- data.frame(H_mM = perturbH, Time_hours = aucPredict) -->
<!-- plot(perturbH, aucPredict) -->
<!-- # (1) Plot fitted linear regression line -->
<!-- ggplot(df_corr_no2, aes(x=H_mM, y=auc)) + -->
<!--   geom_point(size=2.5, shape=21, color = "brown") + -->
<!--   # geom_line(size=1)+ -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Area under curve (NO2-)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and area under nitrite curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_auc_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=0,y=20, label= paste0("y = ", round(coef(fit.no2.quad)[[1]],3), "+", round(coef(fit.no2.quad)[[2]],3),"x+",round(coef(fit.no2.quad)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.no2.quad)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 4.3. 1/AUC -->
<!-- ```{r echo=F} -->
<!-- df_plot2$Time_hours -->
<!-- require(pracma) -->
<!-- colnames(df_plot2) -->
<!-- dim(df_plot2) -->
<!-- head(df_plot2) -->
<!-- df_corr <- df_plot2 %>% select(H_mM, Soil, Time_hours, NO3_mM, NO2_mM) %>% group_by(H_mM, Soil) %>% summarize(auc = trapz(Time_hours, NO3_mM)) %>% ungroup() -->
<!-- df_corr$auc_inv <- 1/(df_corr$auc) -->
<!-- plot(df_corr$H_mM, df_corr$auc_inv) -->
<!-- # remove the outlier 1 point -->
<!-- df_corr %>% filter(auc_inv>1) -->
<!-- dim(df_corr) # 94 -->
<!-- df_corr %<>% filter(!(auc_inv>1)) -->
<!-- dim(df_corr) # 93 -->
<!-- # (1) Plot fitted linear regression line -->
<!-- ggplot(df_corr, aes(x=H_mM, y=auc_inv, color=Soil, group=Soil)) + -->
<!--   geom_point(size=3, shape=21) + -->
<!--   geom_line() + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n 1/AUC of Nitrate dynamics") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Unimodal distribution of 3 soils \n") + -->
<!--   mytheme_2d -->
<!-- ggplot(df_corr, aes(x=H_mM, y=auc_inv, color=Soil, group=Soil)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n 1/AUC of Nitrate dynamics") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Unimodal distribution of 3 soils \n") + -->
<!--   mytheme_2d -->
<!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ``` -->
<!-- ## 5. Get Ir dynamics (Reduction rate of the nitrite) -->
<!-- ## 5.1. I p -->
<!-- ```{r echo=F} -->
<!-- df_plot2 -->
<!-- # Nitrite produced I_produced = A - A(t) -->
<!-- df_plot2 %>% filter(Time_hours == 0) -->
<!-- df_plot2 %>% group_by(H_mM, Soil, Time_hours) -->
<!-- mat_NO3 <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO3_mM") -->
<!-- mat_NO3 <- tibble::column_to_rownames(mat_NO3, var = "H_mM") -->
<!-- mat_NO3 <- as.matrix(mat_NO3)  -->
<!-- class(mat_NO3) -->
<!-- dim(mat_NO3) -->
<!-- # NA -->
<!-- mat_NO3 <- na_to_average(mat_NO3) -->
<!-- mat_NO3[is.na(mat_NO3)] # no more NA -->
<!-- mat_NO3[,2:dim(mat_NO3)[2]] -->
<!-- mat_NO3[,1] - mat_NO3[,1] -->
<!-- mat_NO3_consumed <- mat_NO3 -->
<!-- for (j in 1:dim(mat_NO3_consumed)[2]){ -->
<!--   mat_NO3_consumed[,j] = mat_NO3[,1] - mat_NO3_consumed[,j] -->
<!-- } -->
<!-- mat_NO3_consumed -->
<!-- # make negative values into 0. -->
<!-- mat_NO3_consumed[mat_NO3_consumed<0] <- 0 -->
<!-- mat_NO3_consumed -->
<!-- # This is NO2 produced -->
<!-- mat_NO2_produced = mat_NO3_consumed -->
<!-- df_Ip <- melt(mat_NO2_produced)  -->
<!-- names(df_Ip) <- c("H_mM","Time_hours","NO2_mM") -->
<!-- # plot I p = Nitrite produced (accumulative) = Nitrate reduced  -->
<!-- ggplot(df_Ip, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_p (Nitrite produced) = A0 - A(t) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ``` -->
<!-- ## 5.1. I r -->
<!-- ```{r echo=F} -->
<!-- # plot I r = Nitrite reduced (accumulative) = I p - I -->
<!-- mat_NO2_produced[is.na(mat_NO2_produced)]  -->
<!-- mat_NO2_produced -->
<!-- dim(mat_NO2_produced) -->
<!-- mat_NO2 <- dcast(df_plot2, H_mM ~ Time_hours, value.var = "NO2_mM") -->
<!-- mat_NO2 <- tibble::column_to_rownames(mat_NO2, var = "H_mM") -->
<!-- mat_NO2 <- as.matrix(mat_NO2)  -->
<!-- class(mat_NO2) -->
<!-- dim(mat_NO2) -->
<!-- # Substitute NA to average of NO2_mM of before and after. -->
<!-- mat_NO2[is.na(mat_NO2)]  -->
<!-- for (i in 1:dim(mat_NO2)[1]){ -->
<!--   # print(i) -->
<!--   for (j in 1:dim(mat_NO2)[2]){ -->
<!--     # print(j) -->
<!--     if (is.na(mat_NO2[i,j])){ -->
<!--       mat_NO2[i,j] <- (mat_NO2[i,j-1] + mat_NO2[i,j+1] ) / 2 -->
<!--     } -->
<!--   } -->
<!-- } -->
<!-- mat_NO2[is.na(mat_NO2)] # no more NA -->
<!-- mat_Ir <- mat_NO2_produced - mat_NO2 -->
<!-- df_Ir <- melt(mat_Ir)  -->
<!-- names(df_Ir) <- c("H_mM","Time_hours","NO2_mM") -->
<!-- # plot I r = reduced Nitrite (accumulative) -->
<!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ``` -->
<!-- ## 5.1.1. AAC (area above curve) -->
<!-- ```{r echo=F} -->
<!-- df_Ir$Time_hours -->
<!-- require(pracma) -->
<!-- colnames(df_Ir) -->
<!-- # Let's get the area above curve -->
<!-- # what is the total area? -->
<!-- time_max = max(df_Ir$Time_hours) -->
<!-- upper_limit_mean = mean(df_Ir %>% filter(Time_hours == time_max) %>% filter(NO2_mM > 1.8) %>% select(NO2_mM) %>% unlist()) -->
<!-- total_area = time_max * upper_limit_mean -->
<!-- # correlation -->
<!-- df_corr_aac <- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(aac = total_area - trapz(Time_hours, NO2_mM)) %>% ungroup() -->
<!-- is.na(df_corr_aac) ## no NA -->
<!-- plot(df_corr$H_mM, df_corr_aac$aac) -->
<!-- df_corr_aac$H_mM_2 <- (df_corr_aac$H_mM)^2 -->
<!-- fit.aac <- lm(aac ~ H_mM + H_mM_2, df_corr_aac) -->
<!-- summary(fit.aac) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aacPredict <- predict(fit.aac,list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_aac_quad <- data.frame(H_mM = perturbH, Time_hours = aacPredict) -->
<!-- plot(perturbH, aacPredict) -->
<!-- # (1) Plot fitted quadratic regression line -->
<!-- ggplot(df_corr_aac, aes(x=H_mM, y=aac)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Area above I_r curve (NO2-)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and area under nitrate curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_aac_quad, aes(x = H_mM, y = Time_hours), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=0,y=100, label= paste0("y = ", round(coef(fit.aac)[[1]],3), "+", round(coef(fit.aac)[[2]],3),"x+",round(coef(fit.aac)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aac)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 5.1.2. Average slope (Max NO2_mM / Time) -->
<!-- ```{r echo=F} -->
<!-- df_Ir_av_slope <- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(average_slope = max(NO2_mM) trapz(Time_hours, NO2_mM)) %>% ungroup() -->
<!-- df_Ir %>% select(H_mM, Time_hours, NO2_mM) %>% group_by(H_mM) %>% summarize(max(NO2_mM)) -->
<!-- ## get peaks -->
<!-- # https://stackoverflow.com/questions/55342232/how-to-find-local-maximum-in-r-from-graph  -->
<!-- # plot I r = reduced Nitrite (accumulative) -->
<!-- library(ggpmisc) -->
<!-- ggplot(df_Ir, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   stat_peaks(col = "green", size = 4) + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## plot in a different way -->
<!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 15 & H_mM > -15) -->
<!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # stat_peaks(col = "green", size = 4) + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## plot in a different way -->
<!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 30 & H_mM > -30) -->
<!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # stat_peaks(col = "green", size = 4) + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## plot in a different way -->
<!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 50 & H_mM >-50) -->
<!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # stat_peaks(col = "green", size = 4) + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## plot in a different way -->
<!-- df_Ir_filt <- df_Ir %>% filter(H_mM < 70 & H_mM > -70) -->
<!-- ggplot(df_Ir_filt, aes(x=Time_hours, y=NO2_mM, color=H_mM, group=H_mM)) + -->
<!--   geom_point(size=2.5, shape=16) + -->
<!--   geom_line(size=1.5)+ -->
<!--   # geom_errorbar(aes(ymin=NO2_mM - Std_NO2_mM, ymax=NO2_mM + Std_NO2_mM), width=.05)+ -->
<!--   scale_colour_gradientn(colours = col_pH(100)) + -->
<!--   # scale_color_manual(values=col_pH(4)) + -->
<!--   ylab("NO2- (mM) \n") + -->
<!--   xlab("\n Time (hr)") + -->
<!--   # scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   ggtitle("I_r (Nitrite reduced) = I_p - I(t) \n") + -->
<!--   # stat_peaks(col = "green", size = 4) + -->
<!--   # label -->
<!--   # geom_text(aes(label = round(NO2_mM,3)), size = 3, vjust = -1.5, family="serif", show.legend = FALSE)+ -->
<!--   mytheme_2d + -->
<!--   theme(strip.background = element_rect(colour="black", fill="white", size=0.1)) -->
<!-- ## get peaks coordinates -->
<!-- mat_Ir -->
<!-- x = colnames(mat_Ir) -->
<!-- y = mat_Ir[1,] -->
<!-- x[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)] -->
<!-- y[ggpmisc:::find_peaks(mat_Ir[1,], span = 5)] -->
<!-- plot(x, y) -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   print(i) -->
<!--   x = colnames(mat_Ir) -->
<!--   y = mat_Ir[i,] -->
<!--   x_max = x[ggpmisc:::find_peaks(y, span = 3)] -->
<!--   y_max = y[ggpmisc:::find_peaks(y, span = 3)] -->
<!--   # print(x_max) -->
<!--   print(y_max) -->
<!--   plot(x, y) -->
<!-- } -->
<!-- ## Max -->
<!-- vec_H_mM <- rownames(mat_Ir) -->
<!-- vec_max_NO_mM <- rep(-1, dim(mat_Ir)[1]) -->
<!-- vec_max_time <- rep(-1, dim(mat_Ir)[1]) -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   y = mat_Ir[i,] -->
<!--   max_j = max(y) -->
<!--   # save -->
<!--   vec_max_NO_mM[i] <- max_j -->
<!--   vec_max_time[i] <- which.max(y) -->
<!--   iter_j <- which.max(y) -1 -->
<!--   # get the more early points if the difference with max is not big (10%) -->
<!--   for (k in 1:iter_j){ -->
<!--     print(k) -->
<!--     if (abs(max_j - y[which.max(y)-k])/max_j < 0.1){ -->
<!--       max_j = y[which.max(y)-k] -->
<!--       # print(max_j) -->
<!--       vec_max_NO_mM[i] <- max_j -->
<!--       print(names(max_j)) -->
<!--       vec_max_time[i] <- names(max_j) -->
<!--     } else{ break } -->
<!--   } -->
<!-- } -->
<!-- df_ave_slope <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time) -->
<!-- # check -->
<!-- i = 3 -->
<!-- plot(colnames(mat_Ir), mat_Ir[i,]) -->
<!-- points(vec_max_time[i], vec_max_NO_mM[i], col='red') -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   plot(colnames(mat_Ir), mat_Ir[i,]) -->
<!--   points(vec_max_time[i], vec_max_NO_mM[i], col='red') -->
<!-- } -->
<!-- # get average slope -->
<!-- df_ave_slope$H_mM <- as.numeric(df_ave_slope$H_mM) -->
<!-- df_ave_slope$average_slope <- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours) -->
<!-- # plot -->
<!-- plot(vec_H_mM, df_ave_slope$average_slope) -->
<!-- # quadratic fit -->
<!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 -->
<!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) -->
<!-- summary(fit.aveslope) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) -->
<!-- plot(perturbH, aveslopePredict) -->
<!-- # (1) Plot fitted quadratic regression line -->
<!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ## restrict the samples with maximum over 1.5 mM NO2 -->
<!-- df_ave_slope %<>% filter(NO2_mM > 1.5) -->
<!-- # quadratic fit -->
<!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 -->
<!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) -->
<!-- summary(fit.aveslope) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) -->
<!-- plot(perturbH, aveslopePredict) -->
<!-- # (1) Plot fitted quadratic regression line -->
<!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   # geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   # annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 5.1.3. Got a new idea. Start from the end -->
<!-- ```{r echo=F} -->
<!-- ## Max -->
<!-- vec_H_mM <- rownames(mat_Ir) -->
<!-- vec_max_NO_mM <- rep(-1, dim(mat_Ir)[1]) -->
<!-- vec_max_time <- rep(-1, dim(mat_Ir)[1]) -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   x = colnames(mat_Ir) -->
<!--   y = mat_Ir[i,] -->
<!--   max_j = y[dim(mat_Ir)[2]] -->
<!--   # save -->
<!--   vec_max_NO_mM[i] <- max_j -->
<!--   vec_max_time[i] <- x[dim(mat_Ir)[2]] -->
<!--   iter_j <- dim(mat_Ir)[2] -1 -->
<!--   # get the more early points if the difference with max is not big (10%) -->
<!--   for (k in 1:iter_j){ -->
<!--     print(k) -->
<!--     if ( (y[dim(mat_Ir)[2]-k] > max_j) || (abs(max_j - y[dim(mat_Ir)[2]-k])/max_j < 0.1) ){ -->
<!--       max_j = y[dim(mat_Ir)[2] - k] -->
<!--       print(max_j) -->
<!--       vec_max_NO_mM[i] <- max_j -->
<!--       print(names(max_j)) -->
<!--       vec_max_time[i] <- names(max_j) -->
<!--     } else{ break } -->
<!--   } -->
<!-- } -->
<!-- df_ave_slope <- data.frame(H_mM = vec_H_mM, NO2_mM = vec_max_NO_mM, Time_hours = vec_max_time) -->
<!-- # check -->
<!-- i = 36 -->
<!-- plot(colnames(mat_Ir), mat_Ir[i,]) -->
<!-- points(vec_max_time[i], vec_max_NO_mM[i], col='red') -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   plot(colnames(mat_Ir), mat_Ir[i,]) -->
<!--   points(vec_max_time[i], vec_max_NO_mM[i], col='red') -->
<!-- } -->
<!-- # get average slope -->
<!-- df_ave_slope$H_mM <- as.numeric(df_ave_slope$H_mM) -->
<!-- df_ave_slope$average_slope <- df_ave_slope$NO2_mM / as.numeric(df_ave_slope$Time_hours) -->
<!-- # plot -->
<!-- plot(vec_H_mM, df_ave_slope$average_slope) -->
<!-- # quadratic fit -->
<!-- df_ave_slope$H_mM_2 <- (df_ave_slope$H_mM)^2 -->
<!-- fit.aveslope <- lm(average_slope ~ H_mM + H_mM_2, df_ave_slope) -->
<!-- summary(fit.aveslope) -->
<!-- perturbH <- seq(-100, 100, 0.1) -->
<!-- aveslopePredict <- predict(fit.aveslope, list(H_mM=perturbH, H_mM_2=perturbH^2)) -->
<!-- df_avs_quad <- data.frame(H_mM = perturbH, average_slope = aveslopePredict) -->
<!-- plot(perturbH, aveslopePredict) -->
<!-- # (1) Plot fitted quadratic regression line -->
<!-- ggplot(df_ave_slope, aes(x=H_mM, y=average_slope)) + -->
<!--   geom_point(size=2.5, shape=16, color = "brown") + -->
<!--   # stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 1) + -->
<!--   # geom_errorbar(aes(ymin=NO3_mM - sd_NO3_mM, ymax=NO3_mM + sd_NO3_mM), width=.05)+ -->
<!--   # scale_color_brewer(palette='Set2') + -->
<!--   # scale_color_manual(values = c("maroon2","deepskyblue4"))+ -->
<!--   xlab("Amount of pH perturbation (H+ mM) \n") + -->
<!--   ylab("\n Average slope of I_r curve (NO2_mM/Time_hours)") + -->
<!--   # scale_x_continuous(breaks = seq(0,16,1), limits=c(0, 16))+ -->
<!--   #scale_y_continuous(breaks = seq(0,0.3,0.05), limits=c(0, 0.3))+ -->
<!--   # geom_text_repel(aes(label = round(NO3_mM,3)), size = 3,family="serif", show.legend = FALSE)+ -->
<!--   ggtitle("Correlation with perturbation and average slope of I_r curve \n") + -->
<!--   # geom_abline(slope = 1, intercept=0, show.legend = "y=x")+ -->
<!--   # regression line -->
<!--   geom_line(data = df_avs_quad, aes(x = H_mM, y = average_slope), color = "maroon2", size = 1) + -->
<!--   # show equation -->
<!--   annotate("text",x=0,y=0,label= paste0("y = ", round(coef(fit.aveslope)[[1]],3), "+", round(coef(fit.aveslope)[[2]],3),"x+",round(coef(fit.aveslope)[[3]],3), "x^2", ",  R^2: ", round(summary(fit.aveslope)$r.squared,3)), color = "maroon2") + -->
<!--   mytheme_2d -->
<!-- ``` -->
<!-- ## 5.1.4. T max/2 -->
<!-- ```{r echo=F} -->
<!-- # Michaelis-Menten function (https://rpubs.com/dstoebel/576374) -->
<!-- y = mat_Ir[i,][2:dim(mat_Ir)[2]] -->
<!-- x = as.numeric(names(y)) -->
<!-- mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  -->
<!-- coefficients(mm_fit)[1] -->
<!-- coefficients(mm_fit)[2] # Km -->
<!-- plot(x, y) -->
<!-- curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE) -->
<!-- i=2 -->
<!-- y = mat_Ir[i,][2:dim(mat_Ir)[2]] -->
<!-- x = as.numeric(names(y)) -->
<!-- mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  -->
<!-- coefficients(mm_fit)[1] -->
<!-- coefficients(mm_fit)[2] # Km -->
<!-- plot(x, y) -->
<!-- curve(expr = coefficients(mm_fit)[1]*x/(coefficients(mm_fit)[2]+x), add = TRUE) -->
<!-- # this failed -->
<!-- # get T max/2 -->
<!-- vec_T_0.5max <- rep(-1, dim(mat_Ir)[1]) -->
<!-- for (i in 1:dim(mat_Ir)[1]){ -->
<!--   print(i) -->
<!--   y = mat_Ir[i,][2:dim(mat_Ir)[2]] -->
<!--   x = as.numeric(names(y)) -->
<!--   mm_fit <- nls(y ~ SSmicmen(x, Vmax, Km))  -->
<!--   # T max/2 -->
<!--   vec_T_0.5max[i] <- coefficients(mm_fit)[2]  -->
<!-- } -->
<!-- ``` -->
<!-- ## 6. Correlation between mu -->
<!-- ```{r echo=F} -->
<!-- # From A -->
<!-- df_corr$auc # area under curve -->
<!-- # From I -->
<!-- vec_peakheight # peak height -->
<!-- df_peak_NO2$Time_hours # peak time -->
<!-- df_corr_no2$auc # AUC -->
<!-- # From I_R -->
<!-- df_corr_aac$aac # area above curve -->
<!-- df_ave_slope$average_slope # average slope -->
<!-- # wrap it up -->
<!-- df_mu <- data.frame(A_AUC = df_corr$auc, -->
<!--                     I_peak_height = vec_peakheight, -->
<!--                     I_peak_time = df_peak_NO2$Time_hours, -->
<!--                     I_AUC = df_corr_no2$auc, -->
<!--                     I_R_AAC = df_corr_aac$aac, -->
<!--                     I_R_average_slope = df_ave_slope$average_slope -->
<!--                     ) -->
<!-- pairs(df_mu, main = "mu space correlation", -->
<!--       pch=21, cex.labels = 2, cex=1.5, cex.main=2, col='blue') -->
<!-- df_mu -->
<!-- scaled.dat <- scale(df_mu) -->
<!-- pairs(scaled.dat, main = "(scaled) mu space correlation", -->
<!--       pch=21, cex.labels = 2, cex=1.5, cex.main=2, col='blue') -->
<!-- ``` -->
<!-- ## 6.1. Add ammonia -->
<!-- ```{r echo=F} -->
<!-- # wrap it up -->
<!-- df_mu_amm <- data.frame(A_AUC = df_corr$auc, -->
<!--                     I_peak_height = vec_peakheight, -->
<!--                     I_peak_time = df_peak_NO2$Time_hours, -->
<!--                     I_AUC = df_corr_no2$auc, -->
<!--                     I_R_AAC = df_corr_aac$aac, -->
<!--                     I_R_average_slope = df_ave_slope$average_slope, -->
<!--                     Amm_AUC = am_corr$auc, -->
<!--                     Amm_average_slope  = am_ave_slope$average_slope -->
<!--                     ) -->
<!-- pairs(df_mu_amm, main = "mu space correlation", -->
<!--       pch=21, cex.labels = 1.7, cex=1.5, cex.main=2, col='blue') -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] colorRamps_2.3     rmarkdown_2.9      gridExtra_2.3      ggpubr_0.4.0      
 [5] ggrepel_0.9.1      ape_5.5            openxlsx_4.2.3     devtools_2.4.0    
 [9] usethis_2.0.1      gtools_3.8.2       reshape2_1.4.4     readxl_1.3.1      
[13] magrittr_2.0.1     forcats_0.5.1      stringr_1.4.0      purrr_0.3.4       
[17] readr_1.4.0        tidyr_1.1.3        tibble_3.0.4       tidyverse_1.3.1   
[21] vegan_2.5-7        lattice_0.20-41    permute_0.9-5      RColorBrewer_1.1-2
[25] ggplot2_3.3.5      dplyr_1.0.5        workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0  ggsignif_0.6.2    ellipsis_0.3.2    rio_0.5.27       
 [5] rprojroot_2.0.2   fs_1.5.0          rstudioapi_0.13   farver_2.1.0     
 [9] remotes_2.4.0     fansi_0.4.2       lubridate_1.7.10  xml2_1.3.2       
[13] splines_4.0.3     cachem_1.0.4      knitr_1.37        pkgload_1.2.1    
[17] jsonlite_1.7.2    broom_0.7.9       cluster_2.1.0     dbplyr_2.1.1     
[21] compiler_4.0.3    httr_1.4.2        backports_1.2.1   assertthat_0.2.1 
[25] Matrix_1.2-18     fastmap_1.1.0     cli_3.0.1         later_1.2.0      
[29] htmltools_0.5.1.1 prettyunits_1.1.1 tools_4.0.3       gtable_0.3.0     
[33] glue_1.4.2        Rcpp_1.0.8.3      carData_3.0-4     cellranger_1.1.0 
[37] jquerylib_0.1.4   vctrs_0.3.8       nlme_3.1-149      xfun_0.29        
[41] ps_1.6.0          testthat_3.0.2    rvest_1.0.1       lifecycle_1.0.0  
[45] rstatix_0.7.0     MASS_7.3-53       scales_1.1.1      hms_1.1.0        
[49] promises_1.2.0.1  parallel_4.0.3    yaml_2.2.1        curl_4.3.2       
[53] memoise_2.0.0     sass_0.4.0        stringi_1.5.3     highr_0.9        
[57] desc_1.3.0        pkgbuild_1.2.0    zip_2.1.1         rlang_0.4.10     
[61] pkgconfig_2.0.3   evaluate_0.14     labeling_0.4.2    tidyselect_1.1.1 
[65] processx_3.5.1    plyr_1.8.6        R6_2.5.0          generics_0.1.0   
[69] DBI_1.1.1         pillar_1.6.0      haven_2.4.1       whisker_0.4      
[73] foreign_0.8-80    withr_2.4.2       mgcv_1.8-33       abind_1.4-5      
[77] modelr_0.1.8      crayon_1.4.1      car_3.0-11        utf8_1.1.4       
[81] grid_4.0.3        data.table_1.14.0 callr_3.7.0       git2r_0.28.0     
[85] reprex_2.0.0      digest_0.6.27     httpuv_1.6.0      munsell_0.5.0    
[89] bslib_0.2.5.1     sessioninfo_1.1.1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
